%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}


\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}


% Jupyter Notebook code cell colors
\definecolor{nbsphinxin}{HTML}{307FC1}
\definecolor{nbsphinxout}{HTML}{BF5B3D}
\definecolor{nbsphinx-code-bg}{HTML}{F5F5F5}
\definecolor{nbsphinx-code-border}{HTML}{E0E0E0}
\definecolor{nbsphinx-stderr}{HTML}{FFDDDD}
% ANSI colors for output streams and traceback highlighting
\definecolor{ansi-black}{HTML}{3E424D}
\definecolor{ansi-black-intense}{HTML}{282C36}
\definecolor{ansi-red}{HTML}{E75C58}
\definecolor{ansi-red-intense}{HTML}{B22B31}
\definecolor{ansi-green}{HTML}{00A250}
\definecolor{ansi-green-intense}{HTML}{007427}
\definecolor{ansi-yellow}{HTML}{DDB62B}
\definecolor{ansi-yellow-intense}{HTML}{B27D12}
\definecolor{ansi-blue}{HTML}{208FFB}
\definecolor{ansi-blue-intense}{HTML}{0065CA}
\definecolor{ansi-magenta}{HTML}{D160C4}
\definecolor{ansi-magenta-intense}{HTML}{A03196}
\definecolor{ansi-cyan}{HTML}{60C6C8}
\definecolor{ansi-cyan-intense}{HTML}{258F8F}
\definecolor{ansi-white}{HTML}{C5C1B4}
\definecolor{ansi-white-intense}{HTML}{A1A6B2}
\definecolor{ansi-default-inverse-fg}{HTML}{FFFFFF}
\definecolor{ansi-default-inverse-bg}{HTML}{000000}

% Define an environment for non-plain-text code cell outputs (e.g. images)
\makeatletter
\newenvironment{nbsphinxfancyoutput}{%
    % Avoid fatal error with framed.sty if graphics too long to fit on one page
    \let\sphinxincludegraphics\nbsphinxincludegraphics
    \nbsphinx@image@maxheight\textheight
    \advance\nbsphinx@image@maxheight -2\fboxsep   % default \fboxsep 3pt
    \advance\nbsphinx@image@maxheight -2\fboxrule  % default \fboxrule 0.4pt
    \advance\nbsphinx@image@maxheight -\baselineskip
\def\nbsphinxfcolorbox{\spx@fcolorbox{nbsphinx-code-border}{white}}%
\def\FrameCommand{\nbsphinxfcolorbox\nbsphinxfancyaddprompt\@empty}%
\def\FirstFrameCommand{\nbsphinxfcolorbox\nbsphinxfancyaddprompt\sphinxVerbatim@Continues}%
\def\MidFrameCommand{\nbsphinxfcolorbox\sphinxVerbatim@Continued\sphinxVerbatim@Continues}%
\def\LastFrameCommand{\nbsphinxfcolorbox\sphinxVerbatim@Continued\@empty}%
\MakeFramed{\advance\hsize-\width\@totalleftmargin\z@\linewidth\hsize\@setminipage}%
\lineskip=1ex\lineskiplimit=1ex\raggedright%
}{\par\unskip\@minipagefalse\endMakeFramed}
\makeatother
\newbox\nbsphinxpromptbox
\def\nbsphinxfancyaddprompt{\ifvoid\nbsphinxpromptbox\else
    \kern\fboxrule\kern\fboxsep
    \copy\nbsphinxpromptbox
    \kern-\ht\nbsphinxpromptbox\kern-\dp\nbsphinxpromptbox
    \kern-\fboxsep\kern-\fboxrule\nointerlineskip
    \fi}
\newlength\nbsphinxcodecellspacing
\setlength{\nbsphinxcodecellspacing}{0pt}

% Define support macros for attaching opening and closing lines to notebooks
\newsavebox\nbsphinxbox
\makeatletter
\newcommand{\nbsphinxstartnotebook}[1]{%
    \par
    % measure needed space
    \setbox\nbsphinxbox\vtop{{#1\par}}
    % reserve some space at bottom of page, else start new page
    \needspace{\dimexpr2.5\baselineskip+\ht\nbsphinxbox+\dp\nbsphinxbox}
    % mimick vertical spacing from \section command
      \addpenalty\@secpenalty
      \@tempskipa 3.5ex \@plus 1ex \@minus .2ex\relax
      \addvspace\@tempskipa
      {\Large\@tempskipa\baselineskip
             \advance\@tempskipa-\prevdepth
             \advance\@tempskipa-\ht\nbsphinxbox
             \ifdim\@tempskipa>\z@
               \vskip \@tempskipa
             \fi}
    \unvbox\nbsphinxbox
    % if notebook starts with a \section, prevent it from adding extra space
    \@nobreaktrue\everypar{\@nobreakfalse\everypar{}}%
    % compensate the parskip which will get inserted by next paragraph
    \nobreak\vskip-\parskip
    % do not break here
    \nobreak
}% end of \nbsphinxstartnotebook

\newcommand{\nbsphinxstopnotebook}[1]{%
    \par
    % measure needed space
    \setbox\nbsphinxbox\vbox{{#1\par}}
    \nobreak % it updates page totals
    \dimen@\pagegoal
    \advance\dimen@-\pagetotal \advance\dimen@-\pagedepth
    \advance\dimen@-\ht\nbsphinxbox \advance\dimen@-\dp\nbsphinxbox
    \ifdim\dimen@<\z@
      % little space left
      \unvbox\nbsphinxbox
      \kern-.8\baselineskip
      \nobreak\vskip\z@\@plus1fil
      \penalty100
      \vskip\z@\@plus-1fil
      \kern.8\baselineskip
    \else
      \unvbox\nbsphinxbox
    \fi
}% end of \nbsphinxstopnotebook

% Ensure height of an included graphics fits in nbsphinxfancyoutput frame
\newdimen\nbsphinx@image@maxheight % set in nbsphinxfancyoutput environment
\newcommand*{\nbsphinxincludegraphics}[2][]{%
    \gdef\spx@includegraphics@options{#1}%
    \setbox\spx@image@box\hbox{\includegraphics[#1,draft]{#2}}%
    \in@false
    \ifdim \wd\spx@image@box>\linewidth
      \g@addto@macro\spx@includegraphics@options{,width=\linewidth}%
      \in@true
    \fi
    % no rotation, no need to worry about depth
    \ifdim \ht\spx@image@box>\nbsphinx@image@maxheight
      \g@addto@macro\spx@includegraphics@options{,height=\nbsphinx@image@maxheight}%
      \in@true
    \fi
    \ifin@
      \g@addto@macro\spx@includegraphics@options{,keepaspectratio}%
    \fi
    \setbox\spx@image@box\box\voidb@x % clear memory
    \expandafter\includegraphics\expandafter[\spx@includegraphics@options]{#2}%
}% end of "\MakeFrame"-safe variant of \sphinxincludegraphics
\makeatother

\makeatletter
\renewcommand*\sphinx@verbatim@nolig@list{\do\'\do\`}
\begingroup
\catcode`'=\active
\let\nbsphinx@noligs\@noligs
\g@addto@macro\nbsphinx@noligs{\let'\PYGZsq}
\endgroup
\makeatother
\renewcommand*\sphinxbreaksbeforeactivelist{\do\<\do\"\do\'}
\renewcommand*\sphinxbreaksafteractivelist{\do\.\do\,\do\:\do\;\do\?\do\!\do\/\do\>\do\-}
\makeatletter
\fvset{codes*=\sphinxbreaksattexescapedchars\do\^\^\let\@noligs\nbsphinx@noligs}
\makeatother



\title{MATS}
\date{Jul 19, 2023}
\release{3}
\author{Erin M.\@{} Adkins}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxAtStartPar
\sphinxhref{mailto:erin.adkins@nist.gov}{Erin M. Adkins}

\sphinxAtStartPar
\sphinxstyleemphasis{National Institute of Standards and Technology}

\sphinxAtStartPar
\sphinxstyleemphasis{Last Revision} Jul 19, 2023

\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov/MATS}{This software is available on GitHub} and can be cited with the following DOI \sphinxurl{https://doi.org/10.18434/M32200}

\sphinxAtStartPar
\sphinxstylestrong{The purpose of the MATS project is to develop a NIST\sphinxhyphen{}based multi\sphinxhyphen{}spectrum fitting and analysis tool for spectroscopic data that allows the flexibility to test and adapt to experimental and data\sphinxhyphen{}driven needs.  This software allows for the use of several commonly used spectroscopic line profiles (Voigt, Nelkin\sphinxhyphen{}Ghatak, speed\sphinxhyphen{}dependent Voigt, speed\sphinxhyphen{}dependent Nelkin\sphinxhyphen{}Ghatak, and Hartmann\sphinxhyphen{}Tran) and allows for pressure, temperature, and sample composition constraints to be imposed during fits.  In addition to fitting experimental spectra, MATS can generate simulated spectra, which allows for its use as an error analysis tool.}


\chapter{Contents}
\label{\detokenize{index:contents}}
\sphinxstepscope


\section{Getting Started}
\label{\detokenize{Getting Started:getting-started}}\label{\detokenize{Getting Started::doc}}

\subsection{Set\sphinxhyphen{}Up}
\label{\detokenize{Getting Started:set-up}}
\sphinxAtStartPar
The MATS package was developed and tested using the \sphinxhref{https://www.anaconda.com/}{Anaconda} python environment. Example scripts have typically been written and run using  \sphinxhref{https://jupyter.org/}{jupyter notebooks} (run in jupyter lab), as this allows for code to be run in segments and for iteration on certain code segments.  Any python package distribution should work with the code as long as the dependent packages are installed.


\subsection{Main Packages}
\label{\detokenize{Getting Started:main-packages}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov/MATS}{MATS}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://hitran.org/hapi/}{HITRAN Application Programming Interface (HAPI)}   The v1.2.2.1 was tested and is available in the MATS repository.

\end{itemize}


\subsection{Dependent Packages}
\label{\detokenize{Getting Started:dependent-packages}}
\sphinxAtStartPar
MATS is not written as a basic package, meaning that there are several dependent packages that need to be installed.

\sphinxAtStartPar
If a desired package is not installed then the following command will install it.  Many python package distributions have integrated package managers and required packages can also be installed through that mechanism.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pip} \PYG{n}{install} \PYG{n}{package\PYGZus{}name}
\end{sphinxVerbatim}

\sphinxAtStartPar
There is commonly a delay in the most recent package releases available in python package distribution package managers compared to that available through pip install.  The following command line script will update to the newest release if a package is already installed.  This should only be necessary if there is an error when running MATS with the currently installed version.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{python} \PYG{o}{\PYGZhy{}}\PYG{n}{m} \PYG{n}{pip} \PYG{n}{install} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{upgrade} \PYG{n}{package\PYGZus{}name} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{user}
\end{sphinxVerbatim}

\sphinxAtStartPar
Below are a list of the packages used in MATS.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://www.numpy.org/}{numpy} \sphinxhyphen{} python’s fundamental scientific computing package

\item {} 
\sphinxAtStartPar
\sphinxhref{https://pandas.pydata.org/}{pandas} \sphinxhyphen{} python data structure package

\item {} 
\sphinxAtStartPar
os, sys \sphinxhyphen{} system variables

\item {} 
\sphinxAtStartPar
\sphinxhref{https://lmfit.github.io/lmfit-py/fitting.html}{lmfit} \sphinxhyphen{} non\sphinxhyphen{}linear least\sphinxhyphen{}squares minimization and curve\sphinxhyphen{}fitting for python

\item {} 
\sphinxAtStartPar
\sphinxhref{https://matplotlib.org/}{matplotlib} \sphinxhyphen{} python plotting

\item {} 
\sphinxAtStartPar
\sphinxhref{https://seaborn.pydata.org/}{seaborn} \sphinxhyphen{} pretty plotting

\item {} 
\sphinxAtStartPar
\sphinxhref{https://docs.scipy.org/doc/scipy/reference/fftpack.html}{scipy.fftpack} \sphinxhyphen{} provides fft functionality

\item {} 
\sphinxAtStartPar
\sphinxhref{https://jupyterlab.readthedocs.io/en/stable/}{jupyter lab} \sphinxhyphen{} web\sphinxhyphen{}based user interface for Project Jupyter

\end{itemize}


\subsection{Install}
\label{\detokenize{Getting Started:install}}
\sphinxAtStartPar
Taken from the INSTALL.md file

\sphinxAtStartPar
Basic \sphinxtitleref{MATS} installation instructions.  Note that development is alpha.  In the future, full pypi/conda installs will be made available.

\sphinxAtStartPar
\sphinxstylestrong{Create a conda environment with MATS dependencies}

\sphinxAtStartPar
Download the {[}environment{]}(environment.yml) file.  Note that without setting the environment name below,
by default an evironment \sphinxtitleref{MATS\sphinxhyphen{}env} will be created.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{conda} \PYG{n}{env} \PYG{n}{create} \PYG{o}{\PYGZhy{}}\PYG{n}{n} \PYG{p}{\PYGZob{}}\PYG{n}{optional}\PYG{o}{\PYGZhy{}}\PYG{n}{name}\PYG{o}{\PYGZhy{}}\PYG{n}{of}\PYG{o}{\PYGZhy{}}\PYG{n}{environment}\PYG{p}{\PYGZcb{}} \PYG{o}{\PYGZhy{}}\PYG{n}{f} \PYG{n}{environment}\PYG{o}{.}\PYG{n}{yml}
\end{sphinxVerbatim}

\sphinxAtStartPar
Alternatively, we provide a conda metapackage \sphinxtitleref{mats\sphinxhyphen{}dependencies}, which includes all dependencies for \sphinxtitleref{MATS\textasciigrave{}} (excluding python).
This can be installed using

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{conda} \PYG{n}{install} \PYG{o}{\PYGZhy{}}\PYG{n}{n} \PYG{p}{\PYGZob{}}\PYG{n}{optional}\PYG{o}{\PYGZhy{}}\PYG{n}{name}\PYG{o}{\PYGZhy{}}\PYG{n}{of}\PYG{o}{\PYGZhy{}}\PYG{n}{environment}\PYG{p}{\PYGZcb{}} \PYG{o}{\PYGZhy{}}\PYG{n}{c} \PYG{n}{wpk}\PYG{o}{\PYGZhy{}}\PYG{n}{nist} \PYG{n}{mats}\PYG{o}{\PYGZhy{}}\PYG{n}{dependencies}
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxstylestrong{From source}

\sphinxAtStartPar
After cloning the repo, you can do the following.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pip} \PYG{n}{install} \PYG{o}{.}
\end{sphinxVerbatim}

\sphinxAtStartPar
To install an editable version, use option \sphinxtitleref{\sphinxhyphen{}e}.  To exclude dependencies, use option \sphinxtitleref{\textendash{}no\sphinxhyphen{}deps}

\sphinxAtStartPar
\sphinxstylestrong{With pip from github}

\sphinxAtStartPar
This requires git also be installed.  Downside is that the whole repo (including all examples) are clones.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pip} \PYG{n}{install} \PYG{n}{git}\PYG{o}{+}\PYG{n}{https}\PYG{p}{:}\PYG{o}{/}\PYG{o}{/}\PYG{n}{github}\PYG{o}{.}\PYG{n}{com}\PYG{o}{/}\PYG{n}{wpk}\PYG{o}{\PYGZhy{}}\PYG{n}{nist}\PYG{o}{\PYGZhy{}}\PYG{n}{gov}\PYG{o}{/}\PYG{n}{MATS}\PYG{o}{.}\PYG{n}{git}\PYG{n+nd}{@feature}\PYG{o}{/}\PYG{n}{master}\PYG{o}{\PYGZhy{}}\PYG{n}{reformat}
\end{sphinxVerbatim}

\sphinxAtStartPar
\sphinxstylestrong{With pip from github using wheel}
Note, this is experimental.  Do the following

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pip} \PYG{n}{install} \PYG{n}{https}\PYG{p}{:}\PYG{o}{/}\PYG{o}{/}\PYG{n}{raw}\PYG{o}{.}\PYG{n}{githubusercontent}\PYG{o}{.}\PYG{n}{com}\PYG{o}{/}\PYG{n}{wpk}\PYG{o}{\PYGZhy{}}\PYG{n}{nist}\PYG{o}{\PYGZhy{}}\PYG{n}{gov}\PYG{o}{/}\PYG{n}{MATS}\PYG{o}{/}\PYG{n}{feature}\PYG{o}{/}\PYG{n}{master}\PYG{o}{\PYGZhy{}}\PYG{n}{reformat}\PYG{o}{/}\PYG{n}{wheel}\PYG{o}{/}\PYG{n}{MATS}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{3}\PYG{o}{\PYGZhy{}}\PYG{n}{py3}\PYG{o}{\PYGZhy{}}\PYG{n}{none}\PYG{o}{\PYGZhy{}}\PYG{n+nb}{any}\PYG{o}{.}\PYG{n}{whl}
\end{sphinxVerbatim}

\sphinxAtStartPar
(Note that the actual version will not be 3)

\sphinxAtStartPar
Alternatively, download the wheel and run

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pip} \PYG{n}{install} \PYG{n}{path}\PYG{o}{\PYGZhy{}}\PYG{n}{to}\PYG{o}{\PYGZhy{}}\PYG{n}{wheel}\PYG{o}{.}\PYG{n}{whl}
\end{sphinxVerbatim}

\sphinxstepscope


\section{MATS Version Summary}
\label{\detokenize{MATS Version Summary:mats-version-summary}}\label{\detokenize{MATS Version Summary::doc}}

\subsection{MATS Version 3.0}
\label{\detokenize{MATS Version Summary:mats-version-3-0}}
\sphinxAtStartPar
published on 7/17/2023

\sphinxAtStartPar
\sphinxstylestrong{New Features}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Power law temperature dependence for line\sphinxhyphen{}mixing

\item {} 
\sphinxAtStartPar
Oxygen Theoretical CIA Model based on work of Karman et al

\item {} 
\sphinxAtStartPar
Added ability to select TIPS version

\item {} 
\sphinxAtStartPar
Updated to HAPI v1.2.2.0

\item {} 
\sphinxAtStartPar
Added Example for treating static HITRAN LBL molecules

\item {} 
\sphinxAtStartPar
Added ability to correct ideal gas law w/ compressability factor as calculated with NIST RefProp

\end{itemize}


\subsection{MATS Version 2.1}
\label{\detokenize{MATS Version Summary:mats-version-2-1}}
\sphinxAtStartPar
published on 12/03/2021

\sphinxAtStartPar
\sphinxstylestrong{Focus of version update is cleanup to make package git installable}
\begin{itemize}
\item {} 
\sphinxAtStartPar
renamed python files to lowercase, as this is the preferred python style.

\item {} 
\sphinxAtStartPar
added \_\_init\_\_.py to include the most important functionality at the top level

\item {} 
\sphinxAtStartPar
reformatted constants.  Added codata.py file with includes CODATA dictionary (with values and metadata) and CONSTANTS dictionary (with only values)

\item {} 
\sphinxAtStartPar
Added linelistdata.py file to autoload csv files from MATS/LineList.

\item {} 
\sphinxAtStartPar
Got rid of all \sphinxtitleref{from package import *} commands.  Use explicit imports only

\item {} 
\sphinxAtStartPar
Reworked example notebooks to reflect updates.

\item {} 
\sphinxAtStartPar
Added INSTALL.md file to explain the install options.

\end{itemize}


\subsection{MATS Version 2}
\label{\detokenize{MATS Version Summary:mats-version-2}}
\sphinxAtStartPar
published on 6/10/2021

\sphinxAtStartPar
\sphinxstylestrong{New Features}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Beta correction to the Dicke Narrowing accounting for the hardness of coliisions based on broadener and perturber.

\item {} 
\sphinxAtStartPar
Added CIA functionality.  Currently, can manually enter CIA for each file

\item {} 
\sphinxAtStartPar
Added capability pass non\sphinxhyphen{}fitting columns through the fit definition.  This will help for future addition of bandwide functions or for ease of transforming fit outputs to reported results.

\item {} 
\sphinxAtStartPar
Add ability to weight the spectra

\item {} 
\sphinxAtStartPar
Added beyond HITRAN molecule capabilities, including update of isotope list.

\item {} 
\sphinxAtStartPar
Added preliminary instrument line shape functionality.

\end{itemize}

\sphinxAtStartPar
\sphinxstylestrong{Bug Fixes}
\begin{itemize}
\item {} 
\sphinxAtStartPar
Changed the structure of MATS, so that each class is in its own python file

\item {} 
\sphinxAtStartPar
Changed all constant values to be consistent with CODATA values and hardwired the values to definition in the utility script

\item {} 
\sphinxAtStartPar
Added warning for floating parameters with an initial guess equal to 0.

\item {} 
\sphinxAtStartPar
Changed the etalon\_freq variables to etalon\_period, since that is how it is coded.

\item {} 
\sphinxAtStartPar
Added checks for molecule consistency between dataset and parameter line list.

\item {} 
\sphinxAtStartPar
Ability to simulate at infinite SNR

\item {} 
\sphinxAtStartPar
Indexing error for the spectrum number in baseline parameters

\item {} 
\sphinxAtStartPar
Changed initial baseline guess to 0

\end{itemize}


\subsection{MATS Version 1}
\label{\detokenize{MATS Version Summary:mats-version-1}}
\sphinxAtStartPar
published on 12/27/19

\sphinxstepscope


\section{MATS Summary}
\label{\detokenize{MATS Summary:mats-summary}}\label{\detokenize{MATS Summary::doc}}
\sphinxAtStartPar
In multi\sphinxhyphen{}spectrum fitting, there are a collection of spectra modeled by the same line\sphinxhyphen{}by\sphinxhyphen{}line spectroscopic parameters, but each spectrum might vary in pressure, temperature, and sample composition.

\sphinxAtStartPar
The MATS program is based on {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} objects, which are defined not only by their wavenumber and absorption data, but also information on the spectrum pressure, temperature, baseline characteristics, and sample composition.  In addition to utilizing real spectra, MATS has a {\hyperref[\detokenize{MATS:MATS.spectrum.simulate_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{simulate\_spectrum()}}}}} function, which returns a spectrum object calculated from input simulation parameters.  This is useful for performing error analysis in the same framework as primary data analysis by simply switching from experimental to simulated {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} objects.

\sphinxAtStartPar
These objects are combined to form a {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} object, which is the collection of spectra that are being analyzed together in the multi\sphinxhyphen{}spectrum analysis.

\sphinxAtStartPar
The analysis of spectra in MATS requires an inital spectroscopic linelist.  Details about the format of this linelist and how to generate it using the HITRAN Application Programming Interface are outlined on the  {\hyperref[\detokenize{Generating Parameter Line lists::doc}]{\sphinxcrossref{\DUrole{doc}{Generating Parameter Line lists}}}} page.  A few example line lists have been provided in the \sphinxhref{https://github.com/usnistgov/MATS/tree/master/MATS/Linelists}{Line list folder} , which can be accessed using the {\hyperref[\detokenize{MATS:MATS.linelistdata.LoadLineListData}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{LoadLineListData()}}}}} helper function.  It should be noted that these linelists are provided for use with the examples and to provide an example of line list formatting.  These linelists should not be used as reference data.

\sphinxAtStartPar
There are two files that contain parameters that are fit in this model, one for spectrum dependent parameters (polynomial baseline parameters, etalons, sample composition, and x\sphinxhyphen{}shift term) and the other for line\sphinxhyphen{}by\sphinxhyphen{}line spectroscopic parameters that are common across all spectra.  These files are saved as .csv files with a column for each parameter and with rows corresponding to either the spectrum number or spectral line number.  In addition to the columns containing the values for the fit parameters, there are two additional columns for each fittable parameter called param\_vary and param\_err.  The param\_vary column is a boolean  (True/False) flag that is toggled to indicate whether a given parameter will be varied in the fit.  The param\_err column will be set to zero initially and replaced with the standard error for the parameter determined by the fit results.  Calls of the {\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File}}}}} class not only make these input files, but also set the line shape and define whether a parameter should be varied in the fit and if a parameter should be constrained across all spectra or allowed to vary by spectrum.

\sphinxAtStartPar
Finally, the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}} class fits the spectra.  Additionally, it allows the user to impose constraints on the parameters (min and max values), impose convergence criteria, update background and parameter line lists, and plot fit results.

\sphinxAtStartPar
Below is the sparse documentation for each of the classes and main functions in the MATS project with links to the full documentation provided.


\subsection{Spectrum Class and Objects}
\label{\detokenize{MATS Summary:spectrum-class-and-objects}}

\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\sphinxthistablewithnovlinesstyle
\centering
\begin{tabulary}{\linewidth}[t]{\X{1}{2}\X{1}{2}}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}}(filename{[}, molefraction, ...{]})
&
\sphinxAtStartPar
Spectrum class provides all information describing experimental or simulated spectrum.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.simulate_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{simulate\_spectrum}}}}}(parameter\_linelist{[}, ...{]})
&
\sphinxAtStartPar
Generates a synthetic spectrum, where the output is a spectrum object that can be used in MATS classes.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.calculate_QF}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.calculate\_QF}}}}}()
&
\sphinxAtStartPar
Calculates the quality of fit factor (QF) for a spectrum \sphinxhyphen{} QF = (maximum alpha \sphinxhyphen{} minimum alpha) / std(residuals).
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.fft_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.fft\_spectrum}}}}}()
&
\sphinxAtStartPar
Takes the FFT of the residuals of the spectrum, generates a plot of frequency (cm\sphinxhyphen{}1) versus amplitude (ppm/cm), and prints a dataframe with the 20 highest amplitude frequencies with the FFT frequency (period), amplitude, FFT phase, and frequency (cm\sphinxhyphen{}1).
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.plot_freq_tau}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.plot\_freq\_tau}}}}}()
&
\sphinxAtStartPar
Generates a plot of tau (us) as a function of frequency (MHz).
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.plot_model_residuals}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.plot\_model\_residuals}}}}}()
&
\sphinxAtStartPar
Generates a plot of the alpha and model (ppm/cm) as a function of wavenumber (cm\sphinxhyphen{}1) and on lower plot shows the residuals (ppm/cm) as a function of wavenumber (cm\sphinxhyphen{}1).
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.plot_wave_alpha}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.plot\_wave\_alpha}}}}}()
&
\sphinxAtStartPar
Generates a plot of alpha (ppm/cm) as a function of wavenumber (cm\sphinxhyphen{}1).
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.save_spectrum_info}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.save\_spectrum\_info}}}}}({[}save\_file{]})
&
\sphinxAtStartPar
Saves spectrum information to a pandas dataframe with option to also save as as a csv file.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.segment_wave_alpha}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.segment\_wave\_alpha}}}}}()
&
\sphinxAtStartPar
Defines the wavenumber, alpha, and indices of spectrum that correspond to a given spectrum segment.
\\
\sphinxbottomrule
\end{tabulary}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}


\subsection{Line\sphinxhyphen{}by\sphinxhyphen{}line Model}
\label{\detokenize{MATS Summary:line-by-line-model}}
\sphinxAtStartPar
The line\sphinxhyphen{}by\sphinxhyphen{}line model is based on the HTP code provided in the \sphinxhref{https://hitran.org/hapi/}{HITRAN Application Programming Interface (HAPI)}.  For the most part the conventions and definitions used by HITRAN are used in the MATS program.  However, for some of the advanced line profile parameters the naming convention and temperature dependence is different.  In the sections below, the temperature and pressure dependence of the various parameters is outlined for clarity.  Additionally, MATS uses the CODATA values for calculations.

\sphinxAtStartPar
The Hartmann\sphinxhyphen{}Tran profile limiting cases that correspond to several commonly used line profiles.  These limiting cases are achieved by setting line shape parameters equal to 0. The list below indicates the parameters that are not fixed equal to zero in each of the HTP limiting case line shapes.  For more information about the HTP see the following references: \sphinxhref{https://www.degruyter.com/view/journals/pac/86/12/article-p1931.xml}{Recommended isolated\sphinxhyphen{}line profile for representing high\sphinxhyphen{}resolution spectroscopic transitions (IUPAC Technical Report)} and \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407313002422}{An isolated line\sphinxhyphen{}shape model to go beyond the Voigt profile in spectroscopic databases and radiative transfer codes}

\sphinxAtStartPar
Voigt Profile (VP):  \(\Gamma_{D}, \Gamma_{0}, \Delta_{0}\)

\sphinxAtStartPar
Nelkin\sphinxhyphen{}Ghatak Profile (NGP):  \(\Gamma_{D}, \Gamma_{0}, \Delta_{0}, \nu_{VC}\)

\sphinxAtStartPar
speed\sphinxhyphen{}dependent Voigt Profile (SDVP):  \(\Gamma_{D}, \Gamma_{0}, \Delta_{0}, \Gamma_{2}, \Delta_{2}\)

\sphinxAtStartPar
speed\sphinxhyphen{}dependent Nelkin\sphinxhyphen{}Ghatak Profile (SDNGP):  \(\Gamma_{D}, \Gamma_{0}, \Delta_{0}, \nu_{VC}, \Gamma_{2}, \Delta_{2}\)

\sphinxAtStartPar
Hartmann\sphinxhyphen{}Tran (HTP): \(\Gamma_{D}, \Gamma_{0}, \Delta_{0}, \nu_{VC}, \Gamma_{2}, \Delta_{2}, \eta\)


\subsubsection{Line Intensity}
\label{\detokenize{MATS Summary:line-intensity}}
\sphinxAtStartPar
The line intensity for each line at the experimental temperature is calculated using the EnvironmentDependency\_Intensity function in HAPI.  This function takes as arguments the line intensity at 296 K (\(S(T_{ref})\)), the experimental temperature (\(T\)), the reference temperature 296 K (\(T_{ref}\)), the partition function at the experimental temperature (\(Q(T)\)), the partition function at the reference temperature (\(Q(T_{ref})\)), the lower state energy (\(E"\)), and the line center ((\(\nu\))), and constant (\(c2 = hc/k\)).  The partition functions are calculated using TIPS with the option for the user to select between the three versions available in HAPI TIPS2011, TIPS2017, and TIPS2021. Constants are defined by CODATA values
\begin{equation*}
\begin{split}S(T) = S(T_{ref}) \frac{Q(T_{ref})}{Q(T)}\frac{e^{-c2E"/T}}{e^{-c2E"/T_{ref}}} \frac{1 - e^{-c2\nu / T}}{1 - e^{-c2\nu / T_{ref}}}\end{split}
\end{equation*}

\subsubsection{Doppler Broadening}
\label{\detokenize{MATS Summary:doppler-broadening}}
\sphinxAtStartPar
In MATS, the doppler broadening (\(\Gamma_{D}\))is not a floatable parameter and is calculated based on the experimental temperature (\(T\)), line center (\(\nu\)), and molecular mass (\(m\)).  Constants are defined by CODATA values.  The doppler width is calculated as:
\begin{align*}\!\begin{aligned}
\Gamma_{D} = \sqrt{\frac{2kT \cdot ln(2)}{cMassMol\cdot mc^{2}}} \cdot \nu\\
k = 1.380648813 x 10^{-16} erg K^{-1}\\
cMassMol = 1.66053873x 10^{-24} mol\\
\end{aligned}\end{align*}

\subsubsection{Collisional Half\sphinxhyphen{}Width}
\label{\detokenize{MATS Summary:collisional-half-width}}
\sphinxAtStartPar
The collisional half\sphinxhyphen{}width (\(\Gamma_{0}\)) is a function of both the experimental pressure (\(P\)) and temperature (\(T\)) referenced to \(P_{ref}\) (1 atm) and \(T_{ref}\) (296 K). The contributions from each diluent (k) can be scaled by the diluent composition fraction (\(abun\)) and summed to model the ensemble collisional broadening.  The temperature dependence is modeled as a power law, where \(n\) is the temperature exponent for the collisional width.  The collisional half\sphinxhyphen{}width for each line at experimental temperature and pressure can be represented as:
\begin{equation*}
\begin{split}\Gamma_{0} (P,T) = \sum abun_{k} (\Gamma_{0}^{k} * \frac{P}{P_{ref}} * (\frac{T_{ref}}{T})^{n_{\Gamma_{0}^{k}}})\end{split}
\end{equation*}
\sphinxAtStartPar
In the MATS nomenclature, the collisional half\sphinxhyphen{}width is called gamma0\_diluent and the temperature dependence of the collisional half\sphinxhyphen{}width is called n\_gamma0\_diluent.


\subsubsection{Pressure Shift}
\label{\detokenize{MATS Summary:pressure-shift}}
\sphinxAtStartPar
Just like the collisional half\sphinxhyphen{}width, the pressure shift (\(\Delta_{0}\)) is a function of both the experimental pressure (\(P\)) and temperature (\(T\)) referenced to \(P_{ref}\) (1 atm) and \(T_{ref}\) (296 K). The contributions from each diluent (\(k\)) can be scaled by the diluent composition fraction (\(abun\)) and summed to model the ensemble pressure shift.  Unlike the collisional half\sphinxhyphen{}width, the pressure shift has a linear temperature dependence, where \(n\) represents the temperature dependence of the pressure shift.  The pressure shift for each line at experimental pressure and temperature can be represented as:
\begin{equation*}
\begin{split}\Delta_{0} (P,T) = \sum abun_{k} (\Delta_{0}^{k} +  n_{\Delta_{0}^{k}}\cdot (T - T_{ref}) )\frac{P}{P_{ref}}\end{split}
\end{equation*}
\sphinxAtStartPar
In the MATS nomenclature, the pressure shift is called delta0\_diluent and the temperature dependence of the pressure shift is called n\_delta0\_diluent.


\subsubsection{Speed\sphinxhyphen{}Dependent Broadening}
\label{\detokenize{MATS Summary:speed-dependent-broadening}}
\sphinxAtStartPar
The speed\sphinxhyphen{}dependent mechanism accounts for the speed\sphinxhyphen{}dependence of relaxation rates and is parameterized in the speed\sphinxhyphen{}dependent Voigt (SDVP), speed\sphinxhyphen{}dependent Nelkin\sphinxhyphen{}Ghatak (SDNGP), and Hartmann\sphinxhyphen{}Tran (HTP) profiles.  The speed\sphinxhyphen{}dependent broadening in MATS is tabulated as the ratio \(a{w} = \frac{\Gamma_{2}}{\Gamma_{0}}\), but the actual fitted parameter is \(\Gamma_{2}\).  The temperature dependence of the speed\sphinxhyphen{}dependent broadening is a power law dependence on \(\Gamma_{2}\).  Currently in HITRAN, it is assumed that the \(n_{\Gamma_{0}} = n_{\Gamma_{2}}\), such that \(a_{w}\) is assumed to be temperature independent.  The introduction of \(n_{\Gamma_{2}}\) as a parameter in MATS allows for the option of this assumption to imposed, but the flexibility to explore non\sphinxhyphen{}equivalent temperature dependences between the speed\sphinxhyphen{}dependent and collisional broadening terms.  The contributions from each diluent (\(k\)) can be scaled by the diluent composition fraction (\(abun\)) and summed to model the ensemble speed\sphinxhyphen{}dependent broadening.
\begin{equation*}
\begin{split}\Gamma_{2} (P,T) = \sum abun_{k} (a_{w}^{k} *\Gamma_{0}^{k} * \frac{P}{P_{ref}} * (\frac{T_{ref}}{T})^{n_{\Gamma_{2}^{k}}})\end{split}
\end{equation*}
\sphinxAtStartPar
In the MATS nomenclature, the ratio of the speed\sphinxhyphen{}dependent broadening to the collisional broadening (\(a_{w}\)) is called SD\_gamma\_diluent and the temperature dependence of the speed\sphinxhyphen{}dependent broadening is called n\_gamma2\_diluent.  The difference in the naming structure (SD\_gamma vs gamma2) is chosen to emphasize the difference between the speed\sphinxhyphen{}dependent width being parameterized as a ratio versus as an absolute value.


\subsubsection{Speed\sphinxhyphen{}Dependent Shifting}
\label{\detokenize{MATS Summary:speed-dependent-shifting}}
\sphinxAtStartPar
The speed\sphinxhyphen{}dependent mechanism accounts for the speed\sphinxhyphen{}dependence of relaxation rates and is parameterized in the speed\sphinxhyphen{}dependent Voigt (SDVP), speed\sphinxhyphen{}dependent Nelkin\sphinxhyphen{}Ghatak (SDNGP), and Hartmann\sphinxhyphen{}Tran (HTP) profiles.  The speed\sphinxhyphen{}dependent shift in MATS is tabulated as the ratio \(a{s} = \frac{\Delta_{2}}{\Delta_{0}}\), but the actual fitted parameter is \(\Delta_{2}\).  The temperature dependence of the speed\sphinxhyphen{}dependent shift is modeled with a linear dependence.  Currently, the temperature dependence of the speed\sphinxhyphen{}dependent shift is not parameterized in HITRAN.  The contributions from each diluent (\(k\)) can be scaled by the diluent composition fraction (\(abun\)) and summed to model the ensemble speed\sphinxhyphen{}dependent shift.
\begin{equation*}
\begin{split}\Delta_{2} (P,T) = \sum abun_{k} (a_{s} \cdot \Delta_{0}^{k} +  n_{\Delta_{2}^{k}}\cdot (T - T_{ref}) )\frac{P}{P_{ref}}\end{split}
\end{equation*}
\sphinxAtStartPar
In MATS nomenclature, the ratio of the speed\sphinxhyphen{}dependent shift to the pressure shift (\(a_{s}\)) is called SD\_shift\_diluent and the temperature dependence of the speed\sphinxhyphen{}dependent shift is called n\_delta2\_diluent.  The difference in the naming structure (SD\_delta vs delta2) is chosen to emphasize the difference between the speed\sphinxhyphen{}dependent shift being parameterized as a ratio versus as an absolute value.


\subsubsection{Dicke Narrowing}
\label{\detokenize{MATS Summary:dicke-narrowing}}
\sphinxAtStartPar
The Dicke narrowing mechanism models collisional induced velocity changes and is parameterized in the Nelkin\sphinxhyphen{}Ghatak (NGP), speed\sphinxhyphen{}dependent Nelkin\sphinxhyphen{}Ghatak (SDNGP), and Hartmann\sphinxhyphen{}Tran (HTP) profiles by the term \(\nu_{VC}\). The temperature dependence is modeled as a power law, where \(n\) represents the temperature dependence of the Dicke narrowing term.  If the Dicke narrowing is assumed to behave like the diffusion coefficient, then the temperature dependence theoretically should be 1.  The contributions from each diluent (\(k\)) can be scaled by the diluent composition fraction (\(abun\)) and summed to model the ensemble Dicke narrowing.
\begin{equation*}
\begin{split}\nu_{VC} (P,T) = \sum_{k=i} abun_{k} (\nu_{VC}^{k} * \frac{P}{P_{ref}} * (\frac{T_{ref}}{T})^{n_{\nu{VC}^{k}}})\end{split}
\end{equation*}
\sphinxAtStartPar
In MATS nomenclature, the Dicke narrowing is referred to as nuVC\_diluent and the temperature exponent is n\_nuVC\_diluent.  This differs from the naming convention in HAPI, which changes based on the origin of the Dicke narrowing term (Galatry profile versus HTP).  For simplicity, MATS has adopted a self\sphinxhyphen{}consistent naming convention.


\subsubsection{Correlation Parameter}
\label{\detokenize{MATS Summary:correlation-parameter}}
\sphinxAtStartPar
The correlation parameter (\(\eta\)) models the correlation between velocity and rotation state changes due to collisions and is only parameterized in the Hartmann\sphinxhyphen{}Tran profile (HTP).  Currently, MATS has no temperature or pressure dependence associated with the correlation parameter.  However, contributions from each diluent (\(k\)) can be scaled by the diluent composition fraction (\(abun\)) and summed to model the ensemble correlation parameter.
\begin{equation*}
\begin{split}\eta (k) = \sum abun_{k} \cdot \eta\end{split}
\end{equation*}
\sphinxAtStartPar
In MATS nomenclature, the correlation parameter is referred to as eta\_diluent.


\subsubsection{Line Mixing}
\label{\detokenize{MATS Summary:line-mixing}}
\sphinxAtStartPar
The first order Rosenkranz line mixing (\(Y\)) can be calculated from the imaginary portion of any of the HTP derivative line profiles.  The temperature dependence is modeled as a power law, where \(n\) represents the temperature dependence of the Rosenkranz line\sphinxhyphen{}mixing term.  The contributions from each diluent (\(k\)) can be scaled by the diluent composition fraction (\(abun\)) and  summed to model the ensemble line mixing.
\begin{equation*}
\begin{split}Y(P,T) = \sum_{k = i} abun_{k} (Y^{k} * \frac{P}{P_{ref}} * (\frac{T_{ref}}{T})^{n_{Y^{k}}}\end{split}
\end{equation*}
\sphinxAtStartPar
The line mixing is implemented as:
\begin{equation*}
\begin{split}\alpha = I * (Re{HTP(\Gamma_{D}, \Gamma_{0}, \Delta_{0}, \Gamma_{2}, \Delta_{2}, \nu_{VC}, \eta, \nu)} + Y*Im{HTP(\Gamma_{D}, \Gamma_{0}, \Delta_{0}, \Gamma_{2}, \Delta_{2}, \nu_{VC}, \eta, \nu)})\end{split}
\end{equation*}
\sphinxAtStartPar
In MATS nomenclature, the line mixing parameter is referred to as y\_diluent and the temperature exponent is n\_y\_diluent.  This differs from the naming convention in HAPI, where the parameter name contains information about the corresponding line shape.  MATS does not contain this information in the parameter name.


\subsubsection{Line\sphinxhyphen{}by\sphinxhyphen{}line Models}
\label{\detokenize{MATS Summary:line-by-line-models}}

\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\sphinxthistablewithnovlinesstyle
\centering
\begin{tabulary}{\linewidth}[t]{\X{1}{2}\X{1}{2}}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.HTP_from_DF_select}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{HTP\_from\_DF\_select}}}}}(linelist, waves{[}, ...{]})
&
\sphinxAtStartPar
Calculates the absorbance (ppm/cm) based on input line list, wavenumbers, and spectrum environmental parameters.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.HTP_wBeta_from_DF_select}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{HTP\_wBeta\_from\_DF\_select}}}}}(linelist, waves{[}, ...{]})
&
\sphinxAtStartPar
Calculates the absorbance (ppm/cm) based on input line list, wavenumbers, and spectrum environmental parameters with capability of incorporating the beta correction to the Dicke Narrowing proposed in Analytical\sphinxhyphen{}function correction to the Hartmann\textendash{}Tran profile for more reliable representation of the Dicke\sphinxhyphen{}narrowed molecular spectra.
\\
\sphinxbottomrule
\end{tabulary}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}


\subsection{Dataset Class}
\label{\detokenize{MATS Summary:dataset-class}}

\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\sphinxthistablewithnovlinesstyle
\centering
\begin{tabulary}{\linewidth}[t]{\X{1}{2}\X{1}{2}}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}}(spectra, dataset\_name, param\_linelist)
&
\sphinxAtStartPar
Combines spectrum objects into a Dataset object to enable multi\sphinxhyphen{}spectrum fitting.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.generate_baseline_paramlist}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.generate\_baseline\_paramlist}}}}}()
&
\sphinxAtStartPar
Generates a csv file called dataset\_name + \_baseline\_paramlist, which will be used to generate another csv file that is used for fitting spectrum dependent parameters with columns for spectrum number, segment number, x\_shift, concentration for each molecule in the dataset, baseline terms (a = 0th order term, b = 1st order, etc), and etalon terms (set an amplitude, period, and phase for the number of etalons listed for each spectrum in the Dataset).
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.generate_summary_file}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.generate\_summary\_file}}}}}({[}save\_file{]})
&
\sphinxAtStartPar
Generates a summary file combining spectral information from all spectra in the Dataset.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.get_spectra_extremes}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.get\_spectra\_extremes}}}}}()
&
\sphinxAtStartPar

\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.get_spectrum_extremes}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.get\_spectrum\_extremes}}}}}()
&
\sphinxAtStartPar
Gets the minimum and maximum wavenumber for the entire Dataset.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.average_QF}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.average\_QF}}}}}()
&
\sphinxAtStartPar
Calculates the Average QF from all spectra.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.plot_model_residuals}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.plot\_model\_residuals}}}}}()
&
\sphinxAtStartPar
Generates a plot showing both the model and experimental data as a function of wavenumber in the main plot with a subplot showing the residuals as function of wavenumber.
\\
\sphinxbottomrule
\end{tabulary}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}


\subsection{Generate FitParam File Class}
\label{\detokenize{MATS Summary:generate-fitparam-file-class}}

\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\sphinxthistablewithnovlinesstyle
\centering
\begin{tabulary}{\linewidth}[t]{\X{1}{2}\X{1}{2}}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File}}}}}(dataset, ...{[}, ...{]})
&
\sphinxAtStartPar
Class generates the parameter files used in fitting and sets up fit settings.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.generate_fit_baseline_linelist}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File.generate\_fit\_baseline\_linelist}}}}}({[}...{]})
&
\sphinxAtStartPar
Generates the baseline line list used in fitting and updates the fitting booleans to desired settings.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.generate_fit_param_linelist_from_linelist}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File.generate\_fit\_param\_linelist\_from\_linelist}}}}}({[}...{]})
&
\sphinxAtStartPar
Generates the parameter line list used in fitting and updates the fitting booleans to desired settings.
\\
\sphinxbottomrule
\end{tabulary}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}


\subsection{Fit DataSet Class}
\label{\detokenize{MATS Summary:fit-dataset-class}}

\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\sphinxthistablewithnovlinesstyle
\centering
\begin{tabulary}{\linewidth}[t]{\X{1}{2}\X{1}{2}}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}}(dataset, base\_linelist\_file, ...)
&
\sphinxAtStartPar
Provides the fitting functionality for a Dataset.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.constrained_baseline}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.constrained\_baseline}}}}}(params{[}, ...{]})
&
\sphinxAtStartPar
Imposes baseline constraints when using multiple segments per spectrum, ie all baseline parameters can be the same across the entire spectrum except for the etalon phase, which is allowed to vary per segment.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.fit_data}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.fit\_data}}}}}(params{[}, wing\_cutoff, ...{]})
&
\sphinxAtStartPar
Uses the lmfit minimizer to do the fitting through the simulation model function.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.generate_params}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.generate\_params}}}}}()
&
\sphinxAtStartPar
Generates the lmfit parameter object that will be used in fitting.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.residual_analysis}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.residual\_analysis}}}}}(result{[}, ...{]})
&
\sphinxAtStartPar
Updates the model and residual arrays in each spectrum object with the results of the fit and gives the option of generating the combined absorption and residual plot for each spectrum.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.simulation_model}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.simulation\_model}}}}}(params{[}, ...{]})
&
\sphinxAtStartPar
This is the model used for fitting that includes baseline, resonant absorption, and CIA models.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.update_params}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.update\_params}}}}}(result{[}, ...{]})
&
\sphinxAtStartPar
Updates the baseline and line parameter files based on fit results with the option to write over the file (default) or save as a new file and updates baseline values in the spectrum objects.
\\
\sphinxbottomrule
\end{tabulary}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}


\subsection{Support Modules}
\label{\detokenize{MATS Summary:support-modules}}

\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\sphinxthistablewithnovlinesstyle
\centering
\begin{tabulary}{\linewidth}[t]{\X{1}{2}\X{1}{2}}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.utilities.etalon}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{etalon}}}}}(x, amp, period, phase)
&
\sphinxAtStartPar
Etalon definition
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.utilities.molecularMass}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{molecularMass}}}}}(M, I{[}, isotope\_list{]})
&
\sphinxAtStartPar
molecular mass look\sphinxhyphen{}up based on the HAPI definition adapted to allow used to specify ISO list INPUT PARAMETERS:     M: HITRAN molecule number     I: HITRAN isotopologue number OUTPUT PARAMETERS:     MolMass: molecular mass \sphinxhyphen{}\sphinxhyphen{}\sphinxhyphen{} DESCRIPTION:     Return molecular mass of HITRAN isotolopogue.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.utilities.isotope_list_molecules_isotopes}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{isotope\_list\_molecules\_isotopes}}}}}({[}isotope\_list{]})
&
\sphinxAtStartPar
The HITRAN style isotope list in the format (M,I), this function creates a dictionary from this with M as the keys and lists of I as values.
\\
\sphinxhline
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.utilities.add_to_HITRANstyle_isotope_list}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{add\_to\_HITRANstyle\_isotope\_list}}}}}({[}...{]})
&
\sphinxAtStartPar
Allows for used to add to an existing isotope line list in the HITRAN format
\\
\sphinxbottomrule
\end{tabulary}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}


\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\sphinxthistablewithnovlinesstyle
\centering
\begin{tabulary}{\linewidth}[t]{\X{1}{2}\X{1}{2}}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.linelistdata.LoadLineListData}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{LoadLineListData}}}}}({[}paths{]})
&
\sphinxAtStartPar
Helper class to read in supplied LineList DataFrames
\\
\sphinxbottomrule
\end{tabulary}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}

\sphinxstepscope


\section{Generating Parameter Line lists}
\label{\detokenize{Generating Parameter Line lists:generating-parameter-line-lists}}\label{\detokenize{Generating Parameter Line lists::doc}}

\subsection{Parameter Line list Overview}
\label{\detokenize{Generating Parameter Line lists:parameter-line-list-overview}}\begin{description}
\sphinxlineitem{The MATS program uses a spectroscopic line list that varies from the traditional HITRAN and HAPI formats to accomodate the full HTP parameterization including temperature dependencies for all parameters (other than the correlation parameter and first order line\sphinxhyphen{}mixing).  The spectroscopic line list format has rows that correspond to each molecular transition and columns that correspond to the various HTP parameters.  The necessary column headers are below, where there needs to be a column for each parameter for each diluent (ie. air, self) in the dataset. Additionally, there needs to be a line\sphinxhyphen{}mixing term for each nominal temperature in the dataset, where the nominal temperature is the included as the suffix of the column header.}\begin{itemize}
\item {} 
\sphinxAtStartPar
molec\_id: HITRAN molecule id number

\item {} 
\sphinxAtStartPar
local\_iso\_id: HITRAN local isotope id number

\item {} 
\sphinxAtStartPar
nu: line center (\(cm^{-1}\))

\item {} 
\sphinxAtStartPar
sw : spectral line intensity (\(\frac{cm^{-1}}{molecule \cdot cm^{-2}}\)) at Tref=296K

\item {} 
\sphinxAtStartPar
elower: The lower\sphinxhyphen{}state energy of the transition (\(cm^{-1}\))

\item {} 
\sphinxAtStartPar
gamma0\_diluent: collisional half\sphinxhyphen{}width (\(\frac{cm^{-1}}{atm}\)) at Tref=296K and reference pressure pref=1atm for a given diluent

\item {} 
\sphinxAtStartPar
n\_gamma0\_diluent: coefficient of the temperature dependence of the collisional half\sphinxhyphen{}width

\item {} 
\sphinxAtStartPar
delta0\_diluent: pressure shift (\(\frac{cm^{-1}}{atm}\)) at Tref=296K and pref=1atm of the line position with respect to line center

\item {} 
\sphinxAtStartPar
n\_delta0\_diluent:  the coefficient of the temperature dependence of the pressure shift

\item {} 
\sphinxAtStartPar
SD\_gamma\_diluent: the ratio of the speed dependent width to the collisional half\sphinxhyphen{}width at reference temperature and pressure

\item {} 
\sphinxAtStartPar
n\_gamma2\_diluent: the coefficient of the temperature dependence of the speed\sphinxhyphen{}dependent width NOTE: This is the temperature dependence of the speed\sphinxhyphen{}dependent width not the temperature dependence of the ratio of the speed\sphinxhyphen{}dependent width to the collisional half\sphinxhyphen{}width

\item {} 
\sphinxAtStartPar
SD\_delta\_diluent:  the ratio of the speed\sphinxhyphen{}dependent shift to the collisional shift at reference temperature and pressure

\item {} 
\sphinxAtStartPar
n\_delta2\_diluent: the coefficient of the temperature dependence of the speed\sphinxhyphen{}dependent shift NOTE: This is the temperature dependence of the speed\sphinxhyphen{}dependent shift not the temperature dependence of the ratio of the speed\sphinxhyphen{}dependent shift to the pressure shift

\item {} 
\sphinxAtStartPar
nuVC\_diluent: Dicke narrowing term (\(\frac{cm^{-1}}{atm}\)) at reference temperature and pressure

\item {} 
\sphinxAtStartPar
n\_\_nuVC\_diluent:  coefficient of the temperature dependence of the Dicke narrowing term

\item {} 
\sphinxAtStartPar
eta\_diluent:  correlation parameter

\item {} 
\sphinxAtStartPar
y\_diluent\_nominaltemp: line mixing term (\(\frac{cm^{-1}}{atm}\)) NOTE: As currently written , this doesn’t have a temperature dependence, so there is a different column for each nominal temperature.

\end{itemize}

\end{description}


\subsection{Generating Parameter Line list from HITRAN}
\label{\detokenize{Generating Parameter Line lists:generating-parameter-line-list-from-hitran}}
\sphinxAtStartPar
The parameter line list .csv file can be generated manually, but \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/HITRAN\_to\_Dataframe}{this notebook} generates a parameter list using the \sphinxhref{https://hitran.org/hapi/}{HITRAN Application Programming Interface (HAPI)}.  The overview below walks through this notebook.


\subsubsection{Imports}
\label{\detokenize{Generating Parameter Line lists:imports}}
\sphinxAtStartPar
Import the numpy, pandas, os, and sys packages.  Set pandas dataframe to show all of the rows.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{n}{pd}\PYG{o}{.}\PYG{n}{set\PYGZus{}option}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{display.max\PYGZus{}rows}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+m+mi}{101}\PYG{p}{)}
\PYG{k+kn}{import} \PYG{n+nn}{os}\PYG{o}{,} \PYG{n+nn}{sys}
\end{sphinxVerbatim}

\sphinxAtStartPar
Optional imports include matplotlib and seaborn for plotting.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{import} \PYG{n+nn}{seaborn} \PYG{k}{as} \PYG{n+nn}{sns}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}style}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{whitegrid}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}style}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ticks}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}context}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{poster}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}


\subsubsection{Molecule and Isotope Information}
\label{\detokenize{Generating Parameter Line lists:molecule-and-isotope-information}}
\sphinxAtStartPar
HAPI provides a dictionary called ISO\_ID, where the global isotope id acts as the key, with a sub\sphinxhyphen{}dictionary containing the molecule id, local isotope id, isotope name, relative abundance, mass, and molecule name.  The following command will generate the dictionary as the output.  This provides the necessary global isotope id information for interfacing with HAPI to access HITRAN infomation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{print\PYGZus{}iso\PYGZus{}id}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{The} \PYG{n}{dictionary} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ISO\PYGZus{}ID}\PYG{l+s+s2}{\PYGZdq{}} \PYG{n}{contains} \PYG{n}{information} \PYG{n}{on} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{global}\PYG{l+s+s2}{\PYGZdq{}} \PYG{n}{IDs} \PYG{n}{of} \PYG{n}{isotopologues} \PYG{o+ow}{in} \PYG{n}{HITRAN}

   \PYG{n+nb}{id}            \PYG{n}{M}    \PYG{n}{I}                    \PYG{n}{iso\PYGZus{}name}       \PYG{n}{abundance}       \PYG{n}{mass}        \PYG{n}{mol\PYGZus{}name}
    \PYG{l+m+mi}{1}     \PYG{p}{:}      \PYG{l+m+mi}{1}    \PYG{l+m+mi}{1}                     \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.9973170000}  \PYG{l+m+mf}{18.010565}             \PYG{n}{H2O}
    \PYG{l+m+mi}{2}     \PYG{p}{:}      \PYG{l+m+mi}{1}    \PYG{l+m+mi}{2}                     \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0019998300}  \PYG{l+m+mf}{20.014811}             \PYG{n}{H2O}
    \PYG{l+m+mi}{3}     \PYG{p}{:}      \PYG{l+m+mi}{1}    \PYG{l+m+mi}{3}                     \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0003720000}  \PYG{l+m+mf}{19.014780}             \PYG{n}{H2O}
    \PYG{l+m+mi}{4}     \PYG{p}{:}      \PYG{l+m+mi}{1}    \PYG{l+m+mi}{4}                     \PYG{n}{HD}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0003106900}  \PYG{l+m+mf}{19.016740}             \PYG{n}{H2O}
    \PYG{l+m+mi}{5}     \PYG{p}{:}      \PYG{l+m+mi}{1}    \PYG{l+m+mi}{5}                     \PYG{n}{HD}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000006230}  \PYG{l+m+mf}{21.020985}             \PYG{n}{H2O}
    \PYG{l+m+mi}{6}     \PYG{p}{:}      \PYG{l+m+mi}{1}    \PYG{l+m+mi}{6}                     \PYG{n}{HD}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000001160}  \PYG{l+m+mf}{20.020956}             \PYG{n}{H2O}
  \PYG{l+m+mi}{129}     \PYG{p}{:}      \PYG{l+m+mi}{1}    \PYG{l+m+mi}{7}                     \PYG{n}{D2}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000000242}  \PYG{l+m+mf}{20.022915}             \PYG{n}{H2O}
    \PYG{l+m+mi}{7}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9842040000}  \PYG{l+m+mf}{43.989830}             \PYG{n}{CO2}
    \PYG{l+m+mi}{8}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{2}                 \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.0110570000}  \PYG{l+m+mf}{44.993185}             \PYG{n}{CO2}
    \PYG{l+m+mi}{9}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{3}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0039471000}  \PYG{l+m+mf}{45.994076}             \PYG{n}{CO2}
   \PYG{l+m+mi}{10}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{4}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0007340000}  \PYG{l+m+mf}{44.994045}             \PYG{n}{CO2}
   \PYG{l+m+mi}{11}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{5}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000443400}  \PYG{l+m+mf}{46.997431}             \PYG{n}{CO2}
   \PYG{l+m+mi}{12}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{6}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000082500}  \PYG{l+m+mf}{45.997400}             \PYG{n}{CO2}
   \PYG{l+m+mi}{13}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{7}                 \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.0000039573}  \PYG{l+m+mf}{47.998322}             \PYG{n}{CO2}
   \PYG{l+m+mi}{14}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{8}             \PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000014700}  \PYG{l+m+mf}{46.998291}             \PYG{n}{CO2}
  \PYG{l+m+mi}{121}     \PYG{p}{:}      \PYG{l+m+mi}{2}    \PYG{l+m+mi}{9}                 \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.0000001368}  \PYG{l+m+mf}{45.998262}             \PYG{n}{CO2}
   \PYG{l+m+mi}{15}     \PYG{p}{:}      \PYG{l+m+mi}{2}   \PYG{l+m+mi}{10}                 \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.0000000450}  \PYG{l+m+mf}{49.001675}             \PYG{n}{CO2}
  \PYG{l+m+mi}{120}     \PYG{p}{:}      \PYG{l+m+mi}{2}   \PYG{l+m+mi}{11}             \PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000000165}  \PYG{l+m+mf}{48.001650}             \PYG{n}{CO2}
  \PYG{l+m+mi}{122}     \PYG{p}{:}      \PYG{l+m+mi}{2}   \PYG{l+m+mi}{12}                 \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.0000000015}  \PYG{l+m+mf}{47.001618}             \PYG{n}{CO2}
   \PYG{l+m+mi}{16}     \PYG{p}{:}      \PYG{l+m+mi}{3}    \PYG{l+m+mi}{1}                      \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{3}    \PYG{l+m+mf}{0.9929010000}  \PYG{l+m+mf}{47.984745}              \PYG{n}{O3}
   \PYG{l+m+mi}{17}     \PYG{p}{:}      \PYG{l+m+mi}{3}    \PYG{l+m+mi}{2}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0039819400}  \PYG{l+m+mf}{49.988991}              \PYG{n}{O3}
   \PYG{l+m+mi}{18}     \PYG{p}{:}      \PYG{l+m+mi}{3}    \PYG{l+m+mi}{3}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0019909700}  \PYG{l+m+mf}{49.988991}              \PYG{n}{O3}
   \PYG{l+m+mi}{19}     \PYG{p}{:}      \PYG{l+m+mi}{3}    \PYG{l+m+mi}{4}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0007400000}  \PYG{l+m+mf}{48.988960}              \PYG{n}{O3}
   \PYG{l+m+mi}{20}     \PYG{p}{:}      \PYG{l+m+mi}{3}    \PYG{l+m+mi}{5}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0003700000}  \PYG{l+m+mf}{48.988960}              \PYG{n}{O3}
   \PYG{l+m+mi}{21}     \PYG{p}{:}      \PYG{l+m+mi}{4}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.9903330000}  \PYG{l+m+mf}{44.001062}             \PYG{n}{N2O}
   \PYG{l+m+mi}{22}     \PYG{p}{:}      \PYG{l+m+mi}{4}    \PYG{l+m+mi}{2}             \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{15}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0036409000}  \PYG{l+m+mf}{44.998096}             \PYG{n}{N2O}
   \PYG{l+m+mi}{23}     \PYG{p}{:}      \PYG{l+m+mi}{4}    \PYG{l+m+mi}{3}             \PYG{p}{(}\PYG{l+m+mi}{15}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0036409000}  \PYG{l+m+mf}{44.998096}             \PYG{n}{N2O}
   \PYG{l+m+mi}{24}     \PYG{p}{:}      \PYG{l+m+mi}{4}    \PYG{l+m+mi}{4}                 \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0019858200}  \PYG{l+m+mf}{46.005308}             \PYG{n}{N2O}
   \PYG{l+m+mi}{25}     \PYG{p}{:}      \PYG{l+m+mi}{4}    \PYG{l+m+mi}{5}                 \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0003690000}  \PYG{l+m+mf}{45.005278}             \PYG{n}{N2O}
   \PYG{l+m+mi}{26}     \PYG{p}{:}      \PYG{l+m+mi}{5}    \PYG{l+m+mi}{1}                  \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.9865400000}  \PYG{l+m+mf}{27.994915}              \PYG{n}{CO}
   \PYG{l+m+mi}{27}     \PYG{p}{:}      \PYG{l+m+mi}{5}    \PYG{l+m+mi}{2}                  \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0110800000}  \PYG{l+m+mf}{28.998270}              \PYG{n}{CO}
   \PYG{l+m+mi}{28}     \PYG{p}{:}      \PYG{l+m+mi}{5}    \PYG{l+m+mi}{3}                  \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0019782000}  \PYG{l+m+mf}{29.999161}              \PYG{n}{CO}
   \PYG{l+m+mi}{29}     \PYG{p}{:}      \PYG{l+m+mi}{5}    \PYG{l+m+mi}{4}                  \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0003680000}  \PYG{l+m+mf}{28.999130}              \PYG{n}{CO}
   \PYG{l+m+mi}{30}     \PYG{p}{:}      \PYG{l+m+mi}{5}    \PYG{l+m+mi}{5}                  \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000222200}  \PYG{l+m+mf}{31.002516}              \PYG{n}{CO}
   \PYG{l+m+mi}{31}     \PYG{p}{:}      \PYG{l+m+mi}{5}    \PYG{l+m+mi}{6}                  \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0000041300}  \PYG{l+m+mf}{30.002485}              \PYG{n}{CO}
   \PYG{l+m+mi}{32}     \PYG{p}{:}      \PYG{l+m+mi}{6}    \PYG{l+m+mi}{1}                     \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H4}    \PYG{l+m+mf}{0.9882700000}  \PYG{l+m+mf}{16.031300}             \PYG{n}{CH4}
   \PYG{l+m+mi}{33}     \PYG{p}{:}      \PYG{l+m+mi}{6}    \PYG{l+m+mi}{2}                     \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H4}    \PYG{l+m+mf}{0.0111000000}  \PYG{l+m+mf}{17.034655}             \PYG{n}{CH4}
   \PYG{l+m+mi}{34}     \PYG{p}{:}      \PYG{l+m+mi}{6}    \PYG{l+m+mi}{3}                    \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3D}    \PYG{l+m+mf}{0.0006157500}  \PYG{l+m+mf}{17.037475}             \PYG{n}{CH4}
   \PYG{l+m+mi}{35}     \PYG{p}{:}      \PYG{l+m+mi}{6}    \PYG{l+m+mi}{4}                    \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3D}    \PYG{l+m+mf}{0.0000049203}  \PYG{l+m+mf}{18.040830}             \PYG{n}{CH4}
   \PYG{l+m+mi}{36}     \PYG{p}{:}      \PYG{l+m+mi}{7}    \PYG{l+m+mi}{1}                      \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9952620000}  \PYG{l+m+mf}{31.989830}              \PYG{n}{O2}
   \PYG{l+m+mi}{37}     \PYG{p}{:}      \PYG{l+m+mi}{7}    \PYG{l+m+mi}{2}                  \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0039914100}  \PYG{l+m+mf}{33.994076}              \PYG{n}{O2}
   \PYG{l+m+mi}{38}     \PYG{p}{:}      \PYG{l+m+mi}{7}    \PYG{l+m+mi}{3}                  \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{17}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0007420000}  \PYG{l+m+mf}{32.994045}              \PYG{n}{O2}
   \PYG{l+m+mi}{39}     \PYG{p}{:}      \PYG{l+m+mi}{8}    \PYG{l+m+mi}{1}                  \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.9939740000}  \PYG{l+m+mf}{29.997989}              \PYG{n}{NO}
   \PYG{l+m+mi}{40}     \PYG{p}{:}      \PYG{l+m+mi}{8}    \PYG{l+m+mi}{2}                  \PYG{p}{(}\PYG{l+m+mi}{15}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0036543000}  \PYG{l+m+mf}{30.995023}              \PYG{n}{NO}
   \PYG{l+m+mi}{41}     \PYG{p}{:}      \PYG{l+m+mi}{8}    \PYG{l+m+mi}{3}                  \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0019931200}  \PYG{l+m+mf}{32.002234}              \PYG{n}{NO}
   \PYG{l+m+mi}{42}     \PYG{p}{:}      \PYG{l+m+mi}{9}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9456800000}  \PYG{l+m+mf}{63.961901}             \PYG{n}{SO2}
   \PYG{l+m+mi}{43}     \PYG{p}{:}      \PYG{l+m+mi}{9}    \PYG{l+m+mi}{2}                 \PYG{p}{(}\PYG{l+m+mi}{34}\PYG{n}{S}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.0419500000}  \PYG{l+m+mf}{65.957695}             \PYG{n}{SO2}
   \PYG{l+m+mi}{44}     \PYG{p}{:}     \PYG{l+m+mi}{10}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9916160000}  \PYG{l+m+mf}{45.992904}             \PYG{n}{NO2}
   \PYG{l+m+mi}{45}     \PYG{p}{:}     \PYG{l+m+mi}{11}    \PYG{l+m+mi}{1}                     \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{n}{H3}    \PYG{l+m+mf}{0.9958715000}  \PYG{l+m+mf}{17.026549}             \PYG{n}{NH3}
   \PYG{l+m+mi}{46}     \PYG{p}{:}     \PYG{l+m+mi}{11}    \PYG{l+m+mi}{2}                     \PYG{p}{(}\PYG{l+m+mi}{15}\PYG{n}{N}\PYG{p}{)}\PYG{n}{H3}    \PYG{l+m+mf}{0.0036613000}  \PYG{l+m+mf}{18.023583}             \PYG{n}{NH3}
   \PYG{l+m+mi}{47}     \PYG{p}{:}     \PYG{l+m+mi}{12}    \PYG{l+m+mi}{1}                \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{3}    \PYG{l+m+mf}{0.9891100000}  \PYG{l+m+mf}{62.995644}            \PYG{n}{HNO3}
  \PYG{l+m+mi}{117}     \PYG{p}{:}     \PYG{l+m+mi}{12}    \PYG{l+m+mi}{2}                \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{15}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{3}    \PYG{l+m+mf}{0.0036360000}  \PYG{l+m+mf}{63.992680}            \PYG{n}{HNO3}
   \PYG{l+m+mi}{48}     \PYG{p}{:}     \PYG{l+m+mi}{13}    \PYG{l+m+mi}{1}                      \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{n}{H}    \PYG{l+m+mf}{0.9974730000}  \PYG{l+m+mf}{17.002740}              \PYG{n}{OH}
   \PYG{l+m+mi}{49}     \PYG{p}{:}     \PYG{l+m+mi}{13}    \PYG{l+m+mi}{2}                      \PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}\PYG{n}{H}    \PYG{l+m+mf}{0.0020001400}  \PYG{l+m+mf}{19.006986}              \PYG{n}{OH}
   \PYG{l+m+mi}{50}     \PYG{p}{:}     \PYG{l+m+mi}{13}    \PYG{l+m+mi}{3}                      \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{n}{D}    \PYG{l+m+mf}{0.0001553700}  \PYG{l+m+mf}{18.008915}              \PYG{n}{OH}
   \PYG{l+m+mi}{51}     \PYG{p}{:}     \PYG{l+m+mi}{14}    \PYG{l+m+mi}{1}                      \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{19}\PYG{n}{F}\PYG{p}{)}    \PYG{l+m+mf}{0.9998442500}  \PYG{l+m+mf}{20.006229}              \PYG{n}{HF}
  \PYG{l+m+mi}{110}     \PYG{p}{:}     \PYG{l+m+mi}{14}    \PYG{l+m+mi}{2}                      \PYG{n}{D}\PYG{p}{(}\PYG{l+m+mi}{19}\PYG{n}{F}\PYG{p}{)}    \PYG{l+m+mf}{0.0001150000}  \PYG{l+m+mf}{21.012505}              \PYG{n}{HF}
   \PYG{l+m+mi}{52}     \PYG{p}{:}     \PYG{l+m+mi}{15}    \PYG{l+m+mi}{1}                     \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{35}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.7575870000}  \PYG{l+m+mf}{35.976678}             \PYG{n}{HCl}
   \PYG{l+m+mi}{53}     \PYG{p}{:}     \PYG{l+m+mi}{15}    \PYG{l+m+mi}{2}                     \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{37}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.2422570000}  \PYG{l+m+mf}{37.973729}             \PYG{n}{HCl}
  \PYG{l+m+mi}{107}     \PYG{p}{:}     \PYG{l+m+mi}{15}    \PYG{l+m+mi}{3}                     \PYG{n}{D}\PYG{p}{(}\PYG{l+m+mi}{35}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.0001180050}  \PYG{l+m+mf}{36.982954}             \PYG{n}{HCl}
  \PYG{l+m+mi}{108}     \PYG{p}{:}     \PYG{l+m+mi}{15}    \PYG{l+m+mi}{4}                     \PYG{n}{D}\PYG{p}{(}\PYG{l+m+mi}{37}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.0000377350}  \PYG{l+m+mf}{38.980004}             \PYG{n}{HCl}
   \PYG{l+m+mi}{54}     \PYG{p}{:}     \PYG{l+m+mi}{16}    \PYG{l+m+mi}{1}                     \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{79}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.5067800000}  \PYG{l+m+mf}{79.926160}             \PYG{n}{HBr}
   \PYG{l+m+mi}{55}     \PYG{p}{:}     \PYG{l+m+mi}{16}    \PYG{l+m+mi}{2}                     \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{81}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.4930600000}  \PYG{l+m+mf}{81.924115}             \PYG{n}{HBr}
  \PYG{l+m+mi}{111}     \PYG{p}{:}     \PYG{l+m+mi}{16}    \PYG{l+m+mi}{3}                     \PYG{n}{D}\PYG{p}{(}\PYG{l+m+mi}{79}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.0000582935}  \PYG{l+m+mf}{80.932439}             \PYG{n}{HBr}
  \PYG{l+m+mi}{112}     \PYG{p}{:}     \PYG{l+m+mi}{16}    \PYG{l+m+mi}{4}                     \PYG{n}{D}\PYG{p}{(}\PYG{l+m+mi}{81}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.0000567065}  \PYG{l+m+mf}{82.930392}             \PYG{n}{HBr}
   \PYG{l+m+mi}{56}     \PYG{p}{:}     \PYG{l+m+mi}{17}    \PYG{l+m+mi}{1}                     \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{127}\PYG{n}{I}\PYG{p}{)}    \PYG{l+m+mf}{0.9998442500} \PYG{l+m+mf}{127.912297}              \PYG{n}{HI}
  \PYG{l+m+mi}{113}     \PYG{p}{:}     \PYG{l+m+mi}{17}    \PYG{l+m+mi}{2}                     \PYG{n}{D}\PYG{p}{(}\PYG{l+m+mi}{127}\PYG{n}{I}\PYG{p}{)}    \PYG{l+m+mf}{0.0001150000} \PYG{l+m+mf}{128.918575}              \PYG{n}{HI}
   \PYG{l+m+mi}{57}     \PYG{p}{:}     \PYG{l+m+mi}{18}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{35}\PYG{n}{Cl}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.7559100000}  \PYG{l+m+mf}{50.963768}             \PYG{n}{ClO}
   \PYG{l+m+mi}{58}     \PYG{p}{:}     \PYG{l+m+mi}{18}    \PYG{l+m+mi}{2}                 \PYG{p}{(}\PYG{l+m+mi}{37}\PYG{n}{Cl}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.2417200000}  \PYG{l+m+mf}{52.960819}             \PYG{n}{ClO}
   \PYG{l+m+mi}{59}     \PYG{p}{:}     \PYG{l+m+mi}{19}    \PYG{l+m+mi}{1}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.9373900000}  \PYG{l+m+mf}{59.966986}             \PYG{n}{OCS}
   \PYG{l+m+mi}{60}     \PYG{p}{:}     \PYG{l+m+mi}{19}    \PYG{l+m+mi}{2}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{34}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0415800000}  \PYG{l+m+mf}{61.962780}             \PYG{n}{OCS}
   \PYG{l+m+mi}{61}     \PYG{p}{:}     \PYG{l+m+mi}{19}    \PYG{l+m+mi}{3}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0105300000}  \PYG{l+m+mf}{60.970341}             \PYG{n}{OCS}
   \PYG{l+m+mi}{62}     \PYG{p}{:}     \PYG{l+m+mi}{19}    \PYG{l+m+mi}{4}             \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{33}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0105300000}  \PYG{l+m+mf}{60.966371}             \PYG{n}{OCS}
   \PYG{l+m+mi}{63}     \PYG{p}{:}     \PYG{l+m+mi}{19}    \PYG{l+m+mi}{5}             \PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0018800000}  \PYG{l+m+mf}{61.971231}             \PYG{n}{OCS}
   \PYG{l+m+mi}{64}     \PYG{p}{:}     \PYG{l+m+mi}{20}    \PYG{l+m+mi}{1}                \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.9862400000}  \PYG{l+m+mf}{30.010565}            \PYG{n}{H2CO}
   \PYG{l+m+mi}{65}     \PYG{p}{:}     \PYG{l+m+mi}{20}    \PYG{l+m+mi}{2}                \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0110800000}  \PYG{l+m+mf}{31.013920}            \PYG{n}{H2CO}
   \PYG{l+m+mi}{66}     \PYG{p}{:}     \PYG{l+m+mi}{20}    \PYG{l+m+mi}{3}                \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{18}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.0019776000}  \PYG{l+m+mf}{32.014811}            \PYG{n}{H2CO}
   \PYG{l+m+mi}{67}     \PYG{p}{:}     \PYG{l+m+mi}{21}    \PYG{l+m+mi}{1}                \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{35}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.7557900000}  \PYG{l+m+mf}{51.971593}            \PYG{n}{HOCl}
   \PYG{l+m+mi}{68}     \PYG{p}{:}     \PYG{l+m+mi}{21}    \PYG{l+m+mi}{2}                \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{37}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.2416800000}  \PYG{l+m+mf}{53.968644}            \PYG{n}{HOCl}
   \PYG{l+m+mi}{69}     \PYG{p}{:}     \PYG{l+m+mi}{22}    \PYG{l+m+mi}{1}                      \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9926874000}  \PYG{l+m+mf}{28.006147}              \PYG{n}{N2}
  \PYG{l+m+mi}{118}     \PYG{p}{:}     \PYG{l+m+mi}{22}    \PYG{l+m+mi}{2}                  \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{15}\PYG{n}{N}\PYG{p}{)}    \PYG{l+m+mf}{0.0072535000}  \PYG{l+m+mf}{29.997989}              \PYG{n}{N2}
   \PYG{l+m+mi}{70}     \PYG{p}{:}     \PYG{l+m+mi}{23}    \PYG{l+m+mi}{1}                 \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}    \PYG{l+m+mf}{0.9851100000}  \PYG{l+m+mf}{27.010899}             \PYG{n}{HCN}
   \PYG{l+m+mi}{71}     \PYG{p}{:}     \PYG{l+m+mi}{23}    \PYG{l+m+mi}{2}                 \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}    \PYG{l+m+mf}{0.0110700000}  \PYG{l+m+mf}{28.014254}             \PYG{n}{HCN}
   \PYG{l+m+mi}{72}     \PYG{p}{:}     \PYG{l+m+mi}{23}    \PYG{l+m+mi}{3}                 \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{15}\PYG{n}{N}\PYG{p}{)}    \PYG{l+m+mf}{0.0036217000}  \PYG{l+m+mf}{28.007933}             \PYG{n}{HCN}
   \PYG{l+m+mi}{73}     \PYG{p}{:}     \PYG{l+m+mi}{24}    \PYG{l+m+mi}{1}               \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}\PYG{p}{(}\PYG{l+m+mi}{35}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.7489400000}  \PYG{l+m+mf}{49.992328}           \PYG{n}{CH3Cl}
   \PYG{l+m+mi}{74}     \PYG{p}{:}     \PYG{l+m+mi}{24}    \PYG{l+m+mi}{2}               \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}\PYG{p}{(}\PYG{l+m+mi}{37}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.2394900000}  \PYG{l+m+mf}{51.989379}           \PYG{n}{CH3Cl}
   \PYG{l+m+mi}{75}     \PYG{p}{:}     \PYG{l+m+mi}{25}    \PYG{l+m+mi}{1}                    \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9949520000}  \PYG{l+m+mf}{34.005480}            \PYG{n}{H2O2}
   \PYG{l+m+mi}{76}     \PYG{p}{:}     \PYG{l+m+mi}{26}    \PYG{l+m+mi}{1}                    \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{n}{H2}    \PYG{l+m+mf}{0.9776000000}  \PYG{l+m+mf}{26.015650}            \PYG{n}{C2H2}
   \PYG{l+m+mi}{77}     \PYG{p}{:}     \PYG{l+m+mi}{26}    \PYG{l+m+mi}{2}                \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H2}    \PYG{l+m+mf}{0.0219700000}  \PYG{l+m+mf}{27.019005}            \PYG{n}{C2H2}
  \PYG{l+m+mi}{105}     \PYG{p}{:}     \PYG{l+m+mi}{26}    \PYG{l+m+mi}{3}                    \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{n}{HD}    \PYG{l+m+mf}{0.0003045500}  \PYG{l+m+mf}{27.021825}            \PYG{n}{C2H2}
   \PYG{l+m+mi}{78}     \PYG{p}{:}     \PYG{l+m+mi}{27}    \PYG{l+m+mi}{1}                    \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{n}{H6}    \PYG{l+m+mf}{0.9769900000}  \PYG{l+m+mf}{30.046950}            \PYG{n}{C2H6}
  \PYG{l+m+mi}{106}     \PYG{p}{:}     \PYG{l+m+mi}{27}    \PYG{l+m+mi}{2}              \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}    \PYG{l+m+mf}{0.0219526110}  \PYG{l+m+mf}{31.050305}            \PYG{n}{C2H6}
   \PYG{l+m+mi}{79}     \PYG{p}{:}     \PYG{l+m+mi}{28}    \PYG{l+m+mi}{1}                     \PYG{p}{(}\PYG{l+m+mi}{31}\PYG{n}{P}\PYG{p}{)}\PYG{n}{H3}    \PYG{l+m+mf}{0.9995328300}  \PYG{l+m+mf}{33.997238}             \PYG{n}{PH3}
   \PYG{l+m+mi}{80}     \PYG{p}{:}     \PYG{l+m+mi}{29}    \PYG{l+m+mi}{1}            \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{19}\PYG{n}{F}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9865400000}  \PYG{l+m+mf}{65.991722}            \PYG{n}{COF2}
  \PYG{l+m+mi}{119}     \PYG{p}{:}     \PYG{l+m+mi}{29}    \PYG{l+m+mi}{2}            \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{19}\PYG{n}{F}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.0110834000}  \PYG{l+m+mf}{66.995083}            \PYG{n}{COF2}
  \PYG{l+m+mi}{126}     \PYG{p}{:}     \PYG{l+m+mi}{30}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{19}\PYG{n}{F}\PYG{p}{)}\PYG{l+m+mi}{6}    \PYG{l+m+mf}{0.9501800000} \PYG{l+m+mf}{145.962492}             \PYG{n}{SF6}
   \PYG{l+m+mi}{81}     \PYG{p}{:}     \PYG{l+m+mi}{31}    \PYG{l+m+mi}{1}                     \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.9498800000}  \PYG{l+m+mf}{33.987721}             \PYG{n}{H2S}
   \PYG{l+m+mi}{82}     \PYG{p}{:}     \PYG{l+m+mi}{31}    \PYG{l+m+mi}{2}                     \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{34}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0421400000}  \PYG{l+m+mf}{35.983515}             \PYG{n}{H2S}
   \PYG{l+m+mi}{83}     \PYG{p}{:}     \PYG{l+m+mi}{31}    \PYG{l+m+mi}{3}                     \PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{33}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0074980000}  \PYG{l+m+mf}{34.987105}             \PYG{n}{H2S}
   \PYG{l+m+mi}{84}     \PYG{p}{:}     \PYG{l+m+mi}{32}    \PYG{l+m+mi}{1}           \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{n}{H}    \PYG{l+m+mf}{0.9838980000}  \PYG{l+m+mf}{46.005480}           \PYG{n}{HCOOH}
   \PYG{l+m+mi}{85}     \PYG{p}{:}     \PYG{l+m+mi}{33}    \PYG{l+m+mi}{1}                     \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9951070000}  \PYG{l+m+mf}{32.997655}             \PYG{n}{HO2}
   \PYG{l+m+mi}{86}     \PYG{p}{:}     \PYG{l+m+mi}{34}    \PYG{l+m+mi}{1}                       \PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}    \PYG{l+m+mf}{0.9976280000}  \PYG{l+m+mf}{15.994915}               \PYG{n}{O}
   \PYG{l+m+mi}{87}     \PYG{p}{:}     \PYG{l+m+mi}{36}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{o}{+}    \PYG{l+m+mf}{0.9939740000}  \PYG{l+m+mf}{29.997989}             \PYG{n}{NOp}
   \PYG{l+m+mi}{88}     \PYG{p}{:}     \PYG{l+m+mi}{37}    \PYG{l+m+mi}{1}                \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{79}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.5056000000}  \PYG{l+m+mf}{95.921076}            \PYG{n}{HOBr}
   \PYG{l+m+mi}{89}     \PYG{p}{:}     \PYG{l+m+mi}{37}    \PYG{l+m+mi}{2}                \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{81}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.4919000000}  \PYG{l+m+mf}{97.919027}            \PYG{n}{HOBr}
   \PYG{l+m+mi}{90}     \PYG{p}{:}     \PYG{l+m+mi}{38}    \PYG{l+m+mi}{1}                    \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{n}{H4}    \PYG{l+m+mf}{0.9773000000}  \PYG{l+m+mf}{28.031300}            \PYG{n}{C2H4}
   \PYG{l+m+mi}{91}     \PYG{p}{:}     \PYG{l+m+mi}{38}    \PYG{l+m+mi}{2}              \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H2}\PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H2}    \PYG{l+m+mf}{0.0219600000}  \PYG{l+m+mf}{29.034655}            \PYG{n}{C2H4}
   \PYG{l+m+mi}{92}     \PYG{p}{:}     \PYG{l+m+mi}{39}    \PYG{l+m+mi}{1}               \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{n}{H}    \PYG{l+m+mf}{0.9859300000}  \PYG{l+m+mf}{32.026215}           \PYG{n}{CH3OH}
   \PYG{l+m+mi}{93}     \PYG{p}{:}     \PYG{l+m+mi}{40}    \PYG{l+m+mi}{1}               \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}\PYG{p}{(}\PYG{l+m+mi}{79}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.5013000000}  \PYG{l+m+mf}{93.941811}           \PYG{n}{CH3Br}
   \PYG{l+m+mi}{94}     \PYG{p}{:}     \PYG{l+m+mi}{40}    \PYG{l+m+mi}{2}               \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}\PYG{p}{(}\PYG{l+m+mi}{81}\PYG{n}{Br}\PYG{p}{)}    \PYG{l+m+mf}{0.4876600000}  \PYG{l+m+mf}{95.939764}           \PYG{n}{CH3Br}
   \PYG{l+m+mi}{95}     \PYG{p}{:}     \PYG{l+m+mi}{41}    \PYG{l+m+mi}{1}           \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{n}{H3}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}    \PYG{l+m+mf}{0.9748200000}  \PYG{l+m+mf}{41.026549}           \PYG{n}{CH3CN}
   \PYG{l+m+mi}{96}     \PYG{p}{:}     \PYG{l+m+mi}{42}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{19}\PYG{n}{F}\PYG{p}{)}\PYG{l+m+mi}{4}    \PYG{l+m+mf}{0.9893000000}  \PYG{l+m+mf}{87.993616}             \PYG{n}{CF4}
  \PYG{l+m+mi}{116}     \PYG{p}{:}     \PYG{l+m+mi}{43}    \PYG{l+m+mi}{1}                    \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{l+m+mi}{4}\PYG{n}{H2}    \PYG{l+m+mf}{0.9559980000}  \PYG{l+m+mf}{50.015650}            \PYG{n}{C4H2}
  \PYG{l+m+mi}{109}     \PYG{p}{:}     \PYG{l+m+mi}{44}    \PYG{l+m+mi}{1}                \PYG{n}{H}\PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{l+m+mi}{3}\PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}    \PYG{l+m+mf}{0.9646069000}  \PYG{l+m+mf}{51.010899}            \PYG{n}{HC3N}
  \PYG{l+m+mi}{103}     \PYG{p}{:}     \PYG{l+m+mi}{45}    \PYG{l+m+mi}{1}                          \PYG{n}{H2}    \PYG{l+m+mf}{0.9996880000}   \PYG{l+m+mf}{2.015650}              \PYG{n}{H2}
  \PYG{l+m+mi}{115}     \PYG{p}{:}     \PYG{l+m+mi}{45}    \PYG{l+m+mi}{2}                          \PYG{n}{HD}    \PYG{l+m+mf}{0.0003114320}   \PYG{l+m+mf}{3.021825}              \PYG{n}{H2}
   \PYG{l+m+mi}{97}     \PYG{p}{:}     \PYG{l+m+mi}{46}    \PYG{l+m+mi}{1}                  \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.9396240000}  \PYG{l+m+mf}{43.971036}              \PYG{n}{CS}
   \PYG{l+m+mi}{98}     \PYG{p}{:}     \PYG{l+m+mi}{46}    \PYG{l+m+mi}{2}                  \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{34}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0416817000}  \PYG{l+m+mf}{45.966787}              \PYG{n}{CS}
   \PYG{l+m+mi}{99}     \PYG{p}{:}     \PYG{l+m+mi}{46}    \PYG{l+m+mi}{3}                  \PYG{p}{(}\PYG{l+m+mi}{13}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0105565000}  \PYG{l+m+mf}{44.974368}              \PYG{n}{CS}
  \PYG{l+m+mi}{100}     \PYG{p}{:}     \PYG{l+m+mi}{46}    \PYG{l+m+mi}{4}                  \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{33}\PYG{n}{S}\PYG{p}{)}    \PYG{l+m+mf}{0.0074166800}  \PYG{l+m+mf}{44.970399}              \PYG{n}{CS}
  \PYG{l+m+mi}{114}     \PYG{p}{:}     \PYG{l+m+mi}{47}    \PYG{l+m+mi}{1}                 \PYG{p}{(}\PYG{l+m+mi}{32}\PYG{n}{S}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{l+m+mi}{3}    \PYG{l+m+mf}{0.9423964000}  \PYG{l+m+mf}{79.956820}             \PYG{n}{SO3}
  \PYG{l+m+mi}{123}     \PYG{p}{:}     \PYG{l+m+mi}{48}    \PYG{l+m+mi}{1}                \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{l+m+mi}{2}\PYG{p}{(}\PYG{l+m+mi}{14}\PYG{n}{N}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.9707524330}  \PYG{l+m+mf}{52.006148}            \PYG{n}{C2N2}
  \PYG{l+m+mi}{124}     \PYG{p}{:}     \PYG{l+m+mi}{49}    \PYG{l+m+mi}{1}           \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{35}\PYG{n}{Cl}\PYG{p}{)}\PYG{l+m+mi}{2}    \PYG{l+m+mf}{0.5663917610}  \PYG{l+m+mf}{97.932620}           \PYG{n}{COCl2}
  \PYG{l+m+mi}{125}     \PYG{p}{:}     \PYG{l+m+mi}{49}    \PYG{l+m+mi}{2}      \PYG{p}{(}\PYG{l+m+mi}{12}\PYG{n}{C}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{16}\PYG{n}{O}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{35}\PYG{n}{Cl}\PYG{p}{)}\PYG{p}{(}\PYG{l+m+mi}{37}\PYG{n}{Cl}\PYG{p}{)}    \PYG{l+m+mf}{0.3622352780}  \PYG{l+m+mf}{99.929670}           \PYG{n}{COCl2}
\end{sphinxVerbatim}


\subsubsection{Generate HITRAN and Initial Guess Line lists from HAPI Call}
\label{\detokenize{Generating Parameter Line lists:generate-hitran-and-initial-guess-line-lists-from-hapi-call}}
\sphinxAtStartPar
The next section of the example contains a function and function call where the output is a MATS compatible line list.  The HITRANlinelist\_to\_csv takes a list of global isotope numbers and minimum and maximum wavenumbers as variables with a tablename, filename, temperature, and option to calculate the speed\sphinxhyphen{}dependent broadening as optional parameters.  The spectroscopic data for the isotopes in the global isotope list over the specified wavenumber range will be retrieved from HITRANOnline.  HITRAN breaks\sphinxhyphen{}up the HTP line parameters into temperature regimes, so the temperature specied selects for the most relevant parameters.  The tablename parameter sets the ame of the table generated by the HAPI call, where the filename parameter sets the base for the resulting .csv files.  The HITRANlinelist\_to\_csv function generates two outputs, the first is a HITRAN line list with all data available in HITRAN for the isotopes and spectral range (based on the parsed parameters in the HITRAN\_parameter\_list).  The second file generates the highest order line shape list using the HITRAN values and formats for MATS.  This line list also will fill in temperature dependences and missing broadener information and if calculate\_aw is True calculate the speed dependence based on theory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{def} \PYG{n+nf}{HITRANlinelist\PYGZus{}to\PYGZus{}csv}\PYG{p}{(}\PYG{n}{isotopes}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}wavenumber}\PYG{p}{,} \PYG{n}{maximum\PYGZus{}wavenumber}\PYG{p}{,} \PYG{n}{tablename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{tmp}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{calculate\PYGZus{}aw} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}\PYG{p}{:}

\PYG{+w}{                 }\PYG{l+s+sd}{\PYGZdq{}\PYGZdq{}\PYGZdq{}Generates two .csv files generated information available from HTIRAN.  The first line list matches the information available from HITRAN (\PYGZus{}HITRAN.csv) and the second supplements the HITRAN information with theoretical values and translates into MATS input format (\PYGZus{}initguess.csv)}

\PYG{l+s+sd}{ Outline}

\PYG{l+s+sd}{ 1. Gets a line list from HITRAN and saves all available parameters to filename\PYGZus{}HITRAN.csv}
\PYG{l+s+sd}{ 2. Goes through the data provided from HITRAN and collects the highest order line shape information.}
\PYG{l+s+sd}{ 3.  Where there is missing information for the complete HTP linelist set to 0 or make the following substitutions}
\PYG{l+s+sd}{     \PYGZhy{} for missing diluent information fill values with air}
\PYG{l+s+sd}{     \PYGZhy{} set missing shift temperature dependences equal to 0 (linear temperature dependence)}
\PYG{l+s+sd}{     \PYGZhy{} calculate the SD\PYGZus{}gamma based on theory (if calculate aw = True)}
\PYG{l+s+sd}{     \PYGZhy{} set the gamma\PYGZus{}2 temperature exponent equal to the gamma0 temperature exponent}
\PYG{l+s+sd}{     \PYGZhy{} set the delta\PYGZus{}2 temperature exponent equal to the delta0 temperature exponent}
\PYG{l+s+sd}{     \PYGZhy{} set the dicke narrowing temperature exponent to 1}
\PYG{l+s+sd}{ 4. Save the supplemented and MATS formatted HITRAN information as filename\PYGZus{}initguess.csv}


\PYG{l+s+sd}{ Parameters}
\PYG{l+s+sd}{ \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}}
\PYG{l+s+sd}{ isotopes : list}
\PYG{l+s+sd}{     list of the HITRAN global isotope numbers to include in the HAPI call}
\PYG{l+s+sd}{ minimum\PYGZus{}wavenumber : float}
\PYG{l+s+sd}{     minimum line center (cm\PYGZhy{}1) to include in the HAPI call.}
\PYG{l+s+sd}{ maximum\PYGZus{}wavenumber : float}
\PYG{l+s+sd}{     maximum line center (cm\PYGZhy{}1) to include in the HAPI call.}
\PYG{l+s+sd}{ tablename : str, optional}
\PYG{l+s+sd}{     desired name for table generated from HAPI call. The default is \PYGZsq{}tmp\PYGZsq{}.}
\PYG{l+s+sd}{ temperature : float, optional}
\PYG{l+s+sd}{     Nominal temperature of interest.  HITRAN breaks\PYGZhy{}up the HTP line parameters into temperature regimes.  This allows for selection of the most approriate parameter information. The default is 296.}
\PYG{l+s+sd}{ calculate\PYGZus{}aw : float, optional}
\PYG{l+s+sd}{     Boolean flag to present option to calculate speed\PYGZhy{}dependent shift based on theoretical approximation based on temperature exponent, mass of the absorber, and mass of the perturber}

\PYG{l+s+sd}{ Returns}
\PYG{l+s+sd}{ \PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}\PYGZhy{}}
\PYG{l+s+sd}{ linelist\PYGZus{}select : dataframe}
\PYG{l+s+sd}{     pandas dataframe corresponding to the HITRAN information supplemented by theoretical values/assumptions.}
\PYG{l+s+sd}{ tablename\PYGZus{}HITRAN.csv : .csv file}
\PYG{l+s+sd}{     file corresponding to available HITRAN information}
\PYG{l+s+sd}{ tablename\PYGZus{}initguess.csv : .csv file}
\PYG{l+s+sd}{     file corresponding to available HITRAN information supplemented by theory and assumptions in MATS format}

\PYG{l+s+sd}{ \PYGZdq{}\PYGZdq{}\PYGZdq{}}
\end{sphinxVerbatim}


\subsubsection{Use Example for HITRANlinelist\_to\_csv}
\label{\detokenize{Generating Parameter Line lists:use-example-for-hitranlinelist-to-csv}}
\sphinxAtStartPar
To select the relevant information from HITRAN you will need to provide:
\begin{itemize}
\item {} 
\sphinxAtStartPar
table name (str)

\item {} 
\sphinxAtStartPar
an array containing the global isotope numbers of the molecules/isotopes of interest

\item {} 
\sphinxAtStartPar
the minimum and maximum wavenumbers

\item {} 
\sphinxAtStartPar
the minimum line intensity of interest

\end{itemize}

\sphinxAtStartPar
The example below would generate a HITRAN table named ‘CO’ that contains all \(CO\) isotopes (global isotopes 26 \sphinxhyphen{} 31) and the most abundant \(CO_{2}\) isotope (global isotope 7) in the spectral region between 6200 and 6500 \(cm^{-1}\) that have line intensities greater than 1e\sphinxhyphen{}30 \(\frac{cm^{-1}}{molecule \cdot cm^{-2}}\).   Additionally, it calculates the theoretical aw value and assumes that measurements are at 296 (temperature = 296 is default).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{tablename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{global\PYGZus{}isotopes} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{26}\PYG{p}{,} \PYG{l+m+mi}{27}\PYG{p}{,} \PYG{l+m+mi}{28}\PYG{p}{,} \PYG{l+m+mi}{29}\PYG{p}{,} \PYG{l+m+mi}{30}\PYG{p}{,}\PYG{l+m+mi}{31}\PYG{p}{,}\PYG{l+m+mi}{7}\PYG{p}{]}
\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{6200}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mi}{6500}
\PYG{n}{intensity\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}

\PYG{n}{linelist\PYGZus{}select} \PYG{o}{=} \PYG{p}{(}\PYG{n}{HITRANlinelist\PYGZus{}to\PYGZus{}csv}\PYG{p}{(}\PYG{n}{global\PYGZus{}isotopes}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{tablename} \PYG{o}{=} \PYG{n}{tablename}\PYG{p}{,} \PYG{n}{calculate\PYGZus{}aw} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxstepscope


\section{Examples}
\label{\detokenize{Examples:examples}}\label{\detokenize{Examples::doc}}

\subsection{MATS Example Scripts}
\label{\detokenize{Examples:mats-example-scripts}}
\sphinxAtStartPar
The MATS package includes examples highlighting major features.

\sphinxstepscope


\subsubsection{Fitting Experimental Spectra}
\label{\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}}\label{\detokenize{Fitting Experimental Spectra::doc}}
\sphinxAtStartPar
Provided in the MATS v2 release are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
This example fits isolated Oxygen A\sphinxhyphen{}Band transitions in \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/Experimental\_Spectra}{experimental spectra}


\paragraph{Import Modules and Set\sphinxhyphen{}Up}
\label{\detokenize{Fitting Experimental Spectra:import-modules-and-set-up}}
\sphinxAtStartPar
This example starts with importing modules and setting up file locations

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{os}\PYG{o}{,} \PYG{n+nn}{sys}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{from} \PYG{n+nn}{matplotlib} \PYG{k+kn}{import} \PYG{n}{gridspec}
\PYG{o}{\PYGZpc{}}\PYG{k}{matplotlib} inline
\PYG{k+kn}{import} \PYG{n+nn}{MATS}
\end{sphinxVerbatim}

\sphinxAtStartPar
Optional import of seaborn package for figure generation

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{seaborn} \PYG{k}{as} \PYG{n+nn}{sns}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}style}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{whitegrid}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}style}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ticks}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}context}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{poster}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
If not working within the MATS file structure, change the path to the working directory that contains experimental spectra or to the folder that you want to work in.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{os}\PYG{o}{.}\PYG{n}{chdir}\PYG{p}{(}\PYG{n}{path}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Load Spectra from files}
\label{\detokenize{Fitting Experimental Spectra:load-spectra-from-files}}
\sphinxAtStartPar
There are two options for generating {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} objects.  The first is from a file by instantiating an instance of the class, which is the focus of this example.  The second option is by using the {\hyperref[\detokenize{MATS:MATS.spectrum.simulate_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{simulate\_spectrum()}}}}} function described in the \sphinxhref{https://github.com/usnistgov/MATS/tree/master/MATS/Examples/Synthetic\_Spectra}{fitting synthetic spectra example}.

\sphinxAtStartPar
Before generating {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} objects from your experimental data, it is helpful to set some variables for terms that will be used in all of the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} objects and/or throughout the fitting.  In this example the minimum intensity threshold for simulation (IntensityThreshold), the minimum line intensity of lines  fit in the analysis (Fit\_Intensity), the order of the polynomial used in the baseline fits, and the names of columns used for the absorption, frequency, pressure, and temperature data are defined at the top of the example.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}24} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Corrected Tau (us)}\PYG{l+s+s1}{\PYGZsq{}} \PYG{c+c1}{\PYGZsh{} Mean tau/us}
\PYG{n}{freq\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Total Frequency (Detuning)}\PYG{l+s+s1}{\PYGZsq{}} \PYG{c+c1}{\PYGZsh{} Total Frequency /MHz}
\PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Cavity Pressure /Torr}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Cavity Temperature Side 2 /C}\PYG{l+s+s1}{\PYGZsq{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
After that, 4 instances of the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} class are instantiated from 4 experimental spectra.  In the class instantiation, the mole fraction of the oxygen sample used is defined, the etalon amplitude and period are defined, the sample is confirmed to be at natural abundance, the diluent is set to air, and the columns defined for pressure, temperature, frequency, and absorbance data are set.  All of these spectra were collected at nominally 296K, so the nominal temperature is set to be 296.  This variable is only used for a flag as to whether to allow temperature exponents to be floatable parameters and for generation of line mixing parameters for each nominal temperature as the temperature dependence of line mixing is not currently implemented.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{190510\PYGZus{}2per\PYGZus{}43\PYGZus{}forfit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}} \PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.01949}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.001172}\PYG{p}{,} \PYG{l+m+mf}{1.19574}\PYG{p}{]}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}2} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{190510\PYGZus{}2per\PYGZus{}55\PYGZus{}forfit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}} \PYG{l+m+mi}{7} \PYG{p}{:} \PYG{l+m+mf}{0.01949}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.001172}\PYG{p}{,} \PYG{l+m+mf}{1.19574}\PYG{p}{]}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}3} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{190513\PYGZus{}2per\PYGZus{}82\PYGZus{}forfit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}} \PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.01949}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.001172}\PYG{p}{,} \PYG{l+m+mf}{1.19574}\PYG{p}{]}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}4} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{190514\PYGZus{}2per\PYGZus{}126\PYGZus{}forfit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}} \PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.01949}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.001172}\PYG{p}{,} \PYG{l+m+mf}{1.19574}\PYG{p}{]}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                     \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.plot_wave_alpha}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.plot\_wave\_alpha()}}}}} function can be called to plot any of the spectra.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{plot\PYGZus{}wave\PYGZus{}alpha}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=415\sphinxpxdimen,height=303\sphinxpxdimen]{{plot_wave_alpha}.png}


\paragraph{Generate a Dataset}
\label{\detokenize{Fitting Experimental Spectra:generate-a-dataset}}
\sphinxAtStartPar
If the parameter line list hasn’t been read in from a .csv file, then do that now making sure to switch to the appropriate directories as needed.  This file can be generated following the {\hyperref[\detokenize{Generating Parameter Line lists:generating-parameter-line-lists}]{\sphinxcrossref{\DUrole{std,std-ref}{Generating Parameter Line lists}}}}.  Alternatively, the code block below shows how to use the py:func:\sphinxtitleref{LoadLineListData} function to read in the line list provided for the Oxygen A\sphinxhyphen{}Band.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2\PYGZus{}ABand\PYGZus{}Drouin\PYGZus{}2017\PYGZus{}linelist}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\sphinxAtStartPar
The next step is to combine all desired \sphinxcode{\sphinxupquote{Spectrum}} objects  into a {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} object, where we give the dataset a name and specify the initial parameter line list to use for the fits.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Line Intensity}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} class contains a function to generate a baseline line list, analogous to the one for the parameter line list (done outside of this example), based on the order of the baselines, etalons, molecules, x\sphinxhyphen{}shift parameters, and segments as defined by both the spectrum objects.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Generate Fit Parameter Files}
\label{\detokenize{Fitting Experimental Spectra:generate-fit-parameter-files}}
\sphinxAtStartPar
The next section of code uses the {\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File}}}}} class to define what line shape to use for the initial fits, whether to use line mixing, the minimum line intensity to fit a line, mimimum intensity to included in the simulation, and for each line parameter whether that parameter is going to be constrained across all spectra or whether there will be a parameter for each spectrum (multi\sphinxhyphen{}spectrum vs single\sphinxhyphen{}spectrum fits) on a parameter by parameter basis. In the example below, the SDVP line profile without line mixing will be used to fit lines with line intensities greater than 1e\sphinxhyphen{}24 and the line centers and line intensities will be allowed to float for each line, while all other lines are constrained across all spectra in the dataset. The additional\_columns parameter allows for inclusion of additional columns in the line shape parameter line list to be included in the output file.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDVP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                               \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                               \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                               \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                               \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                               \PYG{n}{additional\PYGZus{}columns} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{trans\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}lower\PYGZus{}quanta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The next step is to generate fit parameter and baseline line lists that include columns that specify whether that parameter should be varied during fitting, in addition to adding uncertainty columns for the fit error for each parameter. For the following example the line centers, line intensities, collisional half\sphinxhyphen{}widths, and speed\sphinxhyphen{}dependent broadening terms will be floated for all main oxygen isotopes for lines where the line intensity is greater than 1e\sphinxhyphen{}24. Additionally, the baseline terms will float, as will the etalon amplitude and phase.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                   \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
These functions will generate .csv files corresponding to these selections, which are read in by the \sphinxcode{\sphinxupquote{Fit\_DataSet}} class instantiation.  This means that edits can be made manually to the .csv files or reading in the .csv, editing, and resaving before the next code segment is run.


\paragraph{Fit Dataset}
\label{\detokenize{Fitting Experimental Spectra:fit-dataset}}
\sphinxAtStartPar
Instantiating the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}} class reads in the information from the baseline and parameter linelists generated in the previous step. It also allows for limits to be placed on the parameters, so that they don’t result in divergent solutions. The example below includes several limits including limiting the line center to be within 0.1 cm\sphinxhyphen{}1 of the initial guess and the Line intensity to be within a factor of 2 of the intial guess. Placing limits on the parameters can be restrictive on the solution and cause the fit to not converge or return NaN for the standard error if it doesn’t allow for a local minima to be found.  The {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}} class also allows for the option to weight\_spectra, currently this is set to False. Later in this example we will explore using weights in fitting.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{weight\PYGZus{}spectra} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
             \PYG{n}{baseline\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,}
             \PYG{n}{molefraction\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mf}{1.1}\PYG{p}{,}
             \PYG{n}{etalon\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{etalon\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{c+c1}{\PYGZsh{}phase is constrained to +/\PYGZhy{} 2pi,}
             \PYG{n}{x\PYGZus{}shift\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift\PYGZus{}limit\PYGZus{}magnitude} \PYG{o}{=} \PYG{l+m+mf}{0.5}\PYG{p}{,}
             \PYG{n}{nu\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nu\PYGZus{}limit\PYGZus{}magnitude} \PYG{o}{=} \PYG{l+m+mf}{0.1}\PYG{p}{,}
             \PYG{n}{sw\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,}
             \PYG{n}{gamma0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{3}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma0\PYGZus{}limit}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{delta0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{SD\PYGZus{}gamma\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{SD\PYGZus{}gamma\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma2\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma2\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{SD\PYGZus{}delta\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{SD\PYGZus{}delta\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta2\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta2\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{nuVC\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{nuVC\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}nuVC\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}nuVC\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{eta\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{eta\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The next step is to generate the \sphinxhref{https://lmfit.github.io/lmfit-py/parameters.html}{lmfit} params dictionary object through the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.generate_params}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.generate\_params()}}}}} function.  This translates baseline and parameter line list .csv files into a lmfit parameter dictionary that is used in the fits.  After the parameters object is generated you can use the keys to set values and impose constraints on individual parameters, if desired.  While this is not coded in the MATS toolkit, it is incredibly powerful as it lets you define min, max, vary, and expression values for any parameter.  In the example below, two additional constraints are imposed on specific fit parameters.  The first constrains all speed\sphinxhyphen{}dependent width parameters to be between the values of 0.01 and 0.25 and the second forces the amplitude of the etalon to be constant across all spectra.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}



\PYG{k}{for} \PYG{n}{param} \PYG{o+ow}{in} \PYG{n}{params}\PYG{p}{:}
             \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                     \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n+nb}{min} \PYG{o}{=} \PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{n+nb}{max} \PYG{o}{=} \PYG{l+m+mf}{0.25}\PYG{p}{)}
             \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                     \PYG{k}{if} \PYG{n}{param} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}
                             \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n}{expr}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The params file is then used to fit the spectra in the dataset using the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.fit_data}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.fit\_data()}}}}} function, where the result is a lmfit result object. The lmft prettyprint function prints the parameter fit results.  Included below is an abbreviated prettyprint output that not only shows the fit result values and standard errors, but also highlights that constraints were imposed on the SD\_gamma (speed dependent broadening) parameters and an expression was imposed on the etalon\_amplitudes.  It also shows that the there is a line intensity reported for every line and spectrum (sw\_spectrum {\color{red}\bfseries{}number\_} line) as the line intensities were not constrained to global fits.  The reported sw shows that the fitted line intensity value is scaled by the minimum fit value.  This scalar term is saved in a column called sw\_scale\_factor for reference.  Scaling the line intensity aids in the fitting as line intensities are so much smaller than other fitted parameters.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{)}
\PYG{n+nb}{print} \PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{o}{.}\PYG{n}{pretty\PYGZus{}print}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Name}                     \PYG{n}{Value}      \PYG{n}{Min}      \PYG{n}{Max}   \PYG{n}{Stderr}     \PYG{n}{Vary}     \PYG{n}{Expr} \PYG{n}{Brute\PYGZus{}Step}
\PYG{n}{Pressure\PYGZus{}1\PYGZus{}1}           \PYG{l+m+mf}{0.07911}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{Pressure\PYGZus{}2\PYGZus{}1}           \PYG{l+m+mf}{0.06556}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{Pressure\PYGZus{}3\PYGZus{}1}           \PYG{l+m+mf}{0.04602}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{Pressure\PYGZus{}4\PYGZus{}1}           \PYG{l+m+mf}{0.02488}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}1}          \PYG{l+m+mi}{0}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}10}         \PYG{l+m+mi}{0}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}13}         \PYG{l+m+mi}{0}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}25}         \PYG{l+m+mi}{0}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}26}         \PYG{l+m+mi}{0}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}1}        \PYG{l+m+mf}{0.1}     \PYG{l+m+mf}{0.01}     \PYG{l+m+mf}{0.25}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}10}    \PYG{l+m+mf}{0.1137}     \PYG{l+m+mf}{0.01}     \PYG{l+m+mf}{0.25} \PYG{l+m+mf}{0.0008273}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}13}    \PYG{l+m+mf}{0.1313}     \PYG{l+m+mf}{0.01}     \PYG{l+m+mf}{0.25} \PYG{l+m+mf}{0.001115}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}25}      \PYG{l+m+mf}{0.09}     \PYG{l+m+mf}{0.01}     \PYG{l+m+mf}{0.25}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}26}       \PYG{l+m+mf}{0.1}     \PYG{l+m+mf}{0.01}     \PYG{l+m+mf}{0.25}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{o}{.} \PYG{o}{.} \PYG{o}{.}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}      \PYG{l+m+mf}{0.001762}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf} \PYG{l+m+mf}{4.007e\PYGZhy{}05}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}2\PYGZus{}1}      \PYG{l+m+mf}{0.001762}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf} \PYG{l+m+mf}{4.007e\PYGZhy{}05}    \PYG{k+kc}{False} \PYG{n}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}3\PYGZus{}1}      \PYG{l+m+mf}{0.001762}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf} \PYG{l+m+mf}{4.007e\PYGZhy{}05}    \PYG{k+kc}{False} \PYG{n}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}4\PYGZus{}1}      \PYG{l+m+mf}{0.001762}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf} \PYG{l+m+mf}{4.007e\PYGZhy{}05}    \PYG{k+kc}{False} \PYG{n}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}freq\PYGZus{}1\PYGZus{}1}        \PYG{l+m+mf}{1.196}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}freq\PYGZus{}2\PYGZus{}1}        \PYG{l+m+mf}{1.196}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}freq\PYGZus{}3\PYGZus{}1}        \PYG{l+m+mf}{1.196}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}freq\PYGZus{}4\PYGZus{}1}        \PYG{l+m+mf}{1.196}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}phase\PYGZus{}1\PYGZus{}1}     \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.3479}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}  \PYG{l+m+mf}{0.04585}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}phase\PYGZus{}2\PYGZus{}1}    \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.09384}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}  \PYG{l+m+mf}{0.04288}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}phase\PYGZus{}3\PYGZus{}1}       \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.04}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}  \PYG{l+m+mf}{0.04446}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{etalon\PYGZus{}1\PYGZus{}phase\PYGZus{}4\PYGZus{}1}      \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.266}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}  \PYG{l+m+mf}{0.04394}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}1}         \PYG{l+m+mf}{0.04}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}10}     \PYG{l+m+mf}{0.04501}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf} \PYG{l+m+mf}{4.919e\PYGZhy{}05}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}13}     \PYG{l+m+mf}{0.04339}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf} \PYG{l+m+mf}{7.531e\PYGZhy{}05}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}25}        \PYG{l+m+mf}{0.04}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}26}        \PYG{l+m+mf}{0.04}     \PYG{o}{\PYGZhy{}}\PYG{n}{inf}      \PYG{n}{inf}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{o}{.} \PYG{o}{.} \PYG{o}{.}
\PYG{n}{sw\PYGZus{}1\PYGZus{}line\PYGZus{}1}              \PYG{l+m+mf}{4.369}    \PYG{l+m+mf}{2.184}    \PYG{l+m+mf}{8.738}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}1\PYGZus{}line\PYGZus{}10}             \PYG{l+m+mf}{4.735}      \PYG{l+m+mf}{2.4}    \PYG{l+m+mf}{9.598} \PYG{l+m+mf}{0.0008558}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}1\PYGZus{}line\PYGZus{}13}             \PYG{l+m+mf}{3.087}    \PYG{l+m+mf}{1.562}    \PYG{l+m+mf}{6.246} \PYG{l+m+mf}{0.0007302}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}1\PYGZus{}line\PYGZus{}25}             \PYG{l+m+mf}{2.083}    \PYG{l+m+mf}{1.042}    \PYG{l+m+mf}{4.166}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}1\PYGZus{}line\PYGZus{}26}             \PYG{l+m+mf}{3.399}    \PYG{l+m+mf}{1.699}    \PYG{l+m+mf}{6.798}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}2\PYGZus{}line\PYGZus{}1}              \PYG{l+m+mf}{4.369}    \PYG{l+m+mf}{2.184}    \PYG{l+m+mf}{8.738}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}2\PYGZus{}line\PYGZus{}10}             \PYG{l+m+mf}{4.752}      \PYG{l+m+mf}{2.4}    \PYG{l+m+mf}{9.598} \PYG{l+m+mf}{0.0006929}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}2\PYGZus{}line\PYGZus{}13}             \PYG{l+m+mf}{3.091}    \PYG{l+m+mf}{1.562}    \PYG{l+m+mf}{6.246} \PYG{l+m+mf}{0.0006913}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}2\PYGZus{}line\PYGZus{}25}             \PYG{l+m+mf}{2.083}    \PYG{l+m+mf}{1.042}    \PYG{l+m+mf}{4.166}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}2\PYGZus{}line\PYGZus{}26}             \PYG{l+m+mf}{3.399}    \PYG{l+m+mf}{1.699}    \PYG{l+m+mf}{6.798}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}3\PYGZus{}line\PYGZus{}1}              \PYG{l+m+mf}{4.369}    \PYG{l+m+mf}{2.184}    \PYG{l+m+mf}{8.738}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}3\PYGZus{}line\PYGZus{}10}             \PYG{l+m+mf}{4.744}      \PYG{l+m+mf}{2.4}    \PYG{l+m+mf}{9.598} \PYG{l+m+mf}{0.0007446}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}3\PYGZus{}line\PYGZus{}13}             \PYG{l+m+mf}{3.095}    \PYG{l+m+mf}{1.562}    \PYG{l+m+mf}{6.246} \PYG{l+m+mf}{0.0007499}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}3\PYGZus{}line\PYGZus{}25}             \PYG{l+m+mf}{2.083}    \PYG{l+m+mf}{1.042}    \PYG{l+m+mf}{4.166}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}3\PYGZus{}line\PYGZus{}26}             \PYG{l+m+mf}{3.399}    \PYG{l+m+mf}{1.699}    \PYG{l+m+mf}{6.798}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}4\PYGZus{}line\PYGZus{}1}              \PYG{l+m+mf}{4.369}    \PYG{l+m+mf}{2.184}    \PYG{l+m+mf}{8.738}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}4\PYGZus{}line\PYGZus{}10}               \PYG{l+m+mf}{4.8}      \PYG{l+m+mf}{2.4}    \PYG{l+m+mf}{9.598} \PYG{l+m+mf}{0.001158}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}4\PYGZus{}line\PYGZus{}13}             \PYG{l+m+mf}{3.118}    \PYG{l+m+mf}{1.562}    \PYG{l+m+mf}{6.246}  \PYG{l+m+mf}{0.00117}     \PYG{k+kc}{True}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}4\PYGZus{}line\PYGZus{}25}             \PYG{l+m+mf}{2.083}    \PYG{l+m+mf}{1.042}    \PYG{l+m+mf}{4.166}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\PYG{n}{sw\PYGZus{}4\PYGZus{}line\PYGZus{}26}             \PYG{l+m+mf}{3.399}    \PYG{l+m+mf}{1.699}    \PYG{l+m+mf}{6.798}        \PYG{l+m+mi}{0}    \PYG{k+kc}{False}     \PYG{k+kc}{None}     \PYG{k+kc}{None}
\end{sphinxVerbatim}

\sphinxAtStartPar
The last segment of code provides residual plots and updates residuals through the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.residual_analysis}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.residual\_analysis()}}}}} and

\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.plot_model_residuals}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.plot\_model\_residuals()}}}}} functions, updates the parameter and baseline line lists through

\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.update_params}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.update\_params()}}}}}, and generates a summary file with the fit results using

\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.dataset.Dataset.generate_summary_file}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset.generate\_summary\_file()}}}}}.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=1029\sphinxpxdimen,height=629\sphinxpxdimen]{{spectra_residuals}.png}

\sphinxAtStartPar
Call to the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.fft_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.fft\_spectrum()}}}}} function takes an FFT of the residuals. If we hadn’t included the etalon, the result of the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.fft_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.fft\_spectrum()}}}}} function would show a peak with the most abundant period being 1.271443 cm\sphinxhyphen{}1 and an amplitude of 0.001364, which were used as the initial guess for the etalon in the spectrum class definitions. The more etalon periods present in the spectral region being fit the more precise the etalon amplitude and frequency determined by the FFT will be.

\sphinxAtStartPar
If we hadn’t included the etalon, the fit residuals and FFT would like the plots below:

\noindent\sphinxincludegraphics[width=1029\sphinxpxdimen,height=629\sphinxpxdimen]{{spectra_residuals_noetalon}.png}

\noindent\sphinxincludegraphics[width=479\sphinxpxdimen,height=303\sphinxpxdimen]{{Noetalon_FFT}.png}

\sphinxAtStartPar
Using these values as the etalon period and amplitude give the fit residuals shown in the use example above and when incorporated the FFT no longer shows a substantial peak at 1.271443 cm\sphinxhyphen{}1.

\noindent\sphinxincludegraphics[width=479\sphinxpxdimen,height=303\sphinxpxdimen]{{etalon_FFT}.png}


\paragraph{Explore the Ability to Weight Spectra}
\label{\detokenize{Fitting Experimental Spectra:explore-the-ability-to-weight-spectra}}
\sphinxAtStartPar
MATS v2 introduced the ability to weight spectra in two different ways, using the stats column defined for each spectrum or weighting the entire spectrum.  In the first example below, we use use the same code that was introduced previously, but set the weight\_spectra variable in the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}} definitions to True.

\sphinxAtStartPar
This weights the contribution to the solution at each point in each spectrum by 1/tau\_stats\_column.  If the tau\_stats\_column is not defined in the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} then this will default to equal weights for all data points.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{weight\PYGZus{}spectra} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
             \PYG{n}{baseline\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,}
             \PYG{n}{molefraction\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mf}{1.1}\PYG{p}{,}
             \PYG{n}{etalon\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{etalon\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{c+c1}{\PYGZsh{}phase is constrained to +/\PYGZhy{} 2pi,}
             \PYG{n}{x\PYGZus{}shift\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift\PYGZus{}limit\PYGZus{}magnitude} \PYG{o}{=} \PYG{l+m+mf}{0.5}\PYG{p}{,}
             \PYG{n}{nu\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nu\PYGZus{}limit\PYGZus{}magnitude} \PYG{o}{=} \PYG{l+m+mf}{0.1}\PYG{p}{,}
             \PYG{n}{sw\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,}
             \PYG{n}{gamma0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{3}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma0\PYGZus{}limit}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{delta0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{SD\PYGZus{}gamma\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{SD\PYGZus{}gamma\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma2\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma2\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{SD\PYGZus{}delta\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{SD\PYGZus{}delta\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta2\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta2\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{nuVC\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{nuVC\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}nuVC\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}nuVC\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{eta\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{eta\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}



\PYG{k}{for} \PYG{n}{param} \PYG{o+ow}{in} \PYG{n}{params}\PYG{p}{:}
 \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
     \PYG{k}{if} \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{vary} \PYG{o}{==} \PYG{k+kc}{True}\PYG{p}{:}
         \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n+nb}{min} \PYG{o}{=} \PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{n+nb}{max} \PYG{o}{=} \PYG{l+m+mf}{0.25}\PYG{p}{)}
 \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
     \PYG{k}{if} \PYG{n}{param} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}
         \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n}{expr}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}




\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{)}
\PYG{n+nb}{print} \PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{o}{.}\PYG{n}{pretty\PYGZus{}print}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}

\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
For this example the statistics are relatively consistent across all points in the dataset, as shown in the plot below, so the fit residuals are very similar with some slight differences in the quality of fits and the parameter fit results.

\noindent\sphinxincludegraphics[width=1029\sphinxpxdimen,height=629\sphinxpxdimen]{{spectra_residuals_ptbypt_weight}.png}

\sphinxAtStartPar
The other option for weighting, that can be used with or without the point by point weighting, is to weight each spectrum by setting the spectrum weight value.  The default for each spectrum is 1.  This value can be set through the weight parameter in the spectrum definition or by using the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.set_weight}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.set\_weight()}}}}} function, which is shown below.

\sphinxAtStartPar
For all weighting options, the residuals are adjusted by the weight factors.  For this example, where the spec\_1 is given a weight of 0 this effectively removed spec\_1 from the solution consideration depicted by the flat residuals for spectrum 190510\_2per\_43\_forfit.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{set\PYGZus{}weight}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDVP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                               \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                               \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
                               \PYG{c+c1}{\PYGZsh{}additional\PYGZus{}columns = [\PYGZsq{}trans\PYGZus{}id\PYGZsq{}, \PYGZsq{}local\PYGZus{}lower\PYGZus{}quanta\PYGZsq{}, \PYGZsq{}m\PYGZsq{}])}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                   \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}



\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{weight\PYGZus{}spectra} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
             \PYG{n}{baseline\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,}
             \PYG{n}{molefraction\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mf}{1.1}\PYG{p}{,}
             \PYG{n}{etalon\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{etalon\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{c+c1}{\PYGZsh{}phase is constrained to +/\PYGZhy{} 2pi,}
             \PYG{n}{x\PYGZus{}shift\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift\PYGZus{}limit\PYGZus{}magnitude} \PYG{o}{=} \PYG{l+m+mf}{0.5}\PYG{p}{,}
             \PYG{n}{nu\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nu\PYGZus{}limit\PYGZus{}magnitude} \PYG{o}{=} \PYG{l+m+mf}{0.1}\PYG{p}{,}
             \PYG{n}{sw\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,}
             \PYG{n}{gamma0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{3}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma0\PYGZus{}limit}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{delta0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta0\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta0\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{SD\PYGZus{}gamma\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{SD\PYGZus{}gamma\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma2\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}gamma2\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{SD\PYGZus{}delta\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{SD\PYGZus{}delta\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta2\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}delta2\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{nuVC\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{nuVC\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{n\PYGZus{}nuVC\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{n\PYGZus{}nuVC\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
             \PYG{n}{eta\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{eta\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}limit} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}limit\PYGZus{}factor} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}



\PYG{k}{for} \PYG{n}{param} \PYG{o+ow}{in} \PYG{n}{params}\PYG{p}{:}
 \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
     \PYG{k}{if} \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{vary} \PYG{o}{==} \PYG{k+kc}{True}\PYG{p}{:}
         \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n+nb}{min} \PYG{o}{=} \PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{n+nb}{max} \PYG{o}{=} \PYG{l+m+mf}{0.25}\PYG{p}{)}
 \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
     \PYG{k}{if} \PYG{n}{param} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}
         \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n}{expr}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}amp\PYGZus{}1\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}




\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{)}
\PYG{n+nb}{print} \PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{o}{.}\PYG{n}{pretty\PYGZus{}print}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}

\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=1016\sphinxpxdimen,height=629\sphinxpxdimen]{{spectra_residuals_ptbypt_spectrum_weight}.png}

\sphinxstepscope


\subsubsection{Fitting Synthetic Spectra}
\label{\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}}\label{\detokenize{Fitting Synthetic Spectra::doc}}
\sphinxAtStartPar
Provided in the MATS v2 release are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
This example \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/Synthetic\_Spectra}{simulates and fits Oxygen A\sphinxhyphen{}Band spectra}


\paragraph{Import Modules and Set\sphinxhyphen{}Up}
\label{\detokenize{Fitting Synthetic Spectra:import-modules-and-set-up}}
\sphinxAtStartPar
This example starts with importing modules and setting up file locations

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{numpy} \PYG{k}{as} \PYG{n+nn}{np}
\PYG{k+kn}{import} \PYG{n+nn}{pandas} \PYG{k}{as} \PYG{n+nn}{pd}
\PYG{k+kn}{import} \PYG{n+nn}{os}
\PYG{k+kn}{import} \PYG{n+nn}{matplotlib}\PYG{n+nn}{.}\PYG{n+nn}{pyplot} \PYG{k}{as} \PYG{n+nn}{plt}
\PYG{k+kn}{from} \PYG{n+nn}{matplotlib} \PYG{k+kn}{import} \PYG{n}{gridspec}
\PYG{k+kn}{import} \PYG{n+nn}{MATS}
\end{sphinxVerbatim}

\sphinxAtStartPar
Optional import of seaborn package for figure generation

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{seaborn} \PYG{k}{as} \PYG{n+nn}{sns}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}style}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{whitegrid}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}style}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ticks}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{sns}\PYG{o}{.}\PYG{n}{set\PYGZus{}context}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{poster}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
If you aren’t using the MATS file strucutre, change the path to the working directory that contains experimental spectra or to the folder that you want to work in.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{os}\PYG{o}{.}\PYG{n}{chdir}\PYG{p}{(}\PYG{n}{path}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Simulate Spectra}
\label{\detokenize{Fitting Synthetic Spectra:simulate-spectra}}
\sphinxAtStartPar
If you are simulating spectra, opposed to reading them in from a file as discussed above, then you can use the {\hyperref[\detokenize{MATS:MATS.spectrum.simulate_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{simulate\_spectrum()}}}}} function.

\sphinxAtStartPar
When simulating spectra, the first step is to read in the reference line list.  This file can be generated following the {\hyperref[\detokenize{Generating Parameter Line lists:generating-parameter-line-lists}]{\sphinxcrossref{\DUrole{std,std-ref}{Generating Parameter Line lists}}}} and read in using pd.read\_csv function. The following code reads in the reference line list using the py:func:\sphinxtitleref{LoadLineListData} function to read in the line list provided for the Oxygen A\sphinxhyphen{}Band.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2\PYGZus{}ABand\PYGZus{}Drouin\PYGZus{}2017\PYGZus{}linelist}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\sphinxAtStartPar
Just as you would do if reading in the experimental spectrum, this example defines some common simulation and fit variables.  In addition to variables that would be used in fitting experimental spectra, the minimum and maximum wavenumbers for the simulation and the simulation wavenumber spacing are defined.  Alternatively, a wavenumbers term can be used and an array can be used as the input defining the x\sphinxhyphen{}axis.  This feature allows the use of a non\sphinxhyphen{}uniform x\sphinxhyphen{}axis and will take precident of the use of wave\_min, wave\_max, and wave\_step parameterization.

\sphinxAtStartPar
The baseline is defined by a polynomial where the array index is the parameter coefficient order, such that the {[}1, 0{]} would correspond to a linear baseline with a slope of 0 and an offset of 1.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}24} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{sample\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.002022}\PYG{p}{\PYGZcb{}}
\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{13155} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mf}{13157.5} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{l+m+mf}{0.005} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wavenumbers} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{l+m+mi}{13155}\PYG{p}{,} \PYG{l+m+mi}{13158}\PYG{p}{,} \PYG{l+m+mf}{0.02}\PYG{p}{)}
\PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{c+c1}{\PYGZsh{}polynomial baseline coefficients where the index is equal to the coefficient order}
\end{sphinxVerbatim}

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.spectrum.simulate_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{simulate\_spectrum()}}}}} function also allows for error to be added in the following ways:
\begin{itemize}
\item {} 
\sphinxAtStartPar
to the absorption axis through signal\sphinxhyphen{}to\sphinxhyphen{}noise ratio (SNR).  The SNR is implemented by adding  gaussian noise to the spectra such that the (maximum alpha \sphinxhyphen{} minimum alpha) / noise is equal to the set SNR.

\item {} 
\sphinxAtStartPar
to the wavenumber axis through the wave\_err parameter.  The wavenumber error is implemented by adding a gaussian noise error of the specified magnitude to the wavenumber axis.

\item {} 
\sphinxAtStartPar
to the mole fraction through the molefraction\_err parameter.  The molefraction error is implemented as a percent error bias on each (could enter a negative percent error to get negative offset).  This mimics the maximum impact that a constant error in sample mole fraction would have.

\item {} 
\sphinxAtStartPar
to the temperature/pressure through the temperature\_err and pressure\_err dictionaries.  In experiments there are generally two type of errors with pressure and temperature measurements.  The first is a constant bias in the reading.  The second type of error is an actual change in the pressure/temperature during the collection of the spectrum.  To account for both error types the pressure\_err and temperature\_err are dictionaries, where the keys correspond to ‘bias/per\_bias’ (bias for temperature and per\_bias for pressure), function (allows ‘linear’ or ‘sine’), and params.  If the function is ‘linear’ then the param keys are ‘m’ and ‘b’ corresponding to the slope and interecept. If the function is ‘sine’ then the param keys are ‘amp’, ‘freq’, and ‘phase’ corresponding to the amplitude, period, and phase of the sine function.  For both temperature and pressure, the pressure/temperature recorded in the simulated spectra output include the average pressure or temperature over the segment (analogous to the frequency of the pressure/temperature measurement in an experiment) and does not include the bias in pressure/temperature as this is would be an unknown in an experiment.

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SNR} \PYG{o}{=} \PYG{l+m+mi}{4000}
\PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}4}
\PYG{n}{temperature\PYGZus{}err} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{bias}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{function}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{k+kc}{None}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{params}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}
\PYG{n}{pressure\PYGZus{}err} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{per\PYGZus{}bias}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+m+mf}{0.01}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{function}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{k+kc}{None}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{params}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}
\PYG{n}{molefraction\PYGZus{}err} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
These parameters and the additional settings for filenames and number of segments can be used to call the {\hyperref[\detokenize{MATS:MATS.spectrum.simulate_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{simulate\_spectrum()}}}}} function setting the output equal to a variable as would be done for generating an instance of the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} class from a .csv file.  This makes it simple to transition code from analysis of experimental spectra to error analysis through simulations.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}1} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wavenumbers} \PYG{o}{=} \PYG{n}{wavenumbers}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}err} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}err}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}err} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{25\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}err} \PYG{o}{=} \PYG{n}{molefraction\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}2} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}err} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}err}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}err} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{50\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}err} \PYG{o}{=} \PYG{n}{molefraction\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}3} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}err} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}err}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}err} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{100\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}err} \PYG{o}{=} \PYG{n}{molefraction\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}4} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}err} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}err}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{200}\PYG{p}{,}
                     \PYG{n}{pressure\PYGZus{}err} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{200\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}err} \PYG{o}{=} \PYG{n}{molefraction\PYGZus{}err}\PYG{p}{,}
                     \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Generate a Dataset}
\label{\detokenize{Fitting Synthetic Spectra:generate-a-dataset}}
\sphinxAtStartPar
The procedure for analysis for both simulating and loading spectrum are the same as described in {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}}. The next step is to combine all desired {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} objects

\sphinxAtStartPar
into a {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} object, where we give the dataset a name and specify the parameter line list to use for analysis/

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Line Intensity}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} class contains a function to generate a baseline line list, analogous to the one for the parameter line list done outside of this example, based on the order of the baseline fit, etalons, molecules, x\sphinxhyphen{}shift parameters, and segments as defined by both the spectrum and dataset objects.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Generate Fit Parameter Files}
\label{\detokenize{Fitting Synthetic Spectra:generate-fit-parameter-files}}
\sphinxAtStartPar
The next section of code uses the {\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File}}}}} class to define what line shape to use for the initial fits, whether to use line mixing, the minimum line intensity to fit a line, mimimum intensity to included in the simulation, and for each line parameter whether that parameter is going to be constrained across all spectra or whether there will be a parameter for each spectrum (multi\sphinxhyphen{}spectrum vs single\sphinxhyphen{}spectrum fits) on a parameter by parameter basis.  In the example below, the SDVP line profile without line mixing will be used to fit lines with line intensities greater than 1e\sphinxhyphen{}24 with no parameters floated.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDVP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                               \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                               \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The next step is to generate fit parameter and baseline line lists that include columns that specify whether that parameter should be varied during fitting, in addition to adding error columns for the fit error for each parameter.  For the following example the line centers, line intensities, collisional half\sphinxhyphen{}widths, and speed\sphinxhyphen{}dependent broadening terms will be floated for all main oxygen isotopes for lines where the line intensity is greater than 1e\sphinxhyphen{}24. None of the baseline parameters are floated

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                   \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                      \PYG{n}{vary\PYGZus{}pressure} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}temperature} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
These functions will generate .csv files corresponding to these selections, which are read in by the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}} class instantiation.  When simulating spectra, if you don’t adjust the baseline and parameter line lists, then your simulated variables are equal to your initial guesses.  This can lead to a local minimum where the fit will not move from the initial guesses.  The last segment of the code shows an example where random uncertainties from the simulated values are applied and then the Parameter line list file is resaved.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Parameter\PYGZus{}LineList} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{index\PYGZus{}col} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)}


\PYG{k}{for} \PYG{n}{index} \PYG{o+ow}{in} \PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZlt{}}\PYG{n}{wave\PYGZus{}max}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{n}{wave\PYGZus{}min}\PYG{p}{)}\PYG{p}{]}\PYG{o}{.}\PYG{n}{index}\PYG{o}{.}\PYG{n}{unique}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
 \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mf}{0.005}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{}adjust by random number scale 0.005 cm\PYGZhy{}1}
 \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mf}{0.01}\PYG{p}{)}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} adjust by random amount at 1\PYGZpc{} scale}
 \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mf}{0.01}\PYG{p}{)}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} adjust by random amount at 1\PYGZpc{} scale}
 \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mf}{0.02}\PYG{p}{)}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} adjust by random amount at 2\PYGZpc{} scale}
 \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{n}{index}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mf}{0.1}\PYG{p}{)}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} adjust by random amount at 10\PYGZpc{} scale}

\PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{to\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Fit Dataset}
\label{\detokenize{Fitting Synthetic Spectra:fit-dataset}}
\sphinxAtStartPar
Instantiating the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}} class reads in the information from the baseline and parameter linelists generated in the previous step.  The \DUrole{xref,std,std-ref}{Fitting Experimental Spectra documentation} gives a more basic example of how to perform a fit.  The example below iterates on a fit, where in the first iteration no parameters are floated, in the second iteration the line centers and line intensities are floated, and in the final iteration the collisional broadening, pressure shift, and speed\sphinxhyphen{}dependent broadening are floated.  Sometimes it is not possible to float all parameters at the same time, specifically with poor initial guesses for line center.  In that case an iterative approach like this is advantageous.

\sphinxAtStartPar
The iteration could also be applied to the lmfit parameter object by setting the the vary term, opposed to through the Parameter\_LineList.  In that approach the final Parameter\_LineList would have the standard uncertainities and fit values saved, but wouldn’t necessarily have the parameter\_vary column appropriately set.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{iteration} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{k}{while} \PYG{n}{iteration} \PYG{o}{\PYGZlt{}}\PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{:}
 \PYG{n+nb}{print} \PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ITERATION }\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n}{iteration}\PYG{p}{)}\PYG{p}{)}
 \PYG{k}{if} \PYG{n}{iteration} \PYG{o}{==} \PYG{l+m+mi}{1}\PYG{p}{:}
     \PYG{n}{Parameter\PYGZus{}LineList} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{index\PYGZus{}col} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)}
     \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZlt{}}\PYG{n}{wave\PYGZus{}max}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{n}{wave\PYGZus{}min}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu\PYGZus{}vary}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{k+kc}{True}
     \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZlt{}}\PYG{n}{wave\PYGZus{}max}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{n}{wave\PYGZus{}min}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw\PYGZus{}vary}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{k+kc}{True}
     \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{to\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
 \PYG{k}{if} \PYG{n}{iteration} \PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{:}
     \PYG{n}{Parameter\PYGZus{}LineList} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{index\PYGZus{}col} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)}
     \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZlt{}}\PYG{n}{wave\PYGZus{}max}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{n}{wave\PYGZus{}min}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air\PYGZus{}vary}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{k+kc}{True}
     \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZlt{}}\PYG{n}{wave\PYGZus{}max}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{n}{wave\PYGZus{}min}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air\PYGZus{}vary}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{k+kc}{True}
     \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZlt{}}\PYG{n}{wave\PYGZus{}max}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{Parameter\PYGZus{}LineList}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{n}{wave\PYGZus{}min}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}vary}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{k+kc}{True}
     \PYG{n}{Parameter\PYGZus{}LineList}\PYG{o}{.}\PYG{n}{to\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}


 \PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
 \PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
 \PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}wavenumbers} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
 \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
 \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
 \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
 \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
 \PYG{n}{iteration}\PYG{o}{+}\PYG{o}{=}\PYG{l+m+mi}{1}
\end{sphinxVerbatim}

\sphinxAtStartPar
The plots below show the results from the various iterations, where the fit residuals decrease with each iteration.

\sphinxAtStartPar
\sphinxstylestrong{Iteration 0}

\noindent\sphinxincludegraphics[width=1022\sphinxpxdimen,height=629\sphinxpxdimen]{{iteration_0}.png}

\sphinxAtStartPar
\sphinxstylestrong{Iteration 1}

\noindent\sphinxincludegraphics[width=1034\sphinxpxdimen,height=629\sphinxpxdimen]{{iteration_1}.png}

\sphinxAtStartPar
\sphinxstylestrong{Iteration 2}

\noindent\sphinxincludegraphics[width=1034\sphinxpxdimen,height=629\sphinxpxdimen]{{iteration_2}.png}

\sphinxstepscope


\subsubsection{Analysis of ASCENDS Line Spectra}
\label{\detokenize{Analysis of ASCENDS Line Spectra:analysis-of-ascends-line-spectra}}\label{\detokenize{Analysis of ASCENDS Line Spectra::doc}}
\sphinxAtStartPar
Provided in the MATS v2 release are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
The 30012\textless{}\textendash{}00001 CO2 band is frequently used for remote sensing applications, where the R16e line in this band centered at 6359.967 cm\sphinxhyphen{}1 is the target line for the ASCENDS \sphinxhyphen{} Active Sensing of CO2 Emissions over Nights, Days, and Seasons mission.  This makes this line an important target for spectroscopic reference data and thus a convenient target for single line spectroscopic studies or examples.

\sphinxAtStartPar
Data originally reported in \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407315001338}{Long, D., et al., Frequency\sphinxhyphen{}agile, rapid scanning cavity ring\sphinxhyphen{}down spectroscopy (FARS\sphinxhyphen{}CRDS) measurements of the (30012)←(00001) near\sphinxhyphen{}infrared carbon dioxide band. Journal of Quantitative Spectroscopy and Radiative Transfer, 2015. 161: p. 35\sphinxhyphen{}40}.  This example is explicitly referenced in the Adkins and Hodges, Assessment of the precision, bias and numerical correlation of fitted parameters obtained by multi\sphinxhyphen{}spectrum fits of the Hartmann\sphinxhyphen{}Tran line profile to simulated spectra, JQSRT (under review).

\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/ASCENDS\_Fitting}{This example}. shows the fitting of real experimental data.  This data has a non\sphinxhyphen{}zero baseline and several etalons that need to be considered in order to effectively fit the spectra.  In addition to fitting the experimental spectra, this example also uses simulations to explore the anticipated impact of improving the SNR of the data and increasing the pressure range of the dataset.


\paragraph{Define Spectra}
\label{\detokenize{Analysis of ASCENDS Line Spectra:define-spectra}}
\sphinxAtStartPar
Module import follows from the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples.  The experimental spectra and the linelists are read\sphinxhyphen{}in in the same manner as described in the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} example.

\sphinxAtStartPar
In this example three etalons are included in the simulation and linear baseine is considered for each spectrum in the dataset.  These fits also make use of the segment column, which allows the spectrum\sphinxhyphen{}specific parameters (baseline, etalons) to be treated on smaller segments of the spectrum, opposed to across the whole spectrum. The segment column is set using the segment\_column parameter in the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} definition and is simply the name of a column in the spectrum .csv that contains integers that are common for continuous segments of the spectrum.   This is especially useful for treatment of faster etalons, which may not maintain a consistent phase over entire spectrum.  Additional constraints can be imposed to help with fit convergence and physical modeling.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{k+kn}{import} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{hapi} \PYG{k}{as} \PYG{n+nn}{hapi}

\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}23} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Alpha}\PYG{l+s+s1}{\PYGZsq{}} \PYG{c+c1}{\PYGZsh{} Mean tau/us}
\PYG{n}{freq\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wavenumber}\PYG{l+s+s1}{\PYGZsq{}} \PYG{c+c1}{\PYGZsh{} Total Frequency /MHz}
\PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressure}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Temperature}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}


\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{JQSRT2021\PYGZus{}SDNGP\PYGZus{}2015}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{isna}\PYG{p}{(}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{etalons} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.004321}\PYG{p}{,}\PYG{l+m+mf}{1.168}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.001377}\PYG{p}{,} \PYG{l+m+mf}{59.38}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.0004578}\PYG{p}{,} \PYG{l+m+mf}{29.75}\PYG{p}{]}\PYG{p}{\PYGZcb{}}


\PYG{n}{spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}56Torr\PYGZus{}03\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}


\PYG{n}{spec\PYGZus{}2} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}83Torr\PYGZus{}02\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}3} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}101Torr\PYGZus{}01\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}4} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}109Torr\PYGZus{}03\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                  \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}5} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}152Torr\PYGZus{}01\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}6} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}186Torr\PYGZus{}03\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}7} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}269Torr\PYGZus{}01\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}8} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{spectrum\PYGZus{}CO2\PYGZus{}Air\PYGZus{}1}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{Ar\PYGZus{}271Torr\PYGZus{}03\PYGZus{}formatted\PYGZus{}R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                                \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{,} \PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Segment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temperature\PYGZus{}column}\PYG{p}{,}
                                                                \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Construct Dataset and Generate Fit Parameters}
\label{\detokenize{Analysis of ASCENDS Line Spectra:construct-dataset-and-generate-fit-parameters}}
\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} object is defined as is outlined in the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples.

\sphinxAtStartPar
The baseline and parameter fit parameter files are generated using the {\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File}}}}} class.  In this example, the line intensity is allowed to float for each spectrum, but all other line shape parameters are constrained to their pressure dependence.  \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407315001338}{Long et al.} fixed the beyond Voigt SDNGP line shape parameters to those reported by \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407315001284}{Lin, H., et al., Cavity ring\sphinxhyphen{}down spectrometer for high\sphinxhyphen{}fidelity molecular absorption measurements. Journal of Quantitative Spectroscopy and Radiative Transfer, 2015. 161: p. 11\sphinxhyphen{}20} , while allowing the VP parameters and line mixing to float.  In this example, we are allowing all SDNGP parameters to float using the \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407315001284}{Lin et al. values} as the initial guess, but only considering spectra in the isolated transition pressure regime, where line mixing can be ignored.

\sphinxAtStartPar
The slope and intercept for the baseline of each spectra are floated, along with the amplitude, period, and phase of each etalon.  Because this example uses the segment function, you can see that each spectrum has 4 segments in the print out of the baseline linelist, so the number of rows in the baseline fit parameter list is four times larger than the number of spectra.  Additional constraints will be added to the etalon parameters in the next section.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{,} \PYG{n}{spec\PYGZus{}5}\PYG{p}{,} \PYG{n}{spec\PYGZus{}6}\PYG{p}{,} \PYG{n}{spec\PYGZus{}7}\PYG{p}{,} \PYG{n}{spec\PYGZus{}8}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{MATS\PYGZus{}Example}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}Generate Baseline Parameter list based on number of etalons in spectra definitions and baseline order}
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}



     \PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDNGP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                                                                       \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                                                                                       \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                                        \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                                        \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                                             \PYG{n}{additional\PYGZus{}columns} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}lower\PYGZus{}quanta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}

     \PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                      \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                      \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                      \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                      \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                      \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                      \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
     \PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}pressure} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}temperature} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                                                                     \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=762\sphinxpxdimen,height=785\sphinxpxdimen]{{ASCENDS_baselineparameters}.png}


\paragraph{Fit Spectra}
\label{\detokenize{Analysis of ASCENDS Line Spectra:fit-spectra}}
\sphinxAtStartPar
The fitting of the dataset is looped to iterate on the fit results.  The result and residual plots are shown after each iteration (with the first and last iteration shown below).  In the first few iterations not all of the etalons are well modeled leading to systematic residuals, iterating on the best fit results helps better model the etalons minimizing the residuals.

\sphinxAtStartPar
Using the segments in a spectrum, allows us to model the spectrum specific parameters by segment opposed to across the whole spectrum.  However, this flexibility can lead to divergent solutions if additional constraints aren’t included.  This example will constrain these parameters using the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.constrained_baseline}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.constrained\_baseline()}}}}} function and by setting constraints directly in the lmfit parameter dictionary.  The {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.constrained_baseline}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.constrained\_baseline()}}}}} function indicates the baseline parameters that should be constrained to be equal across the entire spectrum (parameters that won’t take advantage of the segment structure).  For this example the baseline and etalon periods are constrained to be equal equal across the entire spectrum.

\sphinxAtStartPar
This example also sets a few additional constraints on the etalons.  First, the period of each etalon is set to be equal across all spectra in the dataset.  Second, the phase of the first etalon is constrained to be equal across each spectrum.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{counter} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{iterations} \PYG{o}{=} \PYG{l+m+mi}{5}
\PYG{k}{while} \PYG{n}{counter} \PYG{o}{\PYGZlt{}} \PYG{n}{iterations}\PYG{p}{:}
        \PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
        \PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}




        \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{constrained\PYGZus{}baseline}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{xshift\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{molefraction\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                                \PYG{n}{etalon\PYGZus{}amp\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{etalon\PYGZus{}period\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{etalon\PYGZus{}phase\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                                                                \PYG{n}{pressure\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}segment\PYGZus{}constrained} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}


        \PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{segments}\PYG{o}{.}\PYG{n}{min}\PYG{p}{(}\PYG{p}{)}


        \PYG{k}{for} \PYG{n}{param} \PYG{o+ow}{in} \PYG{n}{params}\PYG{p}{:}

                \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}period\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                        \PYG{c+c1}{\PYGZsh{}Sets the period of etalon\PYGZus{}1 to be equal for all spectra}
                        \PYG{k}{if} \PYG{n}{param} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}period\PYGZus{}1\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
                                \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n}{expr}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}period\PYGZus{}1\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}
                \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}phase\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                        \PYG{c+c1}{\PYGZsh{}Constrains the phase of etalon\PYGZus{}1 across each spectrum}
                        \PYG{n}{start\PYGZus{}} \PYG{o}{=}  \PYG{p}{(}\PYG{n}{param}\PYG{o}{.}\PYG{n}{find}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{)}\PYG{p}{)}
                        \PYG{n}{end\PYGZus{}} \PYG{o}{=}  \PYG{p}{(}\PYG{n}{param}\PYG{o}{.}\PYG{n}{find}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{param}\PYG{o}{.}\PYG{n}{find}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{)} \PYG{o}{+} \PYG{l+m+mi}{1}\PYG{p}{)}\PYG{p}{)}
                        \PYG{n}{spectrum\PYGZus{}number} \PYG{o}{=} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n}{param}\PYG{p}{[}\PYG{n}{start\PYGZus{}} \PYG{o}{+} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{n}{end\PYGZus{}}\PYG{p}{]}\PYG{p}{)}

                        \PYG{k}{if} \PYG{n}{param} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}phase\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{+}\PYG{n}{spectrum\PYGZus{}number} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
                                \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n}{expr}\PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}1\PYGZus{}phase\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{+}\PYG{n}{spectrum\PYGZus{}number} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}

                \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}2\PYGZus{}period\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                        \PYG{c+c1}{\PYGZsh{}Sets the period of etalon\PYGZus{}1 to be equal for all spectra}
                        \PYG{k}{if} \PYG{n}{param} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}2\PYGZus{}period\PYGZus{}1\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
                                \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n}{expr}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}2\PYGZus{}period\PYGZus{}1\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}

                \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}3\PYGZus{}period\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                        \PYG{c+c1}{\PYGZsh{}Sets the period of etalon\PYGZus{}1 to be equal for all spectra}
                        \PYG{k}{if} \PYG{n}{param} \PYG{o}{!=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}3\PYGZus{}period\PYGZus{}1\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{:}
                                \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n}{expr}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{etalon\PYGZus{}3\PYGZus{}period\PYGZus{}1\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{minimum\PYGZus{}segment\PYGZus{}spec\PYGZus{}1}\PYG{p}{)}\PYG{p}{)}\PYG{p}{)}


        \PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{)}


        \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
        \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
        \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
        \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{counter}\PYG{o}{+}\PYG{o}{=}\PYG{l+m+mi}{1}

        \PYG{k}{if} \PYG{n}{counter} \PYG{o}{==} \PYG{n}{iterations}\PYG{p}{:}
                \PYG{n+nb}{print} \PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{o}{.}\PYG{n}{pretty\PYGZus{}print}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
ITERATION 1

\noindent\sphinxincludegraphics[width=1678\sphinxpxdimen,height=629\sphinxpxdimen]{{ASCENDS_Iteration_1}.png}

\sphinxAtStartPar
ITERATION 2

\noindent\sphinxincludegraphics[width=1666\sphinxpxdimen,height=629\sphinxpxdimen]{{ASCENDS_Iteration_5}.png}


\paragraph{Compare to Literature}
\label{\detokenize{Analysis of ASCENDS Line Spectra:compare-to-literature}}
\sphinxAtStartPar
The \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407315001284}{Lin et al.} work reported spectra with QFs between 11000 and 29000 for pressures between 1.33kPa and 26.7kPa.  The current example work was conducted over a slightly higher pressure range of  7.5 to 36 kPa and had QFs of about 1500.

\sphinxAtStartPar
The  \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407315001284}{Lin et al.} work reports uncertainty on the collisional broadening of 0.1\%, the pressure shift of 0.35\%, the speed\sphinxhyphen{}dependent broadening of 0.2\%, the speed\sphinxhyphen{}dependent shift of 20\%, and the Dicke narrowing of 1\%.  The code below compares the results from the \sphinxhref{https://www.sciencedirect.com/science/article/pii/S0022407315001284}{Lin et al.} and current fit example including a plot highlighting the reported relative uncertainty in line shape parameters reported by the fits.  The current example shows much higher fit uncertainties for speed\sphinxhyphen{}dependent width and Dicke narrowing, which indicates that at this QF and pressure range the spectra used in this analysis have a difficult time distinguishing between these narrowing mechanisms.  This motivates a Monte Carlo analysis that explores how improving the SNR and pressure range would decrease the uncertainty in the reported line shape parameters.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Compare\PYGZus{}Results} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}

\PYG{c+c1}{\PYGZsh{}Get Spectra line number from the parameter results list.  Alternatively, could read in the Linelist file and parse the same way as the initial Parameter linelist}
\PYG{k}{for} \PYG{n}{param} \PYG{o+ow}{in} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{:}
        \PYG{k}{if} \PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{)} \PYG{o+ow}{and} \PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{vary} \PYG{o}{==} \PYG{k+kc}{True}\PYG{p}{)}\PYG{p}{:}
                \PYG{n}{line\PYGZus{}number} \PYG{o}{=} \PYG{n}{param}\PYG{o}{.}\PYG{n}{replace}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}


\PYG{k}{for} \PYG{n}{parameter} \PYG{o+ow}{in} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{unique}\PYG{p}{(}\PYG{p}{)}\PYG{p}{:}
        \PYG{c+c1}{\PYGZsh{} Get Lin et al Parameter values and uncertainties}
        \PYG{n}{Compare\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{n}{parameter}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Lin et al. Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}  \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}lower\PYGZus{}quanta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{global\PYGZus{}upper\PYGZus{}quanta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{30012}\PYG{p}{)}\PYG{p}{]}\PYG{p}{[}\PYG{n}{parameter}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
        \PYG{n}{lin\PYGZus{}reported\PYGZus{}uncertainty} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.35}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.2}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{20}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}
        \PYG{n}{Compare\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{n}{parameter}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Lin et al. Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}  \PYG{n}{lin\PYGZus{}reported\PYGZus{}uncertainty}\PYG{p}{[}\PYG{n}{parameter}\PYG{p}{]}

        \PYG{c+c1}{\PYGZsh{} Get Parameter results from fitting}
        \PYG{n}{Compare\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{n}{parameter}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}  \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{n}{parameter} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
        \PYG{n}{Compare\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{n}{parameter}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}  \PYG{l+m+mi}{100}\PYG{o}{*}\PYG{n}{np}\PYG{o}{.}\PYG{n}{abs}\PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{n}{parameter} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr} \PYG{o}{/} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{n}{parameter} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}\PYG{p}{)}


\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Percent Difference (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{o}{*}\PYG{p}{(}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{\PYGZhy{}} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Lin et al. Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)} \PYG{o}{/} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}


\PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mi}{8}\PYG{p}{,}\PYG{l+m+mi}{5}\PYG{p}{)}\PYG{p}{)}


\PYG{n}{plt}\PYG{o}{.}\PYG{n}{semilogy}\PYG{p}{(}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Lin et al. Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{label}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Lin et al. (2015)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{semilogy}\PYG{p}{(}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{,}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{label}\PYG{o}{=}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Current Example}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{ax} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{gca}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xticklabels}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZbs{}}\PYG{l+s+s1}{gamma\PYGZus{}}\PYG{l+s+si}{\PYGZob{}0\PYGZcb{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZbs{}}\PYG{l+s+s1}{delta\PYGZus{}}\PYG{l+s+si}{\PYGZob{}0\PYGZcb{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZdl{}a\PYGZus{}}\PYG{l+s+si}{\PYGZob{}w\PYGZcb{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZdl{}a\PYGZus{}}\PYG{l+s+si}{\PYGZob{}s\PYGZcb{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+se}{\PYGZbs{}\PYGZbs{}}\PYG{l+s+s1}{nu\PYGZus{}}\PYG{l+s+si}{\PYGZob{}VC\PYGZcb{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{p}{]}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{line profile parameter}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}plt.bar(Compare\PYGZus{}Results[\PYGZsq{}Parameters\PYGZsq{}], Compare\PYGZus{}Results[\PYGZsq{}Lin et al. Percent Difference (\PYGZpc{})\PYGZsq{}])}
\PYG{c+c1}{\PYGZsh{}plt.errorbar(Compare\PYGZus{}Results[\PYGZsq{}Parameters\PYGZsq{}], Compare\PYGZus{}Results[\PYGZsq{}Lin et al. Percent Difference (\PYGZpc{})\PYGZsq{}], yerr=Compare\PYGZus{}Results[\PYGZsq{}Lin et al. Uncertainty (\PYGZpc{})\PYGZsq{}], fmt=\PYGZdq{}o\PYGZdq{}, color=\PYGZdq{}r\PYGZdq{})}

\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{Compare\PYGZus{}Results}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=523\sphinxpxdimen,height=404\sphinxpxdimen]{{ASCENDS_FitUncertaintyComparison}.png}


\paragraph{Monte Carlo Analysis}
\label{\detokenize{Analysis of ASCENDS Line Spectra:monte-carlo-analysis}}
\sphinxAtStartPar
The Monte Carlo uncertainty analysis involves running numerous iterations fitting synthetic spectra at experimental conditions.  To simplify the analysis the simulation line list only includes the R16e line and will use the results of the experimental fits as the simulation values.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Generate linelist only using the R16e line and using the results from the experimental fits}
\PYG{n}{SIMULATION\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}lower\PYGZus{}quanta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{R16e}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{global\PYGZus{}upper\PYGZus{}quanta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{30012}\PYG{p}{)}\PYG{p}{]}\PYG{o}{.}\PYG{n}{copy}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{reset\PYGZus{}index}\PYG{p}{(}\PYG{n}{inplace} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}


\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o}{+} \PYG{n}{line\PYGZus{}number}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}air\PYGZus{}296}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\sphinxAtStartPar
Additionally, a dataframe summarizing the pressures and SNR (approximated by the QF from fits to the experimental spectra) are necessary for defining the simulation conditions.  In this analysis we are comparing the synthetic results at experimetnal conditions, to those with three addtional low pressure spectra and with SNR of 5000 (about three times larger than observed in the experimental spectra).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} Pressure and QF of Fit Results for Simulation at Experiment Conditions}
\PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Spectrum Number}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{get\PYGZus{}list\PYGZus{}spectrum\PYGZus{}numbers}\PYG{p}{(}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{spec} \PYG{o+ow}{in} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{spectra}\PYG{p}{:}
        \PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Spectrum Number}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{spec}\PYG{o}{.}\PYG{n}{spectrum\PYGZus{}number}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SNR}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{=}\PYG{n}{spec}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Spectrum Number}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{spec}\PYG{o}{.}\PYG{n}{spectrum\PYGZus{}number}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressures}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{around}\PYG{p}{(}\PYG{n}{spec}\PYG{o}{.}\PYG{n}{get\PYGZus{}pressure\PYGZus{}torr}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}

\PYG{n}{New\PYGZus{}Simulation\PYGZus{}conditions} \PYG{o}{=} \PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions}\PYG{o}{.}\PYG{n}{copy}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{New\PYGZus{}Simulation\PYGZus{}conditions}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SNR}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{5000}
\PYG{n}{new\PYGZus{}pressures} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{n}{data} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressures}\PYG{l+s+s1}{\PYGZsq{}} \PYG{p}{:} \PYG{p}{[}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{25}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SNR}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{p}{[}\PYG{l+m+mi}{5000}\PYG{p}{,} \PYG{l+m+mi}{5000}\PYG{p}{,} \PYG{l+m+mi}{5000}\PYG{p}{]}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
\PYG{n}{New\PYGZus{}Simulation\PYGZus{}conditions} \PYG{o}{=} \PYG{n}{New\PYGZus{}Simulation\PYGZus{}conditions}\PYG{o}{.}\PYG{n}{append}\PYG{p}{(}\PYG{n}{new\PYGZus{}pressures}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
In the Monte Carlo simulation the inital guesses will be perturbed from the simulation value by the uncertainty reported in the experimental fit.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{gamma0\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}
\PYG{n}{delta0\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}
\PYG{n}{aw\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100}\PYG{c+c1}{\PYGZsh{}per}
\PYG{n}{as\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}
\PYG{n}{nuVC\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}
\end{sphinxVerbatim}

\sphinxAtStartPar
We define a function to generate spectrum based on the input simulation conditions (SNR and pressure) and with a given input linelist.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{def} \PYG{n+nf}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{n}{i}\PYG{p}{,} \PYG{n}{simulation\PYGZus{}linelist}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{sample\PYGZus{}molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.0004254}\PYG{p}{\PYGZcb{}}
        \PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mf}{6358.972} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
        \PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mf}{6360.963} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
        \PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{l+m+mf}{0.006772} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
        \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{l+m+mf}{4.67e\PYGZhy{}7}

        \PYG{n}{etalons} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.004321}\PYG{p}{,}\PYG{l+m+mf}{1.168}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.001377}\PYG{p}{,} \PYG{l+m+mf}{59.38}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{p}{[}\PYG{l+m+mf}{0.0004578}\PYG{p}{,} \PYG{l+m+mf}{29.75}\PYG{p}{]}\PYG{p}{\PYGZcb{}}

        \PYG{n}{spec} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}linelist}\PYG{p}{,}
                                                         \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,}
                                                        \PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,}  \PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                                                        \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SNR}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]}\PYG{p}{,} \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalons}\PYG{p}{,}
                                                        \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{23.25}\PYG{p}{,}
                                                        \PYG{n}{pressure} \PYG{o}{=} \PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressures}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]}\PYG{p}{,}
                                                        \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                        \PYG{n}{filename} \PYG{o}{=} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressures}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{n}{i}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                        \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,}
                                                        \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{)}
        \PYG{k}{return} \PYG{n}{spec}
\end{sphinxVerbatim}

\sphinxAtStartPar
The {\color{red}\bfseries{}mc\_} function defined below simply takes the simulation conditions and number of iterations as inputs.  Spectra are simulated at the conditions outlined in the simulation condition dataframe for just the R16e line using the linelist determined from the fits to the experimental spectra.  Again, the inital guesses for the fitting are perturbed randomly at the magnitude of the reported fit error of the fits to the experimental data.  The fits are conducted in the same manner as those to the experimental data without the use of the segment function.  Results from each iteration are saved to a MC\_result .csv file.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{def} \PYG{n+nf}{mc\PYGZus{}}\PYG{p}{(}\PYG{n}{iterations}\PYG{p}{,} \PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{n}{MC\PYGZus{}result\PYGZus{}file}\PYG{p}{)}\PYG{p}{:}
\PYG{+w}{        }\PYG{l+s+sd}{\PYGZsq{}\PYGZsq{}\PYGZsq{}}
\PYG{l+s+sd}{        Runs the MC simulation for given number of iterations using the pressures and SNR defined in simulation condition dataframe for the R16e line and saves results to file}
\PYG{l+s+sd}{        \PYGZsq{}\PYGZsq{}\PYGZsq{}}
        \PYG{n}{MC\PYGZus{}Results} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{p}{)}
        \PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{iterations}\PYG{p}{)}
        \PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
        \PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
        \PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
        \PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
        \PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
        \PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}air\PYGZus{}296}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}air\PYGZus{}296}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}

        \PYG{k}{for} \PYG{n}{iteration} \PYG{o+ow}{in} \PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{iterations}\PYG{p}{)}\PYG{p}{:}

                \PYG{c+c1}{\PYGZsh{}Read in Possible linelists}
                \PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{copy}\PYG{p}{(}\PYG{p}{)}

                \PYG{k}{if} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{)} \PYG{o}{==}\PYG{l+m+mi}{8}\PYG{p}{:}
                        \PYG{n}{spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}2} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}3} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}4} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}5} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}6} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}7} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}8} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{,} \PYG{n}{spec\PYGZus{}5}\PYG{p}{,} \PYG{n}{spec\PYGZus{}6}\PYG{p}{,} \PYG{n}{spec\PYGZus{}7}\PYG{p}{,} \PYG{n}{spec\PYGZus{}8}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Monte Carlo Analysis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{)}
                \PYG{k}{elif} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{)} \PYG{o}{==} \PYG{l+m+mi}{11}\PYG{p}{:}
                        \PYG{n}{spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}2} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}3} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}4} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}5} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}6} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}7} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}8} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}9} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}10} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{spec\PYGZus{}11} \PYG{o}{=} \PYG{n}{gen\PYGZus{}spec}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{SIMULATION\PYGZus{}LINELIST}\PYG{p}{)}
                        \PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{,} \PYG{n}{spec\PYGZus{}5}\PYG{p}{,} \PYG{n}{spec\PYGZus{}6}\PYG{p}{,} \PYG{n}{spec\PYGZus{}7}\PYG{p}{,} \PYG{n}{spec\PYGZus{}8}\PYG{p}{,} \PYG{n}{spec\PYGZus{}9}\PYG{p}{,} \PYG{n}{spec\PYGZus{}10}\PYG{p}{,} \PYG{n}{spec\PYGZus{}11}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Monte Carlo Analysis}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{)}






                \PYG{c+c1}{\PYGZsh{}Generate Baseline Parameter list based on number of etalons in spectra definitions and baseline order}
                \PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}


                \PYG{n}{gamma0\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}
                \PYG{n}{delta0\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}
                \PYG{n}{aw\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100}\PYG{c+c1}{\PYGZsh{}per}
                \PYG{n}{as\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}
                \PYG{n}{nuVC\PYGZus{}err} \PYG{o}{=} \PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{n}{Compare\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameters}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Fit Uncertainty (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{o}{/}\PYG{l+m+mi}{100} \PYG{c+c1}{\PYGZsh{}per}


                \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{size} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{o}{*}\PYG{n}{gamma0\PYGZus{}err}\PYG{p}{)}
                \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{size} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{o}{*}\PYG{n}{delta0\PYGZus{}err}\PYG{p}{)}
                \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{size} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{o}{*}\PYG{n}{aw\PYGZus{}err}\PYG{p}{)}
                \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{size} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{o}{*}\PYG{n}{as\PYGZus{}err}\PYG{p}{)}
                \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{size} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{o}{*}\PYG{n}{nuVC\PYGZus{}err}\PYG{p}{)}



                \PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDNGP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                                                                                  \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                                                                                                  \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                                                   \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                                                   \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                                                        \PYG{n}{additional\PYGZus{}columns} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}lower\PYGZus{}quanta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}



                \PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                                                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}
                \PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}pressure} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}temperature} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                                                                                \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}


                \PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
                \PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}


                \PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{)}

                \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
                \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
                \PYG{c+c1}{\PYGZsh{}SPECTRA.generate\PYGZus{}summary\PYGZus{}file(save\PYGZus{}file = True)}
                \PYG{c+c1}{\PYGZsh{}SPECTRA.plot\PYGZus{}model\PYGZus{}residuals()}


                \PYG{c+c1}{\PYGZsh{}line Shape Parameters}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air\PYGZus{}fit\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air\PYGZus{}fit\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air\PYGZus{}fit\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}fit\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{value}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}fit\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air\PYGZus{}line\PYGZus{}0}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{stderr}



                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}2}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}3}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}3}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}4}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}4}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}5}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}5}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}6}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}6}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}7}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}7}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}8}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}8}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                \PYG{k}{if} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{simulation\PYGZus{}conditions}\PYG{p}{)} \PYG{o}{==} \PYG{l+m+mi}{11}\PYG{p}{:}
                        \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}9}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}9}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                        \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}10}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}10}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}
                        \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{MC\PYGZus{}Results}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==} \PYG{n}{iteration}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{QF\PYGZus{}11}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{spec\PYGZus{}11}\PYG{o}{.}\PYG{n}{calculate\PYGZus{}QF}\PYG{p}{(}\PYG{p}{)}


                \PYG{n}{MC\PYGZus{}Results}\PYG{o}{.}\PYG{n}{to\PYGZus{}csv}\PYG{p}{(}\PYG{n}{MC\PYGZus{}result\PYGZus{}file} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{index} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
Below are the calls to the {\color{red}\bfseries{}mc\_} function at the experimental and new simulation conditions, which will generate and fit 50 iterations of synthetic data for each case.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{mc\PYGZus{}}\PYG{p}{(}\PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{Exp\PYGZus{}Simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Experimental\PYGZus{}Conditions\PYGZus{}MC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{mc\PYGZus{}}\PYG{p}{(}\PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{New\PYGZus{}Simulation\PYGZus{}conditions}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{New\PYGZus{}Conditions\PYGZus{}MC}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The parameter\_eval function summarizes the MC results comparing the fit average and standard deviation across the iterations to the reported fit uncertainty on each iteration.  This also calculates the fractional error compared to the simulation value.  Plots of summary and dataframe are generated for visualization.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k}{def} \PYG{n+nf}{parameter\PYGZus{}eval}\PYG{p}{(}\PYG{n}{df}\PYG{p}{,} \PYG{n}{report\PYGZus{}df}\PYG{p}{)}\PYG{p}{:}
\PYG{+w}{        }\PYG{l+s+sd}{\PYGZsq{}\PYGZsq{}\PYGZsq{}}
\PYG{l+s+sd}{        Summarizes MC results for a given parameter}
\PYG{l+s+sd}{        \PYGZsq{}\PYGZsq{}\PYGZsq{}}
        \PYG{n}{params} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
        \PYG{n}{fig} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{constrained\PYGZus{}layout}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{figsize} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{20}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{spec} \PYG{o}{=} \PYG{n}{gridspec}\PYG{o}{.}\PYG{n}{GridSpec}\PYG{p}{(}\PYG{n}{ncols}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{nrows}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{figure}\PYG{o}{=}\PYG{n}{fig}\PYG{p}{)}
        \PYG{n}{ax1} \PYG{o}{=} \PYG{n}{fig}\PYG{o}{.}\PYG{n}{add\PYGZus{}subplot}\PYG{p}{(}\PYG{n}{spec}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{ax2} \PYG{o}{=} \PYG{n}{fig}\PYG{o}{.}\PYG{n}{add\PYGZus{}subplot}\PYG{p}{(}\PYG{n}{spec}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{1}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{ax3} \PYG{o}{=} \PYG{n}{fig}\PYG{o}{.}\PYG{n}{add\PYGZus{}subplot}\PYG{p}{(}\PYG{n}{spec}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{ax4} \PYG{o}{=} \PYG{n}{fig}\PYG{o}{.}\PYG{n}{add\PYGZus{}subplot}\PYG{p}{(}\PYG{n}{spec}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{3}\PYG{p}{]}\PYG{p}{)}
        \PYG{n}{ax5} \PYG{o}{=} \PYG{n}{fig}\PYG{o}{.}\PYG{n}{add\PYGZus{}subplot}\PYG{p}{(}\PYG{n}{spec}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{,}\PYG{l+m+mi}{4}\PYG{p}{]}\PYG{p}{)}

        \PYG{n}{axes} \PYG{o}{=} \PYG{p}{[}\PYG{n}{ax1}\PYG{p}{,} \PYG{n}{ax2}\PYG{p}{,} \PYG{n}{ax3}\PYG{p}{,} \PYG{n}{ax4}\PYG{p}{,} \PYG{n}{ax5}\PYG{p}{]}
        \PYG{n}{count} \PYG{o}{=}\PYG{l+m+mi}{0}
        \PYG{k}{for} \PYG{n}{simulated\PYGZus{}column} \PYG{o+ow}{in} \PYG{n}{params}\PYG{p}{:}
                \PYG{n}{param\PYGZus{}avg} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{median}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{param\PYGZus{}std} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{std}\PYG{p}{(}\PYG{p}{)}
                \PYG{n}{param\PYGZus{}err\PYGZus{}avg} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}fit\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{median}\PYG{p}{(}\PYG{p}{)}

                \PYG{k}{if} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
                        \PYG{n}{param\PYGZus{}frac\PYGZus{}err} \PYG{o}{=} \PYG{l+m+mi}{0}
                        \PYG{n}{param\PYGZus{}fit\PYGZus{}err} \PYG{o}{=} \PYG{l+m+mi}{0}
                \PYG{k}{else}\PYG{p}{:}
                        \PYG{n}{param\PYGZus{}frac\PYGZus{}err} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{o}{*}\PYG{p}{(}\PYG{n}{param\PYGZus{}avg} \PYG{o}{\PYGZhy{}} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{)} \PYG{o}{/} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
                        \PYG{n}{param\PYGZus{}fit\PYGZus{}err} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{abs}\PYG{p}{(}\PYG{l+m+mi}{100}\PYG{o}{*}\PYG{n}{param\PYGZus{}std} \PYG{o}{/} \PYG{n}{param\PYGZus{}avg}\PYG{p}{)}
                \PYG{k}{if} \PYG{n}{param\PYGZus{}err\PYGZus{}avg} \PYG{o}{==} \PYG{l+m+mi}{0}\PYG{p}{:}
                        \PYG{n}{param\PYGZus{}err\PYGZus{}ratio} \PYG{o}{=} \PYG{l+m+mi}{0}
                \PYG{k}{else}\PYG{p}{:}
                        \PYG{n}{param\PYGZus{}err\PYGZus{}ratio} \PYG{o}{=} \PYG{n}{param\PYGZus{}std}\PYG{o}{/}\PYG{n}{param\PYGZus{}err\PYGZus{}avg}

                \PYG{n}{report\PYGZus{}df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{report\PYGZus{}df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Simulated Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}
                \PYG{n}{report\PYGZus{}df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{report\PYGZus{}df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit Average}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{param\PYGZus{}avg}
                \PYG{n}{report\PYGZus{}df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{report\PYGZus{}df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit StDev}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{param\PYGZus{}std}
                \PYG{n}{report\PYGZus{}df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{report\PYGZus{}df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Error Average}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{param\PYGZus{}err\PYGZus{}avg}
                \PYG{n}{report\PYGZus{}df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{report\PYGZus{}df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fractional Error}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{param\PYGZus{}frac\PYGZus{}err}
                \PYG{n}{report\PYGZus{}df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{report\PYGZus{}df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit Error}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{param\PYGZus{}fit\PYGZus{}err}
                \PYG{n}{report\PYGZus{}df}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{n}{report\PYGZus{}df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Error Ratio}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{param\PYGZus{}err\PYGZus{}ratio}

                \PYG{n}{ax} \PYG{o}{=} \PYG{n}{axes}\PYG{p}{[}\PYG{n}{count}\PYG{p}{]}
                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Iterations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}fit}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{fmt} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{yerr} \PYG{o}{=} \PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column} \PYG{o}{+} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}fit\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}
                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{hlines}\PYG{p}{(}\PYG{n}{param\PYGZus{}avg}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{50}\PYG{p}{)}
                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{hlines}\PYG{p}{(}\PYG{n}{param\PYGZus{}avg} \PYG{o}{+} \PYG{n}{param\PYGZus{}std}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{linestyles} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{dashed}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{hlines}\PYG{p}{(}\PYG{n}{param\PYGZus{}avg} \PYG{o}{\PYGZhy{}} \PYG{n}{param\PYGZus{}std}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{linestyles} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{dashed}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{hlines}\PYG{p}{(}\PYG{n}{df}\PYG{p}{[}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{linestyles} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{dashed}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{colors} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{r}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}title}\PYG{p}{(}\PYG{n}{simulated\PYGZus{}column}\PYG{p}{)}
                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{interations}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

                \PYG{n}{ax}\PYG{o}{.}\PYG{n}{ticklabel\PYGZus{}format}\PYG{p}{(}\PYG{n}{useOffset}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
                \PYG{n}{count}\PYG{o}{+}\PYG{o}{=}\PYG{l+m+mi}{1}
        \PYG{n}{ax1}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

        \PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}

        \PYG{k}{return} \PYG{n}{report\PYGZus{}df}
\end{sphinxVerbatim}

\sphinxAtStartPar
The Experimental parameter result table shows good agreement in the fit uncertainty reported in the experimental data and averaged over the iterations of the MC analysis conducted at experimental conditions.  This also provides a measure of the ratio between the standard deviations in the fit values across the iterations compared to the average fit uncertainty.  Ratios greater than one indicate that the fit uncertainty reported by one iteration might under represent the uncertainty, most likely because of correlation between parameters that aren’t included in the calculation of the fit uncertainty.  This ratio, when greater than one, could act as a correction factor to increase the experiment fit uncertainty.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{exp\PYGZus{}MC} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Experimental\PYGZus{}Conditions\PYGZus{}MC.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{new\PYGZus{}MC} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{New\PYGZus{}Conditions\PYGZus{}MC.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{n}{EXP\PYGZus{}SUMMARY} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Reported Values}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Simulated Value}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit Average}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit StDev}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fractional Error}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit Error}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Error Ratio}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}

\PYG{n}{NEW\PYGZus{}SUMMARY} \PYG{o}{=} \PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{o}{.}\PYG{n}{copy}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{EXP\PYGZus{}SUMMARY} \PYG{o}{=} \PYG{n}{parameter\PYGZus{}eval}\PYG{p}{(}\PYG{n}{exp\PYGZus{}MC}\PYG{p}{,} \PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{)}
\PYG{n}{EXP\PYGZus{}SUMMARY}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=1448\sphinxpxdimen,height=227\sphinxpxdimen]{{ASCENDS_MC_Exp}.png}

\noindent\sphinxincludegraphics[width=769\sphinxpxdimen,height=142\sphinxpxdimen]{{ASCENDS_MC_Exp_Summary}.png}

\sphinxAtStartPar
By comparing the average fit uncertainties over MC simulations in the new conditions (higher SNR and 3 extra pressures) to the experimental conditions indicate that the proposed improvements to the experiments can be expected to reduce the uncertainties in the spectroscopic parameters by factors of 2.8 to 3.7.  The largest improvement would be to the Dicke narrowing parameter, which is aided by not only the higher SNR, but also by the inclusion of additional low\sphinxhyphen{}pressure spectra that fall in a domain where the Dicke narrowing mechanism dominates over the speed\sphinxhyphen{}dependent narrowing.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{NEW\PYGZus{}SUMMARY}\PYG{o}{=} \PYG{n}{parameter\PYGZus{}eval}\PYG{p}{(}\PYG{n}{new\PYGZus{}MC}\PYG{p}{,} \PYG{n}{NEW\PYGZus{}SUMMARY}\PYG{p}{)}
\PYG{n}{NEW\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Improvement Over Experiment}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{EXP\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit Error}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{/}\PYG{n}{NEW\PYGZus{}SUMMARY}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter Fit Error}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}
\PYG{n}{NEW\PYGZus{}SUMMARY}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=1448\sphinxpxdimen,height=227\sphinxpxdimen]{{ASCENDS_MC_New}.png}

\noindent\sphinxincludegraphics[width=890\sphinxpxdimen,height=142\sphinxpxdimen]{{ASCENDS_MC_New_Summary}.png}

\sphinxstepscope


\subsubsection{Applying Analytical Correction to hardness of collisions for HTP}
\label{\detokenize{Applying Analytical Correction to hardness of collisions for HTP:applying-analytical-correction-to-hardness-of-collisions-for-htp}}\label{\detokenize{Applying Analytical Correction to hardness of collisions for HTP::doc}}
\sphinxAtStartPar
Provided in the MATS v2 release are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
Dicke narrowing accounts for effective narrowing of the doppler width because of velocity changing collisions.  The magnitude of this effect is dependent on the hardness of the collisions.  Different line shapes make different assumptions on the hardness of the collision with the Nelkhin\sphinxhyphen{}Ghatak making the assumption that collisions are hard, meaning that the molecular velocity before a collision is forgotten after the collision, and  the Galatry profile assuming collisions are soft, meaning that many collisions are necessary to change the velocity. In reality, the hardness of the collision is related to the perturbers and absorbers being studied and acts on a continuum.  In a \sphinxhref{https://doi.org/10.1016/j.jqsrt.2019.106784}{paper by  Konefał et al.} (citation below), authors map the hard Dicke Narrowing term onto the billiard ball collision model, based on a parameter beta, which is related to the pressure (parameterized as the Dicke narrowing at a pressure relative to the doppler width) and the ratio of the masses between the perturber and absorber.  As Beta can be calculated, this correction provides a more robust modeling of the Dicke Narrowing without the addition of additional floated parameters.

\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2019.106784}{M. Konefał, M. Słowiński, M. Zaborowski, R. Ciuryło, D. Lisak, P. Wcisło, Analytical\sphinxhyphen{}function correction to the Hartmann\textendash{}Tran profile for more reliable representation of the Dicke\sphinxhyphen{}narrowed molecular spectra, Journal of Quantitative Spectroscopy and Radiative Transfer, Volume 242, 2020,106784,ISSN 0022\sphinxhyphen{}4073.}

\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/Beta\_Parameterization}{The Beta correction has been included in the MATS software and the following example provides an example of how to use it.}


\paragraph{Simulate Spectrum Objects}
\label{\detokenize{Applying Analytical Correction to hardness of collisions for HTP:simulate-spectrum-objects}}
\sphinxAtStartPar
Module import follows from the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples with additional details on how to simulate spectra found in {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} and the source documentation.

\sphinxAtStartPar
After the generic parameters are introduced the line list for simulations is read in using the {\hyperref[\detokenize{MATS:MATS.linelistdata.LoadLineListData}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{LoadLineListData()}}}}} function.  In this example we are making some adjustments to the line list before simulating.  In the MATS fitting, simulations are only conducted to some user\sphinxhyphen{}defined range outside of the spectral frequency range (wave\_range = 1.5 \(cm^{-1}\) in the current example).  This same constraint is not imposed when simulating spectra.  This can lead to some far\sphinxhyphen{}wing issues between simulation and fitting if care is not taken to match the wave\_range to the lines in the parameter linelist and the simulation wing\_cutoff magnitude, which should also match the fit wing\_cutoff magnitude. For this example, we are truncating the simulation line list to {[}wave\_min \sphinxhyphen{} wave\_range, wave\_max + wave\_range{]}.  Additionally, values for the Dicke Narrowing air and self parameters are set and the line mixing terms are set to 0.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{k+kn}{from} \PYG{n+nn}{MATS} \PYG{k+kn}{import} \PYG{n}{simulate\PYGZus{}spectrum}

\PYG{c+c1}{\PYGZsh{}Generic Fit Parameters}
\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}25} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{6326} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mi}{6328} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{l+m+mf}{0.005} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{c+c1}{\PYGZsh{}polynomial baseline coefficients where the index is equal to the coefficient order}
\PYG{n}{etalon} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}

\PYG{c+c1}{\PYGZsh{}Read in  linelists}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}30012}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{c+c1}{\PYGZsh{}\PYGZsh{}Adjust the linelist before simualting spectra}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}30012.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{o}{=} \PYG{n}{wave\PYGZus{}min} \PYG{o}{\PYGZhy{}} \PYG{n}{wave\PYGZus{}range}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZlt{}}\PYG{o}{=} \PYG{n}{wave\PYGZus{}min} \PYG{o}{+} \PYG{n}{wave\PYGZus{}range}\PYG{p}{)}\PYG{p}{]}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{\PYGZgt{}} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}\PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}iso\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mf}{0.018596}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{\PYGZgt{}} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}iso\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mf}{0.01993}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{\PYGZgt{}} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}iso\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}self\PYGZus{}296}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{loc}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{\PYGZgt{}} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}iso\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{1}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{==} \PYG{l+m+mi}{2}\PYG{p}{)}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}air\PYGZus{}296}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\end{sphinxVerbatim}

\sphinxAtStartPar
For the purpose of showing the mechanism, we will be simulating data for \(CO_{2}\), air, and 10\% \(CO_{2}\) in air samples between 1 and 500 torr.  The structure of the script follows that outlined in the {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} example.

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.spectrum.simulate_spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{simulate\_spectrum()}}}}} function is used to generate several spectrum class objects.  The beta\_formalism variable is set to True, which simulates the spectra accounting for the change in Dicke Narrowing with the hardness of the collision.  The synthetic spectra span 3 samples: \(CO_{2}\), air, and 10\% \(CO_{2}\) in air.  The air and \(CO_{2}\) samples can use the diluent = ‘air’ and ‘self’ variables as the Diluent variable can be auto\sphinxhyphen{}generated.  However, the 10\% \(CO_{2}\) in air sample requires the explicit definition of the Diluent variable, which is a dictionary of dictionaries where each diluent is a dictionary with composition and mass (m) keys.  This allows the calculation of the mass of perturber, which is necessary for the \(\beta\) implementation.  The use of the self term will generate a warning to consider switching to an explicitly name the broadener.  This warning is to have the user consider if this is causing ambiguity or unanticipated behavior.  Specifically, this could be an issue in multi\sphinxhyphen{}species fits or if the explict broadener and self term are both used.  To remove the use of the self broadening, have the line shape parameter file with the explicit broadener (ie. gamma0\_CO2 instead of gamma0\_self) and use the Diluent sample definition as modeled for the 10\% \(CO_{2}\) in air sample (ie pure \(CO_{2}\) would be Diluent = \{‘CO2’:\{‘composition’:1, ‘m’:43.98983\} \}).

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Synthetic Air Simulation}
\PYG{n+nb}{print} \PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Synthetic Air Samples}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}1} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}1\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{400}\PYG{o}{*}\PYG{l+m+mf}{1e\PYGZhy{}6}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}2} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}10\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{400}\PYG{o}{*}\PYG{l+m+mf}{1e\PYGZhy{}6}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}3} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}100\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{400}\PYG{o}{*}\PYG{l+m+mf}{1e\PYGZhy{}6}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}4} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{250}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}250\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{400}\PYG{o}{*}\PYG{l+m+mf}{1e\PYGZhy{}6}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}5} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}500\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                    \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{400}\PYG{o}{*}\PYG{l+m+mf}{1e\PYGZhy{}6}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}Mixture of CO2 and Air}
\PYG{c+c1}{\PYGZsh{}\PYGZsh{} Note that we had to use the Diluent method opposedc to the diluent method.  Additionally, the use of the self broadening term will generate some warnings}
\PYG{n+nb}{print} \PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mixture Samples}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}6} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mix\PYGZus{}1\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{Diluent} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{43.98983}\PYG{p}{\PYGZcb{}} \PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}} \PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.9}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{28.95734}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}7}\PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mix\PYGZus{}10\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{Diluent} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{43.98983}\PYG{p}{\PYGZcb{}} \PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}} \PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.9}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{28.95734}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}8} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mix\PYGZus{}100\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{Diluent} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{43.98983}\PYG{p}{\PYGZcb{}} \PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}} \PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.9}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{28.95734}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}9} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{250}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mix\PYGZus{}250\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{Diluent} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{43.98983}\PYG{p}{\PYGZcb{}} \PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}} \PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.9}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{28.95734}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}10} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mix\PYGZus{}500\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                          \PYG{n}{Diluent} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{43.98983}\PYG{p}{\PYGZcb{}} \PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}} \PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{0.9}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mf}{28.95734}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mf}{0.1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}Pure CO2.  Notice that the use of the self broadening term will cause some warnings.}
\PYG{n+nb}{print} \PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pure CO2 Samples}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}11} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}1\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}12} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}10\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}13} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}100\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}  \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}14} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{250}\PYG{p}{,}  \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}250\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}15} \PYG{o}{=}  \PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,} \PYG{n}{wave\PYGZus{}error} \PYG{o}{=} \PYG{n}{wave\PYGZus{}error}\PYG{p}{,}
                     \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{500}\PYG{p}{,}  \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}500\PYGZus{}torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                     \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30}\PYG{p}{,} \PYG{n}{num\PYGZus{}segments} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Generate Dataset and Fit Parameter Files}
\label{\detokenize{Applying Analytical Correction to hardness of collisions for HTP:generate-dataset-and-fit-parameter-files}}
\sphinxAtStartPar
Following the procedure outlined in {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} and {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}}.  This example is using the SDNGP without line mixing, is constraining all parameters to multi\sphinxhyphen{}spectrum fits, and is floating the line center, line intensity, collisional broadening, pressure shift, speed\sphinxhyphen{}dependent broadening, and Dicke narrowing terms.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{,} \PYG{n}{spec\PYGZus{}5}\PYG{p}{,} \PYG{n}{spec\PYGZus{}6}\PYG{p}{,} \PYG{n}{spec\PYGZus{}7}\PYG{p}{,} \PYG{n}{spec\PYGZus{}8}\PYG{p}{,} \PYG{n}{spec\PYGZus{}9}\PYG{p}{,} \PYG{n}{spec\PYGZus{}10}\PYG{p}{,} \PYG{n}{spec\PYGZus{}11}\PYG{p}{,} \PYG{n}{spec\PYGZus{}12}\PYG{p}{,} \PYG{n}{spec\PYGZus{}13}\PYG{p}{,} \PYG{n}{spec\PYGZus{}14}\PYG{p}{,} \PYG{n}{spec\PYGZus{}15}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2 Study}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}Generate Baseline Parameter list based on number of etalons in spectra definitions and baseline order}
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDNGP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                               \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                               \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
                               \PYG{c+c1}{\PYGZsh{}additional\PYGZus{}columns = [\PYGZsq{}trans\PYGZus{}id\PYGZsq{}, \PYGZsq{}local\PYGZus{}lower\PYGZus{}quanta\PYGZsq{}, \PYGZsq{}m\PYGZsq{}])}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{n}{Fa}   \PYG{n}{lse}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                   \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Fit Data}
\label{\detokenize{Applying Analytical Correction to hardness of collisions for HTP:fit-data}}
\sphinxAtStartPar
In this first iteration, we are fitting the data without including the beta\_formalism to correct for the hardness of the collisions by setting the beta\_formalism to False in the {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet}}}}}  instance, which is the default.  The results show systematic residuals at the line core based on the large range of pressures.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}24}\PYG{p}{,}
                   \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,}\PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}print (result.params.pretty\PYGZus{}print())}

\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{param\PYGZus{}linelist\PYGZus{}update\PYGZus{}file} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Results without Beta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=960\sphinxpxdimen,height=593\sphinxpxdimen]{{noBeta}.png}

\sphinxAtStartPar
In the second iteration, the beta\_formalism term is set to True.  Unsuprisingly, using the beta correction gives better results as this is how the spectra were simulated.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}24}\PYG{p}{,}
                   \PYG{n}{beta\PYGZus{}formalism} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,}\PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}print (result.params.pretty\PYGZus{}print())}

\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=965\sphinxpxdimen,height=593\sphinxpxdimen]{{Beta}.png}

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.generate_beta_output_file}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Fit\_DataSet.generate\_beta\_output\_file()}}}}} definition can be used to tabulate a value of the math:\sphinxtitleref{beta} parameters that are used for all lines and each spectrum. By default this is saved in a file called Beta Summary File.csv.  This plot shows how this math:\sphinxtitleref{beta}  parameter changes with sample and pressure.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}beta\PYGZus{}output\PYGZus{}file}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{beta\PYGZus{}summary\PYGZus{}file} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta Summary File.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{index\PYGZus{}col} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air\PYGZus{}vary}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{==}\PYG{k+kc}{True}\PYG{p}{]}

\PYG{n}{pressures} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{l+m+mi}{10}\PYG{p}{,} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{l+m+mi}{250}\PYG{p}{,} \PYG{l+m+mi}{500}\PYG{p}{]}
\PYG{n}{air\PYGZus{}beta} \PYG{o}{=} \PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}1}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}3}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,}
         \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}4}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}5}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{]}
\PYG{n}{mix\PYGZus{}beta} \PYG{o}{=} \PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}6}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}7}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}8}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,}
         \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}9}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}10}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{]}
\PYG{n}{CO2\PYGZus{}beta} \PYG{o}{=} \PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}11}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}12}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}13}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,}
         \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}14}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{p}{[}\PYG{n}{beta\PYGZus{}summary\PYGZus{}file}\PYG{o}{.}\PYG{n}{index} \PYG{o}{==} \PYG{l+m+mi}{62}\PYG{p}{]}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Beta\PYGZus{}15}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{]}

\PYG{n}{plt}\PYG{o}{.}\PYG{n}{semilogx}\PYG{p}{(}\PYG{n}{pressures}\PYG{p}{,} \PYG{n}{air\PYGZus{}beta}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{semilogx}\PYG{p}{(}\PYG{n}{pressures}\PYG{p}{,} \PYG{n}{mix\PYGZus{}beta}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mixture}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{semilogx}\PYG{p}{(}\PYG{n}{pressures}\PYG{p}{,} \PYG{n}{CO2\PYGZus{}beta}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{o}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{label} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO\PYGZdl{}\PYGZus{}}\PYG{l+s+si}{\PYGZob{}2\PYGZcb{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressure (Torr)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZdl{}}\PYG{l+s+se}{\PYGZbs{}\PYGZbs{}}\PYG{l+s+s1}{beta\PYGZdl{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{legend}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=396\sphinxpxdimen,height=266\sphinxpxdimen]{{beta_values}.png}

\sphinxstepscope


\subsubsection{Incorporating CIA into fits}
\label{\detokenize{Incorporating CIA into fits:incorporating-cia-into-fits}}\label{\detokenize{Incorporating CIA into fits::doc}}
\sphinxAtStartPar
Provided in the MATS v2 release are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
Collision induced absorption (CIA) is a broadband effect stemming from intermolecular interactions leading to a collisionally\sphinxhyphen{}induced dipole. Traditionally, CIA is treated as the remaining absorption after modeling baseline and resonant absorption. To accomodate this approach, MATS allows for the input of a static CIA for each spectrum to account for in the fitting solution. \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/Generic\_CIA}{The example} below takes synthetic spectra that include CIA, evalutates using the initial line list, generates a CIA for each spectrum, and then fits the line shape parameters.


\paragraph{Generate Spectrum Objects}
\label{\detokenize{Incorporating CIA into fits:generate-spectrum-objects}}
\sphinxAtStartPar
Module import follows from the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples with additional details on how to simulate spectra found in {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} and the source documentation.

\sphinxAtStartPar
The provided spectra were simulated in MATS offline using the \sphinxhref{https://doi.org/10.1016/j.jqsrt.2016.03.037}{Drouin 2017 Multispectrum analysis of the Oxygen A\sphinxhyphen{}Band} line list and also includes the CIA reported in that work. The code below generates {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} objects from these spectra.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}26}\PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}
\PYG{n}{etalon} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
\PYG{n}{sample\PYGZus{}concentration} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}  \PYG{l+m+mf}{0.209}\PYG{p}{\PYGZcb{}}



\PYG{n}{freq\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wavenumber + Noise (cm\PYGZhy{}1)}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Alpha + LM + CIA}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressure (Torr)}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{temp\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Temperature (C)}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{2}



\PYG{n}{spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{100 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}concentration}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                         \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temp\PYGZus{}column}\PYG{p}{,}\PYG{c+c1}{\PYGZsh{}segment\PYGZus{}column = segment\PYGZus{}column,}
                         \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}2} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{200 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}concentration}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                         \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temp\PYGZus{}column}\PYG{p}{,}\PYG{c+c1}{\PYGZsh{}segment\PYGZus{}column = segment\PYGZus{}column,}
                         \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}3} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{400 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}concentration}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                         \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temp\PYGZus{}column}\PYG{p}{,}\PYG{c+c1}{\PYGZsh{}segment\PYGZus{}column = segment\PYGZus{}column,}
                         \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}4} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{700 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}concentration}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                         \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temp\PYGZus{}column}\PYG{p}{,}\PYG{c+c1}{\PYGZsh{}segment\PYGZus{}column = segment\PYGZus{}column,}
                         \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}5} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{1000 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}concentration}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                         \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temp\PYGZus{}column}\PYG{p}{,}\PYG{c+c1}{\PYGZsh{}segment\PYGZus{}column = segment\PYGZus{}column,}
                         \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}6} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{2000 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}concentration}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                         \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temp\PYGZus{}column}\PYG{p}{,}\PYG{c+c1}{\PYGZsh{}segment\PYGZus{}column = segment\PYGZus{}column,}
                         \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}7} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Spectrum}\PYG{p}{(}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{3000 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}concentration}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{diluent} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                         \PYG{n}{etalons} \PYG{o}{=} \PYG{n}{etalon}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}freq} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{frequency\PYGZus{}column} \PYG{o}{=} \PYG{n}{freq\PYGZus{}column}\PYG{p}{,}
                         \PYG{n}{input\PYGZus{}tau} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{n}{tau\PYGZus{}column}\PYG{p}{,} \PYG{n}{tau\PYGZus{}stats\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                         \PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{n}{pressure\PYGZus{}column}\PYG{p}{,} \PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{n}{temp\PYGZus{}column}\PYG{p}{,}\PYG{c+c1}{\PYGZsh{}segment\PYGZus{}column = segment\PYGZus{}column,}
                         \PYG{n}{nominal\PYGZus{}temperature} \PYG{o}{=} \PYG{l+m+mi}{296}\PYG{p}{,} \PYG{n}{x\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}order} \PYG{o}{=} \PYG{n}{order\PYGZus{}baseline\PYGZus{}fit}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Generate Dataset}
\label{\detokenize{Incorporating CIA into fits:generate-dataset}}
\sphinxAtStartPar
A line list generated from the Drouin 2017 Oxygen A\sphinxhyphen{}Band data combined with HITRAN 2016 values is used as the initial guess.  A similar line list was used in the generation of the spectra, with truncation errors being the most like cause of discrepencies.  Spectra are combined to form a {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} object.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2\PYGZus{}ABand\PYGZus{}Drouin\PYGZus{}2017\PYGZus{}linelist}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}

\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{,} \PYG{n}{spec\PYGZus{}5}\PYG{p}{,} \PYG{n}{spec\PYGZus{}6}\PYG{p}{,} \PYG{n}{spec\PYGZus{}7}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CIA Spectra Study}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}Generate Baseline Parameter list based on number of etalons in spectra definitions and baseline order}
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Evaluation}
\label{\detokenize{Incorporating CIA into fits:evaluation}}
\sphinxAtStartPar
Inital fit doesn’t float any parameters.  The residual plots shown are only for the lowest and highest pressures.  In the high pressure residuals, you can see that the unaccounted for CIA is dominating the residuals and wouldn’t allow for optimization of line shape parameters without treatment.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,}
                                \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDVP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                   \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,}
                                   \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                    \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                    \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}


\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                   \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}


\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                    \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}


\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}wavenumbers} \PYG{o}{=} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{xtol} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}7}\PYG{p}{,} \PYG{n}{maxfev} \PYG{o}{=} \PYG{l+m+mi}{2000}\PYG{p}{,} \PYG{n}{ftol} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}7}\PYG{p}{)}

\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=1025\sphinxpxdimen,height=653\sphinxpxdimen]{{CIA_low}.png}

\noindent\sphinxincludegraphics[width=1018\sphinxpxdimen,height=653\sphinxpxdimen]{{CIA_high}.png}

\sphinxAtStartPar
A simple function is defined to  treat the CIA as the smooth residuals.  To do this we will use a filter to smooth over any resonant absorbance and then define the CIA as the negative of what is leftover.  This is done for all spectra making up the dataset.  Note that this defininition includes the {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum.set_cia}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum.set\_cia()}}}}} function to assign this array to a given {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} instance.  Again the plots below show this process for the lowest and highest pressure spectra, where the blue lines are the residuals and the orange lines are the smoothed residuals that will define the magnitude of the CIA.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{scipy}\PYG{n+nn}{.}\PYG{n+nn}{signal} \PYG{k+kn}{import} \PYG{n}{savgol\PYGZus{}filter}

\PYG{k}{def} \PYG{n+nf}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec}\PYG{p}{)}\PYG{p}{:}

 \PYG{n}{waves} \PYG{o}{=} \PYG{n}{spec}\PYG{o}{.}\PYG{n}{wavenumber}
 \PYG{n}{resds} \PYG{o}{=} \PYG{n}{spec}\PYG{o}{.}\PYG{n}{residuals}

 \PYG{n}{CIA} \PYG{o}{=} \PYG{n}{savgol\PYGZus{}filter}\PYG{p}{(}\PYG{n}{resds}\PYG{p}{,} \PYG{l+m+mi}{501}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{} Savitsky Golay Filte that uses a 501 pt sliding window with a first order polynomial to smooth the data.  These settings were arbitrary to smooth out the data.}
 \PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{waves}\PYG{p}{,} \PYG{n}{resds}\PYG{p}{)}
 \PYG{n}{plt}\PYG{o}{.}\PYG{n}{plot}\PYG{p}{(}\PYG{n}{waves}\PYG{p}{,} \PYG{n}{CIA}\PYG{p}{)}
 \PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wavenumbers (\PYGZdl{}cm\PYGZca{}}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZhy{}1\PYGZcb{}\PYGZdl{})}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
 \PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Residuals (ppm/cm)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
 \PYG{n}{plt}\PYG{o}{.}\PYG{n}{title}\PYG{p}{(} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Spectrum }\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{+} \PYG{n+nb}{str}\PYG{p}{(}\PYG{n}{spec}\PYG{o}{.}\PYG{n}{spectrum\PYGZus{}number}\PYG{p}{)}\PYG{p}{)}
 \PYG{n}{plt}\PYG{o}{.}\PYG{n}{show}\PYG{p}{(}\PYG{p}{)}
 \PYG{n}{spec}\PYG{o}{.}\PYG{n}{set\PYGZus{}cia}\PYG{p}{(}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{o}{*}\PYG{n}{CIA}\PYG{p}{)}
\PYG{n}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec\PYGZus{}1}\PYG{p}{)}
\PYG{n}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec\PYGZus{}2}\PYG{p}{)}
\PYG{n}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec\PYGZus{}3}\PYG{p}{)}
\PYG{n}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec\PYGZus{}4}\PYG{p}{)}
\PYG{n}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec\PYGZus{}5}\PYG{p}{)}
\PYG{n}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec\PYGZus{}6}\PYG{p}{)}
\PYG{n}{ad\PYGZus{}hoc\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{spec\PYGZus{}7}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=461\sphinxpxdimen,height=327\sphinxpxdimen]{{Resds_smooth_low}.png}

\noindent\sphinxincludegraphics[width=443\sphinxpxdimen,height=327\sphinxpxdimen]{{Resds_smooth_high}.png}

\sphinxAtStartPar
The spectra are evaluated again and we see that the residuals have improved.  To further improve fits, I would optimize line parameters in small wavenumber sections using the baseline parameter to account for remaining CIA.  These line shape parameters  could then be used in evaluation of the full band (like above) and then the CIA column could be ammended based on those results.  Iterating on these steps should allow for optimization of both the line shape parameters and the CIA.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                    \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}wavenumbers} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{xtol} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}7}\PYG{p}{,} \PYG{n}{maxfev} \PYG{o}{=} \PYG{l+m+mi}{2000}\PYG{p}{,} \PYG{n}{ftol} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}7}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=1025\sphinxpxdimen,height=653\sphinxpxdimen]{{CIA_low_2}.png}

\noindent\sphinxincludegraphics[width=1018\sphinxpxdimen,height=653\sphinxpxdimen]{{CIA_high_2}.png}

\sphinxstepscope


\subsubsection{Including instrument line shapes in simulation and fits}
\label{\detokenize{Including instrument line shapes in simulation and fits:including-instrument-line-shapes-in-simulation-and-fits}}\label{\detokenize{Including instrument line shapes in simulation and fits::doc}}
\sphinxAtStartPar
Provided in the MATS v2 release are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
MATS has added the capability to simulate and fit spectra using an instrument lineshape.  MATS automatically loads the slit functions defined in HAPI, but could use any slit function with the form slit\_function(x, resolution) or slit\_function(x, {[}resolutions{]}).  MATS uses a slight variation of the HAPI convolveSpectrumSame, where the {\color{red}\bfseries{}arange\_} function was redefined to address an integer/float bug in the underlying np.linspace call.  This implementation currently assumes an equal wavenumber spacing based on the application of the slit function calculation.

\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/Instrument\_Lineshape}{The example below simulates the same spectrum and applies the different HAPI instrument line shapes.}   The simulated molefraction is perturbted and then floated during a fit.


\paragraph{Simulate Spectra with ILS}
\label{\detokenize{Including instrument line shapes in simulation and fits:simulate-spectra-with-ils}}
\sphinxAtStartPar
Module import follows from the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples with additional details on how to simulate spectra found in {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} and the source documentation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{k+kn}{import} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{hapi} \PYG{k}{as} \PYG{n+nn}{hapi}

\PYG{c+c1}{\PYGZsh{} Shared Simulation and Fit Parameters}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}24}\PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{segment\PYGZus{}column} \PYG{o}{=} \PYG{k+kc}{None}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{etalon} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}
\PYG{n}{SNR} \PYG{o}{=} \PYG{l+m+mi}{5000}
\PYG{n}{resolution} \PYG{o}{=} \PYG{l+m+mf}{0.25}
\PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}
\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{6320}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mi}{6340}
\PYG{n}{wave\PYGZus{}step} \PYG{o}{=} \PYG{l+m+mf}{0.01}
\end{sphinxVerbatim}

\sphinxAtStartPar
When fitting spectra, the simulated spectral range is the minimum and maximum wavenumbers in your dataset +/\sphinxhyphen{} an extra spectral window defined in this example by the wave\_range parameter.  This is used to truncate the input line list when generating the Parameter linelist used in fitting to increase the fit speed.  However, this trunctation doesn’t occur when simulating spectra, which can lead to far\sphinxhyphen{}wing effects as the default is to calculate each transition to +/\sphinxhyphen{} 25 half\sphinxhyphen{}widths from the line center.  For this example, we are truncating input line list used for the simulation to match the fit simulation window.  Alternatively, the wave\_range parameter could be increased substantially to include these far\sphinxhyphen{}wing impacts.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CO2\PYGZus{}30012}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{\PYGZlt{}}\PYG{o}{=} \PYG{n}{wave\PYGZus{}max} \PYG{o}{+} \PYG{n}{wave\PYGZus{}range}\PYG{p}{)} \PYG{o}{\PYGZam{}} \PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{\PYGZgt{}}\PYG{o}{=} \PYG{n}{wave\PYGZus{}min} \PYG{o}{\PYGZhy{}} \PYG{n}{wave\PYGZus{}step}\PYG{p}{)}\PYG{p}{]}
\end{sphinxVerbatim}

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.spectrum.Spectrum}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Spectrum}}}}} object definitions below show how to use the ILS\_function, ILS\_resolution, and ILS\_wing (defines the length of the ILS slit function to use in the convolution, such that the length is +/\sphinxhyphen{} ILS\_wing / spectrum stepsize) to define the ILS function. The outputs below show the generated spectrum for each ILS\_function

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}rectangular} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                             \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85}\PYG{p}{,}
                             \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{150}\PYG{p}{,}
                              \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Rectangular}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                            \PYG{n}{molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}} \PYG{p}{,} \PYG{n}{ILS\PYGZus{}function} \PYG{o}{=} \PYG{n}{SLIT\PYGZus{}RECTANGULAR}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}resolution} \PYG{o}{=} \PYG{n}{resolution}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}wing} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=395\sphinxpxdimen,height=267\sphinxpxdimen]{{rectangular}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}triangular}\PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                             \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85}\PYG{p}{,}
                             \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{150}\PYG{p}{,}
                              \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Triangular}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                            \PYG{n}{molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}function} \PYG{o}{=} \PYG{n}{SLIT\PYGZus{}TRIANGULAR}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}resolution} \PYG{o}{=} \PYG{n}{resolution}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}wing} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=395\sphinxpxdimen,height=267\sphinxpxdimen]{{triangular}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}gaussian} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                             \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85}\PYG{p}{,}
                             \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{150}\PYG{p}{,}
                              \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Gaussian}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                            \PYG{n}{molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}function} \PYG{o}{=} \PYG{n}{SLIT\PYGZus{}GAUSSIAN}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}resolution} \PYG{o}{=} \PYG{n}{resolution}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}wing} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=395\sphinxpxdimen,height=267\sphinxpxdimen]{{gaussian}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}dispersion} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                             \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85}\PYG{p}{,}
                             \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{150}\PYG{p}{,}
                              \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Dispersion}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                            \PYG{n}{molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}function} \PYG{o}{=} \PYG{n}{SLIT\PYGZus{}DISPERSION}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}resolution} \PYG{o}{=} \PYG{n}{resolution}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}wing} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=401\sphinxpxdimen,height=267\sphinxpxdimen]{{dispersion}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}cosinus} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                             \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85}\PYG{p}{,}
                             \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{150}\PYG{p}{,}
                              \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Cosinus}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                            \PYG{n}{molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}function} \PYG{o}{=} \PYG{n}{SLIT\PYGZus{}COSINUS}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}resolution} \PYG{o}{=} \PYG{n}{resolution}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}wing} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=401\sphinxpxdimen,height=267\sphinxpxdimen]{{cosinus}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}diffraction} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                             \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85}\PYG{p}{,}
                             \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{150}\PYG{p}{,}
                              \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Diffraction}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                            \PYG{n}{molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}function} \PYG{o}{=} \PYG{n}{SLIT\PYGZus{}DIFFRACTION}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}resolution} \PYG{o}{=} \PYG{n}{resolution}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}wing} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=395\sphinxpxdimen,height=268\sphinxpxdimen]{{diffraction}.png}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}michelson} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                             \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85}\PYG{p}{,}
                             \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{150}\PYG{p}{,}
                              \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Michelson}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                            \PYG{n}{molefraction} \PYG{o}{=}  \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mf}{0.01}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}function} \PYG{o}{=} \PYG{n}{SLIT\PYGZus{}MICHELSON}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}resolution} \PYG{o}{=} \PYG{n}{resolution}\PYG{p}{,} \PYG{n}{ILS\PYGZus{}wing} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=394\sphinxpxdimen,height=267\sphinxpxdimen]{{michelson}.png}


\paragraph{Generate Dataset and Fit Parameters}
\label{\detokenize{Including instrument line shapes in simulation and fits:generate-dataset-and-fit-parameters}}
\sphinxAtStartPar
The dataset and fit parameters are generated with instances of the {\hyperref[\detokenize{MATS:MATS.dataset.Dataset}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Dataset}}}}} and

\sphinxAtStartPar
{\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File}}}}} classes. Additionally, the initial CO2 molefraction is perturbed within 2\% of the simulated mole fraction.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[} \PYG{n}{spec\PYGZus{}rectangular}\PYG{p}{,} \PYG{n}{spec\PYGZus{}triangular}\PYG{p}{,} \PYG{n}{spec\PYGZus{}gaussian}\PYG{p}{,} \PYG{n}{spec\PYGZus{}dispersion}\PYG{p}{,} \PYG{n}{spec\PYGZus{}cosinus}\PYG{p}{,}
                            \PYG{n}{spec\PYGZus{}diffraction}\PYG{p}{,} \PYG{n}{spec\PYGZus{}michelson}\PYG{p}{]}\PYG{p}{,}  \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{ILS Study}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{)}
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molefraction\PYGZus{}CO2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molefraction\PYGZus{}CO2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{1} \PYG{o}{+} \PYG{n}{np}\PYG{o}{.}\PYG{n}{random}\PYG{o}{.}\PYG{n}{normal}\PYG{p}{(}\PYG{n}{loc} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{scale} \PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,} \PYG{n}{size} \PYG{o}{=} \PYG{n+nb}{len}\PYG{p}{(}\PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molefraction\PYGZus{}CO2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}\PYG{p}{)}\PYG{o}{*}\PYG{p}{(}\PYG{l+m+mi}{2}\PYG{o}{/}\PYG{l+m+mi}{100}\PYG{p}{)}\PYG{p}{)} \PYG{c+c1}{\PYGZsh{}Adjust the mole fraction of each sample by a mole fraction of random value normally distributed 2\PYGZpc{}}




\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,}
                                        \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDNGP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                           \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,}
                                           \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                            \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                            \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}


\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}gamma2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}delta2} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}n\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}ILS\PYGZus{}res} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
In the {\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.generate_fit_baseline_linelist}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File.generate\_fit\_baseline\_linelist()}}}}} function, in addition to the normal baseline parameters the ILS resolution parameters are also fittable parameters.  Additionally, this is coded in such a way that each ILS\_function has a unique resolution parameter.  This allows for different spectra to have different ILS functions and the baseline linelist table adequately accounts for this. Subset of a {\hyperref[\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.generate_fit_baseline_linelist}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Generate\_FitParam\_File.generate\_fit\_baseline\_linelist()}}}}} output highlights this.  While floating the resolution parameters is possible, it should be done with caution as all other parameters are dependent on this value, so correlation and poor fits are likely.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Spectrum} \PYG{n}{Number}    \PYG{n}{SLIT\PYGZus{}RECTANGULAR\PYGZus{}res\PYGZus{}0}    \PYG{n}{SLIT\PYGZus{}RECTANGULAR\PYGZus{}res\PYGZus{}0\PYGZus{}err}    \PYG{n}{SLIT\PYGZus{}RECTANGULAR\PYGZus{}res\PYGZus{}0\PYGZus{}vary}    \PYG{n}{SLIT\PYGZus{}MICHELSON\PYGZus{}res\PYGZus{}0}
\PYG{l+m+mi}{1}                      \PYG{l+m+mf}{0.25}                                          \PYG{l+m+mi}{0}                                                         \PYG{k+kc}{False}                                                  \PYG{l+m+mi}{0}
\PYG{o}{.} \PYG{o}{.} \PYG{o}{.}
\PYG{l+m+mi}{7}                      \PYG{l+m+mi}{0}                                             \PYG{l+m+mi}{0}                                                         \PYG{k+kc}{False}                                                  \PYG{l+m+mf}{0.25}
\end{sphinxVerbatim}


\paragraph{Fit Data}
\label{\detokenize{Including instrument line shapes in simulation and fits:fit-data}}
\sphinxAtStartPar
In this example only the sample concentration is allowed to float, which is a spectrum dependent parameter.  At this SNR all of the results yield fit mole fractions very similar to the simulated value.  The plot below shows the relative Bias in the fit mole fraction with error bars equal to the relative fit uncertainty for each ILS.  Note that this is only one iteration and not generalizable.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{param} \PYG{o+ow}{in} \PYG{n}{params}\PYG{p}{:}
 \PYG{k}{if} \PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{\PYGZus{}res\PYGZus{}}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{)} \PYG{o+ow}{and} \PYG{p}{(}\PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{vary} \PYG{o}{==} \PYG{k+kc}{True}\PYG{p}{)}\PYG{p}{:}
     \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n+nb}{min} \PYG{o}{=} \PYG{l+m+mf}{0.01}\PYG{p}{)}

\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{)}
\PYG{n+nb}{print} \PYG{p}{(}\PYG{n}{result}\PYG{o}{.}\PYG{n}{params}\PYG{o}{.}\PYG{n}{pretty\PYGZus{}print}\PYG{p}{(}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=630\sphinxpxdimen,height=317\sphinxpxdimen]{{ILS_Function_mole_fraction}.png}

\sphinxstepscope


\subsubsection{Fitting and Simulating isotopes and molecules not in HITRAN}
\label{\detokenize{Fitting and Simulating isotopes and molecules not in HITRAN:fitting-and-simulating-isotopes-and-molecules-not-in-hitran}}\label{\detokenize{Fitting and Simulating isotopes and molecules not in HITRAN::doc}}
\sphinxAtStartPar
Provided in the MATS v2 release are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
HITRAN’s focus is on atmospherically relevent molecules.  This was the original focus of MATS, however recent feature development has allowed for adaption of to include non\sphinxhyphen{}HITRAN molecules.

\sphinxAtStartPar
The example shows how to supplement the HITRAN isotope list and then use that in simulations and fits. This example uses Mercury as an examaple case \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/nonHITRAN\_Molecules}{This example uses Mercury as an example case}.


\paragraph{Append to the HITRAN isotope list to include desired isotopes}
\label{\detokenize{Fitting and Simulating isotopes and molecules not in HITRAN:append-to-the-hitran-isotope-list-to-include-desired-isotopes}}
\sphinxAtStartPar
Module import follows from the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples.

\sphinxAtStartPar
The {\hyperref[\detokenize{MATS:MATS.utilities.add_to_HITRANstyle_isotope_list}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{add\_to\_HITRANstyle\_isotope\_list()}}}}} function allows you to ammend a given isotope list to include additional molecules and/or isotopes using the HITRAN\sphinxhyphen{}style molcule\_id, local\_isotope id, and global isotope id parameterization.  This function has some checks to verify expected behavior
\sphinxhyphen{} if the molecule id and isotope id are in the initial isotope line list, it will return ‘Already entry with that molec\_id and local\_iso\_id.  This result will write over that entry.
\sphinxhyphen{} if the molecule id is in the initial isotope list it will return ‘This is being added as a new isotope of molecule name.  Proceed if that was the intention.
\sphinxhyphen{} if the global isotope id is already in the isotope list it will return, ‘There is another entry with this global isotope id.  Consider changing the value for consistency’

\sphinxAtStartPar
Mercury line shape parameters are not included in the HITRAN database.  The code below uses {\hyperref[\detokenize{MATS:MATS.utilities.add_to_HITRANstyle_isotope_list}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{add\_to\_HITRANstyle\_isotope\_list()}}}}} function to add 10 Mercury isotopes to the HITRAN isotope list.  This is saved in a new isotope list called HITRAN\_Hg\_isolist that can be used in simulating and fitting spectra. The \sphinxhref{https://hitran.org/docs/iso-meta/}{HITRAN isotope list} provides the HITRAN molecule and global ids that are already in use.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{hapi} \PYG{k+kn}{import} \PYG{n}{ISO}
\PYG{k+kn}{from} \PYG{n+nn}{MATS} \PYG{k+kn}{import} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}

\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{ISO}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{200}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{196}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.0015}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=} \PYG{l+m+mf}{195.96581}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{2}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{201}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{198}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.1004}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{197.96674}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{3}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{202}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{199A}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.1694}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{198.96825}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{4}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{203}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{199B}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.1694}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{198.96825}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{5}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{204}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{200}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.2314}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{199.96825}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{6}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{205}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{201a}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.1317}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{200.97028}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{7}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{206}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{201b}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.1317}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{200.97028}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{8}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{207}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{201c}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.1317}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{200.97028}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{9}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{208}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{202}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.2974}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{201.97062}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist} \PYG{o}{=} \PYG{n}{add\PYGZus{}to\PYGZus{}HITRANstyle\PYGZus{}isotope\PYGZus{}list}\PYG{p}{(}\PYG{n}{input\PYGZus{}isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{molec\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{local\PYGZus{}iso\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{10}\PYG{p}{,}
                                                 \PYG{n}{global\PYGZus{}isotope\PYGZus{}id} \PYG{o}{=} \PYG{l+m+mi}{209}\PYG{p}{,} \PYG{n}{iso\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{204}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                 \PYG{n}{abundance} \PYG{o}{=} \PYG{l+m+mf}{0.0682}\PYG{p}{,} \PYG{n}{mass} \PYG{o}{=}\PYG{l+m+mf}{203.97347}\PYG{p}{,} \PYG{n}{mol\PYGZus{}name} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Generate Spectra}
\label{\detokenize{Fitting and Simulating isotopes and molecules not in HITRAN:generate-spectra}}
\sphinxAtStartPar
The simulate spectrum function is used to generate spectra (similar to in the {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} example), but the isotope\_list variable is set to use the isotope list that was just generated opposed to using the default HITRAN isotope list defined in HAPI.

\sphinxAtStartPar
The initial parameter line list should have molecule and local isotope ids that correspond to the new isotope list.  The mole fraction definition should also use the appropriate mole fraction id.  The same would be true for the abundance\_ratio\_MI term if the sample was not at natural abundance.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Hg\PYGZus{}Linelist.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}Generic Fit Parameters}
\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}17} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{tau\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Absoprtion Coefficient (ppm/cm)}\PYG{l+s+s1}{\PYGZsq{}} \PYG{c+c1}{\PYGZsh{} Mean tau/us}
\PYG{n}{freq\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Frequency (cm\PYGZhy{}1)}\PYG{l+s+s1}{\PYGZsq{}} \PYG{c+c1}{\PYGZsh{} Total Frequency /MHz}
\PYG{n}{pressure\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Pressure (Torr)}\PYG{l+s+s1}{\PYGZsq{}}
\PYG{n}{temperature\PYGZus{}column} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Temperature}\PYG{l+s+s1}{\PYGZsq{}}

\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{39411}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mf}{39413.5}
\PYG{n}{wave\PYGZus{}step} \PYG{o}{=} \PYG{l+m+mf}{0.001}
\PYG{n}{SNR} \PYG{o}{=} \PYG{l+m+mi}{1000}


\PYG{n}{spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                     \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}} \PYG{l+m+mi}{100} \PYG{p}{:}\PYG{l+m+mf}{1e\PYGZhy{}9}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{SNR} \PYG{o}{=} \PYG{l+m+mi}{1000}\PYG{p}{,} \PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{100 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}2} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}step}\PYG{p}{,}
                     \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{500}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}} \PYG{l+m+mi}{100} \PYG{p}{:}\PYG{l+m+mf}{1e\PYGZhy{}9}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                     \PYG{n}{isotope\PYGZus{}list} \PYG{o}{=} \PYG{n}{HITRAN\PYGZus{}Hg\PYGZus{}isolist}\PYG{p}{,} \PYG{n}{natural\PYGZus{}abundance} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{SNR} \PYG{o}{=} \PYG{l+m+mi}{1000}\PYG{p}{,}\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{500 Torr}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{plot\PYGZus{}wave\PYGZus{}alpha}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}2}\PYG{o}{.}\PYG{n}{plot\PYGZus{}wave\PYGZus{}alpha}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=440\sphinxpxdimen,height=303\sphinxpxdimen]{{Hg_100Torr}.png}

\noindent\sphinxincludegraphics[width=440\sphinxpxdimen,height=303\sphinxpxdimen]{{Hg_500Torr}.png}


\paragraph{Construct Dataset and Generate Fit Parameters}
\label{\detokenize{Fitting and Simulating isotopes and molecules not in HITRAN:construct-dataset-and-generate-fit-parameters}}
\sphinxAtStartPar
The construction of the dataset and the generate fit parameter classes act as normal.  For the generation of the fit parameter line list, the example below shows how the vary\_variable input dictionaries function with multiple isotopes.  Specifically, in the example below the VP is used as the fitting line shape, where the spectra were simulated with the SDVP.  All parameters are constrained across the spectra and the only parameters being floated are the collisional broadening terms for all isotopes.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Mercury Fitting}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}Generate Baseline Parameter list based on number of etalons in spectra definitions and baseline order}
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}





\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{VP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                               \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                               \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                               \PYG{n}{additional\PYGZus{}columns} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{trans\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}nu} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}sw} \PYG{o}{=}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}gamma0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{True}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{True}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}delta0} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}aw} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}as} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}nuVC} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}eta} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{,}
                                                 \PYG{n}{vary\PYGZus{}linemixing} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:}\PYG{p}{\PYGZob{}}\PYG{l+m+mi}{1}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{2}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{3}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{4}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{5}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{7}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{8}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{9}\PYG{p}{:}\PYG{k+kc}{False}\PYG{p}{,} \PYG{l+m+mi}{10}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{n}{vary\PYGZus{}baseline} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{100}\PYG{p}{:} \PYG{k+kc}{False}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{n}{vary\PYGZus{}xshift} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,}
                                   \PYG{n}{vary\PYGZus{}etalon\PYGZus{}amp}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}period}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{,} \PYG{n}{vary\PYGZus{}etalon\PYGZus{}phase}\PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\end{sphinxVerbatim}


\paragraph{Fit Spectra}
\label{\detokenize{Fitting and Simulating isotopes and molecules not in HITRAN:fit-spectra}}
\sphinxAtStartPar
Fitting follows the same strucutre described in {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}}.  The fit results shown below are the result of fitting spectra simulated with the SDVP with the VP and only allowing the collsional broadening terms to float.  The residuals at both pressures show the expected w\sphinxhyphen{}shaped residuals.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{weight\PYGZus{}spectra} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=998\sphinxpxdimen,height=629\sphinxpxdimen]{{HG_MS_VP}.png}

\sphinxAtStartPar
We can compare the VP fit collisional broadening fit results to the SDVP collisional broadening values used in the simulation.  On the secondary y\sphinxhyphen{}axis is a stem plot showing the line intensity for the lines.  This highlights that the lines that deviated the most from the simulated collisional broadening value were weaker and/or closely spaced to other isotope transitions.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Parameter\PYGZus{}Linelist}\PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{read\PYGZus{}csv}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}Linelist.csv}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{index\PYGZus{}col} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{)}
\PYG{n}{simulated\PYGZus{}gamma0\PYGZus{}air} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]}

\PYG{n}{fig}\PYG{p}{,} \PYG{n}{ax1} \PYG{o}{=} \PYG{n}{plt}\PYG{o}{.}\PYG{n}{subplots}\PYG{p}{(}\PYG{n}{constrained\PYGZus{}layout}\PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{figsize}\PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{9}\PYG{p}{,} \PYG{l+m+mf}{4.5}\PYG{p}{]}\PYG{p}{)}
\PYG{n}{ax2} \PYG{o}{=} \PYG{n}{ax1}\PYG{o}{.}\PYG{n}{twinx}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{ax2}\PYG{o}{.}\PYG{n}{stem}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}Linelist}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{,}\PYG{n}{Parameter\PYGZus{}Linelist}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{o}{*}\PYG{n}{Parameter\PYGZus{}Linelist}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw\PYGZus{}scale\PYGZus{}factor}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{k}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{markerfmt} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{None}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{basefmt} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{None}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{ax1}\PYG{o}{.}\PYG{n}{errorbar}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}Linelist}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{,} \PYG{l+m+mi}{100}\PYG{o}{*}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}Linelist}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values} \PYG{o}{\PYGZhy{}} \PYG{n}{simulated\PYGZus{}gamma0\PYGZus{}air}\PYG{p}{)}\PYG{o}{/} \PYG{n}{simulated\PYGZus{}gamma0\PYGZus{}air}\PYG{p}{,}
          \PYG{n}{yerr} \PYG{o}{=} \PYG{l+m+mi}{100}\PYG{o}{*}\PYG{p}{(}\PYG{n}{Parameter\PYGZus{}Linelist}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air\PYGZus{}err}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{)}\PYG{o}{/} \PYG{n}{Parameter\PYGZus{}Linelist}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}\PYG{p}{,}\PYG{n}{fmt} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{o}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}


\PYG{n}{ax2}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylim}\PYG{p}{(}\PYG{l+m+mf}{1e\PYGZhy{}17}\PYG{p}{,} \PYG{l+m+mf}{1e\PYGZhy{}14}\PYG{p}{)}
\PYG{n}{ax1}\PYG{o}{.}\PYG{n}{ticklabel\PYGZus{}format}\PYG{p}{(}\PYG{n}{axis} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{x}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{useOffset} \PYG{o}{=} \PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{ax2}\PYG{o}{.}\PYG{n}{set\PYGZus{}yscale}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{log}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{ax1}\PYG{o}{.}\PYG{n}{set\PYGZus{}xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wavenumber (cm\PYGZdl{}\PYGZca{}}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZhy{}1\PYGZcb{}\PYGZdl{})}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{ax2}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Line Intensity (cm molec.\PYGZdl{}\PYGZca{}}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZhy{}1\PYGZcb{}\PYGZdl{})}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{ax1}\PYG{o}{.}\PYG{n}{set\PYGZus{}ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Relative Differnece between }\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s1}{ simulated SDVP \PYGZdl{}}\PYG{l+s+se}{\PYGZbs{}\PYGZbs{}}\PYG{l+s+s1}{gamma\PYGZus{}}\PYG{l+s+si}{\PYGZob{}0\PYGZcb{}}\PYG{l+s+s1}{\PYGZdl{} and fit VP (}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{)}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=656\sphinxpxdimen,height=387\sphinxpxdimen]{{Hg_Results}.png}

\sphinxstepscope


\subsubsection{Correcting the Ideal Gas Law with Compressability Factors}
\label{\detokenize{Correcting the Ideal Gas Law with Compressability Factor:correcting-the-ideal-gas-law-with-compressability-factors}}\label{\detokenize{Correcting the Ideal Gas Law with Compressability Factor::doc}}
\sphinxAtStartPar
Provided in the MATS are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
MATS has added the ability to use the compressability factor to correct the ideal gas law.  This uses the NIST RefProp \textless{}\sphinxurl{https://www.nist.gov/srd/refprop}\textgreater{}\textasciigrave{}\_ offline to generate an input file that is the compressability factor for the desired sample composition on a grid of pressures (MPa) and temperatures (K).  This correction is used if a compressability\_file is provided in the Spectrum object generation.  The rest of the script works as normal for experimental or synthetic spectra.

\sphinxAtStartPar
The compressability factor acts on the molecular density, such that
\begin{equation*}
\begin{split}n_{a} = \chi n_{i}(T, p)/Z(T, p), where n_{a} is the actual number density, n_{i} is the ideal-gas number density, and Z is the compressability factor of the mixture.\end{split}
\end{equation*}
\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/Compressability\_Factor}{The example below is a used example that follows the script available at}


\paragraph{Simulate Spectra with Compressability Factor}
\label{\detokenize{Correcting the Ideal Gas Law with Compressability Factor:simulate-spectra-with-compressability-factor}}
\sphinxAtStartPar
Module import follows from the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples with additional details on how to simulate spectra found in {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} and the source documentation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2\PYGZus{}ABand\PYGZus{}Drouin\PYGZus{}2017\PYGZus{}linelist}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}

\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}24} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{sample\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.0025142}\PYG{p}{\PYGZcb{}}
\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{12950} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mi}{13200} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{l+m+mf}{0.005} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wavenumbers} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{l+m+mi}{13158}\PYG{p}{,} \PYG{l+m+mf}{0.02}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The \sphinxcode{\sphinxupquote{Spectrum}} object definitions follows as normal with the addition of the compressability\_file being set to a file that contains the compressability factor as a function of temperature and pressure.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}1} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min}\PYG{o}{=}\PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{o}{=} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space}\PYG{o}{=}\PYG{n}{wave\PYGZus{}space}\PYG{p}{,}
                                         \PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{975}\PYG{p}{,} \PYG{n}{SNR} \PYG{o}{=} \PYG{k+kc}{None}\PYG{p}{,}
                                         \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,}
                                         \PYG{n}{compressability\PYGZus{}file} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{0.25}\PYG{l+s+s1}{\PYGZpc{}}\PYG{l+s+s1}{O2inN2\PYGZus{}297K}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}


\PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{plot\PYGZus{}wave\PYGZus{}alpha}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The rest of the code follows as is described in the {\hyperref[\detokenize{Fitting Experimental Spectra:fitting-experimental-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Experimental Spectra}}}} and {\hyperref[\detokenize{Fitting Synthetic Spectra:fitting-synthetic-spectra}]{\sphinxcrossref{\DUrole{std,std-ref}{Fitting Synthetic Spectra}}}} examples.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Compressability Factor Test}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{)}
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{FITPARAMS} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Generate\PYGZus{}FitParam\PYGZus{}File}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{BASE\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{lineprofile} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SDVP}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{linemixing} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                           \PYG{n}{fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{,} \PYG{n}{threshold\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{IntensityThreshold}\PYG{p}{,} \PYG{n}{sim\PYGZus{}window} \PYG{o}{=} \PYG{n}{wave\PYGZus{}range}\PYG{p}{,}
                                                           \PYG{n}{nu\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{sw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{gamma0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{delta0\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                \PYG{n}{aw\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{as\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,}
                                                                \PYG{n}{nuVC\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{eta\PYGZus{}constrain} \PYG{o}{=}\PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{linemixing\PYGZus{}constrain} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}param\PYGZus{}linelist\PYGZus{}from\PYGZus{}linelist}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{FITPARAMS}\PYG{o}{.}\PYG{n}{generate\PYGZus{}fit\PYGZus{}baseline\PYGZus{}linelist}\PYG{p}{(}\PYG{p}{)}

\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
In the infinite SNR case, the residuals when spectrum are simulated and evaluated using the compressability factor the residuals are the numerical errors.

\noindent\sphinxincludegraphics[width=1375\sphinxpxdimen,height=887\sphinxpxdimen]{{Compressability_True}.png}

\sphinxAtStartPar
if the the compressability factor for a spectrum simulated using the compressability factor is then evaluated with the molecular density calculated using the ideal gas law, then a molecular density error will be observed in the residuals.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{compressability\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{None}
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=1444\sphinxpxdimen,height=887\sphinxpxdimen]{{Compressability_False}.png}

\sphinxstepscope


\subsubsection{HITRAN Line\sphinxhyphen{}by\sphinxhyphen{}Line Static Molecules}
\label{\detokenize{HITRAN Line-by-Line Static Molecules:hitran-line-by-line-static-molecules}}\label{\detokenize{HITRAN Line-by-Line Static Molecules::doc}}
\sphinxAtStartPar
Provided in the MATS are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
MATS has added the capability to simulate and/or fit spectra of molecules that only available in HITRAN as static molecules (not in the typical HITRANOnline) format.

\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/HITRAN\_static\_LBL\_molecules}{The example below ptovides an example of how to simulate spectra using static HITRAN LBL data and is based on the code provided in the HITRAN static LBL molecule example.}


\paragraph{Generating a MATS linelist from a .par file}
\label{\detokenize{HITRAN Line-by-Line Static Molecules:generating-a-mats-linelist-from-a-par-file}}
\sphinxAtStartPar
The .par formatting is provided in Table 1 of the HITRAN 2004 paper.  The code below uses that .par formatting to parse the spectra into the MATS format.

\sphinxAtStartPar
L.S. Rothman, D. Jacquemart, A. Barbe, D. Chris Benner, M. Birk, L.R. Brown, M.R. Carleer, C. Chackerian, K. Chance, L.H. Coudert, V. Dana, V.M. Devi, J.\sphinxhyphen{}M. Flaud, R.R. Gamache, A. Goldman, J.\sphinxhyphen{}M. Hartmann, K.W. Jucks, A.G. Maki, J.\sphinxhyphen{}Y. Mandin, S.T. Massie, J. Orphal, A. Perrin, C.P. Rinsland, M.A.H. Smith, J. Tennyson, R.N. Tolchenov, R.A. Toth, J. Vander Auwera, P. Varanasi, G. Wagner, The HITRAN 2004 molecular spectroscopic database, Journal of Quantitative Spectroscopy and Radiative Transfer, Volume 96, Issue 2, 2005, Pages 139\sphinxhyphen{}204, \sphinxurl{https://doi.org/10.1016/j.jqsrt.2004.10.008}

\sphinxAtStartPar
Users will need to download the .par file provided by HITRAN for desired molecule.

\sphinxAtStartPar
This example uses CF4 and a portion of the HITRAN provided .par file. The entire HITRAN .par file can be used (and has been tested locally), but is above the github file storage recommendation.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{molecule} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CF4}\PYG{l+s+s1}{\PYGZsq{}}

\PYG{n}{filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{42\PYGZus{}hit20\PYGZus{}500\PYGZus{}800.par}\PYG{l+s+s1}{\PYGZsq{}}

\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{pd}\PYG{o}{.}\PYG{n}{DataFrame}\PYG{p}{(}\PYG{p}{)}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{:}\PYG{l+m+mi}{2}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}iso\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{n+nb}{int}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{2}\PYG{p}{:}\PYG{l+m+mi}{3}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{3}\PYG{p}{:}\PYG{l+m+mi}{15}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[}\PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{15}\PYG{p}{:}\PYG{l+m+mi}{25}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{elower}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{45}\PYG{p}{:}\PYG{l+m+mi}{55}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{35}\PYG{p}{:}\PYG{l+m+mi}{40}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{55}\PYG{p}{:}\PYG{l+m+mi}{59}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{59}\PYG{p}{:}\PYG{l+m+mi}{67}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}
\PYG{k}{with} \PYG{n+nb}{open}\PYG{p}{(}\PYG{n}{filename}\PYG{p}{)} \PYG{k}{as} \PYG{n}{file}\PYG{p}{:}
        \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{p}{[} \PYG{n+nb}{float}\PYG{p}{(}\PYG{n}{line}\PYG{p}{[}\PYG{l+m+mi}{40}\PYG{p}{:}\PYG{l+m+mi}{45}\PYG{p}{]}\PYG{p}{)} \PYG{k}{for} \PYG{n}{line} \PYG{o+ow}{in} \PYG{n}{file}\PYG{p}{]}



\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma2\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma2\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{o}{.}\PYG{n}{values}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta2\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta2\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}nuVC\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{eta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{eta\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{0}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}y\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{1} \PYG{c+c1}{\PYGZsh{}used in MATS versions 3 (current development version) and higher}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}y\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]} \PYG{o}{=} \PYG{l+m+mi}{1}\PYG{c+c1}{\PYGZsh{}used in MATS versions 3 (current development version) and higher}

\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{molec\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{local\PYGZus{}iso\PYGZus{}id}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{elower}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta0\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma2\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta2\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}nuVC\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{eta\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}y\PYGZus{}air}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{gamma0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{delta0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta0\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}gamma\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}gamma2\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{SD\PYGZus{}delta\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}delta2\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                                \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nuVC\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}nuVC\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{eta\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{y\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{n\PYGZus{}y\PYGZus{}self}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{]}\PYG{p}{]}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{sort\PYGZus{}values}\PYG{p}{(}\PYG{n}{by} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{inplace} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{c+c1}{\PYGZsh{}PARAM\PYGZus{}LINELIST.to\PYGZus{}csv(molecule + \PYGZsq{}\PYGZus{}ParameterLinelist.csv\PYGZsq{}, index = False) \PYGZsh{}Can save to .csv file, but it is a large file, so you might want to truncate before saving}

\PYG{n}{plt}\PYG{o}{.}\PYG{n}{figure}\PYG{p}{(}\PYG{n}{figsize}\PYG{o}{=}\PYG{p}{(}\PYG{l+m+mi}{8}\PYG{p}{,} \PYG{l+m+mi}{6}\PYG{p}{)}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{stem}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{sw}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}\PYG{p}{,} \PYG{n}{use\PYGZus{}line\PYGZus{}collection} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{,} \PYG{n}{markerfmt} \PYG{o}{=} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{ }\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{yscale}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{log}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{xlabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Wavenumber (cm\PYGZdl{}\PYGZca{}}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZhy{}1\PYGZcb{}\PYGZdl{})}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{plt}\PYG{o}{.}\PYG{n}{ylabel}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Line Intensity }\PYG{l+s+se}{\PYGZbs{}n}\PYG{l+s+s1}{ (cm\PYGZdl{}\PYGZca{}}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZhy{}1\PYGZcb{}\PYGZdl{} / (molecule \PYGZdl{}}\PYG{l+s+s1}{\PYGZbs{}}\PYG{l+s+s1}{cdot\PYGZdl{} cm\PYGZdl{}\PYGZca{}}\PYG{l+s+s1}{\PYGZob{}}\PYG{l+s+s1}{\PYGZhy{}2\PYGZcb{}\PYGZdl{}))}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=818\sphinxpxdimen,height=596\sphinxpxdimen]{{HITRAN_static_LBL}.png}


\paragraph{TIPS Availability}
\label{\detokenize{HITRAN Line-by-Line Static Molecules:tips-availability}}
\sphinxAtStartPar
While these molecules are available in HITRAN, the molecules available in the TIPS\sphinxhyphen{}2011 and TIPS\sphinxhyphen{}2017 partition function libraries differ. For example in hapi TIPS\sphinxhyphen{}2011 has information for CF4, where TIPS\sphinxhyphen{}2017 is an array of zeros. This leads to nan values for the at simulation temperature line intensities. We added the MATS capability to switch from the default TIPS\sphinxhyphen{}2017 to TIPS\sphinxhyphen{}2011, so that this molecule could be appropriately simulated. Below is a summary of molecules, where this might be an issue. For molecules with no TIPS available the TIPS call with through an error and the 296 line intensity will be used.

\sphinxAtStartPar
TIPS 2011 Only

\sphinxAtStartPar
SO3, M = 46, I =1
CF4, M = 42, I = 1
No TIPS data

\sphinxAtStartPar
HNO3, M = 12, I = 2
HF, M = 14, I = 2
HCl, M = 15, I = 3, 4
HBr, M = 16, I = 3, 4
HI, M = 17, I = 2
N2, M = 22, I = 2
COF2, M = 29, I = 2
O, M = 34, I = 1
CH3OH, M = 38, I=1


\paragraph{Simulating in MATS}
\label{\detokenize{HITRAN Line-by-Line Static Molecules:simulating-in-mats}}
\sphinxAtStartPar
Once the MATS linelist file has been generated spectra can be simulated or fit in MATS as normal.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}PARAM\PYGZus{}LINELIST = pd.read\PYGZus{}csv(\PYGZsq{}CF4\PYGZus{}ParameterLinelist.csv\PYGZsq{}) \PYGZsh{} Option to read in so you don\PYGZsq{}t have to generate the file every time}


\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}24} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{1}
\PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{0}\PYG{p}{]} \PYG{c+c1}{\PYGZsh{}polynomial baseline coefficients where the index is equal to the coefficient order}

\PYG{n}{sample\PYGZus{}molefraction} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{42} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}} \PYG{c+c1}{\PYGZsh{}The number here is the molec\PYGZus{}id number can be found https://hitran.org/docs/iso\PYGZhy{}meta/}
\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{500} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mi}{800} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{l+m+mi}{1} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{SNR} \PYG{o}{=} \PYG{k+kc}{None} \PYG{c+c1}{\PYGZsh{}if None, then infinite SNR}
\PYG{n}{temperature} \PYG{o}{=} \PYG{l+m+mf}{22.85} \PYG{c+c1}{\PYGZsh{} celsius}
\PYG{n}{pressure} \PYG{o}{=} \PYG{l+m+mi}{760} \PYG{c+c1}{\PYGZsh{}torr}
\PYG{n}{spectrum\PYGZus{}filename} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Simulate\PYGZus{}CF4}\PYG{l+s+s1}{\PYGZsq{}}

\PYG{n}{spec\PYGZus{}1} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{n}{wave\PYGZus{}space}\PYG{p}{,}
                                                \PYG{n}{SNR} \PYG{o}{=} \PYG{n}{SNR}\PYG{p}{,} \PYG{n}{baseline\PYGZus{}terms} \PYG{o}{=} \PYG{n}{baseline\PYGZus{}terms}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{=} \PYG{n}{temperature}\PYG{p}{,} \PYG{n}{pressure} \PYG{o}{=} \PYG{n}{pressure}\PYG{p}{,}
                                                \PYG{n}{wing\PYGZus{}cutoff} \PYG{o}{=} \PYG{l+m+mi}{5}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}cutoff}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                \PYG{n}{filename} \PYG{o}{=} \PYG{n}{spectrum\PYGZus{}filename}\PYG{p}{,} \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,}\PYG{n}{TIPS} \PYG{o}{=} \PYG{n}{PYTIPS2011}\PYG{p}{)}

\PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{plot\PYGZus{}wave\PYGZus{}alpha}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=609\sphinxpxdimen,height=514\sphinxpxdimen]{{HITRAN_static_LBL_2}.png}

\sphinxstepscope


\subsubsection{Oxygen CIA using theoretical model}
\label{\detokenize{Oxygen CIA using theoretical model:oxygen-cia-using-theoretical-model}}\label{\detokenize{Oxygen CIA using theoretical model::doc}}
\sphinxAtStartPar
Provided in MATS are several examples highlighting MATS capabilities, which can be found in the MATS \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples}{examples folder}.

\sphinxAtStartPar
Karman et al published a theoretical O\_2 \sphinxhyphen{} O\_2 and O\_2 \sphinxhyphen{} N\_2 collision\sphinxhyphen{}induced absorption model based on quantum calculations \sphinxurl{https://doi.org/10.1038/s41557-018-0015-x}. This model is composed of spin\sphinxhyphen{}orbit and exchange mechanisms. In the manuscript Parameterized Model to Approximate Theoretical Collision\sphinxhyphen{}Induced Absorption Band Shapes for O2\sphinxhyphen{}O2 and O2\sphinxhyphen{}N2, Adkins et al present a parameterized representation of the theoretical model reported by Karman et al.

\sphinxAtStartPar
The details of the model can be found in \sphinxurl{https://doi.org/10.1016/j.jqsrt.2023.108732}.

\sphinxAtStartPar
We have included this parameterized model in MATS to allow for use in multi\sphinxhyphen{}spectrum fitting analyses in order to fit monomer and CIA components to the absorption.  \sphinxhref{https://github.com/usnistgov/MATS/tree/master/Examples/O2CIA\_Karman\_etal\_model}{The example below shows a use example for this model.}
\begin{quote}

\sphinxAtStartPar
Most of this example follows from the theoretical spectra generation and fit example. However, this example makes use of the new parameterized CIA model based on the theory published by Karman et al \sphinxurl{https://doi.org/10.1016/j.jqsrt.2023.108732}.
\end{quote}

\sphinxAtStartPar
In the first step we read in the linelist. The one used in the example is that reported in the JQSRT 270 (2021) 107684. However this linelist is for demonstration purposes only, users should refer to that paper for the actual line list.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{from} \PYG{n+nn}{MATS}\PYG{n+nn}{.}\PYG{n+nn}{linelistdata} \PYG{k+kn}{import} \PYG{n}{linelistdata}
\PYG{n}{PARAM\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{linelistdata}\PYG{p}{[}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Singlet\PYGZus{}Delta\PYGZus{}Linelist\PYGZus{}JQSRT\PYGZus{}270\PYGZus{}2021\PYGZus{}107684}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{]}
\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{o}{.}\PYG{n}{sort\PYGZus{}values}\PYG{p}{(}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{nu}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{inplace} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
In this example, we have written a definition that simulates the monomer absorption for a spectrum, simulates a theoretical CIA based on the Karman et al model using the o2\_cia\_karman\_model definition, and then adds the theoretical cia to the simulated monomer absorption in the spectrum alpha data.

\sphinxAtStartPar
The o2\_cia\_karman\_model function requires wavenumber axis, temperature, pressure, and sample composition (O2 and N2 only), in addition to values for the Spin\_orbit magnitude for both O2\sphinxhyphen{}O2 and O2\sphinxhyphen{}N2, exchange magnitude of O2\sphinxhyphen{}O2, temperature dependences for spin orbit (O2\sphinxhyphen{}O2 and O2\sphinxhyphen{}N2, can be constrained later to be equal) and exchange intensities. Additionally, the band being studied a\_band or singlet\_delta needs to be specified.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{wave\PYGZus{}range} \PYG{o}{=} \PYG{l+m+mf}{1.5} \PYG{c+c1}{\PYGZsh{}range outside of experimental x\PYGZhy{}range to simulate}
\PYG{n}{IntensityThreshold} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}30} \PYG{c+c1}{\PYGZsh{}intensities must be above this value to be simulated}
\PYG{n}{Fit\PYGZus{}Intensity} \PYG{o}{=} \PYG{l+m+mf}{1e\PYGZhy{}20} \PYG{c+c1}{\PYGZsh{}intensities must be above this value for the line to be fit}
\PYG{n}{order\PYGZus{}baseline\PYGZus{}fit} \PYG{o}{=} \PYG{l+m+mi}{0}


\PYG{n}{wave\PYGZus{}min} \PYG{o}{=} \PYG{l+m+mi}{7500} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}max} \PYG{o}{=} \PYG{l+m+mi}{8350} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wave\PYGZus{}space} \PYG{o}{=} \PYG{l+m+mf}{0.01} \PYG{c+c1}{\PYGZsh{}cm\PYGZhy{}1}
\PYG{n}{wavenumbers} \PYG{o}{=} \PYG{n}{np}\PYG{o}{.}\PYG{n}{arange}\PYG{p}{(}\PYG{n}{wave\PYGZus{}min}\PYG{p}{,} \PYG{n}{wave\PYGZus{}max}\PYG{p}{,} \PYG{l+m+mf}{0.02}\PYG{p}{)}


\PYG{k}{def} \PYG{n+nf}{sim\PYGZus{}spectra\PYGZus{}with\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{pressure}\PYG{p}{,} \PYG{n}{temperature}\PYG{p}{,} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,} \PYG{n}{filename}\PYG{p}{)}\PYG{p}{:}
        \PYG{n}{Diluent} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{[}\PYG{l+m+mi}{7}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+m+mf}{31.998}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{N2}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{composition}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+m+mi}{1}\PYG{o}{\PYGZhy{}}\PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{[}\PYG{l+m+mi}{7}\PYG{p}{]}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{m}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+m+mf}{28.0134}\PYG{p}{\PYGZcb{}}\PYG{p}{\PYGZcb{}}
        \PYG{n}{spec} \PYG{o}{=}  \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{simulate\PYGZus{}spectrum}\PYG{p}{(}\PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{wavenumbers} \PYG{o}{=} \PYG{n}{wavenumbers}\PYG{p}{,}
                                                 \PYG{n}{temperature} \PYG{o}{=} \PYG{n}{temperature}\PYG{p}{,}  \PYG{n}{pressure} \PYG{o}{=} \PYG{n}{pressure}\PYG{p}{,}
                                                 \PYG{n}{molefraction} \PYG{o}{=} \PYG{n}{sample\PYGZus{}molefraction}\PYG{p}{,} \PYG{n}{Diluent} \PYG{o}{=} \PYG{n}{Diluent}\PYG{p}{,}
                                                 \PYG{n}{filename} \PYG{o}{=} \PYG{n}{filename}\PYG{p}{,} \PYG{n}{wing\PYGZus{}wavenumbers} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,} \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}\PYG{p}{)}
        \PYG{c+c1}{\PYGZsh{} Adjusted from Initial guess derived based HITRAN 2020 reported theoretical CIA}
        \PYG{n}{EXCH\PYGZus{}c}\PYG{p}{,} \PYG{n}{EXCH\PYGZus{}b}\PYG{p}{,} \PYG{n}{EXCH\PYGZus{}a} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mf}{3.63e\PYGZhy{}06}\PYG{p}{,} \PYG{l+m+mf}{0.003}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{]}
        \PYG{n}{SO\PYGZus{}c}\PYG{p}{,} \PYG{n}{SO\PYGZus{}b}\PYG{p}{,} \PYG{n}{SO\PYGZus{}a} \PYG{o}{=}\PYG{p}{[}\PYG{l+m+mf}{1.5e\PYGZhy{}06}\PYG{p}{,} \PYG{l+m+mf}{0.0002}\PYG{p}{,} \PYG{l+m+mi}{1}\PYG{p}{]}
        \PYG{n}{SO\PYGZus{}O2}\PYG{p}{,} \PYG{n}{SO\PYGZus{}N2}\PYG{p}{,} \PYG{n}{EXCH\PYGZus{}O2} \PYG{o}{=} \PYG{p}{[}\PYG{l+m+mi}{40}\PYG{p}{,} \PYG{l+m+mi}{75} \PYG{p}{,}\PYG{l+m+mi}{315}\PYG{p}{]}

        \PYG{n}{CIA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{o2\PYGZus{}cia\PYGZus{}karman\PYGZus{}model}\PYG{p}{(}\PYG{n}{wavenumbers}\PYG{p}{,} \PYG{n}{temperature} \PYG{o}{+} \PYG{l+m+mf}{273.15}\PYG{p}{,} \PYG{n}{pressure}\PYG{o}{/}\PYG{l+m+mi}{760}\PYG{p}{,} \PYG{n}{Diluent}\PYG{p}{,}
                                                        \PYG{n}{SO\PYGZus{}O2}\PYG{p}{,} \PYG{n}{SO\PYGZus{}N2}\PYG{p}{,} \PYG{n}{EXCH\PYGZus{}O2}\PYG{p}{,}
                                                        \PYG{n}{EXCH\PYGZus{}b}\PYG{p}{,} \PYG{n}{EXCH\PYGZus{}c}\PYG{p}{,} \PYG{c+c1}{\PYGZsh{}O2\PYGZhy{}O2}
                                                        \PYG{n}{SO\PYGZus{}b}\PYG{p}{,} \PYG{n}{SO\PYGZus{}c}\PYG{p}{,} \PYG{c+c1}{\PYGZsh{}O2\PYGZhy{}N2}
                                                        \PYG{n}{SO\PYGZus{}b}\PYG{p}{,} \PYG{n}{SO\PYGZus{}c}\PYG{p}{,}
                                                        \PYG{n}{SO\PYGZus{}shift\PYGZus{}O2\PYGZus{}O2} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{SO\PYGZus{}shift\PYGZus{}O2\PYGZus{}N2} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{n}{EXCH\PYGZus{}shift} \PYG{o}{=} \PYG{l+m+mi}{0}\PYG{p}{,}
                                                        \PYG{n}{band} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{singlet\PYGZus{}delta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
        \PYG{n}{spec}\PYG{o}{.}\PYG{n}{alpha} \PYG{o}{+}\PYG{o}{=} \PYG{n}{CIA}
        \PYG{k}{return} \PYG{n}{spec}

\PYG{n}{spec\PYGZus{}1} \PYG{o}{=} \PYG{n}{sim\PYGZus{}spectra\PYGZus{}with\PYGZus{}CIA}\PYG{p}{(}\PYG{l+m+mi}{760}\PYG{p}{,} \PYG{l+m+mf}{22.85}\PYG{p}{,} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.2095}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}296}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}2} \PYG{o}{=} \PYG{n}{sim\PYGZus{}spectra\PYGZus{}with\PYGZus{}CIA}\PYG{p}{(}\PYG{l+m+mi}{760}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.2095}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}273}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}3} \PYG{o}{=} \PYG{n}{sim\PYGZus{}spectra\PYGZus{}with\PYGZus{}CIA}\PYG{p}{(}\PYG{l+m+mi}{760}\PYG{p}{,} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mf}{0.2095}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Air\PYGZus{}323}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}4} \PYG{o}{=} \PYG{n}{sim\PYGZus{}spectra\PYGZus{}with\PYGZus{}CIA}\PYG{p}{(}\PYG{l+m+mi}{760}\PYG{p}{,} \PYG{l+m+mf}{22.85}\PYG{p}{,} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2\PYGZus{}296}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}5} \PYG{o}{=} \PYG{n}{sim\PYGZus{}spectra\PYGZus{}with\PYGZus{}CIA}\PYG{p}{(}\PYG{l+m+mi}{760}\PYG{p}{,} \PYG{l+m+mi}{0}\PYG{p}{,} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2\PYGZus{}273}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{spec\PYGZus{}6} \PYG{o}{=} \PYG{n}{sim\PYGZus{}spectra\PYGZus{}with\PYGZus{}CIA}\PYG{p}{(}\PYG{l+m+mi}{760}\PYG{p}{,} \PYG{l+m+mi}{50}\PYG{p}{,} \PYG{p}{\PYGZob{}}\PYG{l+m+mi}{7} \PYG{p}{:}\PYG{l+m+mi}{1}\PYG{p}{\PYGZcb{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{O2\PYGZus{}323}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}


\PYG{n}{spec\PYGZus{}1}\PYG{o}{.}\PYG{n}{plot\PYGZus{}wave\PYGZus{}alpha}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics[width=606\sphinxpxdimen,height=487\sphinxpxdimen]{{O2_CIA_simualted}.png}


\paragraph{Set\sphinxhyphen{}up for Fitting}
\label{\detokenize{Oxygen CIA using theoretical model:set-up-for-fitting}}
\sphinxAtStartPar
he CIA model is specified in the instantiation of the Dataset class. The CIA\_model is specified through a dictionary specifying both the model and band. Currently, the Karman model is the only model available and the available bands are a\_band and singlet\_delta.

\sphinxAtStartPar
Like the generate\_baseline\_paramlist() definition, the generate\_CIA\_paramlist() generates a dataframe and file that summarizes the parameters based on the MATS preset values from the literature. \sphinxurl{https://doi.org/10.1038/s41557-018-0015-x}, \sphinxurl{https://doi.org/10.1016/j.icarus.2019.02.034}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{}Add all spectrum to a Dataset object}
\PYG{n}{SPECTRA} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Dataset}\PYG{p}{(}\PYG{p}{[}\PYG{n}{spec\PYGZus{}1}\PYG{p}{,} \PYG{n}{spec\PYGZus{}2}\PYG{p}{,} \PYG{n}{spec\PYGZus{}3}\PYG{p}{,} \PYG{n}{spec\PYGZus{}4}\PYG{p}{,} \PYG{n}{spec\PYGZus{}5}\PYG{p}{,} \PYG{n}{spec\PYGZus{}6}\PYG{p}{]}\PYG{p}{,}
                                           \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Experimental Singlet Delta with CIA}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{PARAM\PYGZus{}LINELIST}\PYG{p}{,} \PYG{n}{CIA\PYGZus{}model} \PYG{o}{=} \PYG{p}{\PYGZob{}}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{model}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Karman}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{band}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{:} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{singlet\PYGZus{}delta}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{\PYGZcb{}}\PYG{p}{)}

\PYG{c+c1}{\PYGZsh{}Generate Baseline Parameter list based on number of etalons in spectra definitions and baseline order}
\PYG{n}{BASE\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}baseline\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{CIA\PYGZus{}LINELIST} \PYG{o}{=} \PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}CIA\PYGZus{}paramlist}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{CIA\PYGZus{}LINELIST}
\end{sphinxVerbatim}

\sphinxAtStartPar
In the Generate\_FitParam\_File class, the CIA\_linelist is set to the CIA\_LINELIST generated by the generate\_CIA\_paramlist. The genererate\_fit\_KarmanCIA\_linelist() will then add the uncertainty and boolean vary columns, setting the vary column to vary the parameters indicated in the definition call


\paragraph{Fit the SPECTRA}
\label{\detokenize{Oxygen CIA using theoretical model:fit-the-spectra}}
\sphinxAtStartPar
Finally, in the Fit\_Dataset instance, the CIA\_linelist\_file needs to be specified to indicate what parameters are being adjusted, provide initial values, and to allow uncertainties to be updated.

\sphinxAtStartPar
The CIA parameters can be constrained using the constrained\_CIA parameters() definition. The default option is that the the spin orbit intensity temperature dependence is the same for O2\sphinxhyphen{}O2 and O2\sphinxhyphen{}N2. Additionally, the shift of the spin\sphinxhyphen{}orbit mechanism is normally constrained to be the same for O2\sphinxhyphen{}O2 and O2\sphinxhyphen{}N2. However, either of these constraints can be removed by setting S\_temperature\_dependence\_constrained = False or shift\_constrained = False.

\sphinxAtStartPar
The rest of the fitting runs in the same manner as for monomer\sphinxhyphen{}absorption only MATS fits.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{fit\PYGZus{}data} \PYG{o}{=} \PYG{n}{MATS}\PYG{o}{.}\PYG{n}{Fit\PYGZus{}DataSet}\PYG{p}{(}\PYG{n}{SPECTRA}\PYG{p}{,}\PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Baseline\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{Parameter\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,} \PYG{n}{CIA\PYGZus{}linelist\PYGZus{}file} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{CIA\PYGZus{}LineList}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{,}
                                                        \PYG{n}{minimum\PYGZus{}parameter\PYGZus{}fit\PYGZus{}intensity} \PYG{o}{=} \PYG{n}{Fit\PYGZus{}Intensity}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{generate\PYGZus{}params}\PYG{p}{(}\PYG{p}{)}
\PYG{n}{params} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{constrained\PYGZus{}CIA}\PYG{p}{(}\PYG{n}{params}\PYG{p}{)}
\PYG{k}{for} \PYG{n}{param} \PYG{o+ow}{in} \PYG{n}{params}\PYG{p}{:}
        \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{S\PYGZus{}SO}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n+nb}{min} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}
        \PYG{k}{if} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{S\PYGZus{}EXCH}\PYG{l+s+s1}{\PYGZsq{}} \PYG{o+ow}{in} \PYG{n}{param}\PYG{p}{:}
                \PYG{n}{params}\PYG{p}{[}\PYG{n}{param}\PYG{p}{]}\PYG{o}{.}\PYG{n}{set}\PYG{p}{(}\PYG{n+nb}{min} \PYG{o}{=} \PYG{l+m+mf}{0.00}\PYG{p}{)}

\PYG{n}{result} \PYG{o}{=} \PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{fit\PYGZus{}data}\PYG{p}{(}\PYG{n}{params}\PYG{p}{,} \PYG{n}{wing\PYGZus{}wavenumbers} \PYG{o}{=} \PYG{l+m+mi}{25}\PYG{p}{,}  \PYG{n}{wing\PYGZus{}method} \PYG{o}{=} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{wing\PYGZus{}wavenumbers}\PYG{l+s+s1}{\PYGZsq{}}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{residual\PYGZus{}analysis}\PYG{p}{(}\PYG{n}{result}\PYG{p}{,} \PYG{n}{indv\PYGZus{}resid\PYGZus{}plot}\PYG{o}{=}\PYG{k+kc}{False}\PYG{p}{)}
\PYG{n}{fit\PYGZus{}data}\PYG{o}{.}\PYG{n}{update\PYGZus{}params}\PYG{p}{(}\PYG{n}{result}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{generate\PYGZus{}summary\PYGZus{}file}\PYG{p}{(}\PYG{n}{save\PYGZus{}file} \PYG{o}{=} \PYG{k+kc}{True}\PYG{p}{)}
\PYG{n}{SPECTRA}\PYG{o}{.}\PYG{n}{plot\PYGZus{}model\PYGZus{}residuals}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}

\noindent\sphinxincludegraphics{{example_files/O2_CIA_fit}.png}


\subsection{Publications that used MATS}
\label{\detokenize{Examples:publications-that-used-mats}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2021.107527}{Adkins EM, Long DA, Fleisher AJ, Hodges JT. Near\sphinxhyphen{}infrared cavity ring\sphinxhyphen{}down spectroscopy measurements of nitrous oxide in the (4200)←(0000) and (5000)←(0000) bands. Journal of Quantitative Spectroscopy and Radiative Transfer. 2021;262.}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2021.107669}{Adkins EM, Long DA, Hodges JT. Air\sphinxhyphen{}broadening in near\sphinxhyphen{}infrared carbon dioxide line shapes: Quantifying contributions from O2, N2, and Ar. Journal of Quantitative Spectroscopy and Radiative Transfer. 2021;270.}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2021.107684}{Fleurbaey H, Reed ZD, Adkins EM, Long DA, Hodges JT. High accuracy spectroscopic parameters of the 1.27 µm band of O2 measured with comb\sphinxhyphen{}referenced, cavity ring\sphinxhyphen{}down spectroscopy. Journal of Quantitative Spectroscopy and Radiative Transfer. 2021;270.}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2021.107735}{Hashemi R, Gordon IE, Adkins EM, Hodges JT, Long DA, Birk M, et al. Improvement of the spectroscopic parameters of the air\sphinxhyphen{} and self\sphinxhyphen{}broadened N2O and CO lines for the HITRAN2020 database applications. Journal of Quantitative Spectroscopy and Radiative Transfer. 2021:107735.}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://pubs.rsc.org/en/content/articlepdf/2023/va/d2va00104g}{Chaillot J, Dasari S, Fleurbaey H, Daeron M, Savarino J, and Kassi S. High\sphinxhyphen{}precision laser spectroscopy of H2S forsimultaneous probing of multiple\sphinxhyphen{}sulfur isotopes. Environmental Science Advances. 2023; (2) 78.}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2021.107558}{Hélène Fleurbaey, Roberto Grilli, Didier Mondelain, Samir Kassi, Andrey Yachmenev, Sergei N. Yurchenko, Alain Campargue. Electric\sphinxhyphen{}quadrupole and magnetic\sphinxhyphen{}dipole contributions to the ν2+ν3 band of carbon dioxide near 3.3 µm. Journal of Quantitative Spectroscopy and Radiative Transfer. 2021; 266.}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2021.108004}{H. Fleurbaey, R. Grilli, D. Mondelain, A. Campargue. Measurements of the water vapor continuum absorption by OFCEAS at 3.50 µm and 2.32 µm. Journal of Quantitative Spectroscopy and Radiative Transfer. 2022; 278.}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://doi.org/10.1016/j.jqsrt.2022.108267}{D. Mondelain, A. Campargue, H. Fleurbaey, S. Kassi, S. Vasilchenko. CRDS measurements of air\sphinxhyphen{}broadened lines in the 1.6 µm band of 12CO2: Line shape parameters with their temperature dependence. Journal of Quantitative Spectroscopy and Radiative Transfer. 2022; 288.}

\end{itemize}

\sphinxstepscope


\section{MATS module}
\label{\detokenize{MATS:module-MATS.codata}}\label{\detokenize{MATS:mats-module}}\label{\detokenize{MATS::doc}}\index{module@\spxentry{module}!MATS.codata@\spxentry{MATS.codata}}\index{MATS.codata@\spxentry{MATS.codata}!module@\spxentry{module}}\phantomsection\label{\detokenize{MATS:module-MATS.dataset}}\index{module@\spxentry{module}!MATS.dataset@\spxentry{MATS.dataset}}\index{MATS.dataset@\spxentry{MATS.dataset}!module@\spxentry{module}}\index{Dataset (class in MATS.dataset)@\spxentry{Dataset}\spxextra{class in MATS.dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{MATS.dataset.}}\sphinxbfcode{\sphinxupquote{Dataset}}}{\emph{\DUrole{n}{spectra}}, \emph{\DUrole{n}{dataset\_name}}, \emph{\DUrole{n}{param\_linelist}}, \emph{\DUrole{n}{CIA\_model}\DUrole{o}{=}\DUrole{default_value}{\{\textquotesingle{}band\textquotesingle{}: None, \textquotesingle{}model\textquotesingle{}: None\}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Combines spectrum objects into a Dataset object to enable multi\sphinxhyphen{}spectrum fitting.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{spectra}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} list of spectrum objects to be included in the Dataset. Example {[}spectrum\_1, spectrum\_2, …{]}

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{dataset\_name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Used to provide a name for the Dataset to use when saving files

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{param\_linelist}} (\sphinxstyleliteralemphasis{\sphinxupquote{pandas dataframe}}) \textendash{} Reads in the  parameter linelist used in fitting.  This enables for consistency checks between the input spectra and the parameter line list.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{baseline\_order}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} sets the baseline order for all spectra in the dataset.  This will automatically be set to the maximum baseline order across all spectrum included in the Dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{CIA\_model}} (\sphinxstyleliteralemphasis{\sphinxupquote{dictionary}}) \textendash{} Specifies the model and band of CIA Model to use.  Default is \{‘model’: None, ‘band’:None\}.  Other option is \{‘model’:’Karman’, ‘band’:’a\_band’ or ‘singlet\_delta’\}, which applies for O2\sphinxhyphen{}O2 and O2\sphinxhyphen{}N2 CIA in the Oxygen A and Singlet Delta Bands.

\end{itemize}

\end{description}\end{quote}
\index{average\_QF() (MATS.dataset.Dataset method)@\spxentry{average\_QF()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.average_QF}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{average\_QF}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Calculates the Average QF from all spectra.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{average\_QF} \textendash{} average QF of all spectra in dataset

\sphinxlineitem{Return type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{check\_iso\_list() (MATS.dataset.Dataset method)@\spxentry{check\_iso\_list()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.check_iso_list}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_iso\_list}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Checks to make sure that all molecules are in the isotope\_list and also checks to make sure all spectra use the same isotope list

\end{fulllineitems}

\index{correct\_component\_list() (MATS.dataset.Dataset method)@\spxentry{correct\_component\_list()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.correct_component_list}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{correct\_component\_list}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Corrects so that all spectra and the parameter line list share the same molecules, but the mole fraction is fixed to zero where molecules are not present (called in the initialization of the class).

\end{fulllineitems}

\index{correct\_etalon\_list() (MATS.dataset.Dataset method)@\spxentry{correct\_etalon\_list()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.correct_etalon_list}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{correct\_etalon\_list}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Corrects so that all spectrum share the same number of etalons, but the amplitude and period are fixed to zero where appropriate(called in the initialization of the class).

\end{fulllineitems}

\index{generate\_CIA\_paramlist() (MATS.dataset.Dataset method)@\spxentry{generate\_CIA\_paramlist()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.generate_CIA_paramlist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_CIA\_paramlist}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a csv file called dataset\_name + \_CIA\_paramlist, which will be used to generate another csv file that is used for fitting the broadband CIA that is common across all spectra, where the columns will be dependent on the CIA model used.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{band}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} specifies the band for the CIA model. For the O2 CIA model reported by Karman et al. The options are a\_band and singlet\_delta. The default is None.

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{CIA\_paramlist} \textendash{} dataframe containing information decribing the CIA parameters based on the CIA model chosen.  This dataframe is also saved to a dataframe.  Either file can be edited before making the CIA parameter list used for fitting.  If editting the .csv file will need to regenerate dataframe from .csv.

\sphinxlineitem{Return type}
\sphinxAtStartPar
pandas dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_baseline\_paramlist() (MATS.dataset.Dataset method)@\spxentry{generate\_baseline\_paramlist()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.generate_baseline_paramlist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_baseline\_paramlist}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a csv file called dataset\_name + \_baseline\_paramlist, which will be used to generate another csv file that is used for fitting spectrum dependent parameters with columns for
spectrum number, segment number, x\_shift, concentration for each molecule in the dataset, baseline terms (a = 0th order term, b = 1st order, etc), and etalon terms (set an amplitude, period, and phase for the number of etalons listed for each spectrum in the Dataset).
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{baseline\_paramlist} \textendash{} dataframe containing information describing baseline parameters.  Also saves to .csv.  Either file can be edited before making the baseline parameter list used for fitting.  If editting the .csv file will need to regenerate dataframe from .csv.

\sphinxlineitem{Return type}
\sphinxAtStartPar
dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_summary\_file() (MATS.dataset.Dataset method)@\spxentry{generate\_summary\_file()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.generate_summary_file}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_summary\_file}}}{\emph{\DUrole{n}{save\_file}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a summary file combining spectral information from all spectra in the Dataset.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{save\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then a .csv is generated in addition to the dataframe. The default is False.

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{summary\_file} \textendash{} Summary dataframe comprised of spectral information inculding model and residuals for all spectra in Dataset.

\sphinxlineitem{Return type}
\sphinxAtStartPar
dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_ILS\_function\_dict() (MATS.dataset.Dataset method)@\spxentry{get\_ILS\_function\_dict()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_ILS_function_dict}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_ILS\_function\_dict}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Provides a dictionary of all ILS functions used in the dataset and the number of resolution parameters
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{dataset\_ILS\_list} \textendash{} list of strings matching the name of the ILS functions used in the dataset

\sphinxlineitem{Return type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_baseline\_order() (MATS.dataset.Dataset method)@\spxentry{get\_baseline\_order()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_baseline_order}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_baseline\_order}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_broadener\_list() (MATS.dataset.Dataset method)@\spxentry{get\_broadener\_list()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_broadener_list}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_broadener\_list}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Provides a list of all broadeners in the dataset.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{dataset\_broadener\_list} \textendash{} list of all broadeners in the dataset.

\sphinxlineitem{Return type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_dataset\_name() (MATS.dataset.Dataset method)@\spxentry{get\_dataset\_name()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_dataset_name}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dataset\_name}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_etalons() (MATS.dataset.Dataset method)@\spxentry{get\_etalons()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_etalons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_etalons}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get list of number of etalons for spectra.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{dataset\_etalon\_list} \textendash{} etalon keys across spectra

\sphinxlineitem{Return type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_list\_spectrum\_numbers() (MATS.dataset.Dataset method)@\spxentry{get\_list\_spectrum\_numbers()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_list_spectrum_numbers}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_list\_spectrum\_numbers}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a list of all spectrum\_numbers.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{spec\_num\_list} \textendash{} list of all spectrum numbers used in the dataset.

\sphinxlineitem{Return type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_molecules() (MATS.dataset.Dataset method)@\spxentry{get\_molecules()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_molecules}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_molecules}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get list of molecules in spectra.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{dataset\_molecule\_list} \textendash{} list of molecules in spectra

\sphinxlineitem{Return type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_number\_nominal\_temperatures() (MATS.dataset.Dataset method)@\spxentry{get\_number\_nominal\_temperatures()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_number_nominal_temperatures}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_number\_nominal\_temperatures}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get the number of nominal temperatures in the .
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{num\_nominal\_temperatures} (\sphinxstyleemphasis{int}) \textendash{} number of nominal temperatures in the Dataset

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{nominal\_temperatures} (\sphinxstyleemphasis{list}) \textendash{} list of all nominal temperatures listed in the input spectra

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{get\_number\_spectra() (MATS.dataset.Dataset method)@\spxentry{get\_number\_spectra()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_number_spectra}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_number\_spectra}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_spectra() (MATS.dataset.Dataset method)@\spxentry{get\_spectra()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_spectra}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_spectra}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_spectra\_extremes() (MATS.dataset.Dataset method)@\spxentry{get\_spectra\_extremes()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_spectra_extremes}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_spectra\_extremes}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_spectrum\_extremes() (MATS.dataset.Dataset method)@\spxentry{get\_spectrum\_extremes()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_spectrum_extremes}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_spectrum\_extremes}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Gets the minimum and maximum wavenumber for the entire Dataset.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{wave\_min} (\sphinxstyleemphasis{float}) \textendash{} The minimum wavenumber in all spectra in the Dataset.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{wave\_max} (\sphinxstyleemphasis{float}) \textendash{} The maximum wavenumber in all spectra in the Dataset.

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{get\_spectrum\_filename() (MATS.dataset.Dataset method)@\spxentry{get\_spectrum\_filename()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_spectrum_filename}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_spectrum\_filename}}}{\emph{\DUrole{n}{spectrum\_num}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Gets spectrum filename for spectrum in Dataset.
:param spectrum\_num: Integer assigned to a given spectrum.
:type spectrum\_num: int
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{filename} \textendash{} if spectrum\_num in Dataset then the filename for that the spectrum\_number is returned

\sphinxlineitem{Return type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_spectrum\_pressure() (MATS.dataset.Dataset method)@\spxentry{get\_spectrum\_pressure()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_spectrum_pressure}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_spectrum\_pressure}}}{\emph{\DUrole{n}{spectrum\_num}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Gets spectrum pressure for spectrum in Dataset.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{spectrum\_num}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Integer assigned to a given spectrum.

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{pressure} \textendash{} if spectrum\_num in Dataset then the pressure (torr) for that the spectrum\_number is returned.

\sphinxlineitem{Return type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_spectrum\_temperature() (MATS.dataset.Dataset method)@\spxentry{get\_spectrum\_temperature()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.get_spectrum_temperature}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_spectrum\_temperature}}}{\emph{\DUrole{n}{spectrum\_num}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Gets spectrum temperature for spectrum in Dataset.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{spectrum\_num}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Integer assigned to a given spectrum.

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{temperature} \textendash{} if spectrum\_num in Dataset then the temperature (K) for that the spectrum\_number is returned.

\sphinxlineitem{Return type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{max\_baseline\_order() (MATS.dataset.Dataset method)@\spxentry{max\_baseline\_order()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.max_baseline_order}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{max\_baseline\_order}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
sets the baseline order to be equal to the maximum in any of the included spectra.

\end{fulllineitems}

\index{plot\_model\_residuals() (MATS.dataset.Dataset method)@\spxentry{plot\_model\_residuals()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.plot_model_residuals}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_model\_residuals}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a plot showing both the model and experimental data as a function of wavenumber in the main plot with a subplot showing the residuals as function of wavenumber.

\end{fulllineitems}

\index{renumber\_spectra() (MATS.dataset.Dataset method)@\spxentry{renumber\_spectra()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.renumber_spectra}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{renumber\_spectra}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
renumbers the spectra to be sequential starting at 1 (called in the initialization of the class).

\end{fulllineitems}

\index{set\_baseline\_order() (MATS.dataset.Dataset method)@\spxentry{set\_baseline\_order()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.set_baseline_order}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_baseline\_order}}}{\emph{\DUrole{n}{new\_baseline\_order}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_dataset\_name() (MATS.dataset.Dataset method)@\spxentry{set\_dataset\_name()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.set_dataset_name}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_dataset\_name}}}{\emph{\DUrole{n}{new\_dataset\_name}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_spectra() (MATS.dataset.Dataset method)@\spxentry{set\_spectra()}\spxextra{MATS.dataset.Dataset method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.dataset.Dataset.set_spectra}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_spectra}}}{\emph{\DUrole{n}{new\_spectra}}}{}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}

\phantomsection\label{\detokenize{MATS:module-MATS.fit_dataset}}\index{module@\spxentry{module}!MATS.fit\_dataset@\spxentry{MATS.fit\_dataset}}\index{MATS.fit\_dataset@\spxentry{MATS.fit\_dataset}!module@\spxentry{module}}\index{Fit\_DataSet (class in MATS.fit\_dataset)@\spxentry{Fit\_DataSet}\spxextra{class in MATS.fit\_dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{MATS.fit\_dataset.}}\sphinxbfcode{\sphinxupquote{Fit\_DataSet}}}{\emph{\DUrole{n}{dataset}}, \emph{\DUrole{n}{base\_linelist\_file}}, \emph{\DUrole{n}{param\_linelist\_file}}, \emph{\DUrole{n}{CIA\_linelist\_file}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{minimum\_parameter\_fit\_intensity}\DUrole{o}{=}\DUrole{default_value}{1e\sphinxhyphen{}27}}, \emph{\DUrole{n}{weight\_spectra}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{baseline\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{baseline\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{pressure\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{pressure\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{temperature\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{temperature\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{molefraction\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{molefraction\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{etalon\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{etalon\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{50}}, \emph{\DUrole{n}{x\_shift\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{x\_shift\_limit\_magnitude}\DUrole{o}{=}\DUrole{default_value}{0.1}}, \emph{\DUrole{n}{nu\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{nu\_limit\_magnitude}\DUrole{o}{=}\DUrole{default_value}{0.1}}, \emph{\DUrole{n}{sw\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{sw\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{gamma0\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{gamma0\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{n\_gamma0\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{n\_gamma0\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{delta0\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{delta0\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{n\_delta0\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{n\_delta0\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{SD\_gamma\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{SD\_gamma\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{n\_gamma2\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{n\_gamma2\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{SD\_delta\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{SD\_delta\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{n\_delta2\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{n\_delta2\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{nuVC\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{nuVC\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{n\_nuVC\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{n\_nuVC\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{eta\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{eta\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{linemixing\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{linemixing\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{n\_linemixing\_limit}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{n\_linemixing\_limit\_factor}\DUrole{o}{=}\DUrole{default_value}{10}}, \emph{\DUrole{n}{beta\_formalism}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Provides the fitting functionality for a Dataset.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{dataset}} (\sphinxstyleliteralemphasis{\sphinxupquote{object}}) \textendash{} Dataset Object.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{base\_linelist\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} filename for file containing baseline parameters.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{param\_linelist\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} filename for file containing parmeter parameters.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{CIA\_linelist\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Future Feature: filename for file constraining CIA parameters

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{minimum\_parameter\_fit\_intensity}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} minimum intensity for parameters to be generated for fitting. NOTE: Even if a value is floated in the param\_linelist if it is below this threshold then it won’t be a floated.. The default is 1e\sphinxhyphen{}27.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{weight\_spectra}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}) \textendash{} If True, then the pt by pt percent uncertainty for each spectrum and the spectrum weighting will be used in the calculation of the residuals.  Default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{baseline\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on baseline parameter solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{baseline\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor that the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pressure\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on pressure solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pressure\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{temperature\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on temperature solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{temperature\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on mole fraction solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} DESCRIPTION. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{etalon\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on etalon solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{etalon\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 50. \#phase is constrained to +/\sphinxhyphen{} 2pi

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{x\_shift\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on x\sphinxhyphen{}shift solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{x\_shift\_limit\_magnitude}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The magnitude variables set the +/\sphinxhyphen{} range of the variable in cm\sphinxhyphen{}1. The default is 0.1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nu\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on line center solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nu\_limit\_magnitude}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The magnitude variables set the +/\sphinxhyphen{} range of the variable in cm\sphinxhyphen{}1. The default is 0.1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{sw\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on line intensity solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{sw\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{gamma0\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on collisional half\sphinxhyphen{}width solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{gamma0\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_gamma0\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} DESCIf True, then impose min/max limits on temperature exponent for half width solutions. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_gamma0\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{delta0\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on collisional shift solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{delta0\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_delta0\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on temperature exponent of the collisional shift solutions. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_delta0\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} DESCRIPTION. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{SD\_gamma\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on the aw solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{SD\_gamma\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_gamma2\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on temperature exponent of the speed\sphinxhyphen{}dependent width solutions. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_gamma2\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{SD\_delta\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on as solutions. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{SD\_delta\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_delta2\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on temperature exponent of the speed\sphinxhyphen{}dependent shift solutions. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_delta2\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nuVC\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on dicke narrowing solutions. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nuVC\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_nuVC\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on temperature exponent of dicke narrowing solutions. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{n\_nuVC\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{eta\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on correlation parameter solutions.. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{eta\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{linemixing\_limit}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The factor variable describes the multiplicative factor the value can vary by min = 1/factor * init\_guess, max = factor* init\_guess. NOTE: If the init\_guess for a parameter is equal to zero, then the limits aren’t imposed because 1) then it would constrain the fit to 0 and 2) LMfit won’t let you set min = max.. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{linemixing\_limit\_factor}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then impose min/max limits on line\sphinxhyphen{}mixing solutions.. The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{beta\_formalism}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then the beta correction on the Dicke narrowing is used in the simulation model.

\end{itemize}

\end{description}\end{quote}
\index{constrained\_CIA() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{constrained\_CIA()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.constrained_CIA}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{constrained\_CIA}}}{\emph{\DUrole{n}{params}}, \emph{\DUrole{n}{S\_temperature\_dependence\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{shift\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}}{}
\pysigstopsignatures
\sphinxAtStartPar
All user to set the temperature dependence of the SO mechanism scalar and/or the shift of the SO to be equal for O2\sphinxhyphen{}O2 and O2\_N2
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{params}} (\sphinxstyleliteralemphasis{\sphinxupquote{lmfit parameter object}}) \textendash{} the params object is a dictionary comprised of all parameters translated from dataframes into a dictionary format compatible with lmfit.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{S\_temperature\_dependence\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Constrains the temperature dependence of the SO mechanism scalar to be the same for O2\sphinxhyphen{}N2 and O2\sphinxhyphen{}O2. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{shift\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Constrains the SO shift to be the same for O2\sphinxhyphen{}N2 and O2\sphinxhyphen{}O2. The default is True.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{params} \textendash{} the params object is a dictionary comprised of all parameters translated from dataframes into a dictionary format compatible with lmfit.

\sphinxlineitem{Return type}
\sphinxAtStartPar
lmfit parameter object

\end{description}\end{quote}

\end{fulllineitems}

\index{constrained\_baseline() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{constrained\_baseline()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.constrained_baseline}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{constrained\_baseline}}}{\emph{\DUrole{n}{params}}, \emph{\DUrole{n}{baseline\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{xshift\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{molefraction\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{etalon\_amp\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{etalon\_period\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{etalon\_phase\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{pressure\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{temperature\_segment\_constrained}\DUrole{o}{=}\DUrole{default_value}{True}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Imposes baseline constraints when using multiple segments per spectrum, ie all baseline parameters can be the same across the entire spectrum except for the etalon phase, which is allowed to vary per segment.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{params}} (\sphinxstyleliteralemphasis{\sphinxupquote{lmfit parameter object}}) \textendash{} the params object is a dictionary comprised of all parameters translated from dataframes into a dictionary format compatible with lmfit.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{baseline\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the baseline terms are constrained across each spectrum. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{xshift\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the x\_shift terms are constrained across each spectrum. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the mole fraction for that molecule is constrained across each spectrum. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{etalon\_amp\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the etalon amplitude is constrained across each spectrum. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{etalon\_period\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the etalon period is constrained across each spectrum. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{etalon\_phase\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the etalon phase is constrained across each spectrum. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pressure\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the pressure is constrained across each spectrum. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{temperature\_segment\_constrained}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True means the temperature is constrained across each spectrum. The default is True.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{params} \textendash{} the params object is a dictionary comprised of all parameters translated from dataframes into a dictionary format compatible with lmfit.

\sphinxlineitem{Return type}
\sphinxAtStartPar
lmfit parameter object

\end{description}\end{quote}

\end{fulllineitems}

\index{fit\_data() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{fit\_data()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.fit_data}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{fit\_data}}}{\emph{\DUrole{n}{params}}, \emph{\DUrole{n}{wing\_cutoff}\DUrole{o}{=}\DUrole{default_value}{25}}, \emph{\DUrole{n}{wing\_wavenumbers}\DUrole{o}{=}\DUrole{default_value}{25}}, \emph{\DUrole{n}{wing\_method}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}wing\_cutoff\textquotesingle{}}}, \emph{\DUrole{n}{xtol}\DUrole{o}{=}\DUrole{default_value}{1e\sphinxhyphen{}07}}, \emph{\DUrole{n}{maxfev}\DUrole{o}{=}\DUrole{default_value}{2000}}, \emph{\DUrole{n}{ftol}\DUrole{o}{=}\DUrole{default_value}{1e\sphinxhyphen{}07}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Uses the lmfit minimizer to do the fitting through the simulation model function.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{params}} (\sphinxstyleliteralemphasis{\sphinxupquote{lmfit parameter object}}) \textendash{} the params object is a dictionary comprised of all parameters translated from dataframes into a dictionary format compatible with lmfit.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_cutoff}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of voigt half\sphinxhyphen{}widths to simulate on either side of each line. The default is 50.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_wavenumbers}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of wavenumbers to simulate on either side of each line. The default is 50.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_method}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Provides choice between the wing\_cutoff and wing\_wavenumbers line cut\sphinxhyphen{}off options. The default is ‘wing\_cutoff’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{xtol}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Absolute error in xopt between iterations that is acceptable for convergence. The default is 1e\sphinxhyphen{}7.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{maxfev}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} DESCRIPTION. The default is 2000.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ftol}} (\sphinxstyleliteralemphasis{\sphinxupquote{The maximum number}}\sphinxstyleliteralemphasis{\sphinxupquote{ of }}\sphinxstyleliteralemphasis{\sphinxupquote{calls to the function.}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Absolute error in func(xopt) between iterations that is acceptable for convergence.. The default is 1e\sphinxhyphen{}7.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{result} \textendash{} contains all fit results as LMFit results object.

\sphinxlineitem{Return type}
\sphinxAtStartPar
LMFit result Object

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_beta\_output\_file() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{generate\_beta\_output\_file()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.generate_beta_output_file}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_beta\_output\_file}}}{\emph{\DUrole{n}{beta\_summary\_filename}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a file that summarizes the beta values used in the fitting in the case that beta was used to correct the Dicke narrowing term (beta\_formalism = True).
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{beta\_summary\_filename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Filename to save the beta information. The default is Beta Summary File.

\sphinxlineitem{Return type}
\sphinxAtStartPar
The generated file has the beta values for each line and spectra that were used in the fitting.  Access to this information is critical as the relationship between beta and nuVC is what generates a spoecific nuVC.

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_params() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{generate\_params()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.generate_params}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_params}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates the lmfit parameter object that will be used in fitting.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{params} \textendash{} the parameter object is a dictionary comprised of all parameters translated from dataframes into a dictionary format compatible with lmfit

\sphinxlineitem{Return type}
\sphinxAtStartPar
lmfit parameter object

\end{description}\end{quote}

\end{fulllineitems}

\index{residual\_analysis() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{residual\_analysis()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.residual_analysis}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{residual\_analysis}}}{\emph{\DUrole{n}{result}}, \emph{\DUrole{n}{indv\_resid\_plot}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Updates the model and residual arrays in each spectrum object with the results of the fit and gives the option of generating the combined absorption and residual plot for each spectrum.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{result}} (\sphinxstyleliteralemphasis{\sphinxupquote{LMFit result Object}}) \textendash{} contains all fit results as LMFit results object.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{indv\_resid\_plot}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True if you want to show residual plots for each spectra.. The default is False.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{simulation\_model() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{simulation\_model()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.simulation_model}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{simulation\_model}}}{\emph{\DUrole{n}{params}}, \emph{\DUrole{n}{wing\_cutoff}\DUrole{o}{=}\DUrole{default_value}{25}}, \emph{\DUrole{n}{wing\_wavenumbers}\DUrole{o}{=}\DUrole{default_value}{25}}, \emph{\DUrole{n}{wing\_method}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}wing\_cutoff\textquotesingle{}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
This is the model used for fitting that includes baseline, resonant absorption, and CIA models.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{params}} (\sphinxstyleliteralemphasis{\sphinxupquote{lmfit parameter object}}) \textendash{} the params object is a dictionary comprised of all parameters translated from dataframes into a dictionary format compatible with lmfit.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_cutoff}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of voigt half\sphinxhyphen{}widths to simulate on either side of each line. The default is 25.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_wavenumbers}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of wavenumbers to simulate on either side of each line. The default is 25

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_method}} (\sphinxstyleliteralemphasis{\sphinxupquote{TYPE}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Provides choice between the wing\_cutoff and wing\_wavenumbers line cut\sphinxhyphen{}off options. The default is ‘wing\_cutoff’.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{total\_residuals} \textendash{} residuals for all spectra in Dataset.

\sphinxlineitem{Return type}
\sphinxAtStartPar
array

\end{description}\end{quote}

\end{fulllineitems}

\index{update\_params() (MATS.fit\_dataset.Fit\_DataSet method)@\spxentry{update\_params()}\spxextra{MATS.fit\_dataset.Fit\_DataSet method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.Fit_DataSet.update_params}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update\_params}}}{\emph{\DUrole{n}{result}}, \emph{\DUrole{n}{base\_linelist\_update\_file}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{param\_linelist\_update\_file}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{CIA\_linelist\_update\_file}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Updates the baseline and line parameter files based on fit results with the option to write over the file (default) or save as a new file and updates baseline values in the spectrum objects.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{result}} (\sphinxstyleliteralemphasis{\sphinxupquote{LMFit result Object}}) \textendash{} contains all fit results as LMFit results object.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{base\_linelist\_update\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of file to save the updated baseline parameters. Default is to override the input. The default is None.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{param\_linelist\_update\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of file to save the updated line parameters. Default is to override the input. The default is None.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{cia\_linelist\_update\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of file to save the updated CIA parameters.  Default is to override the input.

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}

\index{HTP\_from\_DF\_select() (in module MATS.fit\_dataset)@\spxentry{HTP\_from\_DF\_select()}\spxextra{in module MATS.fit\_dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.HTP_from_DF_select}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.fit\_dataset.}}\sphinxbfcode{\sphinxupquote{HTP\_from\_DF\_select}}}{\emph{linelist, waves, wing\_cutoff=25, wing\_wavenumbers=25, wing\_method=\textquotesingle{}wing\_cutoff\textquotesingle{}, p=1, T=296, molefraction=\{\}, isotope\_list=\{(1, 1): {[}1, \textquotesingle{}H2(16O)\textquotesingle{}, 0.9973173, 18.01056, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 2): {[}2, \textquotesingle{}H2(18O)\textquotesingle{}, 0.001999827, 20.01481, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 3): {[}3, \textquotesingle{}H2(17O)\textquotesingle{}, 0.0003718841, 19.01478, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 4): {[}4, \textquotesingle{}HD(16O)\textquotesingle{}, 0.0003106928, 19.01674, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 5): {[}5, \textquotesingle{}HD(18O)\textquotesingle{}, 6.230031e\sphinxhyphen{}07, 21.02098, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 6): {[}6, \textquotesingle{}HD(17O)\textquotesingle{}, 1.158526e\sphinxhyphen{}07, 20.02096, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 7): {[}129, \textquotesingle{}D2(16O)\textquotesingle{}, 2.419741e\sphinxhyphen{}08, 20.02292, \textquotesingle{}H2O\textquotesingle{}{]}, (2, 1): {[}7, \textquotesingle{}(12C)(16O)2\textquotesingle{}, 0.9842043, 43.98983, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 2): {[}8, \textquotesingle{}(13C)(16O)2\textquotesingle{}, 0.01105736, 44.99318, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 3): {[}9, \textquotesingle{}(16O)(12C)(18O)\textquotesingle{}, 0.003947066, 45.99408, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 4): {[}10, \textquotesingle{}(16O)(12C)(17O)\textquotesingle{}, 0.000733989, 44.99404, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 5): {[}11, \textquotesingle{}(16O)(13C)(18O)\textquotesingle{}, 4.434456e\sphinxhyphen{}05, 46.99743, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 6): {[}12, \textquotesingle{}(16O)(13C)(17O)\textquotesingle{}, 8.246233e\sphinxhyphen{}06, 45.9974, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 7): {[}13, \textquotesingle{}(12C)(18O)2\textquotesingle{}, 3.95734e\sphinxhyphen{}06, 47.99832, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 8): {[}14, \textquotesingle{}(17O)(12C)(18O)\textquotesingle{}, 1.471799e\sphinxhyphen{}06, 46.99829, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 9): {[}121, \textquotesingle{}(12C)(17O)2\textquotesingle{}, 1.368466e\sphinxhyphen{}07, 45.99826, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 10): {[}15, \textquotesingle{}(13C)(18O)2\textquotesingle{}, 4.446e\sphinxhyphen{}08, 49.00167, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 11): {[}120, \textquotesingle{}(18O)(13C)(17O)\textquotesingle{}, 1.65354e\sphinxhyphen{}08, 48.00165, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 12): {[}122, \textquotesingle{}(13C)(17O)2\textquotesingle{}, 1.537446e\sphinxhyphen{}09, 47.00162, \textquotesingle{}CO2\textquotesingle{}{]}, (3, 1): {[}16, \textquotesingle{}(16O)3\textquotesingle{}, 0.9929009, 47.98474, \textquotesingle{}O3\textquotesingle{}{]}, (3, 2): {[}17, \textquotesingle{}(16O)(16O)(18O)\textquotesingle{}, 0.003981942, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 3): {[}18, \textquotesingle{}(16O)(18O)(16O)\textquotesingle{}, 0.001990971, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 4): {[}19, \textquotesingle{}(16O)(16O)(17O)\textquotesingle{}, 0.0007404746, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (3, 5): {[}20, \textquotesingle{}(16O)(17O)(16O)\textquotesingle{}, 0.0003702373, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (4, 1): {[}21, \textquotesingle{}(14N)2(16O)\textquotesingle{}, 0.9903328, 44.00106, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 2): {[}22, \textquotesingle{}(14N)(15N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 3): {[}23, \textquotesingle{}(15N)(14N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 4): {[}24, \textquotesingle{}(14N)2(18O)\textquotesingle{}, 0.001985822, 46.00531, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 5): {[}25, \textquotesingle{}(14N)2(17O)\textquotesingle{}, 0.0003692797, 45.00528, \textquotesingle{}N2O\textquotesingle{}{]}, (5, 1): {[}26, \textquotesingle{}(12C)(16O)\textquotesingle{}, 0.9865444, 27.99491, \textquotesingle{}CO\textquotesingle{}{]}, (5, 2): {[}27, \textquotesingle{}(13C)(16O)\textquotesingle{}, 0.01108364, 28.99827, \textquotesingle{}CO\textquotesingle{}{]}, (5, 3): {[}28, \textquotesingle{}(12C)(18O)\textquotesingle{}, 0.001978224, 29.99916, \textquotesingle{}CO\textquotesingle{}{]}, (5, 4): {[}29, \textquotesingle{}(12C)(17O)\textquotesingle{}, 0.0003678671, 28.99913, \textquotesingle{}CO\textquotesingle{}{]}, (5, 5): {[}30, \textquotesingle{}(13C)(18O)\textquotesingle{}, 2.2225e\sphinxhyphen{}05, 31.00252, \textquotesingle{}CO\textquotesingle{}{]}, (5, 6): {[}31, \textquotesingle{}(13C)(17O)\textquotesingle{}, 4.13292e\sphinxhyphen{}06, 30.00249, \textquotesingle{}CO\textquotesingle{}{]}, (6, 1): {[}32, \textquotesingle{}(12C)H4\textquotesingle{}, 0.9882741, 16.0313, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 2): {[}33, \textquotesingle{}(13C)H4\textquotesingle{}, 0.01110308, 17.03466, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 3): {[}34, \textquotesingle{}(12C)H3D\textquotesingle{}, 0.0006157511, 17.03748, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 4): {[}35, \textquotesingle{}(13C)H3D\textquotesingle{}, 6.917852e\sphinxhyphen{}06, 18.04083, \textquotesingle{}CH4\textquotesingle{}{]}, (7, 1): {[}36, \textquotesingle{}(16O)2\textquotesingle{}, 0.9952616, 31.98983, \textquotesingle{}O2\textquotesingle{}{]}, (7, 2): {[}37, \textquotesingle{}(16O)(18O)\textquotesingle{}, 0.00399141, 33.99408, \textquotesingle{}O2\textquotesingle{}{]}, (7, 3): {[}38, \textquotesingle{}(16O)(17O)\textquotesingle{}, 0.0007422352, 32.99404, \textquotesingle{}O2\textquotesingle{}{]}, (8, 1): {[}39, \textquotesingle{}(14N)(16O)\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NO\textquotesingle{}{]}, (8, 2): {[}40, \textquotesingle{}(15N)(16O)\textquotesingle{}, 0.003654311, 30.99502, \textquotesingle{}NO\textquotesingle{}{]}, (8, 3): {[}41, \textquotesingle{}(14N)(18O)\textquotesingle{}, 0.001993122, 32.00223, \textquotesingle{}NO\textquotesingle{}{]}, (9, 1): {[}42, \textquotesingle{}(32S)(16O)2\textquotesingle{}, 0.9456777, 63.9619, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 2): {[}43, \textquotesingle{}(34S)(16O)2\textquotesingle{}, 0.04195028, 65.9577, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 3): {[}137, \textquotesingle{}(33S)(16O)2\textquotesingle{}, 0.007464462, 64.96129, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 4): {[}138, \textquotesingle{}(16O)(32S)(18O)\textquotesingle{}, 0.003792558, 65.96615, \textquotesingle{}SO2\textquotesingle{}{]}, (10, 1): {[}44, \textquotesingle{}(14N)(16O)2\textquotesingle{}, 0.991616, 45.9929, \textquotesingle{}NO2\textquotesingle{}{]}, (10, 2): {[}130, \textquotesingle{}(15N)(16O)2\textquotesingle{}, 0.003645643, 46.98994, \textquotesingle{}NO2\textquotesingle{}{]}, (11, 1): {[}45, \textquotesingle{}(14N)H3\textquotesingle{}, 0.9958716, 17.02655, \textquotesingle{}NH3\textquotesingle{}{]}, (11, 2): {[}46, \textquotesingle{}(15N)H3\textquotesingle{}, 0.003661289, 18.02358, \textquotesingle{}NH3\textquotesingle{}{]}, (12, 1): {[}47, \textquotesingle{}H(14N)(16O)3\textquotesingle{}, 0.9891098, 62.99564, \textquotesingle{}HNO3\textquotesingle{}{]}, (12, 2): {[}117, \textquotesingle{}H(15N)(16O)3\textquotesingle{}, 0.003636429, 63.99268, \textquotesingle{}HNO3\textquotesingle{}{]}, (13, 1): {[}48, \textquotesingle{}(16O)H\textquotesingle{}, 0.9974726, 17.00274, \textquotesingle{}OH\textquotesingle{}{]}, (13, 2): {[}49, \textquotesingle{}(18O)H\textquotesingle{}, 0.002000138, 19.00699, \textquotesingle{}OH\textquotesingle{}{]}, (13, 3): {[}50, \textquotesingle{}(16O)D\textquotesingle{}, 0.0001553706, 18.00891, \textquotesingle{}OH\textquotesingle{}{]}, (14, 1): {[}51, \textquotesingle{}H(19F)\textquotesingle{}, 0.9998443, 20.00623, \textquotesingle{}HF\textquotesingle{}{]}, (14, 2): {[}110, \textquotesingle{}D(19F)\textquotesingle{}, 0.000155741, 21.0124, \textquotesingle{}HF\textquotesingle{}{]}, (15, 1): {[}52, \textquotesingle{}H(35Cl)\textquotesingle{}, 0.757587, 35.97668, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 2): {[}53, \textquotesingle{}H(37Cl)\textquotesingle{}, 0.2422573, 37.97373, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 3): {[}107, \textquotesingle{}D(35Cl)\textquotesingle{}, 0.000118005, 36.98285, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 4): {[}108, \textquotesingle{}D(37Cl)\textquotesingle{}, 3.773502e\sphinxhyphen{}05, 38.9799, \textquotesingle{}HCl\textquotesingle{}{]}, (16, 1): {[}54, \textquotesingle{}H(79Br)\textquotesingle{}, 0.5067811, 79.92616, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 2): {[}55, \textquotesingle{}H(81Br)\textquotesingle{}, 0.4930632, 81.92412, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 3): {[}111, \textquotesingle{}D(79Br)\textquotesingle{}, 7.893838e\sphinxhyphen{}05, 80.93234, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 4): {[}112, \textquotesingle{}D(81Br)\textquotesingle{}, 7.680162e\sphinxhyphen{}05, 82.93029, \textquotesingle{}HBr\textquotesingle{}{]}, (17, 1): {[}56, \textquotesingle{}H(127I)\textquotesingle{}, 0.9998443, 127.9123, \textquotesingle{}HI\textquotesingle{}{]}, (17, 2): {[}113, \textquotesingle{}D(127I)\textquotesingle{}, 0.000155741, 128.9185, \textquotesingle{}HI\textquotesingle{}{]}, (18, 1): {[}57, \textquotesingle{}(35Cl)(16O)\textquotesingle{}, 0.7559077, 50.96377, \textquotesingle{}ClO\textquotesingle{}{]}, (18, 2): {[}58, \textquotesingle{}(37Cl)(16O)\textquotesingle{}, 0.2417203, 52.96082, \textquotesingle{}ClO\textquotesingle{}{]}, (19, 1): {[}59, \textquotesingle{}(16O)(12C)(32S)\textquotesingle{}, 0.9373947, 59.96699, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 2): {[}60, \textquotesingle{}(16O)(12C)(34S)\textquotesingle{}, 0.04158284, 61.96278, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 3): {[}61, \textquotesingle{}(16O)(13C)(32S)\textquotesingle{}, 0.01053146, 60.97034, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 4): {[}62, \textquotesingle{}(16O)(12C)(33S)\textquotesingle{}, 0.007399083, 60.96637, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 5): {[}63, \textquotesingle{}(18O)(12C)(32S)\textquotesingle{}, 0.00187967, 61.97123, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 6): {[}135, \textquotesingle{}(16O)(13C)(34S)\textquotesingle{}, 0.0004671757, 62.96614, \textquotesingle{}OCS\textquotesingle{}{]}, (20, 1): {[}64, \textquotesingle{}H2(12C)(16O)\textquotesingle{}, 0.9862371, 30.01056, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 2): {[}65, \textquotesingle{}H2(13C)(16O)\textquotesingle{}, 0.0110802, 31.01392, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 3): {[}66, \textquotesingle{}H2(12C)(18O)\textquotesingle{}, 0.001977609, 32.01481, \textquotesingle{}H2CO\textquotesingle{}{]}, (21, 1): {[}67, \textquotesingle{}H(16O)(35Cl)\textquotesingle{}, 0.75579, 51.97159, \textquotesingle{}HOCl\textquotesingle{}{]}, (21, 2): {[}68, \textquotesingle{}H(16O)(37Cl)\textquotesingle{}, 0.2416826, 53.96864, \textquotesingle{}HOCl\textquotesingle{}{]}, (22, 1): {[}69, \textquotesingle{}(14N)2\textquotesingle{}, 0.9926874, 28.00615, \textquotesingle{}N2\textquotesingle{}{]}, (22, 2): {[}118, \textquotesingle{}(14N)(15N)\textquotesingle{}, 0.007299165, 29.00318, \textquotesingle{}N2\textquotesingle{}{]}, (23, 1): {[}70, \textquotesingle{}H(12C)(14N)\textquotesingle{}, 0.9851143, 27.0109, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 2): {[}71, \textquotesingle{}H(13C)(14N)\textquotesingle{}, 0.01106758, 28.01425, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 3): {[}72, \textquotesingle{}H(12C)(15N)\textquotesingle{}, 0.00362174, 28.00793, \textquotesingle{}HCN\textquotesingle{}{]}, (24, 1): {[}73, \textquotesingle{}(12C)H3(35Cl)\textquotesingle{}, 0.7489369, 49.99233, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (24, 2): {[}74, \textquotesingle{}(12C)H3(37Cl)\textquotesingle{}, 0.2394912, 51.98938, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (25, 1): {[}75, \textquotesingle{}H2(16O)2\textquotesingle{}, 0.9949516, 34.00548, \textquotesingle{}H2O2\textquotesingle{}{]}, (26, 1): {[}76, \textquotesingle{}(12C)2H2\textquotesingle{}, 0.9775989, 26.01565, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 2): {[}77, \textquotesingle{}(12C)(13C)H2\textquotesingle{}, 0.02196629, 27.019, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 3): {[}105, \textquotesingle{}(12C)2HD\textquotesingle{}, 0.0003045499, 27.02182, \textquotesingle{}C2H2\textquotesingle{}{]}, (27, 1): {[}78, \textquotesingle{}(12C)2H6\textquotesingle{}, 0.97699, 30.04695, \textquotesingle{}C2H6\textquotesingle{}{]}, (27, 2): {[}106, \textquotesingle{}(12C)H3(13C)H3\textquotesingle{}, 0.02195261, 31.05031, \textquotesingle{}C2H6\textquotesingle{}{]}, (28, 1): {[}79, \textquotesingle{}(31P)H3\textquotesingle{}, 0.9995329, 33.99724, \textquotesingle{}PH3\textquotesingle{}{]}, (29, 1): {[}80, \textquotesingle{}(12C)(16O)(19F)2\textquotesingle{}, 0.9865444, 65.99172, \textquotesingle{}COF2\textquotesingle{}{]}, (29, 2): {[}119, \textquotesingle{}(13C)(16O)(19F)2\textquotesingle{}, 0.01108366, 66.99508, \textquotesingle{}COF2\textquotesingle{}{]}, (30, 1): {[}126, \textquotesingle{}(32S)(19F)6\textquotesingle{}, 0.95018, 145.9625, \textquotesingle{}SF6\textquotesingle{}{]}, (31, 1): {[}81, \textquotesingle{}H2(32S)\textquotesingle{}, 0.9498841, 33.98772, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 2): {[}82, \textquotesingle{}H2(34S)\textquotesingle{}, 0.04213687, 35.98351, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 3): {[}83, \textquotesingle{}H2(33S)\textquotesingle{}, 0.007497664, 34.9871, \textquotesingle{}H2S\textquotesingle{}{]}, (32, 1): {[}84, \textquotesingle{}H(12C)(16O)(16O)H\textquotesingle{}, 0.9838977, 46.00548, \textquotesingle{}HCOOH\textquotesingle{}{]}, (33, 1): {[}85, \textquotesingle{}H(16O)2\textquotesingle{}, 0.9951066, 32.99766, \textquotesingle{}HO2\textquotesingle{}{]}, (34, 1): {[}86, \textquotesingle{}(16O)\textquotesingle{}, 0.997628, 15.99492, \textquotesingle{}O\textquotesingle{}{]}, (35, 1): {[}127, \textquotesingle{}(35Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.7495702, 96.95667, \textquotesingle{}ClONO2\textquotesingle{}{]}, (35, 2): {[}128, \textquotesingle{}(37Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.2396937, 98.95372, \textquotesingle{}ClONO2\textquotesingle{}{]}, (36, 1): {[}87, \textquotesingle{}(14N)(16O)+\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NOp\textquotesingle{}{]}, (37, 1): {[}88, \textquotesingle{}H(16O)(79Br)\textquotesingle{}, 0.505579, 95.92108, \textquotesingle{}HOBr\textquotesingle{}{]}, (37, 2): {[}89, \textquotesingle{}H(16O)(81Br)\textquotesingle{}, 0.4918937, 97.91903, \textquotesingle{}HOBr\textquotesingle{}{]}, (38, 1): {[}90, \textquotesingle{}(12C)2H4\textquotesingle{}, 0.9772944, 28.0313, \textquotesingle{}C2H4\textquotesingle{}{]}, (38, 2): {[}91, \textquotesingle{}(12C)H2(13C)H2\textquotesingle{}, 0.02195946, 29.03466, \textquotesingle{}C2H4\textquotesingle{}{]}, (39, 1): {[}92, \textquotesingle{}(12C)H3(16O)H\textquotesingle{}, 0.9859299, 32.02622, \textquotesingle{}CH3OH\textquotesingle{}{]}, (40, 1): {[}93, \textquotesingle{}(12C)H3(79Br)\textquotesingle{}, 0.5009946, 93.94181, \textquotesingle{}CH3Br\textquotesingle{}{]}, (40, 2): {[}94, \textquotesingle{}(12C)H3(81Br)\textquotesingle{}, 0.4874334, 95.93976, \textquotesingle{}CH3Br\textquotesingle{}{]}, (41, 1): {[}95, \textquotesingle{}(12C)H3(12C)(14N)\textquotesingle{}, 0.9738662, 41.02655, \textquotesingle{}CH3CN\textquotesingle{}{]}, (42, 1): {[}96, \textquotesingle{}(12C)(19F)4\textquotesingle{}, 0.98889, 87.99362, \textquotesingle{}CF4\textquotesingle{}{]}, (43, 1): {[}116, \textquotesingle{}(12C)4H2\textquotesingle{}, 0.955998, 50.01565, \textquotesingle{}C4H2\textquotesingle{}{]}, (44, 1): {[}109, \textquotesingle{}H(12C)3(14N)\textquotesingle{}, 0.963346, 51.0109, \textquotesingle{}HC3N\textquotesingle{}{]}, (45, 1): {[}103, \textquotesingle{}H2\textquotesingle{}, 0.9996885, 2.01565, \textquotesingle{}H2\textquotesingle{}{]}, (45, 2): {[}115, \textquotesingle{}HD\textquotesingle{}, 0.0003114316, 3.021825, \textquotesingle{}H2\textquotesingle{}{]}, (46, 1): {[}97, \textquotesingle{}(12C)(32S)\textquotesingle{}, 0.9396236, 43.97207, \textquotesingle{}CS\textquotesingle{}{]}, (46, 2): {[}98, \textquotesingle{}(12C)(34S)\textquotesingle{}, 0.04168171, 45.96787, \textquotesingle{}CS\textquotesingle{}{]}, (46, 3): {[}99, \textquotesingle{}(13C)(32S)\textquotesingle{}, 0.0105565, 44.97543, \textquotesingle{}CS\textquotesingle{}{]}, (46, 4): {[}100, \textquotesingle{}(12C)(33S)\textquotesingle{}, 0.007416675, 44.97146, \textquotesingle{}CS\textquotesingle{}{]}, (47, 1): {[}114, \textquotesingle{}(32S)(16O)3\textquotesingle{}, 0.9434345, 79.95682, \textquotesingle{}SO3\textquotesingle{}{]}, (48, 1): {[}123, \textquotesingle{}(12C)2(14N)2\textquotesingle{}, 0.9707524, 52.00615, \textquotesingle{}C2N2\textquotesingle{}{]}, (49, 1): {[}124, \textquotesingle{}(12C)(16O)(35Cl)2\textquotesingle{}, 0.5663918, 97.93262, \textquotesingle{}COCl2\textquotesingle{}{]}, (49, 2): {[}125, \textquotesingle{}(12C)(16O)(35Cl)(37Cl)\textquotesingle{}, 0.362235, 99.92967, \textquotesingle{}COCl2\textquotesingle{}{]}, (50, 1): {[}146, \textquotesingle{}(32S)(16O)\textquotesingle{}, 0.9479262, 47.96699, \textquotesingle{}SO\textquotesingle{}{]}, (50, 2): {[}147, \textquotesingle{}(34S)(16O)\textquotesingle{}, 0.04205002, 49.96278, \textquotesingle{}SO\textquotesingle{}{]}, (50, 3): {[}148, \textquotesingle{}(32S)(18O)\textquotesingle{}, 0.001900788, 49.97123, \textquotesingle{}SO\textquotesingle{}{]}, (51, 1): {[}144, \textquotesingle{}(12C)H3(19F)\textquotesingle{}, 0.988428, 34.02188, \textquotesingle{}CH3F\textquotesingle{}{]}, (52, 1): {[}139, \textquotesingle{}(74Ge)H4\textquotesingle{}, 0.3651724, 77.95248, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 2): {[}140, \textquotesingle{}(72Ge)H4\textquotesingle{}, 0.2741292, 75.95338, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 3): {[}141, \textquotesingle{}(70Ge)H4\textquotesingle{}, 0.2050722, 73.95555, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 4): {[}142, \textquotesingle{}(73Ge)H4\textquotesingle{}, 0.07755167, 76.95476, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 5): {[}143, \textquotesingle{}(76Ge)H4\textquotesingle{}, 0.07755167, 79.9527, \textquotesingle{}GeH4\textquotesingle{}{]}, (53, 1): {[}131, \textquotesingle{}(12C)(32S)2\textquotesingle{}, 0.8928115, 75.94414, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 2): {[}132, \textquotesingle{}(32S)(12C)(34S)\textquotesingle{}, 0.07921026, 77.93994, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 3): {[}133, \textquotesingle{}(32S)(12C)(33S)\textquotesingle{}, 0.01409435, 76.94353, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 4): {[}134, \textquotesingle{}(13C)(32S)2\textquotesingle{}, 0.01003057, 76.9475, \textquotesingle{}CS2\textquotesingle{}{]}, (54, 1): {[}145, \textquotesingle{}(12C)H3(127I)\textquotesingle{}, 0.988428, 141.9279, \textquotesingle{}CH3I\textquotesingle{}{]}, (55, 1): {[}136, \textquotesingle{}(14N)(19F)3\textquotesingle{}, 0.996337, 70.99829, \textquotesingle{}NF3\textquotesingle{}{]}\}, natural\_abundance=True, abundance\_ratio\_MI=\{\}, Diluent=\{\}, diluent=\textquotesingle{}air\textquotesingle{}, IntensityThreshold=1e\sphinxhyphen{}30, TIPS=\textless{}function \textless{}lambda\textgreater{}\textgreater{}, compressability\_factor=1}}{}
\pysigstopsignatures
\sphinxAtStartPar
Calculates the absorbance (ppm/cm) based on input line list, wavenumbers, and spectrum environmental parameters.

\sphinxAtStartPar
Outline
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Uses provided wavenumber axis

\item {} 
\sphinxAtStartPar
Calculates the molecular density from pressure and temperature

\item {} 
\sphinxAtStartPar
Sets up Diluent dictionary if not given as input

\item {} 
\sphinxAtStartPar
Calculates line intensity and doppler width at temperature for all lines

\item {} 
\sphinxAtStartPar
Loops through each line in the line list and loops through each diluent, generating a line parameter at experimental conditions
that is the appropriate ratio of each diluent species corrected for pressure and temperature.  For each line, simulate the line for the given simulation cutoffs and add to cross section

\item {} 
\sphinxAtStartPar
Return wavenumber and cross section arrays

\end{enumerate}
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{linelist}} (\sphinxstyleliteralemphasis{\sphinxupquote{dataframe}}) \textendash{} \begin{description}
\sphinxlineitem{Pandas dataframe with the following column headers, where species corresponds to each diluent in the spectrum objects included in the dataset and nominal temperature corresponds to the nominal temperatures included in the dataset:}
\sphinxAtStartPar
nu = wavenumber of the spectral line transition (cm\sphinxhyphen{}1) in vacuum

\sphinxAtStartPar
sw = The spectral line intensity (cm−1/(molecule⋅cm−2)) at Tref=296K

\sphinxAtStartPar
elower = The lower\sphinxhyphen{}state energy of the transition (cm\sphinxhyphen{}1)

\sphinxAtStartPar
molec\_id = HITRAN molecular ID number

\sphinxAtStartPar
local\_iso\_id = HITRAN isotopologue ID number

\sphinxAtStartPar
gamma\_0\_species = half width at half maximum (HWHM) (cm−1/atm) at Tref=296K and reference pressure pref=1atm for a given diluent (air, self, CO2, etc.)

\sphinxAtStartPar
n\_gamma0\_species = The coefficient of the temperature dependence of the half width

\sphinxAtStartPar
delta\_0\_species = The pressure shift (cm−1/atm) at Tref=296K and pref=1atm of the line position with respect to the vacuum transition wavenumber νij

\sphinxAtStartPar
n\_delta0\_species = the coefficient of the temperature dependence of the pressure shift

\sphinxAtStartPar
SD\_gamma\_species = the ratio of the speed dependent width to the half\sphinxhyphen{}width at reference temperature and pressure

\sphinxAtStartPar
n\_gamma2\_species = the coefficient of the temperature dependence of the speed dependent width NOTE: This is the temperature dependence of the speed dependent width not the ratio of the speed dependence to the half\sphinxhyphen{}width

\sphinxAtStartPar
SD\_delta\_species = the ratio of the speed dependent shift to the collisional shift at reference temperature and pressure

\sphinxAtStartPar
n\_delta2\_species = the coefficient of the temperature dependence of the speed dependent shift NOTE: This is the temperature dependence of the speed dependent shift not the ratio of the speed dependence to the shift

\sphinxAtStartPar
nuVC\_species = dicke narrowing term at reference temperature

\sphinxAtStartPar
n\_nuVC\_species = coefficient of the temperature dependence of the dicke narrowing term

\sphinxAtStartPar
eta\_species = the correlation parameter for the VC and SD effects

\sphinxAtStartPar
y\_species = linemixing term

\sphinxAtStartPar
n\_y\_species = temperature dependence of linemixing term

\end{description}


\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{waves}} (\sphinxstyleliteralemphasis{\sphinxupquote{array}}) \textendash{} 1\sphinxhyphen{}D array comprised of wavenumbers (cm\sphinxhyphen{}1) to use as x\sphinxhyphen{}axis for simulation.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_cutoff}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of half\sphinxhyphen{}widths for each line to be calculated for. The default is 50.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_wavenumbers}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of wavenumber for each line to be calculated. The default is 50.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_method}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} defines which wing cut\sphinxhyphen{}off option to use.  Options are wing\_cutoff or wing\_wavenumbers The default is ‘wing\_cutoff’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{p}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} pressure for simulation in atmospheres. The default is 1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{T}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} temperature for simulation in kelvin. The default is 296.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} takes the form \{molecule\_id : mole fraction, molecule\_id: mole fraction, …\}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{isotope\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} provides opportunity to specify the isotope look\sphinxhyphen{}up table.  Default is ISO, which is from HAPI.  If not using ISO, then must use this format and suggested you use function to add to ISO

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{natural\_abundance}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True indicates the sample is at natural abundance. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{abundance\_ratio\_MI}} (\sphinxstyleliteralemphasis{\sphinxupquote{dictionary}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If sample is not at natural abundance, then the natural abundance defines the enrichment factor compared to natural abundance(ie a sample where the main isotope is the only isotope would have a 1/natural abundance as the enrichment factor). Defined as \{M:\{I:enrichment factor, I: enrichment factor, I: enrichment factor\}, … \}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{Diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} contains the species as the key with the value being the abundance of that diluent in the sample, ie \{‘He’:0.5, ‘self’:0.5\}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If Diluent = \{\}, then this value will be used to set the only diluent to be equal to this diluent. The default is ‘air’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{IntensityThreshold}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} minimum line intensity that will be simulated. The default is 1e\sphinxhyphen{}30.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{TIPS}} (\sphinxstyleliteralemphasis{\sphinxupquote{definition}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} selects the HAPI provided TIPS version to use for the partition function

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{wavenumbers} (\sphinxstyleemphasis{array}) \textendash{} wavenumber axis that should match the input waves

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{xsect} (\sphinxstyleemphasis{array}) \textendash{} simulated cross section as a function of wavenumbers (ppm/cm)

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{HTP\_wBeta\_from\_DF\_select() (in module MATS.fit\_dataset)@\spxentry{HTP\_wBeta\_from\_DF\_select()}\spxextra{in module MATS.fit\_dataset}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.fit_dataset.HTP_wBeta_from_DF_select}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.fit\_dataset.}}\sphinxbfcode{\sphinxupquote{HTP\_wBeta\_from\_DF\_select}}}{\emph{linelist, waves, wing\_cutoff=25, wing\_wavenumbers=25, wing\_method=\textquotesingle{}wing\_cutoff\textquotesingle{}, p=1, T=296, molefraction=\{\}, isotope\_list=\{(1, 1): {[}1, \textquotesingle{}H2(16O)\textquotesingle{}, 0.9973173, 18.01056, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 2): {[}2, \textquotesingle{}H2(18O)\textquotesingle{}, 0.001999827, 20.01481, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 3): {[}3, \textquotesingle{}H2(17O)\textquotesingle{}, 0.0003718841, 19.01478, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 4): {[}4, \textquotesingle{}HD(16O)\textquotesingle{}, 0.0003106928, 19.01674, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 5): {[}5, \textquotesingle{}HD(18O)\textquotesingle{}, 6.230031e\sphinxhyphen{}07, 21.02098, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 6): {[}6, \textquotesingle{}HD(17O)\textquotesingle{}, 1.158526e\sphinxhyphen{}07, 20.02096, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 7): {[}129, \textquotesingle{}D2(16O)\textquotesingle{}, 2.419741e\sphinxhyphen{}08, 20.02292, \textquotesingle{}H2O\textquotesingle{}{]}, (2, 1): {[}7, \textquotesingle{}(12C)(16O)2\textquotesingle{}, 0.9842043, 43.98983, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 2): {[}8, \textquotesingle{}(13C)(16O)2\textquotesingle{}, 0.01105736, 44.99318, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 3): {[}9, \textquotesingle{}(16O)(12C)(18O)\textquotesingle{}, 0.003947066, 45.99408, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 4): {[}10, \textquotesingle{}(16O)(12C)(17O)\textquotesingle{}, 0.000733989, 44.99404, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 5): {[}11, \textquotesingle{}(16O)(13C)(18O)\textquotesingle{}, 4.434456e\sphinxhyphen{}05, 46.99743, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 6): {[}12, \textquotesingle{}(16O)(13C)(17O)\textquotesingle{}, 8.246233e\sphinxhyphen{}06, 45.9974, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 7): {[}13, \textquotesingle{}(12C)(18O)2\textquotesingle{}, 3.95734e\sphinxhyphen{}06, 47.99832, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 8): {[}14, \textquotesingle{}(17O)(12C)(18O)\textquotesingle{}, 1.471799e\sphinxhyphen{}06, 46.99829, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 9): {[}121, \textquotesingle{}(12C)(17O)2\textquotesingle{}, 1.368466e\sphinxhyphen{}07, 45.99826, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 10): {[}15, \textquotesingle{}(13C)(18O)2\textquotesingle{}, 4.446e\sphinxhyphen{}08, 49.00167, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 11): {[}120, \textquotesingle{}(18O)(13C)(17O)\textquotesingle{}, 1.65354e\sphinxhyphen{}08, 48.00165, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 12): {[}122, \textquotesingle{}(13C)(17O)2\textquotesingle{}, 1.537446e\sphinxhyphen{}09, 47.00162, \textquotesingle{}CO2\textquotesingle{}{]}, (3, 1): {[}16, \textquotesingle{}(16O)3\textquotesingle{}, 0.9929009, 47.98474, \textquotesingle{}O3\textquotesingle{}{]}, (3, 2): {[}17, \textquotesingle{}(16O)(16O)(18O)\textquotesingle{}, 0.003981942, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 3): {[}18, \textquotesingle{}(16O)(18O)(16O)\textquotesingle{}, 0.001990971, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 4): {[}19, \textquotesingle{}(16O)(16O)(17O)\textquotesingle{}, 0.0007404746, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (3, 5): {[}20, \textquotesingle{}(16O)(17O)(16O)\textquotesingle{}, 0.0003702373, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (4, 1): {[}21, \textquotesingle{}(14N)2(16O)\textquotesingle{}, 0.9903328, 44.00106, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 2): {[}22, \textquotesingle{}(14N)(15N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 3): {[}23, \textquotesingle{}(15N)(14N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 4): {[}24, \textquotesingle{}(14N)2(18O)\textquotesingle{}, 0.001985822, 46.00531, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 5): {[}25, \textquotesingle{}(14N)2(17O)\textquotesingle{}, 0.0003692797, 45.00528, \textquotesingle{}N2O\textquotesingle{}{]}, (5, 1): {[}26, \textquotesingle{}(12C)(16O)\textquotesingle{}, 0.9865444, 27.99491, \textquotesingle{}CO\textquotesingle{}{]}, (5, 2): {[}27, \textquotesingle{}(13C)(16O)\textquotesingle{}, 0.01108364, 28.99827, \textquotesingle{}CO\textquotesingle{}{]}, (5, 3): {[}28, \textquotesingle{}(12C)(18O)\textquotesingle{}, 0.001978224, 29.99916, \textquotesingle{}CO\textquotesingle{}{]}, (5, 4): {[}29, \textquotesingle{}(12C)(17O)\textquotesingle{}, 0.0003678671, 28.99913, \textquotesingle{}CO\textquotesingle{}{]}, (5, 5): {[}30, \textquotesingle{}(13C)(18O)\textquotesingle{}, 2.2225e\sphinxhyphen{}05, 31.00252, \textquotesingle{}CO\textquotesingle{}{]}, (5, 6): {[}31, \textquotesingle{}(13C)(17O)\textquotesingle{}, 4.13292e\sphinxhyphen{}06, 30.00249, \textquotesingle{}CO\textquotesingle{}{]}, (6, 1): {[}32, \textquotesingle{}(12C)H4\textquotesingle{}, 0.9882741, 16.0313, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 2): {[}33, \textquotesingle{}(13C)H4\textquotesingle{}, 0.01110308, 17.03466, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 3): {[}34, \textquotesingle{}(12C)H3D\textquotesingle{}, 0.0006157511, 17.03748, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 4): {[}35, \textquotesingle{}(13C)H3D\textquotesingle{}, 6.917852e\sphinxhyphen{}06, 18.04083, \textquotesingle{}CH4\textquotesingle{}{]}, (7, 1): {[}36, \textquotesingle{}(16O)2\textquotesingle{}, 0.9952616, 31.98983, \textquotesingle{}O2\textquotesingle{}{]}, (7, 2): {[}37, \textquotesingle{}(16O)(18O)\textquotesingle{}, 0.00399141, 33.99408, \textquotesingle{}O2\textquotesingle{}{]}, (7, 3): {[}38, \textquotesingle{}(16O)(17O)\textquotesingle{}, 0.0007422352, 32.99404, \textquotesingle{}O2\textquotesingle{}{]}, (8, 1): {[}39, \textquotesingle{}(14N)(16O)\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NO\textquotesingle{}{]}, (8, 2): {[}40, \textquotesingle{}(15N)(16O)\textquotesingle{}, 0.003654311, 30.99502, \textquotesingle{}NO\textquotesingle{}{]}, (8, 3): {[}41, \textquotesingle{}(14N)(18O)\textquotesingle{}, 0.001993122, 32.00223, \textquotesingle{}NO\textquotesingle{}{]}, (9, 1): {[}42, \textquotesingle{}(32S)(16O)2\textquotesingle{}, 0.9456777, 63.9619, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 2): {[}43, \textquotesingle{}(34S)(16O)2\textquotesingle{}, 0.04195028, 65.9577, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 3): {[}137, \textquotesingle{}(33S)(16O)2\textquotesingle{}, 0.007464462, 64.96129, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 4): {[}138, \textquotesingle{}(16O)(32S)(18O)\textquotesingle{}, 0.003792558, 65.96615, \textquotesingle{}SO2\textquotesingle{}{]}, (10, 1): {[}44, \textquotesingle{}(14N)(16O)2\textquotesingle{}, 0.991616, 45.9929, \textquotesingle{}NO2\textquotesingle{}{]}, (10, 2): {[}130, \textquotesingle{}(15N)(16O)2\textquotesingle{}, 0.003645643, 46.98994, \textquotesingle{}NO2\textquotesingle{}{]}, (11, 1): {[}45, \textquotesingle{}(14N)H3\textquotesingle{}, 0.9958716, 17.02655, \textquotesingle{}NH3\textquotesingle{}{]}, (11, 2): {[}46, \textquotesingle{}(15N)H3\textquotesingle{}, 0.003661289, 18.02358, \textquotesingle{}NH3\textquotesingle{}{]}, (12, 1): {[}47, \textquotesingle{}H(14N)(16O)3\textquotesingle{}, 0.9891098, 62.99564, \textquotesingle{}HNO3\textquotesingle{}{]}, (12, 2): {[}117, \textquotesingle{}H(15N)(16O)3\textquotesingle{}, 0.003636429, 63.99268, \textquotesingle{}HNO3\textquotesingle{}{]}, (13, 1): {[}48, \textquotesingle{}(16O)H\textquotesingle{}, 0.9974726, 17.00274, \textquotesingle{}OH\textquotesingle{}{]}, (13, 2): {[}49, \textquotesingle{}(18O)H\textquotesingle{}, 0.002000138, 19.00699, \textquotesingle{}OH\textquotesingle{}{]}, (13, 3): {[}50, \textquotesingle{}(16O)D\textquotesingle{}, 0.0001553706, 18.00891, \textquotesingle{}OH\textquotesingle{}{]}, (14, 1): {[}51, \textquotesingle{}H(19F)\textquotesingle{}, 0.9998443, 20.00623, \textquotesingle{}HF\textquotesingle{}{]}, (14, 2): {[}110, \textquotesingle{}D(19F)\textquotesingle{}, 0.000155741, 21.0124, \textquotesingle{}HF\textquotesingle{}{]}, (15, 1): {[}52, \textquotesingle{}H(35Cl)\textquotesingle{}, 0.757587, 35.97668, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 2): {[}53, \textquotesingle{}H(37Cl)\textquotesingle{}, 0.2422573, 37.97373, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 3): {[}107, \textquotesingle{}D(35Cl)\textquotesingle{}, 0.000118005, 36.98285, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 4): {[}108, \textquotesingle{}D(37Cl)\textquotesingle{}, 3.773502e\sphinxhyphen{}05, 38.9799, \textquotesingle{}HCl\textquotesingle{}{]}, (16, 1): {[}54, \textquotesingle{}H(79Br)\textquotesingle{}, 0.5067811, 79.92616, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 2): {[}55, \textquotesingle{}H(81Br)\textquotesingle{}, 0.4930632, 81.92412, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 3): {[}111, \textquotesingle{}D(79Br)\textquotesingle{}, 7.893838e\sphinxhyphen{}05, 80.93234, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 4): {[}112, \textquotesingle{}D(81Br)\textquotesingle{}, 7.680162e\sphinxhyphen{}05, 82.93029, \textquotesingle{}HBr\textquotesingle{}{]}, (17, 1): {[}56, \textquotesingle{}H(127I)\textquotesingle{}, 0.9998443, 127.9123, \textquotesingle{}HI\textquotesingle{}{]}, (17, 2): {[}113, \textquotesingle{}D(127I)\textquotesingle{}, 0.000155741, 128.9185, \textquotesingle{}HI\textquotesingle{}{]}, (18, 1): {[}57, \textquotesingle{}(35Cl)(16O)\textquotesingle{}, 0.7559077, 50.96377, \textquotesingle{}ClO\textquotesingle{}{]}, (18, 2): {[}58, \textquotesingle{}(37Cl)(16O)\textquotesingle{}, 0.2417203, 52.96082, \textquotesingle{}ClO\textquotesingle{}{]}, (19, 1): {[}59, \textquotesingle{}(16O)(12C)(32S)\textquotesingle{}, 0.9373947, 59.96699, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 2): {[}60, \textquotesingle{}(16O)(12C)(34S)\textquotesingle{}, 0.04158284, 61.96278, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 3): {[}61, \textquotesingle{}(16O)(13C)(32S)\textquotesingle{}, 0.01053146, 60.97034, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 4): {[}62, \textquotesingle{}(16O)(12C)(33S)\textquotesingle{}, 0.007399083, 60.96637, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 5): {[}63, \textquotesingle{}(18O)(12C)(32S)\textquotesingle{}, 0.00187967, 61.97123, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 6): {[}135, \textquotesingle{}(16O)(13C)(34S)\textquotesingle{}, 0.0004671757, 62.96614, \textquotesingle{}OCS\textquotesingle{}{]}, (20, 1): {[}64, \textquotesingle{}H2(12C)(16O)\textquotesingle{}, 0.9862371, 30.01056, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 2): {[}65, \textquotesingle{}H2(13C)(16O)\textquotesingle{}, 0.0110802, 31.01392, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 3): {[}66, \textquotesingle{}H2(12C)(18O)\textquotesingle{}, 0.001977609, 32.01481, \textquotesingle{}H2CO\textquotesingle{}{]}, (21, 1): {[}67, \textquotesingle{}H(16O)(35Cl)\textquotesingle{}, 0.75579, 51.97159, \textquotesingle{}HOCl\textquotesingle{}{]}, (21, 2): {[}68, \textquotesingle{}H(16O)(37Cl)\textquotesingle{}, 0.2416826, 53.96864, \textquotesingle{}HOCl\textquotesingle{}{]}, (22, 1): {[}69, \textquotesingle{}(14N)2\textquotesingle{}, 0.9926874, 28.00615, \textquotesingle{}N2\textquotesingle{}{]}, (22, 2): {[}118, \textquotesingle{}(14N)(15N)\textquotesingle{}, 0.007299165, 29.00318, \textquotesingle{}N2\textquotesingle{}{]}, (23, 1): {[}70, \textquotesingle{}H(12C)(14N)\textquotesingle{}, 0.9851143, 27.0109, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 2): {[}71, \textquotesingle{}H(13C)(14N)\textquotesingle{}, 0.01106758, 28.01425, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 3): {[}72, \textquotesingle{}H(12C)(15N)\textquotesingle{}, 0.00362174, 28.00793, \textquotesingle{}HCN\textquotesingle{}{]}, (24, 1): {[}73, \textquotesingle{}(12C)H3(35Cl)\textquotesingle{}, 0.7489369, 49.99233, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (24, 2): {[}74, \textquotesingle{}(12C)H3(37Cl)\textquotesingle{}, 0.2394912, 51.98938, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (25, 1): {[}75, \textquotesingle{}H2(16O)2\textquotesingle{}, 0.9949516, 34.00548, \textquotesingle{}H2O2\textquotesingle{}{]}, (26, 1): {[}76, \textquotesingle{}(12C)2H2\textquotesingle{}, 0.9775989, 26.01565, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 2): {[}77, \textquotesingle{}(12C)(13C)H2\textquotesingle{}, 0.02196629, 27.019, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 3): {[}105, \textquotesingle{}(12C)2HD\textquotesingle{}, 0.0003045499, 27.02182, \textquotesingle{}C2H2\textquotesingle{}{]}, (27, 1): {[}78, \textquotesingle{}(12C)2H6\textquotesingle{}, 0.97699, 30.04695, \textquotesingle{}C2H6\textquotesingle{}{]}, (27, 2): {[}106, \textquotesingle{}(12C)H3(13C)H3\textquotesingle{}, 0.02195261, 31.05031, \textquotesingle{}C2H6\textquotesingle{}{]}, (28, 1): {[}79, \textquotesingle{}(31P)H3\textquotesingle{}, 0.9995329, 33.99724, \textquotesingle{}PH3\textquotesingle{}{]}, (29, 1): {[}80, \textquotesingle{}(12C)(16O)(19F)2\textquotesingle{}, 0.9865444, 65.99172, \textquotesingle{}COF2\textquotesingle{}{]}, (29, 2): {[}119, \textquotesingle{}(13C)(16O)(19F)2\textquotesingle{}, 0.01108366, 66.99508, \textquotesingle{}COF2\textquotesingle{}{]}, (30, 1): {[}126, \textquotesingle{}(32S)(19F)6\textquotesingle{}, 0.95018, 145.9625, \textquotesingle{}SF6\textquotesingle{}{]}, (31, 1): {[}81, \textquotesingle{}H2(32S)\textquotesingle{}, 0.9498841, 33.98772, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 2): {[}82, \textquotesingle{}H2(34S)\textquotesingle{}, 0.04213687, 35.98351, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 3): {[}83, \textquotesingle{}H2(33S)\textquotesingle{}, 0.007497664, 34.9871, \textquotesingle{}H2S\textquotesingle{}{]}, (32, 1): {[}84, \textquotesingle{}H(12C)(16O)(16O)H\textquotesingle{}, 0.9838977, 46.00548, \textquotesingle{}HCOOH\textquotesingle{}{]}, (33, 1): {[}85, \textquotesingle{}H(16O)2\textquotesingle{}, 0.9951066, 32.99766, \textquotesingle{}HO2\textquotesingle{}{]}, (34, 1): {[}86, \textquotesingle{}(16O)\textquotesingle{}, 0.997628, 15.99492, \textquotesingle{}O\textquotesingle{}{]}, (35, 1): {[}127, \textquotesingle{}(35Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.7495702, 96.95667, \textquotesingle{}ClONO2\textquotesingle{}{]}, (35, 2): {[}128, \textquotesingle{}(37Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.2396937, 98.95372, \textquotesingle{}ClONO2\textquotesingle{}{]}, (36, 1): {[}87, \textquotesingle{}(14N)(16O)+\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NOp\textquotesingle{}{]}, (37, 1): {[}88, \textquotesingle{}H(16O)(79Br)\textquotesingle{}, 0.505579, 95.92108, \textquotesingle{}HOBr\textquotesingle{}{]}, (37, 2): {[}89, \textquotesingle{}H(16O)(81Br)\textquotesingle{}, 0.4918937, 97.91903, \textquotesingle{}HOBr\textquotesingle{}{]}, (38, 1): {[}90, \textquotesingle{}(12C)2H4\textquotesingle{}, 0.9772944, 28.0313, \textquotesingle{}C2H4\textquotesingle{}{]}, (38, 2): {[}91, \textquotesingle{}(12C)H2(13C)H2\textquotesingle{}, 0.02195946, 29.03466, \textquotesingle{}C2H4\textquotesingle{}{]}, (39, 1): {[}92, \textquotesingle{}(12C)H3(16O)H\textquotesingle{}, 0.9859299, 32.02622, \textquotesingle{}CH3OH\textquotesingle{}{]}, (40, 1): {[}93, \textquotesingle{}(12C)H3(79Br)\textquotesingle{}, 0.5009946, 93.94181, \textquotesingle{}CH3Br\textquotesingle{}{]}, (40, 2): {[}94, \textquotesingle{}(12C)H3(81Br)\textquotesingle{}, 0.4874334, 95.93976, \textquotesingle{}CH3Br\textquotesingle{}{]}, (41, 1): {[}95, \textquotesingle{}(12C)H3(12C)(14N)\textquotesingle{}, 0.9738662, 41.02655, \textquotesingle{}CH3CN\textquotesingle{}{]}, (42, 1): {[}96, \textquotesingle{}(12C)(19F)4\textquotesingle{}, 0.98889, 87.99362, \textquotesingle{}CF4\textquotesingle{}{]}, (43, 1): {[}116, \textquotesingle{}(12C)4H2\textquotesingle{}, 0.955998, 50.01565, \textquotesingle{}C4H2\textquotesingle{}{]}, (44, 1): {[}109, \textquotesingle{}H(12C)3(14N)\textquotesingle{}, 0.963346, 51.0109, \textquotesingle{}HC3N\textquotesingle{}{]}, (45, 1): {[}103, \textquotesingle{}H2\textquotesingle{}, 0.9996885, 2.01565, \textquotesingle{}H2\textquotesingle{}{]}, (45, 2): {[}115, \textquotesingle{}HD\textquotesingle{}, 0.0003114316, 3.021825, \textquotesingle{}H2\textquotesingle{}{]}, (46, 1): {[}97, \textquotesingle{}(12C)(32S)\textquotesingle{}, 0.9396236, 43.97207, \textquotesingle{}CS\textquotesingle{}{]}, (46, 2): {[}98, \textquotesingle{}(12C)(34S)\textquotesingle{}, 0.04168171, 45.96787, \textquotesingle{}CS\textquotesingle{}{]}, (46, 3): {[}99, \textquotesingle{}(13C)(32S)\textquotesingle{}, 0.0105565, 44.97543, \textquotesingle{}CS\textquotesingle{}{]}, (46, 4): {[}100, \textquotesingle{}(12C)(33S)\textquotesingle{}, 0.007416675, 44.97146, \textquotesingle{}CS\textquotesingle{}{]}, (47, 1): {[}114, \textquotesingle{}(32S)(16O)3\textquotesingle{}, 0.9434345, 79.95682, \textquotesingle{}SO3\textquotesingle{}{]}, (48, 1): {[}123, \textquotesingle{}(12C)2(14N)2\textquotesingle{}, 0.9707524, 52.00615, \textquotesingle{}C2N2\textquotesingle{}{]}, (49, 1): {[}124, \textquotesingle{}(12C)(16O)(35Cl)2\textquotesingle{}, 0.5663918, 97.93262, \textquotesingle{}COCl2\textquotesingle{}{]}, (49, 2): {[}125, \textquotesingle{}(12C)(16O)(35Cl)(37Cl)\textquotesingle{}, 0.362235, 99.92967, \textquotesingle{}COCl2\textquotesingle{}{]}, (50, 1): {[}146, \textquotesingle{}(32S)(16O)\textquotesingle{}, 0.9479262, 47.96699, \textquotesingle{}SO\textquotesingle{}{]}, (50, 2): {[}147, \textquotesingle{}(34S)(16O)\textquotesingle{}, 0.04205002, 49.96278, \textquotesingle{}SO\textquotesingle{}{]}, (50, 3): {[}148, \textquotesingle{}(32S)(18O)\textquotesingle{}, 0.001900788, 49.97123, \textquotesingle{}SO\textquotesingle{}{]}, (51, 1): {[}144, \textquotesingle{}(12C)H3(19F)\textquotesingle{}, 0.988428, 34.02188, \textquotesingle{}CH3F\textquotesingle{}{]}, (52, 1): {[}139, \textquotesingle{}(74Ge)H4\textquotesingle{}, 0.3651724, 77.95248, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 2): {[}140, \textquotesingle{}(72Ge)H4\textquotesingle{}, 0.2741292, 75.95338, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 3): {[}141, \textquotesingle{}(70Ge)H4\textquotesingle{}, 0.2050722, 73.95555, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 4): {[}142, \textquotesingle{}(73Ge)H4\textquotesingle{}, 0.07755167, 76.95476, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 5): {[}143, \textquotesingle{}(76Ge)H4\textquotesingle{}, 0.07755167, 79.9527, \textquotesingle{}GeH4\textquotesingle{}{]}, (53, 1): {[}131, \textquotesingle{}(12C)(32S)2\textquotesingle{}, 0.8928115, 75.94414, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 2): {[}132, \textquotesingle{}(32S)(12C)(34S)\textquotesingle{}, 0.07921026, 77.93994, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 3): {[}133, \textquotesingle{}(32S)(12C)(33S)\textquotesingle{}, 0.01409435, 76.94353, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 4): {[}134, \textquotesingle{}(13C)(32S)2\textquotesingle{}, 0.01003057, 76.9475, \textquotesingle{}CS2\textquotesingle{}{]}, (54, 1): {[}145, \textquotesingle{}(12C)H3(127I)\textquotesingle{}, 0.988428, 141.9279, \textquotesingle{}CH3I\textquotesingle{}{]}, (55, 1): {[}136, \textquotesingle{}(14N)(19F)3\textquotesingle{}, 0.996337, 70.99829, \textquotesingle{}NF3\textquotesingle{}{]}\}, natural\_abundance=True, abundance\_ratio\_MI=\{\}, Diluent=\{\}, diluent=\textquotesingle{}air\textquotesingle{}, IntensityThreshold=1e\sphinxhyphen{}30, TIPS=\textless{}function \textless{}lambda\textgreater{}\textgreater{}, compressability\_factor=1}}{}
\pysigstopsignatures
\sphinxAtStartPar
Calculates the absorbance (ppm/cm) based on input line list, wavenumbers, and spectrum environmental parameters with capability of incorporating the beta correction to the Dicke Narrowing proposed in Analytical\sphinxhyphen{}function correction to the Hartmann\textendash{}Tran profile for more reliable representation of the Dicke\sphinxhyphen{}narrowed molecular spectra.

\sphinxAtStartPar
Outline
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Uses provided wavenumber axis

\item {} 
\sphinxAtStartPar
Calculates the molecular density from pressure and temperature

\item {} 
\sphinxAtStartPar
Sets up Diluent dictionary if not given as input

\item {} 
\sphinxAtStartPar
Calculates line intensity and doppler width at temperature for all lines

\item {} 
\sphinxAtStartPar
Loops through each line in the line list and loops through each diluent, generating a line parameter at experimental conditions
that is the appropriate ratio of each diluent species corrected for pressure and temperature.  For each line, simulate the line for the given simulation cutoffs and add to cross section

\item {} 
\sphinxAtStartPar
Return wavenumber and cross section arrays

\end{enumerate}
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{linelist}} (\sphinxstyleliteralemphasis{\sphinxupquote{dataframe}}) \textendash{} \begin{description}
\sphinxlineitem{Pandas dataframe with the following column headers, where species corresponds to each diluent in the spectrum objects included in the dataset and nominal temperature corresponds to the nominal temperatures included in the dataset:}
\sphinxAtStartPar
nu = wavenumber of the spectral line transition (cm\sphinxhyphen{}1) in vacuum

\sphinxAtStartPar
sw = The spectral line intensity (cm−1/(molecule⋅cm−2)) at Tref=296K

\sphinxAtStartPar
elower = The lower\sphinxhyphen{}state energy of the transition (cm\sphinxhyphen{}1)

\sphinxAtStartPar
molec\_id = HITRAN molecular ID number

\sphinxAtStartPar
local\_iso\_id = HITRAN isotopologue ID number

\sphinxAtStartPar
gamma\_0\_species = half width at half maximum (HWHM) (cm−1/atm) at Tref=296K and reference pressure pref=1atm for a given diluent (air, self, CO2, etc.)

\sphinxAtStartPar
n\_gamma0\_species = The coefficient of the temperature dependence of the half width

\sphinxAtStartPar
delta\_0\_species = The pressure shift (cm−1/atm) at Tref=296K and pref=1atm of the line position with respect to the vacuum transition wavenumber νij

\sphinxAtStartPar
n\_delta0\_species = the coefficient of the temperature dependence of the pressure shift

\sphinxAtStartPar
SD\_gamma\_species = the ratio of the speed dependent width to the half\sphinxhyphen{}width at reference temperature and pressure

\sphinxAtStartPar
n\_gamma2\_species = the coefficient of the temperature dependence of the speed dependent width NOTE: This is the temperature dependence of the speed dependent width not the ratio of the speed dependence to the half\sphinxhyphen{}width

\sphinxAtStartPar
SD\_delta\_species = the ratio of the speed dependent shift to the collisional shift at reference temperature and pressure

\sphinxAtStartPar
n\_delta2\_species = the coefficient of the temperature dependence of the speed dependent shift NOTE: This is the temperature dependence of the speed dependent shift not the ratio of the speed dependence to the shift

\sphinxAtStartPar
nuVC\_species = dicke narrowing term at reference temperature

\sphinxAtStartPar
n\_nuVC\_species = coefficient of the temperature dependence of the dicke narrowing term

\sphinxAtStartPar
eta\_species = the correlation parameter for the VC and SD effects

\sphinxAtStartPar
y\_species = linemixing term

\sphinxAtStartPar
n\_y\_species = temperature dependence of line mixing term

\end{description}


\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{waves}} (\sphinxstyleliteralemphasis{\sphinxupquote{array}}) \textendash{} 1\sphinxhyphen{}D array comprised of wavenumbers (cm\sphinxhyphen{}1) to use as x\sphinxhyphen{}axis for simulation.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_cutoff}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of half\sphinxhyphen{}widths for each line to be calculated for. The default is 50.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_wavenumbers}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of wavenumber for each line to be calculated. The default is 50.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_method}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} defines which wing cut\sphinxhyphen{}off option to use.  Options are wing\_cutoff or wing\_wavenumbers The default is ‘wing\_cutoff’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{p}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} pressure for simulation in atmospheres. The default is 1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{T}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} temperature for simulation in kelvin. The default is 296.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} takes the form \{molecule\_id : mole fraction, molecule\_id: mole fraction, …\}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{isotope\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} provides opportunity to specify the isotope look\sphinxhyphen{}up table.  Default is ISO, which is from HAPI.  If not using ISO, then must use this format and suggested you use function to add to ISO

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{natural\_abundance}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True indicates the sample is at natural abundance. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{abundance\_ratio\_MI}} (\sphinxstyleliteralemphasis{\sphinxupquote{dictionary}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If sample is not at natural abundance, then the natural abundance defines the enrichment factor compared to natural abundance(ie a sample where the main isotope is the only isotope would have a 1/natural abundance as the enrichment factor). Defined as \{M:\{I:enrichment factor, I: enrichment factor, I: enrichment factor\}, … \}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{Diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} contains the species as the key with the value being the abundance of that diluent in the sample, ie \{‘He’:0.5, ‘self’:0.5\}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If Diluent = \{\}, then this value will be used to set the only diluent to be equal to this diluent. The default is ‘air’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{IntensityThreshold}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} minimum line intensity that will be simulated. The default is 1e\sphinxhyphen{}30.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{TIPS}} (\sphinxstyleliteralemphasis{\sphinxupquote{definition}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} selects the HAPI provided TIPS version to use for the partition function

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{wavenumbers} (\sphinxstyleemphasis{array}) \textendash{} wavenumber axis that should match the input waves

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{xsect} (\sphinxstyleemphasis{array}) \textendash{} simulated cross section as a function of wavenumbers (ppm/cm)

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\phantomsection\label{\detokenize{MATS:module-MATS.generate_fitparam_file}}\index{module@\spxentry{module}!MATS.generate\_fitparam\_file@\spxentry{MATS.generate\_fitparam\_file}}\index{MATS.generate\_fitparam\_file@\spxentry{MATS.generate\_fitparam\_file}!module@\spxentry{module}}\index{Generate\_FitParam\_File (class in MATS.generate\_fitparam\_file)@\spxentry{Generate\_FitParam\_File}\spxextra{class in MATS.generate\_fitparam\_file}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{MATS.generate\_fitparam\_file.}}\sphinxbfcode{\sphinxupquote{Generate\_FitParam\_File}}}{\emph{\DUrole{n}{dataset}}, \emph{\DUrole{n}{param\_linelist}}, \emph{\DUrole{n}{base\_linelist}}, \emph{\DUrole{n}{CIA\_linelist}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{lineprofile}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}VP\textquotesingle{}}}, \emph{\DUrole{n}{linemixing}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{threshold\_intensity}\DUrole{o}{=}\DUrole{default_value}{1e\sphinxhyphen{}30}}, \emph{\DUrole{n}{fit\_intensity}\DUrole{o}{=}\DUrole{default_value}{1e\sphinxhyphen{}26}}, \emph{\DUrole{n}{fit\_window}\DUrole{o}{=}\DUrole{default_value}{1.5}}, \emph{\DUrole{n}{sim\_window}\DUrole{o}{=}\DUrole{default_value}{5}}, \emph{\DUrole{n}{param\_linelist\_savename}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}Parameter\_LineList\textquotesingle{}}}, \emph{\DUrole{n}{base\_linelist\_savename}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}Baseline\_LineList\textquotesingle{}}}, \emph{\DUrole{n}{CIA\_linelist\_savename}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}CIA\_LineList\textquotesingle{}}}, \emph{\DUrole{n}{nu\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{sw\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{gamma0\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{delta0\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{aw\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{as\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{nuVC\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{eta\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{linemixing\_constrain}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{additional\_columns}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Class generates the parameter files used in fitting and sets up fit settings.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{dataset}} (\sphinxstyleliteralemphasis{\sphinxupquote{object}}) \textendash{} Dataset object to be used in the fits

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{param\_linelist}} (\sphinxstyleliteralemphasis{\sphinxupquote{dataframe}}) \textendash{} parameter linelist dataframe name, where the dataframe has the appropriate column headers

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{base\_linelist}} (\sphinxstyleliteralemphasis{\sphinxupquote{dataframe}}) \textendash{} baseline parameter dataframe name generated from the dataset.generate\_baseline\_paramlist() function.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{CIA\_linelist}} (\sphinxstyleliteralemphasis{\sphinxupquote{dataframe}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Future Function: CIA linelist dataframe name generated from the dataset.generate\_CIA\_paramlist() function.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{lineprofile}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} lineprofile to use for the simulation. This sets values in the parameter linelist to 0 and forces them to not vary unless manually changed. Default values is VP, voigt profile. Options are VP, SDVP, NGP, SDNGP, HTP.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{linemixing}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} If False, then all linemixing parameters are set to 0 and not able to float.  Default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{threshold\_intensity}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} This is the minimum line intensity that will be simulated. Default value is 1e\sphinxhyphen{}30.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{fit\_intensity}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} This is the minimum line intensity for which parameters will be set to float. This can be overwritten manually. Default value is 1e\sphinxhyphen{}26.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{fit\_window}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} currently not used

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{sim\_window}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} This is the region outside of the wavelength region of the dataset that will be simulated for analysis. Default value is 5 cm\sphinxhyphen{}1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{base\_linelist\_savename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} filename that the baseline linelist will be saved as. Default is Baseline\_LineList

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{param\_linelist\_savename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} filename that the parameter linelist will be saved as. Default is Parameter\_LineList.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{CIA\_linelist\_save\_name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Future Feature: filename that the CIA linelist will be saved as. Default is CIA\_LineList.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nu\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the line centers will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{sw\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the line intensities will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{gamma0\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the collisional width will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{delta0\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the shift will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{aw\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the speed dependent width will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{as\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the speed dependent shift will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nuVC\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the dicke narrowing term will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{eta\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the correlation parameter will be a global variable across all spectra. False generates a new value for each spectrum in the dataset.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{linemixing\_constrain}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}) \textendash{} True indicates that the first\sphinxhyphen{}order linemixing term will be a global variable across all spectra. False generates a new value for each spectrum in the datas

\end{itemize}

\end{description}\end{quote}
\index{generate\_fit\_KarmanCIA\_linelist() (MATS.generate\_fitparam\_file.Generate\_FitParam\_File method)@\spxentry{generate\_fit\_KarmanCIA\_linelist()}\spxextra{MATS.generate\_fitparam\_file.Generate\_FitParam\_File method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.generate_fit_KarmanCIA_linelist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_fit\_KarmanCIA\_linelist}}}{\emph{\DUrole{n}{vary\_S\_SO}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_S\_EXCH}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_EXCH\_temp}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_SO\_temp}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_EXCH\_shift}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_SO\_shift}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{generate\_fit\_baseline\_linelist() (MATS.generate\_fitparam\_file.Generate\_FitParam\_File method)@\spxentry{generate\_fit\_baseline\_linelist()}\spxextra{MATS.generate\_fitparam\_file.Generate\_FitParam\_File method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.generate_fit_baseline_linelist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_fit\_baseline\_linelist}}}{\emph{\DUrole{n}{vary\_baseline}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_pressure}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_temperature}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_molefraction}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_xshift}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_etalon\_amp}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_etalon\_period}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_etalon\_phase}\DUrole{o}{=}\DUrole{default_value}{False}}, \emph{\DUrole{n}{vary\_ILS\_res}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates the baseline line list used in fitting and updates the fitting booleans to desired settings.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_baseline}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then sets all baseline parameters for all spectra to float. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_pressure}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then the pressures for all spectra are floated.  This should be used with caution as the impact these parameters have on other floated parameters might lead to an unstable solution. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_temperature}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then the temperatures for all spectra are floated. This should be used with caution as the impact these parameters have on other floated parameters might lead to an unstable solution.The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_molefraction}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} keys to dictionary correspond to molecule id where the value is boolean flag, which dictates whether to float the mole fraction. The default is \{\}. Example: \{7: True, 1: False\}

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_xshift}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then sets x\sphinxhyphen{}shift parameters for all spectra to float. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_etalon\_amp}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then sets etalon amplitude parameters for all spectra to float. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_etalon\_period}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then sets etalon period parameters for all spectra to float. . The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_etalon\_phase}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then sets etalon phase parameters for all spectra to float.. The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_ILS\_res}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If True, then sets ILS resolution parameters for all spectra to float.. The default is False.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{base\_linelist\_df} \textendash{} returns dataframe based on baseline line list with addition of a vary and err column for every floatable parameter.  The vary columns are defined by the inputs.  The err columns will be populated from fit results.  The dataframe is also saved as a .csv file..

\sphinxlineitem{Return type}
\sphinxAtStartPar
dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{generate\_fit\_param\_linelist\_from\_linelist() (MATS.generate\_fitparam\_file.Generate\_FitParam\_File method)@\spxentry{generate\_fit\_param\_linelist\_from\_linelist()}\spxextra{MATS.generate\_fitparam\_file.Generate\_FitParam\_File method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.generate_fit_param_linelist_from_linelist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_fit\_param\_linelist\_from\_linelist}}}{\emph{\DUrole{n}{vary\_nu}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_sw}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_gamma0}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_n\_gamma0}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_delta0}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_n\_delta0}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_aw}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_n\_gamma2}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_as}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_n\_delta2}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_nuVC}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_n\_nuVC}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_eta}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_linemixing}\DUrole{o}{=}\DUrole{default_value}{\{\}}}, \emph{\DUrole{n}{vary\_n\_linemixing}\DUrole{o}{=}\DUrole{default_value}{\{\}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates the parameter line list used in fitting and updates the fitting booleans to desired settings.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_nu}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope line centers should be floated.  Each key in the dictionary corresponds to a molecule, where the values are a dictionary corresponding to local\_iso\_ids as keys and boolean values acting as boolean flags. True means float the nu for that molecule and isotope. Example vary\_nu = \{7:\{1:True, 2: False, 3: False\}, 1:\{1:False, 2: False, 3: False\}\}, would indicate that line centers of all main oxygen isotopes would be floated (if the line intensity is greater than the fit intensity), where the line centers for all other O2 isotopes and all water isotopes would not be varied. If these value are left blank, then all variable would have to be manually switched to float. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_sw}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope line intensities should be floated.  Follows nu\_vary example.  The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_gamma0}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope collisional half\sphinxhyphen{}width should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_n\_gamma0}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope coefficient of the temperature dependence of the half width  should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_delta0}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope pressure shift should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_n\_delta0}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope coefficient of the temperature dependence of the pressure shift should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_aw}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope ratio of the speed dependent width to the half\sphinxhyphen{}width should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_n\_gamma2}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope coefficient of the temperature dependence of the speed dependent width should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_as}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope the ratio of the speed dependent shift to the shift should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_n\_delta2}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope coefficient of the temperature dependence of the speed dependent shift should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_nuVC}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope dicke narrowing should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_n\_nuVC}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope  coefficient of the temperature dependence of the dicke narrowing should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_eta}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope correlation parameter for the VC and SD effects should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_linemixing}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope first\sphinxhyphen{}order line\sphinxhyphen{}mixing should be floated.  Follows nu\_vary example.  . The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{vary\_n\_linemixing}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Dictionary of dictionaries setting whether the molecule and isotope temperature dependence for the first\sphinxhyphen{}order line\sphinxhyphen{}mixing should be floated.  Follows nu\_vary example.  . The default is \{\}.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{param\_linelist\_df} \textendash{} returns dataframe based on parameter line list with addition of a vary and err column for every floatable parameter.  The vary columns are defined by the inputs and the fit\_intensity value.  The err columns will be populated from fit results.  The dataframe is also saved as a .csv file.  line intensity will be normalized by the fit\_intensity (set to the sw\_scale\_factor). The term ‘sw’ is now equal to the normalized value, such that in the simulation ‘sw’{\color{red}\bfseries{}*}sw\_scale\_factor is used for the line intensity. Because line intensities are so small it is difficult to fit the value without normalization.

\sphinxlineitem{Return type}
\sphinxAtStartPar
dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_CIA\_linelist() (MATS.generate\_fitparam\_file.Generate\_FitParam\_File method)@\spxentry{get\_CIA\_linelist()}\spxextra{MATS.generate\_fitparam\_file.Generate\_FitParam\_File method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.get_CIA_linelist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_CIA\_linelist}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_base\_linelist() (MATS.generate\_fitparam\_file.Generate\_FitParam\_File method)@\spxentry{get\_base\_linelist()}\spxextra{MATS.generate\_fitparam\_file.Generate\_FitParam\_File method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.get_base_linelist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_base\_linelist}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_dataset() (MATS.generate\_fitparam\_file.Generate\_FitParam\_File method)@\spxentry{get\_dataset()}\spxextra{MATS.generate\_fitparam\_file.Generate\_FitParam\_File method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.get_dataset}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_dataset}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_param\_linelist() (MATS.generate\_fitparam\_file.Generate\_FitParam\_File method)@\spxentry{get\_param\_linelist()}\spxextra{MATS.generate\_fitparam\_file.Generate\_FitParam\_File method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.generate_fitparam_file.Generate_FitParam_File.get_param_linelist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_param\_linelist}}}{}{}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}

\phantomsection\label{\detokenize{MATS:module-MATS.linelistdata}}\index{module@\spxentry{module}!MATS.linelistdata@\spxentry{MATS.linelistdata}}\index{MATS.linelistdata@\spxentry{MATS.linelistdata}!module@\spxentry{module}}
\sphinxAtStartPar
Routines to load example data locally or from web
\index{LoadLineListData (class in MATS.linelistdata)@\spxentry{LoadLineListData}\spxextra{class in MATS.linelistdata}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.linelistdata.LoadLineListData}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{MATS.linelistdata.}}\sphinxbfcode{\sphinxupquote{LoadLineListData}}}{\emph{\DUrole{n}{paths}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Helper class to read in supplied LineList DataFrames
\index{names (MATS.linelistdata.LoadLineListData attribute)@\spxentry{names}\spxextra{MATS.linelistdata.LoadLineListData attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.linelistdata.LoadLineListData.names}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{names}}}
\pysigstopsignatures
\sphinxAtStartPar
names of LineList files available
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}



\begin{fulllineitems}

\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{\_\_getitem\_\_~:~get~file~DataFrame~by~position/name}}}
\pysigstopsignatures
\sphinxAtStartPar
\sphinxtitleref{self{[}x{]}} returns a copy of the linelist frame corresponding to
\sphinxtitleref{self.names{[}x{]}} if \sphinxtitleref{x} is an integer, otherwise returns Frame corresponding to \sphinxtitleref{name = x}

\end{fulllineitems}

\index{names (MATS.linelistdata.LoadLineListData property)@\spxentry{names}\spxextra{MATS.linelistdata.LoadLineListData property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:id0}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{names}}}
\pysigstopsignatures
\end{fulllineitems}

\index{paths (MATS.linelistdata.LoadLineListData property)@\spxentry{paths}\spxextra{MATS.linelistdata.LoadLineListData property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.linelistdata.LoadLineListData.paths}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{paths}}}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}

\phantomsection\label{\detokenize{MATS:module-MATS.spectrum}}\index{module@\spxentry{module}!MATS.spectrum@\spxentry{MATS.spectrum}}\index{MATS.spectrum@\spxentry{MATS.spectrum}!module@\spxentry{module}}\index{Spectrum (class in MATS.spectrum)@\spxentry{Spectrum}\spxextra{class in MATS.spectrum}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{MATS.spectrum.}}\sphinxbfcode{\sphinxupquote{Spectrum}}}{\emph{filename, molefraction=\{\}, natural\_abundance=True, isotope\_list=\{(1, 1): {[}1, \textquotesingle{}H2(16O)\textquotesingle{}, 0.9973173, 18.01056, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 2): {[}2, \textquotesingle{}H2(18O)\textquotesingle{}, 0.001999827, 20.01481, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 3): {[}3, \textquotesingle{}H2(17O)\textquotesingle{}, 0.0003718841, 19.01478, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 4): {[}4, \textquotesingle{}HD(16O)\textquotesingle{}, 0.0003106928, 19.01674, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 5): {[}5, \textquotesingle{}HD(18O)\textquotesingle{}, 6.230031e\sphinxhyphen{}07, 21.02098, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 6): {[}6, \textquotesingle{}HD(17O)\textquotesingle{}, 1.158526e\sphinxhyphen{}07, 20.02096, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 7): {[}129, \textquotesingle{}D2(16O)\textquotesingle{}, 2.419741e\sphinxhyphen{}08, 20.02292, \textquotesingle{}H2O\textquotesingle{}{]}, (2, 1): {[}7, \textquotesingle{}(12C)(16O)2\textquotesingle{}, 0.9842043, 43.98983, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 2): {[}8, \textquotesingle{}(13C)(16O)2\textquotesingle{}, 0.01105736, 44.99318, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 3): {[}9, \textquotesingle{}(16O)(12C)(18O)\textquotesingle{}, 0.003947066, 45.99408, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 4): {[}10, \textquotesingle{}(16O)(12C)(17O)\textquotesingle{}, 0.000733989, 44.99404, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 5): {[}11, \textquotesingle{}(16O)(13C)(18O)\textquotesingle{}, 4.434456e\sphinxhyphen{}05, 46.99743, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 6): {[}12, \textquotesingle{}(16O)(13C)(17O)\textquotesingle{}, 8.246233e\sphinxhyphen{}06, 45.9974, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 7): {[}13, \textquotesingle{}(12C)(18O)2\textquotesingle{}, 3.95734e\sphinxhyphen{}06, 47.99832, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 8): {[}14, \textquotesingle{}(17O)(12C)(18O)\textquotesingle{}, 1.471799e\sphinxhyphen{}06, 46.99829, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 9): {[}121, \textquotesingle{}(12C)(17O)2\textquotesingle{}, 1.368466e\sphinxhyphen{}07, 45.99826, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 10): {[}15, \textquotesingle{}(13C)(18O)2\textquotesingle{}, 4.446e\sphinxhyphen{}08, 49.00167, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 11): {[}120, \textquotesingle{}(18O)(13C)(17O)\textquotesingle{}, 1.65354e\sphinxhyphen{}08, 48.00165, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 12): {[}122, \textquotesingle{}(13C)(17O)2\textquotesingle{}, 1.537446e\sphinxhyphen{}09, 47.00162, \textquotesingle{}CO2\textquotesingle{}{]}, (3, 1): {[}16, \textquotesingle{}(16O)3\textquotesingle{}, 0.9929009, 47.98474, \textquotesingle{}O3\textquotesingle{}{]}, (3, 2): {[}17, \textquotesingle{}(16O)(16O)(18O)\textquotesingle{}, 0.003981942, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 3): {[}18, \textquotesingle{}(16O)(18O)(16O)\textquotesingle{}, 0.001990971, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 4): {[}19, \textquotesingle{}(16O)(16O)(17O)\textquotesingle{}, 0.0007404746, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (3, 5): {[}20, \textquotesingle{}(16O)(17O)(16O)\textquotesingle{}, 0.0003702373, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (4, 1): {[}21, \textquotesingle{}(14N)2(16O)\textquotesingle{}, 0.9903328, 44.00106, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 2): {[}22, \textquotesingle{}(14N)(15N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 3): {[}23, \textquotesingle{}(15N)(14N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 4): {[}24, \textquotesingle{}(14N)2(18O)\textquotesingle{}, 0.001985822, 46.00531, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 5): {[}25, \textquotesingle{}(14N)2(17O)\textquotesingle{}, 0.0003692797, 45.00528, \textquotesingle{}N2O\textquotesingle{}{]}, (5, 1): {[}26, \textquotesingle{}(12C)(16O)\textquotesingle{}, 0.9865444, 27.99491, \textquotesingle{}CO\textquotesingle{}{]}, (5, 2): {[}27, \textquotesingle{}(13C)(16O)\textquotesingle{}, 0.01108364, 28.99827, \textquotesingle{}CO\textquotesingle{}{]}, (5, 3): {[}28, \textquotesingle{}(12C)(18O)\textquotesingle{}, 0.001978224, 29.99916, \textquotesingle{}CO\textquotesingle{}{]}, (5, 4): {[}29, \textquotesingle{}(12C)(17O)\textquotesingle{}, 0.0003678671, 28.99913, \textquotesingle{}CO\textquotesingle{}{]}, (5, 5): {[}30, \textquotesingle{}(13C)(18O)\textquotesingle{}, 2.2225e\sphinxhyphen{}05, 31.00252, \textquotesingle{}CO\textquotesingle{}{]}, (5, 6): {[}31, \textquotesingle{}(13C)(17O)\textquotesingle{}, 4.13292e\sphinxhyphen{}06, 30.00249, \textquotesingle{}CO\textquotesingle{}{]}, (6, 1): {[}32, \textquotesingle{}(12C)H4\textquotesingle{}, 0.9882741, 16.0313, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 2): {[}33, \textquotesingle{}(13C)H4\textquotesingle{}, 0.01110308, 17.03466, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 3): {[}34, \textquotesingle{}(12C)H3D\textquotesingle{}, 0.0006157511, 17.03748, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 4): {[}35, \textquotesingle{}(13C)H3D\textquotesingle{}, 6.917852e\sphinxhyphen{}06, 18.04083, \textquotesingle{}CH4\textquotesingle{}{]}, (7, 1): {[}36, \textquotesingle{}(16O)2\textquotesingle{}, 0.9952616, 31.98983, \textquotesingle{}O2\textquotesingle{}{]}, (7, 2): {[}37, \textquotesingle{}(16O)(18O)\textquotesingle{}, 0.00399141, 33.99408, \textquotesingle{}O2\textquotesingle{}{]}, (7, 3): {[}38, \textquotesingle{}(16O)(17O)\textquotesingle{}, 0.0007422352, 32.99404, \textquotesingle{}O2\textquotesingle{}{]}, (8, 1): {[}39, \textquotesingle{}(14N)(16O)\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NO\textquotesingle{}{]}, (8, 2): {[}40, \textquotesingle{}(15N)(16O)\textquotesingle{}, 0.003654311, 30.99502, \textquotesingle{}NO\textquotesingle{}{]}, (8, 3): {[}41, \textquotesingle{}(14N)(18O)\textquotesingle{}, 0.001993122, 32.00223, \textquotesingle{}NO\textquotesingle{}{]}, (9, 1): {[}42, \textquotesingle{}(32S)(16O)2\textquotesingle{}, 0.9456777, 63.9619, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 2): {[}43, \textquotesingle{}(34S)(16O)2\textquotesingle{}, 0.04195028, 65.9577, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 3): {[}137, \textquotesingle{}(33S)(16O)2\textquotesingle{}, 0.007464462, 64.96129, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 4): {[}138, \textquotesingle{}(16O)(32S)(18O)\textquotesingle{}, 0.003792558, 65.96615, \textquotesingle{}SO2\textquotesingle{}{]}, (10, 1): {[}44, \textquotesingle{}(14N)(16O)2\textquotesingle{}, 0.991616, 45.9929, \textquotesingle{}NO2\textquotesingle{}{]}, (10, 2): {[}130, \textquotesingle{}(15N)(16O)2\textquotesingle{}, 0.003645643, 46.98994, \textquotesingle{}NO2\textquotesingle{}{]}, (11, 1): {[}45, \textquotesingle{}(14N)H3\textquotesingle{}, 0.9958716, 17.02655, \textquotesingle{}NH3\textquotesingle{}{]}, (11, 2): {[}46, \textquotesingle{}(15N)H3\textquotesingle{}, 0.003661289, 18.02358, \textquotesingle{}NH3\textquotesingle{}{]}, (12, 1): {[}47, \textquotesingle{}H(14N)(16O)3\textquotesingle{}, 0.9891098, 62.99564, \textquotesingle{}HNO3\textquotesingle{}{]}, (12, 2): {[}117, \textquotesingle{}H(15N)(16O)3\textquotesingle{}, 0.003636429, 63.99268, \textquotesingle{}HNO3\textquotesingle{}{]}, (13, 1): {[}48, \textquotesingle{}(16O)H\textquotesingle{}, 0.9974726, 17.00274, \textquotesingle{}OH\textquotesingle{}{]}, (13, 2): {[}49, \textquotesingle{}(18O)H\textquotesingle{}, 0.002000138, 19.00699, \textquotesingle{}OH\textquotesingle{}{]}, (13, 3): {[}50, \textquotesingle{}(16O)D\textquotesingle{}, 0.0001553706, 18.00891, \textquotesingle{}OH\textquotesingle{}{]}, (14, 1): {[}51, \textquotesingle{}H(19F)\textquotesingle{}, 0.9998443, 20.00623, \textquotesingle{}HF\textquotesingle{}{]}, (14, 2): {[}110, \textquotesingle{}D(19F)\textquotesingle{}, 0.000155741, 21.0124, \textquotesingle{}HF\textquotesingle{}{]}, (15, 1): {[}52, \textquotesingle{}H(35Cl)\textquotesingle{}, 0.757587, 35.97668, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 2): {[}53, \textquotesingle{}H(37Cl)\textquotesingle{}, 0.2422573, 37.97373, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 3): {[}107, \textquotesingle{}D(35Cl)\textquotesingle{}, 0.000118005, 36.98285, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 4): {[}108, \textquotesingle{}D(37Cl)\textquotesingle{}, 3.773502e\sphinxhyphen{}05, 38.9799, \textquotesingle{}HCl\textquotesingle{}{]}, (16, 1): {[}54, \textquotesingle{}H(79Br)\textquotesingle{}, 0.5067811, 79.92616, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 2): {[}55, \textquotesingle{}H(81Br)\textquotesingle{}, 0.4930632, 81.92412, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 3): {[}111, \textquotesingle{}D(79Br)\textquotesingle{}, 7.893838e\sphinxhyphen{}05, 80.93234, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 4): {[}112, \textquotesingle{}D(81Br)\textquotesingle{}, 7.680162e\sphinxhyphen{}05, 82.93029, \textquotesingle{}HBr\textquotesingle{}{]}, (17, 1): {[}56, \textquotesingle{}H(127I)\textquotesingle{}, 0.9998443, 127.9123, \textquotesingle{}HI\textquotesingle{}{]}, (17, 2): {[}113, \textquotesingle{}D(127I)\textquotesingle{}, 0.000155741, 128.9185, \textquotesingle{}HI\textquotesingle{}{]}, (18, 1): {[}57, \textquotesingle{}(35Cl)(16O)\textquotesingle{}, 0.7559077, 50.96377, \textquotesingle{}ClO\textquotesingle{}{]}, (18, 2): {[}58, \textquotesingle{}(37Cl)(16O)\textquotesingle{}, 0.2417203, 52.96082, \textquotesingle{}ClO\textquotesingle{}{]}, (19, 1): {[}59, \textquotesingle{}(16O)(12C)(32S)\textquotesingle{}, 0.9373947, 59.96699, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 2): {[}60, \textquotesingle{}(16O)(12C)(34S)\textquotesingle{}, 0.04158284, 61.96278, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 3): {[}61, \textquotesingle{}(16O)(13C)(32S)\textquotesingle{}, 0.01053146, 60.97034, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 4): {[}62, \textquotesingle{}(16O)(12C)(33S)\textquotesingle{}, 0.007399083, 60.96637, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 5): {[}63, \textquotesingle{}(18O)(12C)(32S)\textquotesingle{}, 0.00187967, 61.97123, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 6): {[}135, \textquotesingle{}(16O)(13C)(34S)\textquotesingle{}, 0.0004671757, 62.96614, \textquotesingle{}OCS\textquotesingle{}{]}, (20, 1): {[}64, \textquotesingle{}H2(12C)(16O)\textquotesingle{}, 0.9862371, 30.01056, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 2): {[}65, \textquotesingle{}H2(13C)(16O)\textquotesingle{}, 0.0110802, 31.01392, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 3): {[}66, \textquotesingle{}H2(12C)(18O)\textquotesingle{}, 0.001977609, 32.01481, \textquotesingle{}H2CO\textquotesingle{}{]}, (21, 1): {[}67, \textquotesingle{}H(16O)(35Cl)\textquotesingle{}, 0.75579, 51.97159, \textquotesingle{}HOCl\textquotesingle{}{]}, (21, 2): {[}68, \textquotesingle{}H(16O)(37Cl)\textquotesingle{}, 0.2416826, 53.96864, \textquotesingle{}HOCl\textquotesingle{}{]}, (22, 1): {[}69, \textquotesingle{}(14N)2\textquotesingle{}, 0.9926874, 28.00615, \textquotesingle{}N2\textquotesingle{}{]}, (22, 2): {[}118, \textquotesingle{}(14N)(15N)\textquotesingle{}, 0.007299165, 29.00318, \textquotesingle{}N2\textquotesingle{}{]}, (23, 1): {[}70, \textquotesingle{}H(12C)(14N)\textquotesingle{}, 0.9851143, 27.0109, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 2): {[}71, \textquotesingle{}H(13C)(14N)\textquotesingle{}, 0.01106758, 28.01425, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 3): {[}72, \textquotesingle{}H(12C)(15N)\textquotesingle{}, 0.00362174, 28.00793, \textquotesingle{}HCN\textquotesingle{}{]}, (24, 1): {[}73, \textquotesingle{}(12C)H3(35Cl)\textquotesingle{}, 0.7489369, 49.99233, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (24, 2): {[}74, \textquotesingle{}(12C)H3(37Cl)\textquotesingle{}, 0.2394912, 51.98938, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (25, 1): {[}75, \textquotesingle{}H2(16O)2\textquotesingle{}, 0.9949516, 34.00548, \textquotesingle{}H2O2\textquotesingle{}{]}, (26, 1): {[}76, \textquotesingle{}(12C)2H2\textquotesingle{}, 0.9775989, 26.01565, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 2): {[}77, \textquotesingle{}(12C)(13C)H2\textquotesingle{}, 0.02196629, 27.019, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 3): {[}105, \textquotesingle{}(12C)2HD\textquotesingle{}, 0.0003045499, 27.02182, \textquotesingle{}C2H2\textquotesingle{}{]}, (27, 1): {[}78, \textquotesingle{}(12C)2H6\textquotesingle{}, 0.97699, 30.04695, \textquotesingle{}C2H6\textquotesingle{}{]}, (27, 2): {[}106, \textquotesingle{}(12C)H3(13C)H3\textquotesingle{}, 0.02195261, 31.05031, \textquotesingle{}C2H6\textquotesingle{}{]}, (28, 1): {[}79, \textquotesingle{}(31P)H3\textquotesingle{}, 0.9995329, 33.99724, \textquotesingle{}PH3\textquotesingle{}{]}, (29, 1): {[}80, \textquotesingle{}(12C)(16O)(19F)2\textquotesingle{}, 0.9865444, 65.99172, \textquotesingle{}COF2\textquotesingle{}{]}, (29, 2): {[}119, \textquotesingle{}(13C)(16O)(19F)2\textquotesingle{}, 0.01108366, 66.99508, \textquotesingle{}COF2\textquotesingle{}{]}, (30, 1): {[}126, \textquotesingle{}(32S)(19F)6\textquotesingle{}, 0.95018, 145.9625, \textquotesingle{}SF6\textquotesingle{}{]}, (31, 1): {[}81, \textquotesingle{}H2(32S)\textquotesingle{}, 0.9498841, 33.98772, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 2): {[}82, \textquotesingle{}H2(34S)\textquotesingle{}, 0.04213687, 35.98351, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 3): {[}83, \textquotesingle{}H2(33S)\textquotesingle{}, 0.007497664, 34.9871, \textquotesingle{}H2S\textquotesingle{}{]}, (32, 1): {[}84, \textquotesingle{}H(12C)(16O)(16O)H\textquotesingle{}, 0.9838977, 46.00548, \textquotesingle{}HCOOH\textquotesingle{}{]}, (33, 1): {[}85, \textquotesingle{}H(16O)2\textquotesingle{}, 0.9951066, 32.99766, \textquotesingle{}HO2\textquotesingle{}{]}, (34, 1): {[}86, \textquotesingle{}(16O)\textquotesingle{}, 0.997628, 15.99492, \textquotesingle{}O\textquotesingle{}{]}, (35, 1): {[}127, \textquotesingle{}(35Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.7495702, 96.95667, \textquotesingle{}ClONO2\textquotesingle{}{]}, (35, 2): {[}128, \textquotesingle{}(37Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.2396937, 98.95372, \textquotesingle{}ClONO2\textquotesingle{}{]}, (36, 1): {[}87, \textquotesingle{}(14N)(16O)+\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NOp\textquotesingle{}{]}, (37, 1): {[}88, \textquotesingle{}H(16O)(79Br)\textquotesingle{}, 0.505579, 95.92108, \textquotesingle{}HOBr\textquotesingle{}{]}, (37, 2): {[}89, \textquotesingle{}H(16O)(81Br)\textquotesingle{}, 0.4918937, 97.91903, \textquotesingle{}HOBr\textquotesingle{}{]}, (38, 1): {[}90, \textquotesingle{}(12C)2H4\textquotesingle{}, 0.9772944, 28.0313, \textquotesingle{}C2H4\textquotesingle{}{]}, (38, 2): {[}91, \textquotesingle{}(12C)H2(13C)H2\textquotesingle{}, 0.02195946, 29.03466, \textquotesingle{}C2H4\textquotesingle{}{]}, (39, 1): {[}92, \textquotesingle{}(12C)H3(16O)H\textquotesingle{}, 0.9859299, 32.02622, \textquotesingle{}CH3OH\textquotesingle{}{]}, (40, 1): {[}93, \textquotesingle{}(12C)H3(79Br)\textquotesingle{}, 0.5009946, 93.94181, \textquotesingle{}CH3Br\textquotesingle{}{]}, (40, 2): {[}94, \textquotesingle{}(12C)H3(81Br)\textquotesingle{}, 0.4874334, 95.93976, \textquotesingle{}CH3Br\textquotesingle{}{]}, (41, 1): {[}95, \textquotesingle{}(12C)H3(12C)(14N)\textquotesingle{}, 0.9738662, 41.02655, \textquotesingle{}CH3CN\textquotesingle{}{]}, (42, 1): {[}96, \textquotesingle{}(12C)(19F)4\textquotesingle{}, 0.98889, 87.99362, \textquotesingle{}CF4\textquotesingle{}{]}, (43, 1): {[}116, \textquotesingle{}(12C)4H2\textquotesingle{}, 0.955998, 50.01565, \textquotesingle{}C4H2\textquotesingle{}{]}, (44, 1): {[}109, \textquotesingle{}H(12C)3(14N)\textquotesingle{}, 0.963346, 51.0109, \textquotesingle{}HC3N\textquotesingle{}{]}, (45, 1): {[}103, \textquotesingle{}H2\textquotesingle{}, 0.9996885, 2.01565, \textquotesingle{}H2\textquotesingle{}{]}, (45, 2): {[}115, \textquotesingle{}HD\textquotesingle{}, 0.0003114316, 3.021825, \textquotesingle{}H2\textquotesingle{}{]}, (46, 1): {[}97, \textquotesingle{}(12C)(32S)\textquotesingle{}, 0.9396236, 43.97207, \textquotesingle{}CS\textquotesingle{}{]}, (46, 2): {[}98, \textquotesingle{}(12C)(34S)\textquotesingle{}, 0.04168171, 45.96787, \textquotesingle{}CS\textquotesingle{}{]}, (46, 3): {[}99, \textquotesingle{}(13C)(32S)\textquotesingle{}, 0.0105565, 44.97543, \textquotesingle{}CS\textquotesingle{}{]}, (46, 4): {[}100, \textquotesingle{}(12C)(33S)\textquotesingle{}, 0.007416675, 44.97146, \textquotesingle{}CS\textquotesingle{}{]}, (47, 1): {[}114, \textquotesingle{}(32S)(16O)3\textquotesingle{}, 0.9434345, 79.95682, \textquotesingle{}SO3\textquotesingle{}{]}, (48, 1): {[}123, \textquotesingle{}(12C)2(14N)2\textquotesingle{}, 0.9707524, 52.00615, \textquotesingle{}C2N2\textquotesingle{}{]}, (49, 1): {[}124, \textquotesingle{}(12C)(16O)(35Cl)2\textquotesingle{}, 0.5663918, 97.93262, \textquotesingle{}COCl2\textquotesingle{}{]}, (49, 2): {[}125, \textquotesingle{}(12C)(16O)(35Cl)(37Cl)\textquotesingle{}, 0.362235, 99.92967, \textquotesingle{}COCl2\textquotesingle{}{]}, (50, 1): {[}146, \textquotesingle{}(32S)(16O)\textquotesingle{}, 0.9479262, 47.96699, \textquotesingle{}SO\textquotesingle{}{]}, (50, 2): {[}147, \textquotesingle{}(34S)(16O)\textquotesingle{}, 0.04205002, 49.96278, \textquotesingle{}SO\textquotesingle{}{]}, (50, 3): {[}148, \textquotesingle{}(32S)(18O)\textquotesingle{}, 0.001900788, 49.97123, \textquotesingle{}SO\textquotesingle{}{]}, (51, 1): {[}144, \textquotesingle{}(12C)H3(19F)\textquotesingle{}, 0.988428, 34.02188, \textquotesingle{}CH3F\textquotesingle{}{]}, (52, 1): {[}139, \textquotesingle{}(74Ge)H4\textquotesingle{}, 0.3651724, 77.95248, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 2): {[}140, \textquotesingle{}(72Ge)H4\textquotesingle{}, 0.2741292, 75.95338, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 3): {[}141, \textquotesingle{}(70Ge)H4\textquotesingle{}, 0.2050722, 73.95555, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 4): {[}142, \textquotesingle{}(73Ge)H4\textquotesingle{}, 0.07755167, 76.95476, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 5): {[}143, \textquotesingle{}(76Ge)H4\textquotesingle{}, 0.07755167, 79.9527, \textquotesingle{}GeH4\textquotesingle{}{]}, (53, 1): {[}131, \textquotesingle{}(12C)(32S)2\textquotesingle{}, 0.8928115, 75.94414, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 2): {[}132, \textquotesingle{}(32S)(12C)(34S)\textquotesingle{}, 0.07921026, 77.93994, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 3): {[}133, \textquotesingle{}(32S)(12C)(33S)\textquotesingle{}, 0.01409435, 76.94353, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 4): {[}134, \textquotesingle{}(13C)(32S)2\textquotesingle{}, 0.01003057, 76.9475, \textquotesingle{}CS2\textquotesingle{}{]}, (54, 1): {[}145, \textquotesingle{}(12C)H3(127I)\textquotesingle{}, 0.988428, 141.9279, \textquotesingle{}CH3I\textquotesingle{}{]}, (55, 1): {[}136, \textquotesingle{}(14N)(19F)3\textquotesingle{}, 0.996337, 70.99829, \textquotesingle{}NF3\textquotesingle{}{]}\}, diluent=\textquotesingle{}air\textquotesingle{}, Diluent=\{\}, abundance\_ratio\_MI=\{\}, spectrum\_number=1, input\_freq=True, input\_tau=True, pressure\_column=\textquotesingle{}Cavity Pressure /Torr\textquotesingle{}, temperature\_column=\textquotesingle{}Cavity Temperature Side 2 /C\textquotesingle{}, frequency\_column=\textquotesingle{}Total Frequency /MHz\textquotesingle{}, tau\_column=\textquotesingle{}Mean tau/us\textquotesingle{}, tau\_stats\_column=None, segment\_column=None, etalons=\{\}, nominal\_temperature=296, x\_shift=0, baseline\_order=1, weight=1, ILS\_function=None, ILS\_resolution=0.1, ILS\_wing=10, TIPS=\textless{}function \textless{}lambda\textgreater{}\textgreater{}, compressability\_file=None}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Spectrum class provides all information describing experimental or simulated spectrum.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{filename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} file containing spectrum with .csv extension. File extension is not included in the name

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} mole fraction of each molecule in spectra in the format \{molec\_id: mole fraction (out of 1), molec\_id: molefraction, … \}

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{natural\_abundance}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} flag for if the molecular species in the spectrum are at natural abundance

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{abundance\_ratio\_MI}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} if not at natural abundance sets the enhancement factor for each molecule and isotope in the following format \{molec\_id:\{iso\_id: enhancement, iso\_id: enhancement\}, … \}.  The enhancement is the ratio of the new abundance to the natural abundance.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{isotope\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} provides opportunity to specify the isotope look\sphinxhyphen{}up table.  Default is ISO, which is from HAPI.  If not using ISO, then must use this format and suggested you use function to add to ISO

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} sets the diluent for the sample.  Default = ‘air’

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{Diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} sets the diluent for the sample if there are a combination of several diluents. Format \{‘he’: 0.5, ‘air’: 0.5). NOTE: the line parameter file must have parameters that correspond to the diluent (ie gamma0\_he, and gamma0\_air). Additionally, the contribution from all diluents must sum to 1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{spectrum\_number}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} sets a number for the spectrum that will correspond to fit parameters. This is set in the Dataset class, so should not need to be defined manually

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{input\_freq}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True indicates that the frequency\_column is in MHz. False indicates that the frequency column is in wavenumbers.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{input\_tau}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} True indicates that the tau\_column is in us. False indicates that the tau column is in 1/c*tau (ppm/cm).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pressure\_column}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of the pressure column in input file. Column should be in torr. Default is Cavity Pressure /Torr.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{temperature\_column}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of the temperature column in input file. Column should be in celsius. Default is Cavity Temperature Side 2 /C.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{frequency\_column}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of the frequency column in input file. Column should be in MHz or Wavenumbers (with appropriate flag for input\_freq). NOTE: This is not a detuning axis. Default is Total Frequency /MHz.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{tau\_column}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of the tau column in input file. Column should be in us or in 1/ctau (ppm/cm) (with appropriate flag for input\_tau). Default is Mean tau/us.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{tau\_stats\_column}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of column containing the pt by pt error as a percent of the y\sphinxhyphen{}axis. Default is ‘tau rel. std. dev./\%

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{segment\_column}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Name of column with segment numbers in input file. This column allows spectrum background parameters to be treated in segments across the spectrum. If None then the entire spectrum will share the same segment.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{etalons}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Allows you to define etalons by an amplitude and period (1/ frequency in cm\sphinxhyphen{}1). Default is no etalons. Input is dictionary with keys being a number and the value being an array with the first index being the amplitude and the second being the period.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nominal\_temperature}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} nominal temperature indicates the approximate temperature of the spectrum. When combining spectra into a dataset this acts as a flag to whether temperature dependence terms should be parameters that can be fit or whether they should act as constants.  Default = 296

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{x\_shift}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} value in wavenumbers of the x shift for the spectrum axis. This is a fittable parameter. Be careful in using this parameter as floating multiple parameters with similar effects cause fits to not converge (ie. unconstrained line centers + x\_shift or fits with line center, shifts, and x\_shifts terms without enough lines per spectrum to isolate the different effects). Floating this term works best if you have a good initial guess. Default = 0

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{baseline\_order}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} sets the baseline order for this spectrum. Allows you to change the baseline order across the broader dataset.()

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{weight}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}\sphinxstyleliteralemphasis{\sphinxupquote{,}}) \textendash{} set the weighting to used for the spectrum if using weighted fits.  This can be used to weight a whole spectrum in addition to the pt by pt weighting using the stats column.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ILS\_function}} (\sphinxstyleliteralemphasis{\sphinxupquote{string}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Default is None and means that no instrument line shape is used in the fitting.
Function can be: SLIT\_MICHELSON, SLIT\_DIFFRACTION, SLIT\_COSINUS, SLIT\_DISPERSION, SLIT\_GAUSSIAN, SLIT\_TRIANGULAR, SLIT\_RECTANGULAR corresponding to the ILS functions defined in HAPI or a user defined function.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ILS\_resolution}} (\sphinxstyleliteralemphasis{\sphinxupquote{float/array}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Resolution is a float or array of ILS resolutions in wavenumbers.  The SlitFunctions defined in HAPI have 1 resolution, but this opens the option for the user defined function to be comprised of several functions with varying resolutions.  Default is 0.1 cm\sphinxhyphen{}1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ILS\_wing}} (\sphinxstyleliteralemphasis{\sphinxupquote{float/array}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} AF\_wing is the a float or array consisting of the range the ILS is calculted over in cm\sphinxhyphen{}1.  This is a single value fort he HAPI slit functions, but could be an array of multiple values for user\sphinxhyphen{}defined functions.  Default is 10 cm\sphinxhyphen{}1

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{TIPS}} (\sphinxstyleliteralemphasis{\sphinxupquote{definition}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} selects the HAPI provided TIPS version to use for the partition function

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{compressability\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} compressability\_file is the name of a file that contains the compressability information, generated by the NIST Refprop program with the rows containing compressability (used to correct the ideal gas law) as a function pressure (MPa) and the columns as a function of temperature (K).  Pressure and temperature are header/index in the given axis.

\end{itemize}

\end{description}\end{quote}
\index{calculate\_QF() (MATS.spectrum.Spectrum method)@\spxentry{calculate\_QF()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.calculate_QF}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{calculate\_QF}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Calculates the quality of fit factor (QF) for a spectrum \sphinxhyphen{} QF = (maximum alpha \sphinxhyphen{} minimum alpha) / std(residuals).
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
QF.

\sphinxlineitem{Return type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{diluent\_sum\_check() (MATS.spectrum.Spectrum method)@\spxentry{diluent\_sum\_check()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.diluent_sum_check}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{diluent\_sum\_check}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Checks that if multiple broadeners are used that the contributions sum to one.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
Warning if the diluents don’t sum to one

\sphinxlineitem{Return type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{fft\_spectrum() (MATS.spectrum.Spectrum method)@\spxentry{fft\_spectrum()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.fft_spectrum}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{fft\_spectrum}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Takes the FFT of the residuals of the spectrum, generates a plot of frequency (cm\sphinxhyphen{}1) versus amplitude (ppm/cm), and prints a dataframe with the 20 highest amplitude frequencies with the FFT frequency (period), amplitude, FFT phase, and frequency (cm\sphinxhyphen{}1).

\end{fulllineitems}

\index{get\_Diluent() (MATS.spectrum.Spectrum method)@\spxentry{get\_Diluent()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_Diluent}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_Diluent}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_abundance\_ratio\_MI() (MATS.spectrum.Spectrum method)@\spxentry{get\_abundance\_ratio\_MI()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_abundance_ratio_MI}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_abundance\_ratio\_MI}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_alpha() (MATS.spectrum.Spectrum method)@\spxentry{get\_alpha()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_alpha}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_alpha}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_background() (MATS.spectrum.Spectrum method)@\spxentry{get\_background()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_background}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_background}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_cia() (MATS.spectrum.Spectrum method)@\spxentry{get\_cia()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_cia}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_cia}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_diluent() (MATS.spectrum.Spectrum method)@\spxentry{get\_diluent()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_diluent}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_diluent}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_etalons() (MATS.spectrum.Spectrum method)@\spxentry{get\_etalons()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_etalons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_etalons}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_filename() (MATS.spectrum.Spectrum method)@\spxentry{get\_filename()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_filename}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_filename}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_frequency() (MATS.spectrum.Spectrum method)@\spxentry{get\_frequency()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_frequency}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_frequency}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_model() (MATS.spectrum.Spectrum method)@\spxentry{get\_model()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_model}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_model}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_molefraction() (MATS.spectrum.Spectrum method)@\spxentry{get\_molefraction()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_molefraction}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_molefraction}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_natural\_abundance() (MATS.spectrum.Spectrum method)@\spxentry{get\_natural\_abundance()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_natural_abundance}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_natural\_abundance}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_nominal\_temperature() (MATS.spectrum.Spectrum method)@\spxentry{get\_nominal\_temperature()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_nominal_temperature}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_nominal\_temperature}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_pressure() (MATS.spectrum.Spectrum method)@\spxentry{get\_pressure()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_pressure}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_pressure}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_pressure\_torr() (MATS.spectrum.Spectrum method)@\spxentry{get\_pressure\_torr()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_pressure_torr}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_pressure\_torr}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_residuals() (MATS.spectrum.Spectrum method)@\spxentry{get\_residuals()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_residuals}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_residuals}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_spectrum\_number() (MATS.spectrum.Spectrum method)@\spxentry{get\_spectrum\_number()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_spectrum_number}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_spectrum\_number}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_tau() (MATS.spectrum.Spectrum method)@\spxentry{get\_tau()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_tau}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_tau}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_tau\_stats() (MATS.spectrum.Spectrum method)@\spxentry{get\_tau\_stats()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_tau_stats}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_tau\_stats}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_temperature() (MATS.spectrum.Spectrum method)@\spxentry{get\_temperature()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_temperature}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_temperature}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_temperature\_C() (MATS.spectrum.Spectrum method)@\spxentry{get\_temperature\_C()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_temperature_C}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_temperature\_C}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_wavenumber() (MATS.spectrum.Spectrum method)@\spxentry{get\_wavenumber()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.get_wavenumber}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_wavenumber}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{plot\_freq\_tau() (MATS.spectrum.Spectrum method)@\spxentry{plot\_freq\_tau()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.plot_freq_tau}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_freq\_tau}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a plot of tau (us) as a function of frequency (MHz).

\end{fulllineitems}

\index{plot\_model\_residuals() (MATS.spectrum.Spectrum method)@\spxentry{plot\_model\_residuals()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.plot_model_residuals}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_model\_residuals}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a plot of the alpha and model (ppm/cm) as a function of wavenumber (cm\sphinxhyphen{}1) and on lower plot shows the residuals (ppm/cm) as a function of wavenumber (cm\sphinxhyphen{}1).

\end{fulllineitems}

\index{plot\_wave\_alpha() (MATS.spectrum.Spectrum method)@\spxentry{plot\_wave\_alpha()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.plot_wave_alpha}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_wave\_alpha}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a plot of alpha (ppm/cm) as a function of wavenumber (cm\sphinxhyphen{}1).

\end{fulllineitems}

\index{save\_spectrum\_info() (MATS.spectrum.Spectrum method)@\spxentry{save\_spectrum\_info()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.save_spectrum_info}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{save\_spectrum\_info}}}{\emph{\DUrole{n}{save\_file}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Saves spectrum information to a pandas dataframe with option to also save as as a csv file.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{save\_file}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If False, then only a dataframe is created. If True, then a csv file will be generated with the name filename + ‘\_saved.csv’. The default is False.

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{new\_file} \textendash{} returns a pandas dataframe with columns related to the spectrum information

\sphinxlineitem{Return type}
\sphinxAtStartPar
dataframe

\end{description}\end{quote}

\end{fulllineitems}

\index{segment\_wave\_alpha() (MATS.spectrum.Spectrum method)@\spxentry{segment\_wave\_alpha()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.segment_wave_alpha}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{segment\_wave\_alpha}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Defines the wavenumber, alpha, and indices of spectrum that correspond to a given spectrum segment.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{wavenumber\_segments} (\sphinxstyleemphasis{dict}) \textendash{} dictionary where the key corresponds to a segment number and the values correspond to the wavenumbers for that segment

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{alpha\_segments} (\sphinxstyleemphasis{dict}) \textendash{} dictionary where the key corresponds to a segment number and the values correspond to the alpha values for that segment.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{indices\_segments} (\sphinxstyleemphasis{dict}) \textendash{} dictionary where the key corresponds to a segment number and the values correspond to the array indices for that segment.

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\index{set\_Diluent() (MATS.spectrum.Spectrum method)@\spxentry{set\_Diluent()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_Diluent}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_Diluent}}}{\emph{\DUrole{n}{new\_Diluent}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_abundance\_ration\_MI() (MATS.spectrum.Spectrum method)@\spxentry{set\_abundance\_ration\_MI()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_abundance_ration_MI}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_abundance\_ration\_MI}}}{\emph{\DUrole{n}{new\_abundance\_ratio\_MI}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_background() (MATS.spectrum.Spectrum method)@\spxentry{set\_background()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_background}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_background}}}{\emph{\DUrole{n}{new\_background}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_cia() (MATS.spectrum.Spectrum method)@\spxentry{set\_cia()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_cia}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_cia}}}{\emph{\DUrole{n}{new\_cia}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_diluent() (MATS.spectrum.Spectrum method)@\spxentry{set\_diluent()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_diluent}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_diluent}}}{\emph{\DUrole{n}{new\_diluent}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_etalons() (MATS.spectrum.Spectrum method)@\spxentry{set\_etalons()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_etalons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_etalons}}}{\emph{\DUrole{n}{new\_etalons}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_frequency\_column() (MATS.spectrum.Spectrum method)@\spxentry{set\_frequency\_column()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_frequency_column}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_frequency\_column}}}{\emph{\DUrole{n}{new\_frequency\_column}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_model() (MATS.spectrum.Spectrum method)@\spxentry{set\_model()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_model}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_model}}}{\emph{\DUrole{n}{new\_model}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_molefraction() (MATS.spectrum.Spectrum method)@\spxentry{set\_molefraction()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_molefraction}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_molefraction}}}{\emph{\DUrole{n}{new\_molefraction}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_natural\_abundance() (MATS.spectrum.Spectrum method)@\spxentry{set\_natural\_abundance()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_natural_abundance}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_natural\_abundance}}}{\emph{\DUrole{n}{new\_natural\_abundance}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_nominal\_temperature() (MATS.spectrum.Spectrum method)@\spxentry{set\_nominal\_temperature()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_nominal_temperature}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_nominal\_temperature}}}{\emph{\DUrole{n}{new\_nominal\_temperature}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_pressure\_column() (MATS.spectrum.Spectrum method)@\spxentry{set\_pressure\_column()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_pressure_column}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_pressure\_column}}}{\emph{\DUrole{n}{new\_pressure\_column}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_residuals() (MATS.spectrum.Spectrum method)@\spxentry{set\_residuals()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_residuals}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_residuals}}}{\emph{\DUrole{n}{new\_residuals}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_spectrum\_number() (MATS.spectrum.Spectrum method)@\spxentry{set\_spectrum\_number()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_spectrum_number}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_spectrum\_number}}}{\emph{\DUrole{n}{new\_spectrum\_number}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_tau\_column() (MATS.spectrum.Spectrum method)@\spxentry{set\_tau\_column()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_tau_column}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_tau\_column}}}{\emph{\DUrole{n}{new\_tau\_column}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_tau\_stats\_column() (MATS.spectrum.Spectrum method)@\spxentry{set\_tau\_stats\_column()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_tau_stats_column}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_tau\_stats\_column}}}{\emph{\DUrole{n}{new\_tau\_stats\_column}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_temperature\_column() (MATS.spectrum.Spectrum method)@\spxentry{set\_temperature\_column()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_temperature_column}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_temperature\_column}}}{\emph{\DUrole{n}{new\_temperature\_column}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_weight() (MATS.spectrum.Spectrum method)@\spxentry{set\_weight()}\spxextra{MATS.spectrum.Spectrum method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.Spectrum.set_weight}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_weight}}}{\emph{\DUrole{n}{new\_weight}}}{}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}

\index{simulate\_spectrum() (in module MATS.spectrum)@\spxentry{simulate\_spectrum()}\spxextra{in module MATS.spectrum}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.spectrum.simulate_spectrum}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.spectrum.}}\sphinxbfcode{\sphinxupquote{simulate\_spectrum}}}{\emph{parameter\_linelist, wave\_min=None, wave\_max=None, wave\_space=None, wavenumbers={[}{]}, wave\_error=0, SNR=None, baseline\_terms={[}0{]}, temperature=25, temperature\_err=\{\textquotesingle{}bias\textquotesingle{}: 0, \textquotesingle{}function\textquotesingle{}: None, \textquotesingle{}params\textquotesingle{}: \{\}\}, pressure=760, pressure\_err=\{\textquotesingle{}function\textquotesingle{}: None, \textquotesingle{}params\textquotesingle{}: \{\}, \textquotesingle{}per\_bias\textquotesingle{}: 0\}, wing\_cutoff=25, wing\_wavenumbers=25, wing\_method=\textquotesingle{}wing\_cutoff\textquotesingle{}, filename=\textquotesingle{}temp\textquotesingle{}, molefraction=\{\}, molefraction\_err=\{\}, isotope\_list=\{(1, 1): {[}1, \textquotesingle{}H2(16O)\textquotesingle{}, 0.9973173, 18.01056, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 2): {[}2, \textquotesingle{}H2(18O)\textquotesingle{}, 0.001999827, 20.01481, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 3): {[}3, \textquotesingle{}H2(17O)\textquotesingle{}, 0.0003718841, 19.01478, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 4): {[}4, \textquotesingle{}HD(16O)\textquotesingle{}, 0.0003106928, 19.01674, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 5): {[}5, \textquotesingle{}HD(18O)\textquotesingle{}, 6.230031e\sphinxhyphen{}07, 21.02098, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 6): {[}6, \textquotesingle{}HD(17O)\textquotesingle{}, 1.158526e\sphinxhyphen{}07, 20.02096, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 7): {[}129, \textquotesingle{}D2(16O)\textquotesingle{}, 2.419741e\sphinxhyphen{}08, 20.02292, \textquotesingle{}H2O\textquotesingle{}{]}, (2, 1): {[}7, \textquotesingle{}(12C)(16O)2\textquotesingle{}, 0.9842043, 43.98983, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 2): {[}8, \textquotesingle{}(13C)(16O)2\textquotesingle{}, 0.01105736, 44.99318, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 3): {[}9, \textquotesingle{}(16O)(12C)(18O)\textquotesingle{}, 0.003947066, 45.99408, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 4): {[}10, \textquotesingle{}(16O)(12C)(17O)\textquotesingle{}, 0.000733989, 44.99404, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 5): {[}11, \textquotesingle{}(16O)(13C)(18O)\textquotesingle{}, 4.434456e\sphinxhyphen{}05, 46.99743, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 6): {[}12, \textquotesingle{}(16O)(13C)(17O)\textquotesingle{}, 8.246233e\sphinxhyphen{}06, 45.9974, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 7): {[}13, \textquotesingle{}(12C)(18O)2\textquotesingle{}, 3.95734e\sphinxhyphen{}06, 47.99832, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 8): {[}14, \textquotesingle{}(17O)(12C)(18O)\textquotesingle{}, 1.471799e\sphinxhyphen{}06, 46.99829, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 9): {[}121, \textquotesingle{}(12C)(17O)2\textquotesingle{}, 1.368466e\sphinxhyphen{}07, 45.99826, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 10): {[}15, \textquotesingle{}(13C)(18O)2\textquotesingle{}, 4.446e\sphinxhyphen{}08, 49.00167, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 11): {[}120, \textquotesingle{}(18O)(13C)(17O)\textquotesingle{}, 1.65354e\sphinxhyphen{}08, 48.00165, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 12): {[}122, \textquotesingle{}(13C)(17O)2\textquotesingle{}, 1.537446e\sphinxhyphen{}09, 47.00162, \textquotesingle{}CO2\textquotesingle{}{]}, (3, 1): {[}16, \textquotesingle{}(16O)3\textquotesingle{}, 0.9929009, 47.98474, \textquotesingle{}O3\textquotesingle{}{]}, (3, 2): {[}17, \textquotesingle{}(16O)(16O)(18O)\textquotesingle{}, 0.003981942, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 3): {[}18, \textquotesingle{}(16O)(18O)(16O)\textquotesingle{}, 0.001990971, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 4): {[}19, \textquotesingle{}(16O)(16O)(17O)\textquotesingle{}, 0.0007404746, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (3, 5): {[}20, \textquotesingle{}(16O)(17O)(16O)\textquotesingle{}, 0.0003702373, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (4, 1): {[}21, \textquotesingle{}(14N)2(16O)\textquotesingle{}, 0.9903328, 44.00106, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 2): {[}22, \textquotesingle{}(14N)(15N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 3): {[}23, \textquotesingle{}(15N)(14N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 4): {[}24, \textquotesingle{}(14N)2(18O)\textquotesingle{}, 0.001985822, 46.00531, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 5): {[}25, \textquotesingle{}(14N)2(17O)\textquotesingle{}, 0.0003692797, 45.00528, \textquotesingle{}N2O\textquotesingle{}{]}, (5, 1): {[}26, \textquotesingle{}(12C)(16O)\textquotesingle{}, 0.9865444, 27.99491, \textquotesingle{}CO\textquotesingle{}{]}, (5, 2): {[}27, \textquotesingle{}(13C)(16O)\textquotesingle{}, 0.01108364, 28.99827, \textquotesingle{}CO\textquotesingle{}{]}, (5, 3): {[}28, \textquotesingle{}(12C)(18O)\textquotesingle{}, 0.001978224, 29.99916, \textquotesingle{}CO\textquotesingle{}{]}, (5, 4): {[}29, \textquotesingle{}(12C)(17O)\textquotesingle{}, 0.0003678671, 28.99913, \textquotesingle{}CO\textquotesingle{}{]}, (5, 5): {[}30, \textquotesingle{}(13C)(18O)\textquotesingle{}, 2.2225e\sphinxhyphen{}05, 31.00252, \textquotesingle{}CO\textquotesingle{}{]}, (5, 6): {[}31, \textquotesingle{}(13C)(17O)\textquotesingle{}, 4.13292e\sphinxhyphen{}06, 30.00249, \textquotesingle{}CO\textquotesingle{}{]}, (6, 1): {[}32, \textquotesingle{}(12C)H4\textquotesingle{}, 0.9882741, 16.0313, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 2): {[}33, \textquotesingle{}(13C)H4\textquotesingle{}, 0.01110308, 17.03466, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 3): {[}34, \textquotesingle{}(12C)H3D\textquotesingle{}, 0.0006157511, 17.03748, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 4): {[}35, \textquotesingle{}(13C)H3D\textquotesingle{}, 6.917852e\sphinxhyphen{}06, 18.04083, \textquotesingle{}CH4\textquotesingle{}{]}, (7, 1): {[}36, \textquotesingle{}(16O)2\textquotesingle{}, 0.9952616, 31.98983, \textquotesingle{}O2\textquotesingle{}{]}, (7, 2): {[}37, \textquotesingle{}(16O)(18O)\textquotesingle{}, 0.00399141, 33.99408, \textquotesingle{}O2\textquotesingle{}{]}, (7, 3): {[}38, \textquotesingle{}(16O)(17O)\textquotesingle{}, 0.0007422352, 32.99404, \textquotesingle{}O2\textquotesingle{}{]}, (8, 1): {[}39, \textquotesingle{}(14N)(16O)\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NO\textquotesingle{}{]}, (8, 2): {[}40, \textquotesingle{}(15N)(16O)\textquotesingle{}, 0.003654311, 30.99502, \textquotesingle{}NO\textquotesingle{}{]}, (8, 3): {[}41, \textquotesingle{}(14N)(18O)\textquotesingle{}, 0.001993122, 32.00223, \textquotesingle{}NO\textquotesingle{}{]}, (9, 1): {[}42, \textquotesingle{}(32S)(16O)2\textquotesingle{}, 0.9456777, 63.9619, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 2): {[}43, \textquotesingle{}(34S)(16O)2\textquotesingle{}, 0.04195028, 65.9577, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 3): {[}137, \textquotesingle{}(33S)(16O)2\textquotesingle{}, 0.007464462, 64.96129, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 4): {[}138, \textquotesingle{}(16O)(32S)(18O)\textquotesingle{}, 0.003792558, 65.96615, \textquotesingle{}SO2\textquotesingle{}{]}, (10, 1): {[}44, \textquotesingle{}(14N)(16O)2\textquotesingle{}, 0.991616, 45.9929, \textquotesingle{}NO2\textquotesingle{}{]}, (10, 2): {[}130, \textquotesingle{}(15N)(16O)2\textquotesingle{}, 0.003645643, 46.98994, \textquotesingle{}NO2\textquotesingle{}{]}, (11, 1): {[}45, \textquotesingle{}(14N)H3\textquotesingle{}, 0.9958716, 17.02655, \textquotesingle{}NH3\textquotesingle{}{]}, (11, 2): {[}46, \textquotesingle{}(15N)H3\textquotesingle{}, 0.003661289, 18.02358, \textquotesingle{}NH3\textquotesingle{}{]}, (12, 1): {[}47, \textquotesingle{}H(14N)(16O)3\textquotesingle{}, 0.9891098, 62.99564, \textquotesingle{}HNO3\textquotesingle{}{]}, (12, 2): {[}117, \textquotesingle{}H(15N)(16O)3\textquotesingle{}, 0.003636429, 63.99268, \textquotesingle{}HNO3\textquotesingle{}{]}, (13, 1): {[}48, \textquotesingle{}(16O)H\textquotesingle{}, 0.9974726, 17.00274, \textquotesingle{}OH\textquotesingle{}{]}, (13, 2): {[}49, \textquotesingle{}(18O)H\textquotesingle{}, 0.002000138, 19.00699, \textquotesingle{}OH\textquotesingle{}{]}, (13, 3): {[}50, \textquotesingle{}(16O)D\textquotesingle{}, 0.0001553706, 18.00891, \textquotesingle{}OH\textquotesingle{}{]}, (14, 1): {[}51, \textquotesingle{}H(19F)\textquotesingle{}, 0.9998443, 20.00623, \textquotesingle{}HF\textquotesingle{}{]}, (14, 2): {[}110, \textquotesingle{}D(19F)\textquotesingle{}, 0.000155741, 21.0124, \textquotesingle{}HF\textquotesingle{}{]}, (15, 1): {[}52, \textquotesingle{}H(35Cl)\textquotesingle{}, 0.757587, 35.97668, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 2): {[}53, \textquotesingle{}H(37Cl)\textquotesingle{}, 0.2422573, 37.97373, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 3): {[}107, \textquotesingle{}D(35Cl)\textquotesingle{}, 0.000118005, 36.98285, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 4): {[}108, \textquotesingle{}D(37Cl)\textquotesingle{}, 3.773502e\sphinxhyphen{}05, 38.9799, \textquotesingle{}HCl\textquotesingle{}{]}, (16, 1): {[}54, \textquotesingle{}H(79Br)\textquotesingle{}, 0.5067811, 79.92616, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 2): {[}55, \textquotesingle{}H(81Br)\textquotesingle{}, 0.4930632, 81.92412, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 3): {[}111, \textquotesingle{}D(79Br)\textquotesingle{}, 7.893838e\sphinxhyphen{}05, 80.93234, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 4): {[}112, \textquotesingle{}D(81Br)\textquotesingle{}, 7.680162e\sphinxhyphen{}05, 82.93029, \textquotesingle{}HBr\textquotesingle{}{]}, (17, 1): {[}56, \textquotesingle{}H(127I)\textquotesingle{}, 0.9998443, 127.9123, \textquotesingle{}HI\textquotesingle{}{]}, (17, 2): {[}113, \textquotesingle{}D(127I)\textquotesingle{}, 0.000155741, 128.9185, \textquotesingle{}HI\textquotesingle{}{]}, (18, 1): {[}57, \textquotesingle{}(35Cl)(16O)\textquotesingle{}, 0.7559077, 50.96377, \textquotesingle{}ClO\textquotesingle{}{]}, (18, 2): {[}58, \textquotesingle{}(37Cl)(16O)\textquotesingle{}, 0.2417203, 52.96082, \textquotesingle{}ClO\textquotesingle{}{]}, (19, 1): {[}59, \textquotesingle{}(16O)(12C)(32S)\textquotesingle{}, 0.9373947, 59.96699, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 2): {[}60, \textquotesingle{}(16O)(12C)(34S)\textquotesingle{}, 0.04158284, 61.96278, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 3): {[}61, \textquotesingle{}(16O)(13C)(32S)\textquotesingle{}, 0.01053146, 60.97034, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 4): {[}62, \textquotesingle{}(16O)(12C)(33S)\textquotesingle{}, 0.007399083, 60.96637, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 5): {[}63, \textquotesingle{}(18O)(12C)(32S)\textquotesingle{}, 0.00187967, 61.97123, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 6): {[}135, \textquotesingle{}(16O)(13C)(34S)\textquotesingle{}, 0.0004671757, 62.96614, \textquotesingle{}OCS\textquotesingle{}{]}, (20, 1): {[}64, \textquotesingle{}H2(12C)(16O)\textquotesingle{}, 0.9862371, 30.01056, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 2): {[}65, \textquotesingle{}H2(13C)(16O)\textquotesingle{}, 0.0110802, 31.01392, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 3): {[}66, \textquotesingle{}H2(12C)(18O)\textquotesingle{}, 0.001977609, 32.01481, \textquotesingle{}H2CO\textquotesingle{}{]}, (21, 1): {[}67, \textquotesingle{}H(16O)(35Cl)\textquotesingle{}, 0.75579, 51.97159, \textquotesingle{}HOCl\textquotesingle{}{]}, (21, 2): {[}68, \textquotesingle{}H(16O)(37Cl)\textquotesingle{}, 0.2416826, 53.96864, \textquotesingle{}HOCl\textquotesingle{}{]}, (22, 1): {[}69, \textquotesingle{}(14N)2\textquotesingle{}, 0.9926874, 28.00615, \textquotesingle{}N2\textquotesingle{}{]}, (22, 2): {[}118, \textquotesingle{}(14N)(15N)\textquotesingle{}, 0.007299165, 29.00318, \textquotesingle{}N2\textquotesingle{}{]}, (23, 1): {[}70, \textquotesingle{}H(12C)(14N)\textquotesingle{}, 0.9851143, 27.0109, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 2): {[}71, \textquotesingle{}H(13C)(14N)\textquotesingle{}, 0.01106758, 28.01425, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 3): {[}72, \textquotesingle{}H(12C)(15N)\textquotesingle{}, 0.00362174, 28.00793, \textquotesingle{}HCN\textquotesingle{}{]}, (24, 1): {[}73, \textquotesingle{}(12C)H3(35Cl)\textquotesingle{}, 0.7489369, 49.99233, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (24, 2): {[}74, \textquotesingle{}(12C)H3(37Cl)\textquotesingle{}, 0.2394912, 51.98938, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (25, 1): {[}75, \textquotesingle{}H2(16O)2\textquotesingle{}, 0.9949516, 34.00548, \textquotesingle{}H2O2\textquotesingle{}{]}, (26, 1): {[}76, \textquotesingle{}(12C)2H2\textquotesingle{}, 0.9775989, 26.01565, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 2): {[}77, \textquotesingle{}(12C)(13C)H2\textquotesingle{}, 0.02196629, 27.019, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 3): {[}105, \textquotesingle{}(12C)2HD\textquotesingle{}, 0.0003045499, 27.02182, \textquotesingle{}C2H2\textquotesingle{}{]}, (27, 1): {[}78, \textquotesingle{}(12C)2H6\textquotesingle{}, 0.97699, 30.04695, \textquotesingle{}C2H6\textquotesingle{}{]}, (27, 2): {[}106, \textquotesingle{}(12C)H3(13C)H3\textquotesingle{}, 0.02195261, 31.05031, \textquotesingle{}C2H6\textquotesingle{}{]}, (28, 1): {[}79, \textquotesingle{}(31P)H3\textquotesingle{}, 0.9995329, 33.99724, \textquotesingle{}PH3\textquotesingle{}{]}, (29, 1): {[}80, \textquotesingle{}(12C)(16O)(19F)2\textquotesingle{}, 0.9865444, 65.99172, \textquotesingle{}COF2\textquotesingle{}{]}, (29, 2): {[}119, \textquotesingle{}(13C)(16O)(19F)2\textquotesingle{}, 0.01108366, 66.99508, \textquotesingle{}COF2\textquotesingle{}{]}, (30, 1): {[}126, \textquotesingle{}(32S)(19F)6\textquotesingle{}, 0.95018, 145.9625, \textquotesingle{}SF6\textquotesingle{}{]}, (31, 1): {[}81, \textquotesingle{}H2(32S)\textquotesingle{}, 0.9498841, 33.98772, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 2): {[}82, \textquotesingle{}H2(34S)\textquotesingle{}, 0.04213687, 35.98351, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 3): {[}83, \textquotesingle{}H2(33S)\textquotesingle{}, 0.007497664, 34.9871, \textquotesingle{}H2S\textquotesingle{}{]}, (32, 1): {[}84, \textquotesingle{}H(12C)(16O)(16O)H\textquotesingle{}, 0.9838977, 46.00548, \textquotesingle{}HCOOH\textquotesingle{}{]}, (33, 1): {[}85, \textquotesingle{}H(16O)2\textquotesingle{}, 0.9951066, 32.99766, \textquotesingle{}HO2\textquotesingle{}{]}, (34, 1): {[}86, \textquotesingle{}(16O)\textquotesingle{}, 0.997628, 15.99492, \textquotesingle{}O\textquotesingle{}{]}, (35, 1): {[}127, \textquotesingle{}(35Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.7495702, 96.95667, \textquotesingle{}ClONO2\textquotesingle{}{]}, (35, 2): {[}128, \textquotesingle{}(37Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.2396937, 98.95372, \textquotesingle{}ClONO2\textquotesingle{}{]}, (36, 1): {[}87, \textquotesingle{}(14N)(16O)+\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NOp\textquotesingle{}{]}, (37, 1): {[}88, \textquotesingle{}H(16O)(79Br)\textquotesingle{}, 0.505579, 95.92108, \textquotesingle{}HOBr\textquotesingle{}{]}, (37, 2): {[}89, \textquotesingle{}H(16O)(81Br)\textquotesingle{}, 0.4918937, 97.91903, \textquotesingle{}HOBr\textquotesingle{}{]}, (38, 1): {[}90, \textquotesingle{}(12C)2H4\textquotesingle{}, 0.9772944, 28.0313, \textquotesingle{}C2H4\textquotesingle{}{]}, (38, 2): {[}91, \textquotesingle{}(12C)H2(13C)H2\textquotesingle{}, 0.02195946, 29.03466, \textquotesingle{}C2H4\textquotesingle{}{]}, (39, 1): {[}92, \textquotesingle{}(12C)H3(16O)H\textquotesingle{}, 0.9859299, 32.02622, \textquotesingle{}CH3OH\textquotesingle{}{]}, (40, 1): {[}93, \textquotesingle{}(12C)H3(79Br)\textquotesingle{}, 0.5009946, 93.94181, \textquotesingle{}CH3Br\textquotesingle{}{]}, (40, 2): {[}94, \textquotesingle{}(12C)H3(81Br)\textquotesingle{}, 0.4874334, 95.93976, \textquotesingle{}CH3Br\textquotesingle{}{]}, (41, 1): {[}95, \textquotesingle{}(12C)H3(12C)(14N)\textquotesingle{}, 0.9738662, 41.02655, \textquotesingle{}CH3CN\textquotesingle{}{]}, (42, 1): {[}96, \textquotesingle{}(12C)(19F)4\textquotesingle{}, 0.98889, 87.99362, \textquotesingle{}CF4\textquotesingle{}{]}, (43, 1): {[}116, \textquotesingle{}(12C)4H2\textquotesingle{}, 0.955998, 50.01565, \textquotesingle{}C4H2\textquotesingle{}{]}, (44, 1): {[}109, \textquotesingle{}H(12C)3(14N)\textquotesingle{}, 0.963346, 51.0109, \textquotesingle{}HC3N\textquotesingle{}{]}, (45, 1): {[}103, \textquotesingle{}H2\textquotesingle{}, 0.9996885, 2.01565, \textquotesingle{}H2\textquotesingle{}{]}, (45, 2): {[}115, \textquotesingle{}HD\textquotesingle{}, 0.0003114316, 3.021825, \textquotesingle{}H2\textquotesingle{}{]}, (46, 1): {[}97, \textquotesingle{}(12C)(32S)\textquotesingle{}, 0.9396236, 43.97207, \textquotesingle{}CS\textquotesingle{}{]}, (46, 2): {[}98, \textquotesingle{}(12C)(34S)\textquotesingle{}, 0.04168171, 45.96787, \textquotesingle{}CS\textquotesingle{}{]}, (46, 3): {[}99, \textquotesingle{}(13C)(32S)\textquotesingle{}, 0.0105565, 44.97543, \textquotesingle{}CS\textquotesingle{}{]}, (46, 4): {[}100, \textquotesingle{}(12C)(33S)\textquotesingle{}, 0.007416675, 44.97146, \textquotesingle{}CS\textquotesingle{}{]}, (47, 1): {[}114, \textquotesingle{}(32S)(16O)3\textquotesingle{}, 0.9434345, 79.95682, \textquotesingle{}SO3\textquotesingle{}{]}, (48, 1): {[}123, \textquotesingle{}(12C)2(14N)2\textquotesingle{}, 0.9707524, 52.00615, \textquotesingle{}C2N2\textquotesingle{}{]}, (49, 1): {[}124, \textquotesingle{}(12C)(16O)(35Cl)2\textquotesingle{}, 0.5663918, 97.93262, \textquotesingle{}COCl2\textquotesingle{}{]}, (49, 2): {[}125, \textquotesingle{}(12C)(16O)(35Cl)(37Cl)\textquotesingle{}, 0.362235, 99.92967, \textquotesingle{}COCl2\textquotesingle{}{]}, (50, 1): {[}146, \textquotesingle{}(32S)(16O)\textquotesingle{}, 0.9479262, 47.96699, \textquotesingle{}SO\textquotesingle{}{]}, (50, 2): {[}147, \textquotesingle{}(34S)(16O)\textquotesingle{}, 0.04205002, 49.96278, \textquotesingle{}SO\textquotesingle{}{]}, (50, 3): {[}148, \textquotesingle{}(32S)(18O)\textquotesingle{}, 0.001900788, 49.97123, \textquotesingle{}SO\textquotesingle{}{]}, (51, 1): {[}144, \textquotesingle{}(12C)H3(19F)\textquotesingle{}, 0.988428, 34.02188, \textquotesingle{}CH3F\textquotesingle{}{]}, (52, 1): {[}139, \textquotesingle{}(74Ge)H4\textquotesingle{}, 0.3651724, 77.95248, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 2): {[}140, \textquotesingle{}(72Ge)H4\textquotesingle{}, 0.2741292, 75.95338, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 3): {[}141, \textquotesingle{}(70Ge)H4\textquotesingle{}, 0.2050722, 73.95555, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 4): {[}142, \textquotesingle{}(73Ge)H4\textquotesingle{}, 0.07755167, 76.95476, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 5): {[}143, \textquotesingle{}(76Ge)H4\textquotesingle{}, 0.07755167, 79.9527, \textquotesingle{}GeH4\textquotesingle{}{]}, (53, 1): {[}131, \textquotesingle{}(12C)(32S)2\textquotesingle{}, 0.8928115, 75.94414, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 2): {[}132, \textquotesingle{}(32S)(12C)(34S)\textquotesingle{}, 0.07921026, 77.93994, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 3): {[}133, \textquotesingle{}(32S)(12C)(33S)\textquotesingle{}, 0.01409435, 76.94353, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 4): {[}134, \textquotesingle{}(13C)(32S)2\textquotesingle{}, 0.01003057, 76.9475, \textquotesingle{}CS2\textquotesingle{}{]}, (54, 1): {[}145, \textquotesingle{}(12C)H3(127I)\textquotesingle{}, 0.988428, 141.9279, \textquotesingle{}CH3I\textquotesingle{}{]}, (55, 1): {[}136, \textquotesingle{}(14N)(19F)3\textquotesingle{}, 0.996337, 70.99829, \textquotesingle{}NF3\textquotesingle{}{]}\}, natural\_abundance=True, abundance\_ratio\_MI=\{\}, diluent=\textquotesingle{}air\textquotesingle{}, Diluent=\{\}, nominal\_temperature=296, etalons=\{\}, x\_shift=0, IntensityThreshold=1e\sphinxhyphen{}30, num\_segments=1, beta\_formalism=False, ILS\_function=None, ILS\_resolution=0.1, ILS\_wing=10, TIPS=\textless{}function \textless{}lambda\textgreater{}\textgreater{}, compressability\_file=None}}{}
\pysigstopsignatures
\sphinxAtStartPar
Generates a synthetic spectrum, where the output is a spectrum object that can be used in MATS classes.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{parameter\_linelist}} (\sphinxstyleliteralemphasis{\sphinxupquote{dataframe}}) \textendash{} linelist following the convention of the linelists used for the HTP\_from\_DF\_select.  Note that there will need to be a linemixing column for each nominal temperature, which you will have to do manually (ie y\_air\_296, y\_self\_296).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wavenumbers}} (\sphinxstyleliteralemphasis{\sphinxupquote{array}}\sphinxstyleliteralemphasis{\sphinxupquote{ of }}\sphinxstyleliteralemphasis{\sphinxupquote{floats}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} array of wavenumbers for the simulation (cm\sphinxhyphen{}1).  If provided, then this axis will be used.  If wavenumbers = None, then the wave\_min, wave\_max, and wave\_space will be used to calculate wavenumber grid.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wave\_min}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} minimum wavenumber for the simulation (cm\sphinxhyphen{}1)

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wave\_max}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} maximum wavenumber for the simulation (cm\sphinxhyphen{}1).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wave\_space}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} wavenumber spacing for the simulation (cm\sphinxhyphen{}1).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wave\_error}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} absolute error on the wavenumber axis (cm\sphinxhyphen{}1) to include in simulations. The default is 0.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{SNR}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Signal to noise ratio to impose on the simulated spectrum. The default is None.  If SNR is none there is no noise on the simulation.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{baseline\_terms}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} polynomial baseline coefficients where the index is equal to the coefficient order, ie. {[}0, 1, 2{]} would correspond to baseline = 0 + 1*(wavenumber \sphinxhyphen{} minimum wavenumber) + 2*(wavenumber \sphinxhyphen{} minimum wavenumber)\textasciicircum{}2. The default is {[}0{]}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{temperature}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} temperature for simulation in celsius. The default is 25.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{temperature\_err}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} possible keys include ‘bias’, ‘function’, and ‘params’. The bias indicates the absolute bias in Celsius of the temperature reading, which will be added to the input temperature. Function can be ‘linear’ with params ‘m’ and ‘b’ or ‘sine’ with parameters ‘amp’, ‘phase’, and ‘phase’. These define a function that is added to both the bias and set temperature as a function of the wavenumber. Note: if ‘function’ key is not equal to None, then there also needs to be a params key to define the function.. The default is \{‘bias’: 0, ‘function’: None, ‘params’: \{\}\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pressure}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} pressure for simulation in torr. The default is 760.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pressure\_err}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} possible keys include bias, function, and params. The bias indicates the percent bias in of the pressure reading, which will be added to the input pressure. Function can be ‘linear’ with params ‘m’ and ‘b’ or ‘sine’ with parameters ‘amp’, ‘phase’, and ‘phase’. These define a function that is added to both the bias and set pressure as a function of the wavenumber. Note: if ‘function’ key is not equal to None, then there also needs to be a params key to define the function.. The default is \{‘per\_bias’: 0, ‘function’: None, ‘params’: \{\}\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_cutoff}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of voigt half\sphinxhyphen{}widths to simulate on either side of each line. The default is 25.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_wavenumbers}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} number of wavenumbers to simulate on either side of each line. The default is 25.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{wing\_method}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Provides choice between the wing\_cutoff and wing\_wavenumbers line cut\sphinxhyphen{}off options. The default is ‘wing\_cutoff’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{filename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} allows you to pick the output filename for the simulated spectra. The default is ‘temp’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} mole fraction of each molecule to be simulated in the spectrum in the format \{molec\_id: mole fraction (out of 1), molec\_id: molefraction, … \}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molefraction\_err}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} percent error in the mole fraction of each molecule to be simulated in the spectrum in the format \{molec\_id: percent error in mole fraction, molec\_id: percent error in mole fraction, … \}. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{natural\_abundance}} (\sphinxstyleliteralemphasis{\sphinxupquote{bool}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} flag for if the spectrum contains data at natural abundance. The default is True.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{abundance\_ratio\_MI}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} if not at natural abundance sets the enhancement factor for each molecule and isotope in the following format \{molec\_id:\{iso\_id: enhancement, iso\_id: enhancement\}, … \}. The default is \{\}.  The enhancement is the ratio of the new abundance to the natural abundance.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{isotope\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} provides opportunity to specify the isotope look\sphinxhyphen{}up table.  Default is ISO, which is from HAPI.  If not using ISO, then must use this format and suggested you use function to add to ISO

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} sets the diluent for the sample if only using one broadener. The default is ‘air’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{Diluent}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} sets the diluent for the sample if there are a combination of several. Format \{‘he’: 0.5, ‘air’: 0.5). NOTE: the line parameter file must have parameters that correspond to the diluent (ie gamma0\_he, and gamma0\_air). Additionally, the contribution from all diluents must sum to 1.. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nominal\_temperature}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} nominal temperature indicates the approximate temperature of the spectrum. When combining spectra into a dataset this acts as a flag to whether temperature dependence terms should be parameters that can be fit or whether they should act as constants.. The default is 296.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{etalons}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Allows you to define etalons by an amplitude and period (1/ frequency in cm\sphinxhyphen{}1). Default is no etalons. Input is dictionary with keys being a number and the value being an array with the first index being the amplitude and the second being the period.. The default is \{\}.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{x\_shift}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} value in wavenumbers of the x shift for the spectrum axis.. The default is 0.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{IntensityThreshold}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} minimum line intensity to use in the simulation. The default is 1e\sphinxhyphen{}30.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{num\_segments}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Number of segments in the file, which is implemented labeling the segment column into equal sequential se . The default is 10.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{beta\_formalism}} (\sphinxstyleliteralemphasis{\sphinxupquote{boolean}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Indicates whether the beta correction for Dicke Narrowing should be used.  The default is False.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ILS\_function}} (\sphinxstyleliteralemphasis{\sphinxupquote{string}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Default is None and means that no instrument line shape is used in the fitting.
Function can be: SLIT\_MICHELSON, SLIT\_DIFFRACTION, SLIT\_COSINUS, SLIT\_DISPERSION, SLIT\_GAUSSIAN, SLIT\_TRIANGULAR, SLIT\_RECTANGULAR corresponding to the ILS functions defined in HAPI or a user defined function.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ILS\_resolution}} (\sphinxstyleliteralemphasis{\sphinxupquote{float/array}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Resolution is a float or array of ILS resolutions in wavenumbers.  The SlitFunctions defined in HAPI have 1 resolution, but this opens the option for the user defined function to be comprised of several functions with varying resolutions.  Default is 0.1 cm\sphinxhyphen{}1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ILS\_wing}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} AF\_wing is the a float consisting of the range the ILS is calculted over in cm\sphinxhyphen{}1. Default is 10 cm\sphinxhyphen{}1

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{TIPS}} (\sphinxstyleliteralemphasis{\sphinxupquote{definition}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} selects the HAPI provided TIPS version to use for the partition function

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{spectrum\_file} (\sphinxstyleemphasis{.csv}) \textendash{} File that contains the simulated wavenumber axis, noisy wavenumber axis, absorbance data, noisy absorbance data, percent noise, pressure (torr), and temperature (C). The filename will correspond to the filename parameter, which has a default value of temp. The pressure and temperature columns will include whatever functional change there is to the pressure or temperature, but not the bias offset. This is coded to match how this error would manifest in experiments.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{spectrum\_object} (\sphinxstyleemphasis{object}) \textendash{} Outputs a Spectrum class object. This makes it so the you can easily switch between reading in an experimental spectrum and simulated a synthetic spectrum by simply switching out whether the spectrum object is defined through the class definition or through the simulate\_spectrum function.

\end{itemize}


\end{description}\end{quote}

\end{fulllineitems}

\phantomsection\label{\detokenize{MATS:module-MATS.utilities}}\index{module@\spxentry{module}!MATS.utilities@\spxentry{MATS.utilities}}\index{MATS.utilities@\spxentry{MATS.utilities}!module@\spxentry{module}}\index{add\_to\_HITRANstyle\_isotope\_list() (in module MATS.utilities)@\spxentry{add\_to\_HITRANstyle\_isotope\_list()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.add_to_HITRANstyle_isotope_list}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{add\_to\_HITRANstyle\_isotope\_list}}}{\emph{\DUrole{n}{input\_isotope\_list}\DUrole{o}{=}\DUrole{default_value}{\{(1, 1): {[}1, \textquotesingle{}H2(16O)\textquotesingle{}, 0.9973173, 18.01056, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 2): {[}2, \textquotesingle{}H2(18O)\textquotesingle{}, 0.001999827, 20.01481, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 3): {[}3, \textquotesingle{}H2(17O)\textquotesingle{}, 0.0003718841, 19.01478, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 4): {[}4, \textquotesingle{}HD(16O)\textquotesingle{}, 0.0003106928, 19.01674, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 5): {[}5, \textquotesingle{}HD(18O)\textquotesingle{}, 6.230031e\sphinxhyphen{}07, 21.02098, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 6): {[}6, \textquotesingle{}HD(17O)\textquotesingle{}, 1.158526e\sphinxhyphen{}07, 20.02096, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 7): {[}129, \textquotesingle{}D2(16O)\textquotesingle{}, 2.419741e\sphinxhyphen{}08, 20.02292, \textquotesingle{}H2O\textquotesingle{}{]}, (2, 1): {[}7, \textquotesingle{}(12C)(16O)2\textquotesingle{}, 0.9842043, 43.98983, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 2): {[}8, \textquotesingle{}(13C)(16O)2\textquotesingle{}, 0.01105736, 44.99318, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 3): {[}9, \textquotesingle{}(16O)(12C)(18O)\textquotesingle{}, 0.003947066, 45.99408, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 4): {[}10, \textquotesingle{}(16O)(12C)(17O)\textquotesingle{}, 0.000733989, 44.99404, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 5): {[}11, \textquotesingle{}(16O)(13C)(18O)\textquotesingle{}, 4.434456e\sphinxhyphen{}05, 46.99743, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 6): {[}12, \textquotesingle{}(16O)(13C)(17O)\textquotesingle{}, 8.246233e\sphinxhyphen{}06, 45.9974, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 7): {[}13, \textquotesingle{}(12C)(18O)2\textquotesingle{}, 3.95734e\sphinxhyphen{}06, 47.99832, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 8): {[}14, \textquotesingle{}(17O)(12C)(18O)\textquotesingle{}, 1.471799e\sphinxhyphen{}06, 46.99829, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 9): {[}121, \textquotesingle{}(12C)(17O)2\textquotesingle{}, 1.368466e\sphinxhyphen{}07, 45.99826, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 10): {[}15, \textquotesingle{}(13C)(18O)2\textquotesingle{}, 4.446e\sphinxhyphen{}08, 49.00167, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 11): {[}120, \textquotesingle{}(18O)(13C)(17O)\textquotesingle{}, 1.65354e\sphinxhyphen{}08, 48.00165, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 12): {[}122, \textquotesingle{}(13C)(17O)2\textquotesingle{}, 1.537446e\sphinxhyphen{}09, 47.00162, \textquotesingle{}CO2\textquotesingle{}{]}, (3, 1): {[}16, \textquotesingle{}(16O)3\textquotesingle{}, 0.9929009, 47.98474, \textquotesingle{}O3\textquotesingle{}{]}, (3, 2): {[}17, \textquotesingle{}(16O)(16O)(18O)\textquotesingle{}, 0.003981942, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 3): {[}18, \textquotesingle{}(16O)(18O)(16O)\textquotesingle{}, 0.001990971, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 4): {[}19, \textquotesingle{}(16O)(16O)(17O)\textquotesingle{}, 0.0007404746, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (3, 5): {[}20, \textquotesingle{}(16O)(17O)(16O)\textquotesingle{}, 0.0003702373, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (4, 1): {[}21, \textquotesingle{}(14N)2(16O)\textquotesingle{}, 0.9903328, 44.00106, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 2): {[}22, \textquotesingle{}(14N)(15N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 3): {[}23, \textquotesingle{}(15N)(14N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 4): {[}24, \textquotesingle{}(14N)2(18O)\textquotesingle{}, 0.001985822, 46.00531, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 5): {[}25, \textquotesingle{}(14N)2(17O)\textquotesingle{}, 0.0003692797, 45.00528, \textquotesingle{}N2O\textquotesingle{}{]}, (5, 1): {[}26, \textquotesingle{}(12C)(16O)\textquotesingle{}, 0.9865444, 27.99491, \textquotesingle{}CO\textquotesingle{}{]}, (5, 2): {[}27, \textquotesingle{}(13C)(16O)\textquotesingle{}, 0.01108364, 28.99827, \textquotesingle{}CO\textquotesingle{}{]}, (5, 3): {[}28, \textquotesingle{}(12C)(18O)\textquotesingle{}, 0.001978224, 29.99916, \textquotesingle{}CO\textquotesingle{}{]}, (5, 4): {[}29, \textquotesingle{}(12C)(17O)\textquotesingle{}, 0.0003678671, 28.99913, \textquotesingle{}CO\textquotesingle{}{]}, (5, 5): {[}30, \textquotesingle{}(13C)(18O)\textquotesingle{}, 2.2225e\sphinxhyphen{}05, 31.00252, \textquotesingle{}CO\textquotesingle{}{]}, (5, 6): {[}31, \textquotesingle{}(13C)(17O)\textquotesingle{}, 4.13292e\sphinxhyphen{}06, 30.00249, \textquotesingle{}CO\textquotesingle{}{]}, (6, 1): {[}32, \textquotesingle{}(12C)H4\textquotesingle{}, 0.9882741, 16.0313, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 2): {[}33, \textquotesingle{}(13C)H4\textquotesingle{}, 0.01110308, 17.03466, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 3): {[}34, \textquotesingle{}(12C)H3D\textquotesingle{}, 0.0006157511, 17.03748, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 4): {[}35, \textquotesingle{}(13C)H3D\textquotesingle{}, 6.917852e\sphinxhyphen{}06, 18.04083, \textquotesingle{}CH4\textquotesingle{}{]}, (7, 1): {[}36, \textquotesingle{}(16O)2\textquotesingle{}, 0.9952616, 31.98983, \textquotesingle{}O2\textquotesingle{}{]}, (7, 2): {[}37, \textquotesingle{}(16O)(18O)\textquotesingle{}, 0.00399141, 33.99408, \textquotesingle{}O2\textquotesingle{}{]}, (7, 3): {[}38, \textquotesingle{}(16O)(17O)\textquotesingle{}, 0.0007422352, 32.99404, \textquotesingle{}O2\textquotesingle{}{]}, (8, 1): {[}39, \textquotesingle{}(14N)(16O)\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NO\textquotesingle{}{]}, (8, 2): {[}40, \textquotesingle{}(15N)(16O)\textquotesingle{}, 0.003654311, 30.99502, \textquotesingle{}NO\textquotesingle{}{]}, (8, 3): {[}41, \textquotesingle{}(14N)(18O)\textquotesingle{}, 0.001993122, 32.00223, \textquotesingle{}NO\textquotesingle{}{]}, (9, 1): {[}42, \textquotesingle{}(32S)(16O)2\textquotesingle{}, 0.9456777, 63.9619, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 2): {[}43, \textquotesingle{}(34S)(16O)2\textquotesingle{}, 0.04195028, 65.9577, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 3): {[}137, \textquotesingle{}(33S)(16O)2\textquotesingle{}, 0.007464462, 64.96129, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 4): {[}138, \textquotesingle{}(16O)(32S)(18O)\textquotesingle{}, 0.003792558, 65.96615, \textquotesingle{}SO2\textquotesingle{}{]}, (10, 1): {[}44, \textquotesingle{}(14N)(16O)2\textquotesingle{}, 0.991616, 45.9929, \textquotesingle{}NO2\textquotesingle{}{]}, (10, 2): {[}130, \textquotesingle{}(15N)(16O)2\textquotesingle{}, 0.003645643, 46.98994, \textquotesingle{}NO2\textquotesingle{}{]}, (11, 1): {[}45, \textquotesingle{}(14N)H3\textquotesingle{}, 0.9958716, 17.02655, \textquotesingle{}NH3\textquotesingle{}{]}, (11, 2): {[}46, \textquotesingle{}(15N)H3\textquotesingle{}, 0.003661289, 18.02358, \textquotesingle{}NH3\textquotesingle{}{]}, (12, 1): {[}47, \textquotesingle{}H(14N)(16O)3\textquotesingle{}, 0.9891098, 62.99564, \textquotesingle{}HNO3\textquotesingle{}{]}, (12, 2): {[}117, \textquotesingle{}H(15N)(16O)3\textquotesingle{}, 0.003636429, 63.99268, \textquotesingle{}HNO3\textquotesingle{}{]}, (13, 1): {[}48, \textquotesingle{}(16O)H\textquotesingle{}, 0.9974726, 17.00274, \textquotesingle{}OH\textquotesingle{}{]}, (13, 2): {[}49, \textquotesingle{}(18O)H\textquotesingle{}, 0.002000138, 19.00699, \textquotesingle{}OH\textquotesingle{}{]}, (13, 3): {[}50, \textquotesingle{}(16O)D\textquotesingle{}, 0.0001553706, 18.00891, \textquotesingle{}OH\textquotesingle{}{]}, (14, 1): {[}51, \textquotesingle{}H(19F)\textquotesingle{}, 0.9998443, 20.00623, \textquotesingle{}HF\textquotesingle{}{]}, (14, 2): {[}110, \textquotesingle{}D(19F)\textquotesingle{}, 0.000155741, 21.0124, \textquotesingle{}HF\textquotesingle{}{]}, (15, 1): {[}52, \textquotesingle{}H(35Cl)\textquotesingle{}, 0.757587, 35.97668, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 2): {[}53, \textquotesingle{}H(37Cl)\textquotesingle{}, 0.2422573, 37.97373, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 3): {[}107, \textquotesingle{}D(35Cl)\textquotesingle{}, 0.000118005, 36.98285, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 4): {[}108, \textquotesingle{}D(37Cl)\textquotesingle{}, 3.773502e\sphinxhyphen{}05, 38.9799, \textquotesingle{}HCl\textquotesingle{}{]}, (16, 1): {[}54, \textquotesingle{}H(79Br)\textquotesingle{}, 0.5067811, 79.92616, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 2): {[}55, \textquotesingle{}H(81Br)\textquotesingle{}, 0.4930632, 81.92412, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 3): {[}111, \textquotesingle{}D(79Br)\textquotesingle{}, 7.893838e\sphinxhyphen{}05, 80.93234, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 4): {[}112, \textquotesingle{}D(81Br)\textquotesingle{}, 7.680162e\sphinxhyphen{}05, 82.93029, \textquotesingle{}HBr\textquotesingle{}{]}, (17, 1): {[}56, \textquotesingle{}H(127I)\textquotesingle{}, 0.9998443, 127.9123, \textquotesingle{}HI\textquotesingle{}{]}, (17, 2): {[}113, \textquotesingle{}D(127I)\textquotesingle{}, 0.000155741, 128.9185, \textquotesingle{}HI\textquotesingle{}{]}, (18, 1): {[}57, \textquotesingle{}(35Cl)(16O)\textquotesingle{}, 0.7559077, 50.96377, \textquotesingle{}ClO\textquotesingle{}{]}, (18, 2): {[}58, \textquotesingle{}(37Cl)(16O)\textquotesingle{}, 0.2417203, 52.96082, \textquotesingle{}ClO\textquotesingle{}{]}, (19, 1): {[}59, \textquotesingle{}(16O)(12C)(32S)\textquotesingle{}, 0.9373947, 59.96699, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 2): {[}60, \textquotesingle{}(16O)(12C)(34S)\textquotesingle{}, 0.04158284, 61.96278, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 3): {[}61, \textquotesingle{}(16O)(13C)(32S)\textquotesingle{}, 0.01053146, 60.97034, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 4): {[}62, \textquotesingle{}(16O)(12C)(33S)\textquotesingle{}, 0.007399083, 60.96637, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 5): {[}63, \textquotesingle{}(18O)(12C)(32S)\textquotesingle{}, 0.00187967, 61.97123, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 6): {[}135, \textquotesingle{}(16O)(13C)(34S)\textquotesingle{}, 0.0004671757, 62.96614, \textquotesingle{}OCS\textquotesingle{}{]}, (20, 1): {[}64, \textquotesingle{}H2(12C)(16O)\textquotesingle{}, 0.9862371, 30.01056, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 2): {[}65, \textquotesingle{}H2(13C)(16O)\textquotesingle{}, 0.0110802, 31.01392, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 3): {[}66, \textquotesingle{}H2(12C)(18O)\textquotesingle{}, 0.001977609, 32.01481, \textquotesingle{}H2CO\textquotesingle{}{]}, (21, 1): {[}67, \textquotesingle{}H(16O)(35Cl)\textquotesingle{}, 0.75579, 51.97159, \textquotesingle{}HOCl\textquotesingle{}{]}, (21, 2): {[}68, \textquotesingle{}H(16O)(37Cl)\textquotesingle{}, 0.2416826, 53.96864, \textquotesingle{}HOCl\textquotesingle{}{]}, (22, 1): {[}69, \textquotesingle{}(14N)2\textquotesingle{}, 0.9926874, 28.00615, \textquotesingle{}N2\textquotesingle{}{]}, (22, 2): {[}118, \textquotesingle{}(14N)(15N)\textquotesingle{}, 0.007299165, 29.00318, \textquotesingle{}N2\textquotesingle{}{]}, (23, 1): {[}70, \textquotesingle{}H(12C)(14N)\textquotesingle{}, 0.9851143, 27.0109, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 2): {[}71, \textquotesingle{}H(13C)(14N)\textquotesingle{}, 0.01106758, 28.01425, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 3): {[}72, \textquotesingle{}H(12C)(15N)\textquotesingle{}, 0.00362174, 28.00793, \textquotesingle{}HCN\textquotesingle{}{]}, (24, 1): {[}73, \textquotesingle{}(12C)H3(35Cl)\textquotesingle{}, 0.7489369, 49.99233, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (24, 2): {[}74, \textquotesingle{}(12C)H3(37Cl)\textquotesingle{}, 0.2394912, 51.98938, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (25, 1): {[}75, \textquotesingle{}H2(16O)2\textquotesingle{}, 0.9949516, 34.00548, \textquotesingle{}H2O2\textquotesingle{}{]}, (26, 1): {[}76, \textquotesingle{}(12C)2H2\textquotesingle{}, 0.9775989, 26.01565, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 2): {[}77, \textquotesingle{}(12C)(13C)H2\textquotesingle{}, 0.02196629, 27.019, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 3): {[}105, \textquotesingle{}(12C)2HD\textquotesingle{}, 0.0003045499, 27.02182, \textquotesingle{}C2H2\textquotesingle{}{]}, (27, 1): {[}78, \textquotesingle{}(12C)2H6\textquotesingle{}, 0.97699, 30.04695, \textquotesingle{}C2H6\textquotesingle{}{]}, (27, 2): {[}106, \textquotesingle{}(12C)H3(13C)H3\textquotesingle{}, 0.02195261, 31.05031, \textquotesingle{}C2H6\textquotesingle{}{]}, (28, 1): {[}79, \textquotesingle{}(31P)H3\textquotesingle{}, 0.9995329, 33.99724, \textquotesingle{}PH3\textquotesingle{}{]}, (29, 1): {[}80, \textquotesingle{}(12C)(16O)(19F)2\textquotesingle{}, 0.9865444, 65.99172, \textquotesingle{}COF2\textquotesingle{}{]}, (29, 2): {[}119, \textquotesingle{}(13C)(16O)(19F)2\textquotesingle{}, 0.01108366, 66.99508, \textquotesingle{}COF2\textquotesingle{}{]}, (30, 1): {[}126, \textquotesingle{}(32S)(19F)6\textquotesingle{}, 0.95018, 145.9625, \textquotesingle{}SF6\textquotesingle{}{]}, (31, 1): {[}81, \textquotesingle{}H2(32S)\textquotesingle{}, 0.9498841, 33.98772, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 2): {[}82, \textquotesingle{}H2(34S)\textquotesingle{}, 0.04213687, 35.98351, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 3): {[}83, \textquotesingle{}H2(33S)\textquotesingle{}, 0.007497664, 34.9871, \textquotesingle{}H2S\textquotesingle{}{]}, (32, 1): {[}84, \textquotesingle{}H(12C)(16O)(16O)H\textquotesingle{}, 0.9838977, 46.00548, \textquotesingle{}HCOOH\textquotesingle{}{]}, (33, 1): {[}85, \textquotesingle{}H(16O)2\textquotesingle{}, 0.9951066, 32.99766, \textquotesingle{}HO2\textquotesingle{}{]}, (34, 1): {[}86, \textquotesingle{}(16O)\textquotesingle{}, 0.997628, 15.99492, \textquotesingle{}O\textquotesingle{}{]}, (35, 1): {[}127, \textquotesingle{}(35Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.7495702, 96.95667, \textquotesingle{}ClONO2\textquotesingle{}{]}, (35, 2): {[}128, \textquotesingle{}(37Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.2396937, 98.95372, \textquotesingle{}ClONO2\textquotesingle{}{]}, (36, 1): {[}87, \textquotesingle{}(14N)(16O)+\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NOp\textquotesingle{}{]}, (37, 1): {[}88, \textquotesingle{}H(16O)(79Br)\textquotesingle{}, 0.505579, 95.92108, \textquotesingle{}HOBr\textquotesingle{}{]}, (37, 2): {[}89, \textquotesingle{}H(16O)(81Br)\textquotesingle{}, 0.4918937, 97.91903, \textquotesingle{}HOBr\textquotesingle{}{]}, (38, 1): {[}90, \textquotesingle{}(12C)2H4\textquotesingle{}, 0.9772944, 28.0313, \textquotesingle{}C2H4\textquotesingle{}{]}, (38, 2): {[}91, \textquotesingle{}(12C)H2(13C)H2\textquotesingle{}, 0.02195946, 29.03466, \textquotesingle{}C2H4\textquotesingle{}{]}, (39, 1): {[}92, \textquotesingle{}(12C)H3(16O)H\textquotesingle{}, 0.9859299, 32.02622, \textquotesingle{}CH3OH\textquotesingle{}{]}, (40, 1): {[}93, \textquotesingle{}(12C)H3(79Br)\textquotesingle{}, 0.5009946, 93.94181, \textquotesingle{}CH3Br\textquotesingle{}{]}, (40, 2): {[}94, \textquotesingle{}(12C)H3(81Br)\textquotesingle{}, 0.4874334, 95.93976, \textquotesingle{}CH3Br\textquotesingle{}{]}, (41, 1): {[}95, \textquotesingle{}(12C)H3(12C)(14N)\textquotesingle{}, 0.9738662, 41.02655, \textquotesingle{}CH3CN\textquotesingle{}{]}, (42, 1): {[}96, \textquotesingle{}(12C)(19F)4\textquotesingle{}, 0.98889, 87.99362, \textquotesingle{}CF4\textquotesingle{}{]}, (43, 1): {[}116, \textquotesingle{}(12C)4H2\textquotesingle{}, 0.955998, 50.01565, \textquotesingle{}C4H2\textquotesingle{}{]}, (44, 1): {[}109, \textquotesingle{}H(12C)3(14N)\textquotesingle{}, 0.963346, 51.0109, \textquotesingle{}HC3N\textquotesingle{}{]}, (45, 1): {[}103, \textquotesingle{}H2\textquotesingle{}, 0.9996885, 2.01565, \textquotesingle{}H2\textquotesingle{}{]}, (45, 2): {[}115, \textquotesingle{}HD\textquotesingle{}, 0.0003114316, 3.021825, \textquotesingle{}H2\textquotesingle{}{]}, (46, 1): {[}97, \textquotesingle{}(12C)(32S)\textquotesingle{}, 0.9396236, 43.97207, \textquotesingle{}CS\textquotesingle{}{]}, (46, 2): {[}98, \textquotesingle{}(12C)(34S)\textquotesingle{}, 0.04168171, 45.96787, \textquotesingle{}CS\textquotesingle{}{]}, (46, 3): {[}99, \textquotesingle{}(13C)(32S)\textquotesingle{}, 0.0105565, 44.97543, \textquotesingle{}CS\textquotesingle{}{]}, (46, 4): {[}100, \textquotesingle{}(12C)(33S)\textquotesingle{}, 0.007416675, 44.97146, \textquotesingle{}CS\textquotesingle{}{]}, (47, 1): {[}114, \textquotesingle{}(32S)(16O)3\textquotesingle{}, 0.9434345, 79.95682, \textquotesingle{}SO3\textquotesingle{}{]}, (48, 1): {[}123, \textquotesingle{}(12C)2(14N)2\textquotesingle{}, 0.9707524, 52.00615, \textquotesingle{}C2N2\textquotesingle{}{]}, (49, 1): {[}124, \textquotesingle{}(12C)(16O)(35Cl)2\textquotesingle{}, 0.5663918, 97.93262, \textquotesingle{}COCl2\textquotesingle{}{]}, (49, 2): {[}125, \textquotesingle{}(12C)(16O)(35Cl)(37Cl)\textquotesingle{}, 0.362235, 99.92967, \textquotesingle{}COCl2\textquotesingle{}{]}, (50, 1): {[}146, \textquotesingle{}(32S)(16O)\textquotesingle{}, 0.9479262, 47.96699, \textquotesingle{}SO\textquotesingle{}{]}, (50, 2): {[}147, \textquotesingle{}(34S)(16O)\textquotesingle{}, 0.04205002, 49.96278, \textquotesingle{}SO\textquotesingle{}{]}, (50, 3): {[}148, \textquotesingle{}(32S)(18O)\textquotesingle{}, 0.001900788, 49.97123, \textquotesingle{}SO\textquotesingle{}{]}, (51, 1): {[}144, \textquotesingle{}(12C)H3(19F)\textquotesingle{}, 0.988428, 34.02188, \textquotesingle{}CH3F\textquotesingle{}{]}, (52, 1): {[}139, \textquotesingle{}(74Ge)H4\textquotesingle{}, 0.3651724, 77.95248, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 2): {[}140, \textquotesingle{}(72Ge)H4\textquotesingle{}, 0.2741292, 75.95338, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 3): {[}141, \textquotesingle{}(70Ge)H4\textquotesingle{}, 0.2050722, 73.95555, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 4): {[}142, \textquotesingle{}(73Ge)H4\textquotesingle{}, 0.07755167, 76.95476, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 5): {[}143, \textquotesingle{}(76Ge)H4\textquotesingle{}, 0.07755167, 79.9527, \textquotesingle{}GeH4\textquotesingle{}{]}, (53, 1): {[}131, \textquotesingle{}(12C)(32S)2\textquotesingle{}, 0.8928115, 75.94414, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 2): {[}132, \textquotesingle{}(32S)(12C)(34S)\textquotesingle{}, 0.07921026, 77.93994, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 3): {[}133, \textquotesingle{}(32S)(12C)(33S)\textquotesingle{}, 0.01409435, 76.94353, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 4): {[}134, \textquotesingle{}(13C)(32S)2\textquotesingle{}, 0.01003057, 76.9475, \textquotesingle{}CS2\textquotesingle{}{]}, (54, 1): {[}145, \textquotesingle{}(12C)H3(127I)\textquotesingle{}, 0.988428, 141.9279, \textquotesingle{}CH3I\textquotesingle{}{]}, (55, 1): {[}136, \textquotesingle{}(14N)(19F)3\textquotesingle{}, 0.996337, 70.99829, \textquotesingle{}NF3\textquotesingle{}{]}\}}}, \emph{\DUrole{n}{molec\_id}\DUrole{o}{=}\DUrole{default_value}{100}}, \emph{\DUrole{n}{local\_iso\_id}\DUrole{o}{=}\DUrole{default_value}{1}}, \emph{\DUrole{n}{global\_isotope\_id}\DUrole{o}{=}\DUrole{default_value}{200}}, \emph{\DUrole{n}{iso\_name}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{abundance}\DUrole{o}{=}\DUrole{default_value}{1}}, \emph{\DUrole{n}{mass}\DUrole{o}{=}\DUrole{default_value}{1}}, \emph{\DUrole{n}{mol\_name}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Allows for used to add to an existing isotope line list in the HITRAN format
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{input\_isotope\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{dictionary}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Isotope list dictionary in HAPI format. The default is ISO.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{molec\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} molec\_id for new isotope entry. The default is 100.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{local\_iso\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} local isotope id number for new isotope entry. The default is 1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{global\_isotope\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} global isotope id for new istope entry. The default is 200.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{iso\_name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} isotoope name for new isotope entry. The default is ‘’.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{abundance}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} relative abundance of new isotope entry. The default is 1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{mass}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Mass of isotope in g. The default is 1.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{mol\_name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} name of the molecule for new istope entry. The default is ‘’.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{output\_isotope\_list} \textendash{} Isotope list with additionall entry

\sphinxlineitem{Return type}
\sphinxAtStartPar
dictionary

\end{description}\end{quote}

\end{fulllineitems}

\index{arange\_() (in module MATS.utilities)@\spxentry{arange\_()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.arange_}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{arange\_}}}{\emph{\DUrole{n}{lower}}, \emph{\DUrole{n}{upper}}, \emph{\DUrole{n}{step}}}{}
\pysigstopsignatures
\sphinxAtStartPar
originally from HAPI 1.1.0.9.6, but corrects npnt to be int

\end{fulllineitems}

\index{convolveSpectrumSame() (in module MATS.utilities)@\spxentry{convolveSpectrumSame()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.convolveSpectrumSame}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{convolveSpectrumSame}}}{\emph{\DUrole{n}{Omega}}, \emph{\DUrole{n}{CrossSection}}, \emph{\DUrole{n}{Resolution=0.1}}, \emph{\DUrole{n}{AF\_wing=10.0}}, \emph{\DUrole{n}{SlitFunction=\textless{}function SLIT\_RECTANGULAR\textgreater{}}}, \emph{\DUrole{n}{Wavenumber=None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Convolves cross section with a slit function with given parameters.
Originally from HAPI 1.1.0.9.6 with correction to {\color{red}\bfseries{}arange\_} to prevent float/int error

\end{fulllineitems}

\index{etalon() (in module MATS.utilities)@\spxentry{etalon()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.etalon}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{etalon}}}{\emph{\DUrole{n}{x}}, \emph{\DUrole{n}{amp}}, \emph{\DUrole{n}{period}}, \emph{\DUrole{n}{phase}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Etalon definition
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{x}} (\sphinxstyleliteralemphasis{\sphinxupquote{array}}) \textendash{} array of floats used to define the x\sphinxhyphen{}axis

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{amp}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} amplitude of the etalon.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{period}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} period of the etalon.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{phase}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} phase of the etalon.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{etalon} \textendash{} etalon as a function of input x\sphinxhyphen{}axis, amplitude, period, and phase.

\sphinxlineitem{Return type}
\sphinxAtStartPar
array

\end{description}\end{quote}

\end{fulllineitems}

\index{hasNumbers() (in module MATS.utilities)@\spxentry{hasNumbers()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.hasNumbers}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{hasNumbers}}}{\emph{\DUrole{n}{inputString}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Determines whether there are numbers in a string
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{inputString}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} string for analysis

\sphinxlineitem{Returns}
\sphinxAtStartPar
Returns True if the there are numbers in a string

\sphinxlineitem{Return type}
\sphinxAtStartPar
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{isotope\_list\_molecules\_isotopes() (in module MATS.utilities)@\spxentry{isotope\_list\_molecules\_isotopes()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.isotope_list_molecules_isotopes}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{isotope\_list\_molecules\_isotopes}}}{\emph{\DUrole{n}{isotope\_list}\DUrole{o}{=}\DUrole{default_value}{\{(1, 1): {[}1, \textquotesingle{}H2(16O)\textquotesingle{}, 0.9973173, 18.01056, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 2): {[}2, \textquotesingle{}H2(18O)\textquotesingle{}, 0.001999827, 20.01481, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 3): {[}3, \textquotesingle{}H2(17O)\textquotesingle{}, 0.0003718841, 19.01478, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 4): {[}4, \textquotesingle{}HD(16O)\textquotesingle{}, 0.0003106928, 19.01674, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 5): {[}5, \textquotesingle{}HD(18O)\textquotesingle{}, 6.230031e\sphinxhyphen{}07, 21.02098, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 6): {[}6, \textquotesingle{}HD(17O)\textquotesingle{}, 1.158526e\sphinxhyphen{}07, 20.02096, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 7): {[}129, \textquotesingle{}D2(16O)\textquotesingle{}, 2.419741e\sphinxhyphen{}08, 20.02292, \textquotesingle{}H2O\textquotesingle{}{]}, (2, 1): {[}7, \textquotesingle{}(12C)(16O)2\textquotesingle{}, 0.9842043, 43.98983, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 2): {[}8, \textquotesingle{}(13C)(16O)2\textquotesingle{}, 0.01105736, 44.99318, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 3): {[}9, \textquotesingle{}(16O)(12C)(18O)\textquotesingle{}, 0.003947066, 45.99408, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 4): {[}10, \textquotesingle{}(16O)(12C)(17O)\textquotesingle{}, 0.000733989, 44.99404, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 5): {[}11, \textquotesingle{}(16O)(13C)(18O)\textquotesingle{}, 4.434456e\sphinxhyphen{}05, 46.99743, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 6): {[}12, \textquotesingle{}(16O)(13C)(17O)\textquotesingle{}, 8.246233e\sphinxhyphen{}06, 45.9974, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 7): {[}13, \textquotesingle{}(12C)(18O)2\textquotesingle{}, 3.95734e\sphinxhyphen{}06, 47.99832, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 8): {[}14, \textquotesingle{}(17O)(12C)(18O)\textquotesingle{}, 1.471799e\sphinxhyphen{}06, 46.99829, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 9): {[}121, \textquotesingle{}(12C)(17O)2\textquotesingle{}, 1.368466e\sphinxhyphen{}07, 45.99826, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 10): {[}15, \textquotesingle{}(13C)(18O)2\textquotesingle{}, 4.446e\sphinxhyphen{}08, 49.00167, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 11): {[}120, \textquotesingle{}(18O)(13C)(17O)\textquotesingle{}, 1.65354e\sphinxhyphen{}08, 48.00165, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 12): {[}122, \textquotesingle{}(13C)(17O)2\textquotesingle{}, 1.537446e\sphinxhyphen{}09, 47.00162, \textquotesingle{}CO2\textquotesingle{}{]}, (3, 1): {[}16, \textquotesingle{}(16O)3\textquotesingle{}, 0.9929009, 47.98474, \textquotesingle{}O3\textquotesingle{}{]}, (3, 2): {[}17, \textquotesingle{}(16O)(16O)(18O)\textquotesingle{}, 0.003981942, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 3): {[}18, \textquotesingle{}(16O)(18O)(16O)\textquotesingle{}, 0.001990971, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 4): {[}19, \textquotesingle{}(16O)(16O)(17O)\textquotesingle{}, 0.0007404746, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (3, 5): {[}20, \textquotesingle{}(16O)(17O)(16O)\textquotesingle{}, 0.0003702373, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (4, 1): {[}21, \textquotesingle{}(14N)2(16O)\textquotesingle{}, 0.9903328, 44.00106, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 2): {[}22, \textquotesingle{}(14N)(15N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 3): {[}23, \textquotesingle{}(15N)(14N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 4): {[}24, \textquotesingle{}(14N)2(18O)\textquotesingle{}, 0.001985822, 46.00531, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 5): {[}25, \textquotesingle{}(14N)2(17O)\textquotesingle{}, 0.0003692797, 45.00528, \textquotesingle{}N2O\textquotesingle{}{]}, (5, 1): {[}26, \textquotesingle{}(12C)(16O)\textquotesingle{}, 0.9865444, 27.99491, \textquotesingle{}CO\textquotesingle{}{]}, (5, 2): {[}27, \textquotesingle{}(13C)(16O)\textquotesingle{}, 0.01108364, 28.99827, \textquotesingle{}CO\textquotesingle{}{]}, (5, 3): {[}28, \textquotesingle{}(12C)(18O)\textquotesingle{}, 0.001978224, 29.99916, \textquotesingle{}CO\textquotesingle{}{]}, (5, 4): {[}29, \textquotesingle{}(12C)(17O)\textquotesingle{}, 0.0003678671, 28.99913, \textquotesingle{}CO\textquotesingle{}{]}, (5, 5): {[}30, \textquotesingle{}(13C)(18O)\textquotesingle{}, 2.2225e\sphinxhyphen{}05, 31.00252, \textquotesingle{}CO\textquotesingle{}{]}, (5, 6): {[}31, \textquotesingle{}(13C)(17O)\textquotesingle{}, 4.13292e\sphinxhyphen{}06, 30.00249, \textquotesingle{}CO\textquotesingle{}{]}, (6, 1): {[}32, \textquotesingle{}(12C)H4\textquotesingle{}, 0.9882741, 16.0313, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 2): {[}33, \textquotesingle{}(13C)H4\textquotesingle{}, 0.01110308, 17.03466, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 3): {[}34, \textquotesingle{}(12C)H3D\textquotesingle{}, 0.0006157511, 17.03748, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 4): {[}35, \textquotesingle{}(13C)H3D\textquotesingle{}, 6.917852e\sphinxhyphen{}06, 18.04083, \textquotesingle{}CH4\textquotesingle{}{]}, (7, 1): {[}36, \textquotesingle{}(16O)2\textquotesingle{}, 0.9952616, 31.98983, \textquotesingle{}O2\textquotesingle{}{]}, (7, 2): {[}37, \textquotesingle{}(16O)(18O)\textquotesingle{}, 0.00399141, 33.99408, \textquotesingle{}O2\textquotesingle{}{]}, (7, 3): {[}38, \textquotesingle{}(16O)(17O)\textquotesingle{}, 0.0007422352, 32.99404, \textquotesingle{}O2\textquotesingle{}{]}, (8, 1): {[}39, \textquotesingle{}(14N)(16O)\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NO\textquotesingle{}{]}, (8, 2): {[}40, \textquotesingle{}(15N)(16O)\textquotesingle{}, 0.003654311, 30.99502, \textquotesingle{}NO\textquotesingle{}{]}, (8, 3): {[}41, \textquotesingle{}(14N)(18O)\textquotesingle{}, 0.001993122, 32.00223, \textquotesingle{}NO\textquotesingle{}{]}, (9, 1): {[}42, \textquotesingle{}(32S)(16O)2\textquotesingle{}, 0.9456777, 63.9619, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 2): {[}43, \textquotesingle{}(34S)(16O)2\textquotesingle{}, 0.04195028, 65.9577, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 3): {[}137, \textquotesingle{}(33S)(16O)2\textquotesingle{}, 0.007464462, 64.96129, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 4): {[}138, \textquotesingle{}(16O)(32S)(18O)\textquotesingle{}, 0.003792558, 65.96615, \textquotesingle{}SO2\textquotesingle{}{]}, (10, 1): {[}44, \textquotesingle{}(14N)(16O)2\textquotesingle{}, 0.991616, 45.9929, \textquotesingle{}NO2\textquotesingle{}{]}, (10, 2): {[}130, \textquotesingle{}(15N)(16O)2\textquotesingle{}, 0.003645643, 46.98994, \textquotesingle{}NO2\textquotesingle{}{]}, (11, 1): {[}45, \textquotesingle{}(14N)H3\textquotesingle{}, 0.9958716, 17.02655, \textquotesingle{}NH3\textquotesingle{}{]}, (11, 2): {[}46, \textquotesingle{}(15N)H3\textquotesingle{}, 0.003661289, 18.02358, \textquotesingle{}NH3\textquotesingle{}{]}, (12, 1): {[}47, \textquotesingle{}H(14N)(16O)3\textquotesingle{}, 0.9891098, 62.99564, \textquotesingle{}HNO3\textquotesingle{}{]}, (12, 2): {[}117, \textquotesingle{}H(15N)(16O)3\textquotesingle{}, 0.003636429, 63.99268, \textquotesingle{}HNO3\textquotesingle{}{]}, (13, 1): {[}48, \textquotesingle{}(16O)H\textquotesingle{}, 0.9974726, 17.00274, \textquotesingle{}OH\textquotesingle{}{]}, (13, 2): {[}49, \textquotesingle{}(18O)H\textquotesingle{}, 0.002000138, 19.00699, \textquotesingle{}OH\textquotesingle{}{]}, (13, 3): {[}50, \textquotesingle{}(16O)D\textquotesingle{}, 0.0001553706, 18.00891, \textquotesingle{}OH\textquotesingle{}{]}, (14, 1): {[}51, \textquotesingle{}H(19F)\textquotesingle{}, 0.9998443, 20.00623, \textquotesingle{}HF\textquotesingle{}{]}, (14, 2): {[}110, \textquotesingle{}D(19F)\textquotesingle{}, 0.000155741, 21.0124, \textquotesingle{}HF\textquotesingle{}{]}, (15, 1): {[}52, \textquotesingle{}H(35Cl)\textquotesingle{}, 0.757587, 35.97668, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 2): {[}53, \textquotesingle{}H(37Cl)\textquotesingle{}, 0.2422573, 37.97373, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 3): {[}107, \textquotesingle{}D(35Cl)\textquotesingle{}, 0.000118005, 36.98285, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 4): {[}108, \textquotesingle{}D(37Cl)\textquotesingle{}, 3.773502e\sphinxhyphen{}05, 38.9799, \textquotesingle{}HCl\textquotesingle{}{]}, (16, 1): {[}54, \textquotesingle{}H(79Br)\textquotesingle{}, 0.5067811, 79.92616, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 2): {[}55, \textquotesingle{}H(81Br)\textquotesingle{}, 0.4930632, 81.92412, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 3): {[}111, \textquotesingle{}D(79Br)\textquotesingle{}, 7.893838e\sphinxhyphen{}05, 80.93234, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 4): {[}112, \textquotesingle{}D(81Br)\textquotesingle{}, 7.680162e\sphinxhyphen{}05, 82.93029, \textquotesingle{}HBr\textquotesingle{}{]}, (17, 1): {[}56, \textquotesingle{}H(127I)\textquotesingle{}, 0.9998443, 127.9123, \textquotesingle{}HI\textquotesingle{}{]}, (17, 2): {[}113, \textquotesingle{}D(127I)\textquotesingle{}, 0.000155741, 128.9185, \textquotesingle{}HI\textquotesingle{}{]}, (18, 1): {[}57, \textquotesingle{}(35Cl)(16O)\textquotesingle{}, 0.7559077, 50.96377, \textquotesingle{}ClO\textquotesingle{}{]}, (18, 2): {[}58, \textquotesingle{}(37Cl)(16O)\textquotesingle{}, 0.2417203, 52.96082, \textquotesingle{}ClO\textquotesingle{}{]}, (19, 1): {[}59, \textquotesingle{}(16O)(12C)(32S)\textquotesingle{}, 0.9373947, 59.96699, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 2): {[}60, \textquotesingle{}(16O)(12C)(34S)\textquotesingle{}, 0.04158284, 61.96278, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 3): {[}61, \textquotesingle{}(16O)(13C)(32S)\textquotesingle{}, 0.01053146, 60.97034, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 4): {[}62, \textquotesingle{}(16O)(12C)(33S)\textquotesingle{}, 0.007399083, 60.96637, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 5): {[}63, \textquotesingle{}(18O)(12C)(32S)\textquotesingle{}, 0.00187967, 61.97123, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 6): {[}135, \textquotesingle{}(16O)(13C)(34S)\textquotesingle{}, 0.0004671757, 62.96614, \textquotesingle{}OCS\textquotesingle{}{]}, (20, 1): {[}64, \textquotesingle{}H2(12C)(16O)\textquotesingle{}, 0.9862371, 30.01056, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 2): {[}65, \textquotesingle{}H2(13C)(16O)\textquotesingle{}, 0.0110802, 31.01392, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 3): {[}66, \textquotesingle{}H2(12C)(18O)\textquotesingle{}, 0.001977609, 32.01481, \textquotesingle{}H2CO\textquotesingle{}{]}, (21, 1): {[}67, \textquotesingle{}H(16O)(35Cl)\textquotesingle{}, 0.75579, 51.97159, \textquotesingle{}HOCl\textquotesingle{}{]}, (21, 2): {[}68, \textquotesingle{}H(16O)(37Cl)\textquotesingle{}, 0.2416826, 53.96864, \textquotesingle{}HOCl\textquotesingle{}{]}, (22, 1): {[}69, \textquotesingle{}(14N)2\textquotesingle{}, 0.9926874, 28.00615, \textquotesingle{}N2\textquotesingle{}{]}, (22, 2): {[}118, \textquotesingle{}(14N)(15N)\textquotesingle{}, 0.007299165, 29.00318, \textquotesingle{}N2\textquotesingle{}{]}, (23, 1): {[}70, \textquotesingle{}H(12C)(14N)\textquotesingle{}, 0.9851143, 27.0109, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 2): {[}71, \textquotesingle{}H(13C)(14N)\textquotesingle{}, 0.01106758, 28.01425, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 3): {[}72, \textquotesingle{}H(12C)(15N)\textquotesingle{}, 0.00362174, 28.00793, \textquotesingle{}HCN\textquotesingle{}{]}, (24, 1): {[}73, \textquotesingle{}(12C)H3(35Cl)\textquotesingle{}, 0.7489369, 49.99233, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (24, 2): {[}74, \textquotesingle{}(12C)H3(37Cl)\textquotesingle{}, 0.2394912, 51.98938, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (25, 1): {[}75, \textquotesingle{}H2(16O)2\textquotesingle{}, 0.9949516, 34.00548, \textquotesingle{}H2O2\textquotesingle{}{]}, (26, 1): {[}76, \textquotesingle{}(12C)2H2\textquotesingle{}, 0.9775989, 26.01565, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 2): {[}77, \textquotesingle{}(12C)(13C)H2\textquotesingle{}, 0.02196629, 27.019, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 3): {[}105, \textquotesingle{}(12C)2HD\textquotesingle{}, 0.0003045499, 27.02182, \textquotesingle{}C2H2\textquotesingle{}{]}, (27, 1): {[}78, \textquotesingle{}(12C)2H6\textquotesingle{}, 0.97699, 30.04695, \textquotesingle{}C2H6\textquotesingle{}{]}, (27, 2): {[}106, \textquotesingle{}(12C)H3(13C)H3\textquotesingle{}, 0.02195261, 31.05031, \textquotesingle{}C2H6\textquotesingle{}{]}, (28, 1): {[}79, \textquotesingle{}(31P)H3\textquotesingle{}, 0.9995329, 33.99724, \textquotesingle{}PH3\textquotesingle{}{]}, (29, 1): {[}80, \textquotesingle{}(12C)(16O)(19F)2\textquotesingle{}, 0.9865444, 65.99172, \textquotesingle{}COF2\textquotesingle{}{]}, (29, 2): {[}119, \textquotesingle{}(13C)(16O)(19F)2\textquotesingle{}, 0.01108366, 66.99508, \textquotesingle{}COF2\textquotesingle{}{]}, (30, 1): {[}126, \textquotesingle{}(32S)(19F)6\textquotesingle{}, 0.95018, 145.9625, \textquotesingle{}SF6\textquotesingle{}{]}, (31, 1): {[}81, \textquotesingle{}H2(32S)\textquotesingle{}, 0.9498841, 33.98772, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 2): {[}82, \textquotesingle{}H2(34S)\textquotesingle{}, 0.04213687, 35.98351, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 3): {[}83, \textquotesingle{}H2(33S)\textquotesingle{}, 0.007497664, 34.9871, \textquotesingle{}H2S\textquotesingle{}{]}, (32, 1): {[}84, \textquotesingle{}H(12C)(16O)(16O)H\textquotesingle{}, 0.9838977, 46.00548, \textquotesingle{}HCOOH\textquotesingle{}{]}, (33, 1): {[}85, \textquotesingle{}H(16O)2\textquotesingle{}, 0.9951066, 32.99766, \textquotesingle{}HO2\textquotesingle{}{]}, (34, 1): {[}86, \textquotesingle{}(16O)\textquotesingle{}, 0.997628, 15.99492, \textquotesingle{}O\textquotesingle{}{]}, (35, 1): {[}127, \textquotesingle{}(35Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.7495702, 96.95667, \textquotesingle{}ClONO2\textquotesingle{}{]}, (35, 2): {[}128, \textquotesingle{}(37Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.2396937, 98.95372, \textquotesingle{}ClONO2\textquotesingle{}{]}, (36, 1): {[}87, \textquotesingle{}(14N)(16O)+\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NOp\textquotesingle{}{]}, (37, 1): {[}88, \textquotesingle{}H(16O)(79Br)\textquotesingle{}, 0.505579, 95.92108, \textquotesingle{}HOBr\textquotesingle{}{]}, (37, 2): {[}89, \textquotesingle{}H(16O)(81Br)\textquotesingle{}, 0.4918937, 97.91903, \textquotesingle{}HOBr\textquotesingle{}{]}, (38, 1): {[}90, \textquotesingle{}(12C)2H4\textquotesingle{}, 0.9772944, 28.0313, \textquotesingle{}C2H4\textquotesingle{}{]}, (38, 2): {[}91, \textquotesingle{}(12C)H2(13C)H2\textquotesingle{}, 0.02195946, 29.03466, \textquotesingle{}C2H4\textquotesingle{}{]}, (39, 1): {[}92, \textquotesingle{}(12C)H3(16O)H\textquotesingle{}, 0.9859299, 32.02622, \textquotesingle{}CH3OH\textquotesingle{}{]}, (40, 1): {[}93, \textquotesingle{}(12C)H3(79Br)\textquotesingle{}, 0.5009946, 93.94181, \textquotesingle{}CH3Br\textquotesingle{}{]}, (40, 2): {[}94, \textquotesingle{}(12C)H3(81Br)\textquotesingle{}, 0.4874334, 95.93976, \textquotesingle{}CH3Br\textquotesingle{}{]}, (41, 1): {[}95, \textquotesingle{}(12C)H3(12C)(14N)\textquotesingle{}, 0.9738662, 41.02655, \textquotesingle{}CH3CN\textquotesingle{}{]}, (42, 1): {[}96, \textquotesingle{}(12C)(19F)4\textquotesingle{}, 0.98889, 87.99362, \textquotesingle{}CF4\textquotesingle{}{]}, (43, 1): {[}116, \textquotesingle{}(12C)4H2\textquotesingle{}, 0.955998, 50.01565, \textquotesingle{}C4H2\textquotesingle{}{]}, (44, 1): {[}109, \textquotesingle{}H(12C)3(14N)\textquotesingle{}, 0.963346, 51.0109, \textquotesingle{}HC3N\textquotesingle{}{]}, (45, 1): {[}103, \textquotesingle{}H2\textquotesingle{}, 0.9996885, 2.01565, \textquotesingle{}H2\textquotesingle{}{]}, (45, 2): {[}115, \textquotesingle{}HD\textquotesingle{}, 0.0003114316, 3.021825, \textquotesingle{}H2\textquotesingle{}{]}, (46, 1): {[}97, \textquotesingle{}(12C)(32S)\textquotesingle{}, 0.9396236, 43.97207, \textquotesingle{}CS\textquotesingle{}{]}, (46, 2): {[}98, \textquotesingle{}(12C)(34S)\textquotesingle{}, 0.04168171, 45.96787, \textquotesingle{}CS\textquotesingle{}{]}, (46, 3): {[}99, \textquotesingle{}(13C)(32S)\textquotesingle{}, 0.0105565, 44.97543, \textquotesingle{}CS\textquotesingle{}{]}, (46, 4): {[}100, \textquotesingle{}(12C)(33S)\textquotesingle{}, 0.007416675, 44.97146, \textquotesingle{}CS\textquotesingle{}{]}, (47, 1): {[}114, \textquotesingle{}(32S)(16O)3\textquotesingle{}, 0.9434345, 79.95682, \textquotesingle{}SO3\textquotesingle{}{]}, (48, 1): {[}123, \textquotesingle{}(12C)2(14N)2\textquotesingle{}, 0.9707524, 52.00615, \textquotesingle{}C2N2\textquotesingle{}{]}, (49, 1): {[}124, \textquotesingle{}(12C)(16O)(35Cl)2\textquotesingle{}, 0.5663918, 97.93262, \textquotesingle{}COCl2\textquotesingle{}{]}, (49, 2): {[}125, \textquotesingle{}(12C)(16O)(35Cl)(37Cl)\textquotesingle{}, 0.362235, 99.92967, \textquotesingle{}COCl2\textquotesingle{}{]}, (50, 1): {[}146, \textquotesingle{}(32S)(16O)\textquotesingle{}, 0.9479262, 47.96699, \textquotesingle{}SO\textquotesingle{}{]}, (50, 2): {[}147, \textquotesingle{}(34S)(16O)\textquotesingle{}, 0.04205002, 49.96278, \textquotesingle{}SO\textquotesingle{}{]}, (50, 3): {[}148, \textquotesingle{}(32S)(18O)\textquotesingle{}, 0.001900788, 49.97123, \textquotesingle{}SO\textquotesingle{}{]}, (51, 1): {[}144, \textquotesingle{}(12C)H3(19F)\textquotesingle{}, 0.988428, 34.02188, \textquotesingle{}CH3F\textquotesingle{}{]}, (52, 1): {[}139, \textquotesingle{}(74Ge)H4\textquotesingle{}, 0.3651724, 77.95248, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 2): {[}140, \textquotesingle{}(72Ge)H4\textquotesingle{}, 0.2741292, 75.95338, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 3): {[}141, \textquotesingle{}(70Ge)H4\textquotesingle{}, 0.2050722, 73.95555, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 4): {[}142, \textquotesingle{}(73Ge)H4\textquotesingle{}, 0.07755167, 76.95476, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 5): {[}143, \textquotesingle{}(76Ge)H4\textquotesingle{}, 0.07755167, 79.9527, \textquotesingle{}GeH4\textquotesingle{}{]}, (53, 1): {[}131, \textquotesingle{}(12C)(32S)2\textquotesingle{}, 0.8928115, 75.94414, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 2): {[}132, \textquotesingle{}(32S)(12C)(34S)\textquotesingle{}, 0.07921026, 77.93994, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 3): {[}133, \textquotesingle{}(32S)(12C)(33S)\textquotesingle{}, 0.01409435, 76.94353, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 4): {[}134, \textquotesingle{}(13C)(32S)2\textquotesingle{}, 0.01003057, 76.9475, \textquotesingle{}CS2\textquotesingle{}{]}, (54, 1): {[}145, \textquotesingle{}(12C)H3(127I)\textquotesingle{}, 0.988428, 141.9279, \textquotesingle{}CH3I\textquotesingle{}{]}, (55, 1): {[}136, \textquotesingle{}(14N)(19F)3\textquotesingle{}, 0.996337, 70.99829, \textquotesingle{}NF3\textquotesingle{}{]}\}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
The HITRAN style isotope list in the format (M,I), this function creates a dictionary from this with M as the keys and lists of I as values.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{isotope\_list}} (\sphinxstyleliteralemphasis{\sphinxupquote{dictionary}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} HITRAN style isotope list. The default is ISO.

\sphinxlineitem{Returns}
\sphinxAtStartPar
\sphinxstylestrong{molecule\_isotope\_dictionary} \textendash{} Dictionary with the molecules in an isotope list with the available I values.

\sphinxlineitem{Return type}
\sphinxAtStartPar
dictionary

\end{description}\end{quote}

\end{fulllineitems}

\index{max\_iter() (in module MATS.utilities)@\spxentry{max\_iter()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.max_iter}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{max\_iter}}}{\emph{\DUrole{n}{pars}}, \emph{\DUrole{n}{iter}}, \emph{\DUrole{n}{resid}}, \emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kws}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{molecularMass() (in module MATS.utilities)@\spxentry{molecularMass()}\spxextra{in module MATS.utilities}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{MATS:MATS.utilities.molecularMass}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{MATS.utilities.}}\sphinxbfcode{\sphinxupquote{molecularMass}}}{\emph{\DUrole{n}{M}}, \emph{\DUrole{n}{I}}, \emph{\DUrole{n}{isotope\_list}\DUrole{o}{=}\DUrole{default_value}{\{(1, 1): {[}1, \textquotesingle{}H2(16O)\textquotesingle{}, 0.9973173, 18.01056, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 2): {[}2, \textquotesingle{}H2(18O)\textquotesingle{}, 0.001999827, 20.01481, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 3): {[}3, \textquotesingle{}H2(17O)\textquotesingle{}, 0.0003718841, 19.01478, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 4): {[}4, \textquotesingle{}HD(16O)\textquotesingle{}, 0.0003106928, 19.01674, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 5): {[}5, \textquotesingle{}HD(18O)\textquotesingle{}, 6.230031e\sphinxhyphen{}07, 21.02098, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 6): {[}6, \textquotesingle{}HD(17O)\textquotesingle{}, 1.158526e\sphinxhyphen{}07, 20.02096, \textquotesingle{}H2O\textquotesingle{}{]}, (1, 7): {[}129, \textquotesingle{}D2(16O)\textquotesingle{}, 2.419741e\sphinxhyphen{}08, 20.02292, \textquotesingle{}H2O\textquotesingle{}{]}, (2, 1): {[}7, \textquotesingle{}(12C)(16O)2\textquotesingle{}, 0.9842043, 43.98983, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 2): {[}8, \textquotesingle{}(13C)(16O)2\textquotesingle{}, 0.01105736, 44.99318, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 3): {[}9, \textquotesingle{}(16O)(12C)(18O)\textquotesingle{}, 0.003947066, 45.99408, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 4): {[}10, \textquotesingle{}(16O)(12C)(17O)\textquotesingle{}, 0.000733989, 44.99404, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 5): {[}11, \textquotesingle{}(16O)(13C)(18O)\textquotesingle{}, 4.434456e\sphinxhyphen{}05, 46.99743, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 6): {[}12, \textquotesingle{}(16O)(13C)(17O)\textquotesingle{}, 8.246233e\sphinxhyphen{}06, 45.9974, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 7): {[}13, \textquotesingle{}(12C)(18O)2\textquotesingle{}, 3.95734e\sphinxhyphen{}06, 47.99832, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 8): {[}14, \textquotesingle{}(17O)(12C)(18O)\textquotesingle{}, 1.471799e\sphinxhyphen{}06, 46.99829, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 9): {[}121, \textquotesingle{}(12C)(17O)2\textquotesingle{}, 1.368466e\sphinxhyphen{}07, 45.99826, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 10): {[}15, \textquotesingle{}(13C)(18O)2\textquotesingle{}, 4.446e\sphinxhyphen{}08, 49.00167, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 11): {[}120, \textquotesingle{}(18O)(13C)(17O)\textquotesingle{}, 1.65354e\sphinxhyphen{}08, 48.00165, \textquotesingle{}CO2\textquotesingle{}{]}, (2, 12): {[}122, \textquotesingle{}(13C)(17O)2\textquotesingle{}, 1.537446e\sphinxhyphen{}09, 47.00162, \textquotesingle{}CO2\textquotesingle{}{]}, (3, 1): {[}16, \textquotesingle{}(16O)3\textquotesingle{}, 0.9929009, 47.98474, \textquotesingle{}O3\textquotesingle{}{]}, (3, 2): {[}17, \textquotesingle{}(16O)(16O)(18O)\textquotesingle{}, 0.003981942, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 3): {[}18, \textquotesingle{}(16O)(18O)(16O)\textquotesingle{}, 0.001990971, 49.98899, \textquotesingle{}O3\textquotesingle{}{]}, (3, 4): {[}19, \textquotesingle{}(16O)(16O)(17O)\textquotesingle{}, 0.0007404746, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (3, 5): {[}20, \textquotesingle{}(16O)(17O)(16O)\textquotesingle{}, 0.0003702373, 48.98896, \textquotesingle{}O3\textquotesingle{}{]}, (4, 1): {[}21, \textquotesingle{}(14N)2(16O)\textquotesingle{}, 0.9903328, 44.00106, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 2): {[}22, \textquotesingle{}(14N)(15N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 3): {[}23, \textquotesingle{}(15N)(14N)(16O)\textquotesingle{}, 0.003640926, 44.9981, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 4): {[}24, \textquotesingle{}(14N)2(18O)\textquotesingle{}, 0.001985822, 46.00531, \textquotesingle{}N2O\textquotesingle{}{]}, (4, 5): {[}25, \textquotesingle{}(14N)2(17O)\textquotesingle{}, 0.0003692797, 45.00528, \textquotesingle{}N2O\textquotesingle{}{]}, (5, 1): {[}26, \textquotesingle{}(12C)(16O)\textquotesingle{}, 0.9865444, 27.99491, \textquotesingle{}CO\textquotesingle{}{]}, (5, 2): {[}27, \textquotesingle{}(13C)(16O)\textquotesingle{}, 0.01108364, 28.99827, \textquotesingle{}CO\textquotesingle{}{]}, (5, 3): {[}28, \textquotesingle{}(12C)(18O)\textquotesingle{}, 0.001978224, 29.99916, \textquotesingle{}CO\textquotesingle{}{]}, (5, 4): {[}29, \textquotesingle{}(12C)(17O)\textquotesingle{}, 0.0003678671, 28.99913, \textquotesingle{}CO\textquotesingle{}{]}, (5, 5): {[}30, \textquotesingle{}(13C)(18O)\textquotesingle{}, 2.2225e\sphinxhyphen{}05, 31.00252, \textquotesingle{}CO\textquotesingle{}{]}, (5, 6): {[}31, \textquotesingle{}(13C)(17O)\textquotesingle{}, 4.13292e\sphinxhyphen{}06, 30.00249, \textquotesingle{}CO\textquotesingle{}{]}, (6, 1): {[}32, \textquotesingle{}(12C)H4\textquotesingle{}, 0.9882741, 16.0313, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 2): {[}33, \textquotesingle{}(13C)H4\textquotesingle{}, 0.01110308, 17.03466, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 3): {[}34, \textquotesingle{}(12C)H3D\textquotesingle{}, 0.0006157511, 17.03748, \textquotesingle{}CH4\textquotesingle{}{]}, (6, 4): {[}35, \textquotesingle{}(13C)H3D\textquotesingle{}, 6.917852e\sphinxhyphen{}06, 18.04083, \textquotesingle{}CH4\textquotesingle{}{]}, (7, 1): {[}36, \textquotesingle{}(16O)2\textquotesingle{}, 0.9952616, 31.98983, \textquotesingle{}O2\textquotesingle{}{]}, (7, 2): {[}37, \textquotesingle{}(16O)(18O)\textquotesingle{}, 0.00399141, 33.99408, \textquotesingle{}O2\textquotesingle{}{]}, (7, 3): {[}38, \textquotesingle{}(16O)(17O)\textquotesingle{}, 0.0007422352, 32.99404, \textquotesingle{}O2\textquotesingle{}{]}, (8, 1): {[}39, \textquotesingle{}(14N)(16O)\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NO\textquotesingle{}{]}, (8, 2): {[}40, \textquotesingle{}(15N)(16O)\textquotesingle{}, 0.003654311, 30.99502, \textquotesingle{}NO\textquotesingle{}{]}, (8, 3): {[}41, \textquotesingle{}(14N)(18O)\textquotesingle{}, 0.001993122, 32.00223, \textquotesingle{}NO\textquotesingle{}{]}, (9, 1): {[}42, \textquotesingle{}(32S)(16O)2\textquotesingle{}, 0.9456777, 63.9619, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 2): {[}43, \textquotesingle{}(34S)(16O)2\textquotesingle{}, 0.04195028, 65.9577, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 3): {[}137, \textquotesingle{}(33S)(16O)2\textquotesingle{}, 0.007464462, 64.96129, \textquotesingle{}SO2\textquotesingle{}{]}, (9, 4): {[}138, \textquotesingle{}(16O)(32S)(18O)\textquotesingle{}, 0.003792558, 65.96615, \textquotesingle{}SO2\textquotesingle{}{]}, (10, 1): {[}44, \textquotesingle{}(14N)(16O)2\textquotesingle{}, 0.991616, 45.9929, \textquotesingle{}NO2\textquotesingle{}{]}, (10, 2): {[}130, \textquotesingle{}(15N)(16O)2\textquotesingle{}, 0.003645643, 46.98994, \textquotesingle{}NO2\textquotesingle{}{]}, (11, 1): {[}45, \textquotesingle{}(14N)H3\textquotesingle{}, 0.9958716, 17.02655, \textquotesingle{}NH3\textquotesingle{}{]}, (11, 2): {[}46, \textquotesingle{}(15N)H3\textquotesingle{}, 0.003661289, 18.02358, \textquotesingle{}NH3\textquotesingle{}{]}, (12, 1): {[}47, \textquotesingle{}H(14N)(16O)3\textquotesingle{}, 0.9891098, 62.99564, \textquotesingle{}HNO3\textquotesingle{}{]}, (12, 2): {[}117, \textquotesingle{}H(15N)(16O)3\textquotesingle{}, 0.003636429, 63.99268, \textquotesingle{}HNO3\textquotesingle{}{]}, (13, 1): {[}48, \textquotesingle{}(16O)H\textquotesingle{}, 0.9974726, 17.00274, \textquotesingle{}OH\textquotesingle{}{]}, (13, 2): {[}49, \textquotesingle{}(18O)H\textquotesingle{}, 0.002000138, 19.00699, \textquotesingle{}OH\textquotesingle{}{]}, (13, 3): {[}50, \textquotesingle{}(16O)D\textquotesingle{}, 0.0001553706, 18.00891, \textquotesingle{}OH\textquotesingle{}{]}, (14, 1): {[}51, \textquotesingle{}H(19F)\textquotesingle{}, 0.9998443, 20.00623, \textquotesingle{}HF\textquotesingle{}{]}, (14, 2): {[}110, \textquotesingle{}D(19F)\textquotesingle{}, 0.000155741, 21.0124, \textquotesingle{}HF\textquotesingle{}{]}, (15, 1): {[}52, \textquotesingle{}H(35Cl)\textquotesingle{}, 0.757587, 35.97668, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 2): {[}53, \textquotesingle{}H(37Cl)\textquotesingle{}, 0.2422573, 37.97373, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 3): {[}107, \textquotesingle{}D(35Cl)\textquotesingle{}, 0.000118005, 36.98285, \textquotesingle{}HCl\textquotesingle{}{]}, (15, 4): {[}108, \textquotesingle{}D(37Cl)\textquotesingle{}, 3.773502e\sphinxhyphen{}05, 38.9799, \textquotesingle{}HCl\textquotesingle{}{]}, (16, 1): {[}54, \textquotesingle{}H(79Br)\textquotesingle{}, 0.5067811, 79.92616, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 2): {[}55, \textquotesingle{}H(81Br)\textquotesingle{}, 0.4930632, 81.92412, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 3): {[}111, \textquotesingle{}D(79Br)\textquotesingle{}, 7.893838e\sphinxhyphen{}05, 80.93234, \textquotesingle{}HBr\textquotesingle{}{]}, (16, 4): {[}112, \textquotesingle{}D(81Br)\textquotesingle{}, 7.680162e\sphinxhyphen{}05, 82.93029, \textquotesingle{}HBr\textquotesingle{}{]}, (17, 1): {[}56, \textquotesingle{}H(127I)\textquotesingle{}, 0.9998443, 127.9123, \textquotesingle{}HI\textquotesingle{}{]}, (17, 2): {[}113, \textquotesingle{}D(127I)\textquotesingle{}, 0.000155741, 128.9185, \textquotesingle{}HI\textquotesingle{}{]}, (18, 1): {[}57, \textquotesingle{}(35Cl)(16O)\textquotesingle{}, 0.7559077, 50.96377, \textquotesingle{}ClO\textquotesingle{}{]}, (18, 2): {[}58, \textquotesingle{}(37Cl)(16O)\textquotesingle{}, 0.2417203, 52.96082, \textquotesingle{}ClO\textquotesingle{}{]}, (19, 1): {[}59, \textquotesingle{}(16O)(12C)(32S)\textquotesingle{}, 0.9373947, 59.96699, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 2): {[}60, \textquotesingle{}(16O)(12C)(34S)\textquotesingle{}, 0.04158284, 61.96278, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 3): {[}61, \textquotesingle{}(16O)(13C)(32S)\textquotesingle{}, 0.01053146, 60.97034, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 4): {[}62, \textquotesingle{}(16O)(12C)(33S)\textquotesingle{}, 0.007399083, 60.96637, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 5): {[}63, \textquotesingle{}(18O)(12C)(32S)\textquotesingle{}, 0.00187967, 61.97123, \textquotesingle{}OCS\textquotesingle{}{]}, (19, 6): {[}135, \textquotesingle{}(16O)(13C)(34S)\textquotesingle{}, 0.0004671757, 62.96614, \textquotesingle{}OCS\textquotesingle{}{]}, (20, 1): {[}64, \textquotesingle{}H2(12C)(16O)\textquotesingle{}, 0.9862371, 30.01056, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 2): {[}65, \textquotesingle{}H2(13C)(16O)\textquotesingle{}, 0.0110802, 31.01392, \textquotesingle{}H2CO\textquotesingle{}{]}, (20, 3): {[}66, \textquotesingle{}H2(12C)(18O)\textquotesingle{}, 0.001977609, 32.01481, \textquotesingle{}H2CO\textquotesingle{}{]}, (21, 1): {[}67, \textquotesingle{}H(16O)(35Cl)\textquotesingle{}, 0.75579, 51.97159, \textquotesingle{}HOCl\textquotesingle{}{]}, (21, 2): {[}68, \textquotesingle{}H(16O)(37Cl)\textquotesingle{}, 0.2416826, 53.96864, \textquotesingle{}HOCl\textquotesingle{}{]}, (22, 1): {[}69, \textquotesingle{}(14N)2\textquotesingle{}, 0.9926874, 28.00615, \textquotesingle{}N2\textquotesingle{}{]}, (22, 2): {[}118, \textquotesingle{}(14N)(15N)\textquotesingle{}, 0.007299165, 29.00318, \textquotesingle{}N2\textquotesingle{}{]}, (23, 1): {[}70, \textquotesingle{}H(12C)(14N)\textquotesingle{}, 0.9851143, 27.0109, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 2): {[}71, \textquotesingle{}H(13C)(14N)\textquotesingle{}, 0.01106758, 28.01425, \textquotesingle{}HCN\textquotesingle{}{]}, (23, 3): {[}72, \textquotesingle{}H(12C)(15N)\textquotesingle{}, 0.00362174, 28.00793, \textquotesingle{}HCN\textquotesingle{}{]}, (24, 1): {[}73, \textquotesingle{}(12C)H3(35Cl)\textquotesingle{}, 0.7489369, 49.99233, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (24, 2): {[}74, \textquotesingle{}(12C)H3(37Cl)\textquotesingle{}, 0.2394912, 51.98938, \textquotesingle{}CH3Cl\textquotesingle{}{]}, (25, 1): {[}75, \textquotesingle{}H2(16O)2\textquotesingle{}, 0.9949516, 34.00548, \textquotesingle{}H2O2\textquotesingle{}{]}, (26, 1): {[}76, \textquotesingle{}(12C)2H2\textquotesingle{}, 0.9775989, 26.01565, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 2): {[}77, \textquotesingle{}(12C)(13C)H2\textquotesingle{}, 0.02196629, 27.019, \textquotesingle{}C2H2\textquotesingle{}{]}, (26, 3): {[}105, \textquotesingle{}(12C)2HD\textquotesingle{}, 0.0003045499, 27.02182, \textquotesingle{}C2H2\textquotesingle{}{]}, (27, 1): {[}78, \textquotesingle{}(12C)2H6\textquotesingle{}, 0.97699, 30.04695, \textquotesingle{}C2H6\textquotesingle{}{]}, (27, 2): {[}106, \textquotesingle{}(12C)H3(13C)H3\textquotesingle{}, 0.02195261, 31.05031, \textquotesingle{}C2H6\textquotesingle{}{]}, (28, 1): {[}79, \textquotesingle{}(31P)H3\textquotesingle{}, 0.9995329, 33.99724, \textquotesingle{}PH3\textquotesingle{}{]}, (29, 1): {[}80, \textquotesingle{}(12C)(16O)(19F)2\textquotesingle{}, 0.9865444, 65.99172, \textquotesingle{}COF2\textquotesingle{}{]}, (29, 2): {[}119, \textquotesingle{}(13C)(16O)(19F)2\textquotesingle{}, 0.01108366, 66.99508, \textquotesingle{}COF2\textquotesingle{}{]}, (30, 1): {[}126, \textquotesingle{}(32S)(19F)6\textquotesingle{}, 0.95018, 145.9625, \textquotesingle{}SF6\textquotesingle{}{]}, (31, 1): {[}81, \textquotesingle{}H2(32S)\textquotesingle{}, 0.9498841, 33.98772, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 2): {[}82, \textquotesingle{}H2(34S)\textquotesingle{}, 0.04213687, 35.98351, \textquotesingle{}H2S\textquotesingle{}{]}, (31, 3): {[}83, \textquotesingle{}H2(33S)\textquotesingle{}, 0.007497664, 34.9871, \textquotesingle{}H2S\textquotesingle{}{]}, (32, 1): {[}84, \textquotesingle{}H(12C)(16O)(16O)H\textquotesingle{}, 0.9838977, 46.00548, \textquotesingle{}HCOOH\textquotesingle{}{]}, (33, 1): {[}85, \textquotesingle{}H(16O)2\textquotesingle{}, 0.9951066, 32.99766, \textquotesingle{}HO2\textquotesingle{}{]}, (34, 1): {[}86, \textquotesingle{}(16O)\textquotesingle{}, 0.997628, 15.99492, \textquotesingle{}O\textquotesingle{}{]}, (35, 1): {[}127, \textquotesingle{}(35Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.7495702, 96.95667, \textquotesingle{}ClONO2\textquotesingle{}{]}, (35, 2): {[}128, \textquotesingle{}(37Cl)(16O)(14N)(16O)2\textquotesingle{}, 0.2396937, 98.95372, \textquotesingle{}ClONO2\textquotesingle{}{]}, (36, 1): {[}87, \textquotesingle{}(14N)(16O)+\textquotesingle{}, 0.9939737, 29.99799, \textquotesingle{}NOp\textquotesingle{}{]}, (37, 1): {[}88, \textquotesingle{}H(16O)(79Br)\textquotesingle{}, 0.505579, 95.92108, \textquotesingle{}HOBr\textquotesingle{}{]}, (37, 2): {[}89, \textquotesingle{}H(16O)(81Br)\textquotesingle{}, 0.4918937, 97.91903, \textquotesingle{}HOBr\textquotesingle{}{]}, (38, 1): {[}90, \textquotesingle{}(12C)2H4\textquotesingle{}, 0.9772944, 28.0313, \textquotesingle{}C2H4\textquotesingle{}{]}, (38, 2): {[}91, \textquotesingle{}(12C)H2(13C)H2\textquotesingle{}, 0.02195946, 29.03466, \textquotesingle{}C2H4\textquotesingle{}{]}, (39, 1): {[}92, \textquotesingle{}(12C)H3(16O)H\textquotesingle{}, 0.9859299, 32.02622, \textquotesingle{}CH3OH\textquotesingle{}{]}, (40, 1): {[}93, \textquotesingle{}(12C)H3(79Br)\textquotesingle{}, 0.5009946, 93.94181, \textquotesingle{}CH3Br\textquotesingle{}{]}, (40, 2): {[}94, \textquotesingle{}(12C)H3(81Br)\textquotesingle{}, 0.4874334, 95.93976, \textquotesingle{}CH3Br\textquotesingle{}{]}, (41, 1): {[}95, \textquotesingle{}(12C)H3(12C)(14N)\textquotesingle{}, 0.9738662, 41.02655, \textquotesingle{}CH3CN\textquotesingle{}{]}, (42, 1): {[}96, \textquotesingle{}(12C)(19F)4\textquotesingle{}, 0.98889, 87.99362, \textquotesingle{}CF4\textquotesingle{}{]}, (43, 1): {[}116, \textquotesingle{}(12C)4H2\textquotesingle{}, 0.955998, 50.01565, \textquotesingle{}C4H2\textquotesingle{}{]}, (44, 1): {[}109, \textquotesingle{}H(12C)3(14N)\textquotesingle{}, 0.963346, 51.0109, \textquotesingle{}HC3N\textquotesingle{}{]}, (45, 1): {[}103, \textquotesingle{}H2\textquotesingle{}, 0.9996885, 2.01565, \textquotesingle{}H2\textquotesingle{}{]}, (45, 2): {[}115, \textquotesingle{}HD\textquotesingle{}, 0.0003114316, 3.021825, \textquotesingle{}H2\textquotesingle{}{]}, (46, 1): {[}97, \textquotesingle{}(12C)(32S)\textquotesingle{}, 0.9396236, 43.97207, \textquotesingle{}CS\textquotesingle{}{]}, (46, 2): {[}98, \textquotesingle{}(12C)(34S)\textquotesingle{}, 0.04168171, 45.96787, \textquotesingle{}CS\textquotesingle{}{]}, (46, 3): {[}99, \textquotesingle{}(13C)(32S)\textquotesingle{}, 0.0105565, 44.97543, \textquotesingle{}CS\textquotesingle{}{]}, (46, 4): {[}100, \textquotesingle{}(12C)(33S)\textquotesingle{}, 0.007416675, 44.97146, \textquotesingle{}CS\textquotesingle{}{]}, (47, 1): {[}114, \textquotesingle{}(32S)(16O)3\textquotesingle{}, 0.9434345, 79.95682, \textquotesingle{}SO3\textquotesingle{}{]}, (48, 1): {[}123, \textquotesingle{}(12C)2(14N)2\textquotesingle{}, 0.9707524, 52.00615, \textquotesingle{}C2N2\textquotesingle{}{]}, (49, 1): {[}124, \textquotesingle{}(12C)(16O)(35Cl)2\textquotesingle{}, 0.5663918, 97.93262, \textquotesingle{}COCl2\textquotesingle{}{]}, (49, 2): {[}125, \textquotesingle{}(12C)(16O)(35Cl)(37Cl)\textquotesingle{}, 0.362235, 99.92967, \textquotesingle{}COCl2\textquotesingle{}{]}, (50, 1): {[}146, \textquotesingle{}(32S)(16O)\textquotesingle{}, 0.9479262, 47.96699, \textquotesingle{}SO\textquotesingle{}{]}, (50, 2): {[}147, \textquotesingle{}(34S)(16O)\textquotesingle{}, 0.04205002, 49.96278, \textquotesingle{}SO\textquotesingle{}{]}, (50, 3): {[}148, \textquotesingle{}(32S)(18O)\textquotesingle{}, 0.001900788, 49.97123, \textquotesingle{}SO\textquotesingle{}{]}, (51, 1): {[}144, \textquotesingle{}(12C)H3(19F)\textquotesingle{}, 0.988428, 34.02188, \textquotesingle{}CH3F\textquotesingle{}{]}, (52, 1): {[}139, \textquotesingle{}(74Ge)H4\textquotesingle{}, 0.3651724, 77.95248, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 2): {[}140, \textquotesingle{}(72Ge)H4\textquotesingle{}, 0.2741292, 75.95338, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 3): {[}141, \textquotesingle{}(70Ge)H4\textquotesingle{}, 0.2050722, 73.95555, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 4): {[}142, \textquotesingle{}(73Ge)H4\textquotesingle{}, 0.07755167, 76.95476, \textquotesingle{}GeH4\textquotesingle{}{]}, (52, 5): {[}143, \textquotesingle{}(76Ge)H4\textquotesingle{}, 0.07755167, 79.9527, \textquotesingle{}GeH4\textquotesingle{}{]}, (53, 1): {[}131, \textquotesingle{}(12C)(32S)2\textquotesingle{}, 0.8928115, 75.94414, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 2): {[}132, \textquotesingle{}(32S)(12C)(34S)\textquotesingle{}, 0.07921026, 77.93994, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 3): {[}133, \textquotesingle{}(32S)(12C)(33S)\textquotesingle{}, 0.01409435, 76.94353, \textquotesingle{}CS2\textquotesingle{}{]}, (53, 4): {[}134, \textquotesingle{}(13C)(32S)2\textquotesingle{}, 0.01003057, 76.9475, \textquotesingle{}CS2\textquotesingle{}{]}, (54, 1): {[}145, \textquotesingle{}(12C)H3(127I)\textquotesingle{}, 0.988428, 141.9279, \textquotesingle{}CH3I\textquotesingle{}{]}, (55, 1): {[}136, \textquotesingle{}(14N)(19F)3\textquotesingle{}, 0.996337, 70.99829, \textquotesingle{}NF3\textquotesingle{}{]}\}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
molecular mass look\sphinxhyphen{}up based on the HAPI definition adapted to allow used to specify ISO list
INPUT PARAMETERS:
\begin{quote}

\sphinxAtStartPar
M: HITRAN molecule number
I: HITRAN isotopologue number
\end{quote}
\begin{description}
\sphinxlineitem{OUTPUT PARAMETERS:}
\sphinxAtStartPar
MolMass: molecular mass

\end{description}

\sphinxAtStartPar
—
DESCRIPTION:
\begin{quote}

\sphinxAtStartPar
Return molecular mass of HITRAN isotolopogue.
\end{quote}

\end{fulllineitems}



\chapter{Legal}
\label{\detokenize{index:legal}}
\sphinxAtStartPar
This software is subject to the \sphinxhref{https://www.nist.gov/director/licensing}{NIST Software License} (revised as of August 2018).


\chapter{Contact}
\label{\detokenize{index:contact}}
\sphinxAtStartPar
Erin Adkins
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{mailto:erin.adkins@nist.gov}{Email}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://www.nist.gov/people/erin-adkins}{NIST Staff Page}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/eadkins333}{GitHub profile}

\end{itemize}


\chapter{Links}
\label{\detokenize{index:links}}
\sphinxAtStartPar
\sphinxhref{https://github.com/usnistgov}{NIST GitHub Organization}

\sphinxAtStartPar
\sphinxhref{https://www.nist.gov/mml/csd/optical-measurements-group}{NIST Optical Measurements Group}

\sphinxAtStartPar
\sphinxhref{http://nist.gov}{NIST Home Page}


\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{search}

\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{m}
\item\relax\sphinxstyleindexentry{MATS.codata}\sphinxstyleindexpageref{MATS:\detokenize{module-MATS.codata}}
\item\relax\sphinxstyleindexentry{MATS.dataset}\sphinxstyleindexpageref{MATS:\detokenize{module-MATS.dataset}}
\item\relax\sphinxstyleindexentry{MATS.fit\_dataset}\sphinxstyleindexpageref{MATS:\detokenize{module-MATS.fit_dataset}}
\item\relax\sphinxstyleindexentry{MATS.generate\_fitparam\_file}\sphinxstyleindexpageref{MATS:\detokenize{module-MATS.generate_fitparam_file}}
\item\relax\sphinxstyleindexentry{MATS.linelistdata}\sphinxstyleindexpageref{MATS:\detokenize{module-MATS.linelistdata}}
\item\relax\sphinxstyleindexentry{MATS.spectrum}\sphinxstyleindexpageref{MATS:\detokenize{module-MATS.spectrum}}
\item\relax\sphinxstyleindexentry{MATS.utilities}\sphinxstyleindexpageref{MATS:\detokenize{module-MATS.utilities}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}